%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{times}
\expandafter\ifx\csname T@LGR\endcsname\relax
\else
% LGR was declared as font encoding
  \substitutefont{LGR}{\rmdefault}{cmr}
  \substitutefont{LGR}{\sfdefault}{cmss}
  \substitutefont{LGR}{\ttdefault}{cmtt}
\fi
\expandafter\ifx\csname T@X2\endcsname\relax
  \expandafter\ifx\csname T@T2A\endcsname\relax
  \else
  % T2A was declared as font encoding
    \substitutefont{T2A}{\rmdefault}{cmr}
    \substitutefont{T2A}{\sfdefault}{cmss}
    \substitutefont{T2A}{\ttdefault}{cmtt}
  \fi
\else
% X2 was declared as font encoding
  \substitutefont{X2}{\rmdefault}{cmr}
  \substitutefont{X2}{\sfdefault}{cmss}
  \substitutefont{X2}{\ttdefault}{cmtt}
\fi


\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{1}



\title{Clinical Knowledge Graph Documentation}
\date{May 11, 2020}
\release{1.0b1 BETA}
\author{Alberto Santos, Ana Rita Colaço, Annelaura B. Nielsen}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\ifdefined\shorthandoff
  \ifnum\catcode`\=\string=\active\shorthandoff{=}\fi
  \ifnum\catcode`\"=\active\shorthandoff{"}\fi
\fi

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


This web page contains the documentation for the Python code using \sphinxstylestrong{Sphinx}.

\noindent{\hspace*{\fill}\sphinxincludegraphics{{./src/report_manager/assets/banner}.jpg}\hspace*{\fill}}

\noindent{\hspace*{\fill}\sphinxincludegraphics{{banner}.jpg}\hspace*{\fill}}


\chapter{\sphinxstylestrong{Clinical Knowledge Graph}}
\label{\detokenize{INTRO:clinical-knowledge-graph}}\label{\detokenize{INTRO::doc}}
\sphinxstyleemphasis{version: 1.0b1} \sphinxstyleemphasis{BETA}

A Python project that allows you to analyse proteomics and clinical data, and integrate and mine knowledge from multiple biomedical databases widely used nowadays.
\begin{itemize}
\item {} 
Documentation: \sphinxurl{https://CKG.readthedocs.io}

\item {} 
GitHub: \sphinxurl{https://github.com/MannLabs/CKG}

\item {} 
Free and open source software: \sphinxhref{https://github.com/MannLabs/CKG/LICENSE.rst}{MIT license}

\item {} 
Reference: \sphinxurl{https://www.biorxiv.org/content/10.1101/2020.05.09.084897v1}

\end{itemize}


\section{Abstract}
\label{\detokenize{INTRO:abstract}}
\noindent{\hspace*{\fill}\sphinxincludegraphics{{./src/report_manager/assets/abstract}.png}\hspace*{\fill}}

\noindent{\hspace*{\fill}\sphinxincludegraphics{{abstract_figure}.png}\hspace*{\fill}}

The promise of precision medicine is to deliver personalized treatment based on the unique physiology of each patient. This concept was fueled by the genomic revolution, but it is now evident that integrating other types of omics data, like proteomics, into the clinical decision\sphinxhyphen{}making process will be essential to accomplish precision medicine goals. However, quantity and diversity of biomedical data, and the spread of clinically relevant knowledge across myriad biomedical databases and publications makes this exceptionally difficult. To address this, we developed the Clinical Knowledge Graph (CKG), an open source platform currently comprised of more than 16 million nodes and 220 million relationships to represent relevant experimental data, public databases and the literature. The CKG also incorporates the latest statistical and machine learning algorithms, drastically accelerating analysis and interpretation of typical proteomics workflows. We use several biomarker studies to illustrate how the CKG may support, enrich and accelerate clinical decision\sphinxhyphen{}making.


\section{Cloning and installing}
\label{\detokenize{INTRO:cloning-and-installing}}
The setting up of the CKG includes several steps and might take a few hours (if you are building the database from scratch). However, we have prepared documentation and manuals that will guide through every step.
To get a copy of the GitHub repository on your local machine, please open a terminal windown and run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git clone https://github.com/MannLabs/CKG.git
\end{sphinxVerbatim}

This will create a new folder named “CKG” on your current location. To access the documentation, use the ReadTheDocs link above, or open the html version stored in the \sphinxstyleemphasis{CKG} folder \sphinxtitleref{CKG/docs/build/html/index.html}. After this, follow the instructions in “First Steps” and “Getting Started”.

\begin{sphinxadmonition}{warning}{Warning:}
If git is not installed in your machine, please follow this \sphinxhref{https://www.atlassian.com/git/tutorials/install-git}{tutorial} to install it.
\end{sphinxadmonition}


\section{Features}
\label{\detokenize{INTRO:features}}\begin{itemize}
\item {} 
Cross\sphinxhyphen{}platform: Mac, and Linux are officially supported.

\item {} 
Docker container runs all neccessary steps to setup the CKG.

\end{itemize}


\section{Disclaimer}
\label{\detokenize{INTRO:disclaimer}}
This resource is intended for research purposes and must not substitute a doctor’s medical judgement or healthcare professional advice.


\section{Important Note}
\label{\detokenize{INTRO:important-note}}
The databases provided within the Clinical Knowledge Graph (CKG) have their own licenses and the use of CKG still requires compliance with these data use restrictions. Please, visit the data sources directly for more information:


\begin{savenotes}\sphinxatlongtablestart\begin{longtable}[c]{|l|l|l|l|}
\hline

\endfirsthead

\multicolumn{4}{c}%
{\makebox[0pt]{\sphinxtablecontinued{\tablename\ \thetable{} \textendash{} continued from previous page}}}\\
\hline

\endhead

\hline
\multicolumn{4}{r}{\makebox[0pt][r]{\sphinxtablecontinued{Continued on next page}}}\\
\endfoot

\endlastfoot

Source type
&
Source
&
URL
&
Reference
\\
\hline
Database
&
UniProt
&
\sphinxurl{https://www.uniprot.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/29425356}
\\
\hline
Database
&
TISSUES
&
\sphinxurl{https://tissues.jensenlab.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/29617745}
\\
\hline
Database
&
STRING
&
\sphinxurl{https://string-db.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/30476243}
\\
\hline
Database
&
STITCH
&
\sphinxurl{http://stitch.embl.de/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/26590256}
\\
\hline
Database
&
SMPDB
&
\sphinxurl{https://smpdb.ca/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/24203708}
\\
\hline
Database
&
SIGNOR
&
\sphinxurl{https://signor.uniroma2.it/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/31665520}
\\
\hline
Database
&
SIDER
&
\sphinxurl{http://sideeffects.embl.de/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/26481350}
\\
\hline
Database
&
RefSeq
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/refseq/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/26553804}
\\
\hline
Database
&
Reactome
&
\sphinxurl{https://reactome.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/31691815}
\\
\hline
Database
&
PhosphoSitePlus
&
\sphinxurl{https://www.phosphosite.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/25514926}
\\
\hline
Database
&
Pfam
&
\sphinxurl{https://pfam.xfam.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/30357350}
\\
\hline
Database
&
OncoKB
&
\sphinxurl{https://www.oncokb.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/28890946}
\\
\hline
Database
&
MutationDs
&
\sphinxurl{https://www.ebi.ac.uk/intact/resources/datasets\#mutationDs}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/30602777}
\\
\hline
Database
&
Intact
&
\sphinxurl{https://www.ebi.ac.uk/intact/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/24234451}
\\
\hline
Database
&
HPA
&
\sphinxurl{https://www.proteinatlas.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/21572409}
\\
\hline
Database
&
HMDB
&
\sphinxurl{https://hmdb.ca/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/29140435}
\\
\hline
Database
&
HGNC
&
\sphinxurl{https://www.genenames.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/30304474}
\\
\hline
Database
&
GwasCatalog
&
\sphinxurl{https://www.ebi.ac.uk/gwas/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/30445434}
\\
\hline
Database
&
FooDB
&
\sphinxurl{https://foodb.ca/}
&\\
\hline
Database
&
DrugBank
&
\sphinxurl{https://www.drugbank.ca/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/29126136}
\\
\hline
Database
&
DisGeNET
&
\sphinxurl{https://www.disgenet.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/25877637}
\\
\hline
Database
&
DISEASES
&
\sphinxurl{https://diseases.jensenlab.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/25484339}
\\
\hline
Database
&
DGIdb
&
\sphinxurl{http://www.dgidb.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/29156001}
\\
\hline
Database
&
CORUM
&
\sphinxurl{https://mips.helmholtz-muenchen.de/corum/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/30357367}
\\
\hline
Database
&
Cancer Genome Interpreter
&
\sphinxurl{https://www.cancergenomeinterpreter.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/29592813}
\\
\hline&&&\\
\hline
Ontology
&
Disease Ontology
&
\sphinxurl{https://disease-ontology.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/30407550}
\\
\hline
Ontology
&
Brenda Tissue Ontology
&
\sphinxurl{https://www.brenda-enzymes.org/ontology.php?ontology\_id=3}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/25378310}
\\
\hline
Ontology
&
Experimental Factor Ontology
&
\sphinxurl{https://www.ebi.ac.uk/efo/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/20200009}
\\
\hline
Ontology
&
Gene Ontology
&
\sphinxurl{http://geneontology.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/27899567}
\\
\hline
Ontology
&
Human Phenotype Ontology
&
\sphinxurl{https://hpo.jax.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/27899602}
\\
\hline
Ontology
&
SNOMED\sphinxhyphen{}CT
&
\sphinxurl{http://www.snomed.org/}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/27332304}
\\
\hline
Ontology
&
Protein Modification Ontology
&
\sphinxurl{https://www.ebi.ac.uk/ols/ontologies/mod}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/23482073}
\\
\hline
Ontology
&
Molecular Interactions Ontology
&
\sphinxurl{https://www.ebi.ac.uk/ols/ontologies/mi}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/23482073}
\\
\hline
Ontology
&
Mass Spectrometry Ontology
&
\sphinxurl{https://www.ebi.ac.uk/ols/ontologies/ms}
&
\sphinxurl{https://www.ncbi.nlm.nih.gov/pubmed/23482073}
\\
\hline
\end{longtable}\sphinxatlongtableend\end{savenotes}


\chapter{First steps}
\label{\detokenize{index:first-steps}}
Are you new to the Clinical Knowledge Graph? Learn about how to use it and
all the possibilities.
\begin{itemize}
\item {} 
\sphinxstylestrong{Getting started}:
{\hyperref[\detokenize{intro/getting-started-with-requirements::doc}]{\sphinxcrossref{\DUrole{doc}{With Requirements}}}} |
{\hyperref[\detokenize{intro/getting-started-with-neo4j::doc}]{\sphinxcrossref{\DUrole{doc}{With Neo4j}}}} |
{\hyperref[\detokenize{intro/getting-started-with-build::doc}]{\sphinxcrossref{\DUrole{doc}{With Clinical Knowledge Graph}}}} |
{\hyperref[\detokenize{intro/getting-started-with-windows::doc}]{\sphinxcrossref{\DUrole{doc}{With Windows}}}}

\end{itemize}


\section{Getting Started with some Requirements}
\label{\detokenize{intro/getting-started-with-requirements:getting-started-with-some-requirements}}\label{\detokenize{intro/getting-started-with-requirements::doc}}
The following instructions on installation of software requirements and setting up the Clinical Knowledge graph, are optimized for operating systems MacOS and Linux. For more detailed instructions on how to set up the CKG in Windows, pleaso go to {\hyperref[\detokenize{intro/getting-started-with-windows:windows-installation}]{\sphinxcrossref{\DUrole{std,std-ref}{Getting started with Windows}}}}.


\subsection{Java}
\label{\detokenize{intro/getting-started-with-requirements:java}}
Before starting setting up Neo4j and, later on, the Clinical Knowledge Graph, it is very important that you have \sphinxstyleemphasis{Java} installed in your machine.

Different versions of a Neo4j database can have different requirements. For example, Neo4j 3.5 versions require Oracle Java 8, while Neo4j 4.0 versions already require Oracle Java 11.
When using a new version of Neo4j, always remember to read the respective Operations Manual, and check for the software requirements.

To check if you already have \sphinxstylestrong{Java SE Development Kit} installed, run \sphinxcode{\sphinxupquote{java \sphinxhyphen{}version}} in your terminal window. This should print out three lines similar to the following, with possible variation in the version:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{java} \PYG{n}{version} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{1.8.0\PYGZus{}171}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{Java}\PYG{p}{(}\PYG{n}{TM}\PYG{p}{)} \PYG{n}{SE} \PYG{n}{Runtime} \PYG{n}{Environment} \PYG{p}{(}\PYG{n}{build} \PYG{l+m+mf}{1.8}\PYG{o}{.}\PYG{l+m+mi}{0}\PYG{n}{\PYGZus{}171}\PYG{o}{\PYGZhy{}}\PYG{n}{b11}\PYG{p}{)}
\PYG{n}{Java} \PYG{n}{HotSpot}\PYG{p}{(}\PYG{n}{TM}\PYG{p}{)} \PYG{l+m+mi}{64}\PYG{o}{\PYGZhy{}}\PYG{n}{Bit} \PYG{n}{Server} \PYG{n}{VM} \PYG{p}{(}\PYG{n}{build} \PYG{l+m+mf}{25.171}\PYG{o}{\PYGZhy{}}\PYG{n}{b11}\PYG{p}{,} \PYG{n}{mixed} \PYG{n}{mode}\PYG{p}{)}
\end{sphinxVerbatim}

Running \sphinxcode{\sphinxupquote{/usr/libexec/java\_home}} in the terminal should print out a path like \sphinxcode{\sphinxupquote{/Library/Java/JavaVirtualMachines/jdk1.8.0\_171.jdk/Contents/Home}}. Otherwise, please follow the steps below:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Go to \sphinxcode{\sphinxupquote{https://www.oracle.com/java/technologies/javase\sphinxhyphen{}downloads.html}} and download the version that fits your Neo4j version and OS requirements.

\item {} 
Install the package.

\item {} 
Run \sphinxcode{\sphinxupquote{/usr/libexec/java\_home}} in the terminal to make sure the \sphinxstyleemphasis{Java} package has been installed in \sphinxcode{\sphinxupquote{/Library/Java/JavaVirtualMachines/}}.

\end{enumerate}


\subsection{R}
\label{\detokenize{intro/getting-started-with-requirements:r}}
Another essential package for the functioning of the Clinical Knowledge Graph is R.

Make sure you have installed \sphinxstylestrong{R version \textgreater{}= 3.5.2}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} R \PYGZhy{}\PYGZhy{}version
\end{sphinxVerbatim}

And that R is installed in \sphinxcode{\sphinxupquote{/usr/local/bin/R}}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} which R
\end{sphinxVerbatim}

To install the neccessary R packages, simply initiate R (terminal or shell) and run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{install}\PYG{o}{.}\PYG{n}{packages}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{BiocManager}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{BiocManager}\PYG{p}{:}\PYG{p}{:}\PYG{n}{install}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{BiocManager}\PYG{p}{:}\PYG{p}{:}\PYG{n}{install}\PYG{p}{(}\PYG{n}{c}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{AnnotationDbi}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{GO.db}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{preprocessCore}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{impute}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{install}\PYG{o}{.}\PYG{n}{packages}\PYG{p}{(}\PYG{n}{c}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{flashClust}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{WGCNA}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{samr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{,} \PYG{n}{dependencies}\PYG{o}{=}\PYG{n}{TRUE}\PYG{p}{,} \PYG{n}{repos}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{http://cran.rstudio.com/}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
If you need to install R, follow \sphinxhref{https://web.stanford.edu/~kjytay/courses/stats32-aut2018/Session\%201/Installation\%20for\%20Mac.html}{these} tutorial.
\end{sphinxadmonition}

\begin{sphinxadmonition}{warning}{Warning:}
In Mac OS, make sure you have \sphinxstylestrong{XQuartz} installed.
\end{sphinxadmonition}

Now that you are all set, you can move on and start with Neo4j.


\section{Getting Started with Neo4j}
\label{\detokenize{intro/getting-started-with-neo4j:getting-started-with-neo4j}}\label{\detokenize{intro/getting-started-with-neo4j:id1}}\label{\detokenize{intro/getting-started-with-neo4j::doc}}
Getting started with Neo4j is easy.

First download a copy of the Neo4j desktop version from the \sphinxhref{https://neo4j.com/download/}{Neo4j download page}.
The Community Edition of the software is free but a sign up is required.
Once the file has downloaded, you can install Neo4j by following the instructions automatically opened in the browser.

\noindent\sphinxincludegraphics[width=0.320\linewidth]{{neo4j_app1}.png}

\noindent\sphinxincludegraphics[width=0.320\linewidth]{{neo4j_app2}.png}

\noindent\sphinxincludegraphics[width=0.320\linewidth]{{neo4j_app3}.png}

Open the Neo4j Desktop App and create a database by clicking \sphinxguilabel{Add graph}, followed by \sphinxguilabel{Create a Local Graph}, using the password “neo4j”.
Now that your database is created:

\noindent\sphinxincludegraphics[width=0.320\linewidth]{{neo4j_app4}.png}

\noindent\sphinxincludegraphics[width=0.320\linewidth]{{neo4j_app5}.png}

\noindent\sphinxincludegraphics[width=0.320\linewidth]{{neo4j_app6}.png}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Click \sphinxguilabel{Manage} and then \sphinxguilabel{Plugins}. Install “\sphinxstylestrong{APOC}” and “\sphinxstylestrong{GRAPH ALGORITHMS}”.

\item {} 
Click the tab \sphinxguilabel{Settings}, and comment the option \sphinxcode{\sphinxupquote{dbms.directories.import=import}} by adding \sphinxcode{\sphinxupquote{\#}} at the beginning of the line.

\item {} 
Click \sphinxguilabel{Apply} at the bottom of the window.

\item {} 
Start the Graph by clicking the play sign, at the top of the window.

\end{enumerate}

If the database starts and no errors are reported in the tab \sphinxguilabel{Logs}, you are redy go to!


\subsection{Add Neo4j graph database to \sphinxstyleemphasis{.bashrc}}
\label{\detokenize{intro/getting-started-with-neo4j:add-neo4j-graph-database-to-bashrc}}
In order run the graph database, add the path to the database to your \sphinxcode{\sphinxupquote{.bashrc}} (or \sphinxcode{\sphinxupquote{.bash\_profile}}) file.

To find out which of the files your machine uses, go to the terminal and type \sphinxcode{\sphinxupquote{more \textasciitilde{}/.bash}} and double press the tab key on your keyboard for auto\sphinxhyphen{}complete.
Immediately below, multiple filenames will be printed, check if among those, is \sphinxcode{\sphinxupquote{.bashrc}} or \sphinxcode{\sphinxupquote{.bash\_profile}}.

\begin{sphinxadmonition}{note}{Note:}
The bash file can be name \sphinxcode{\sphinxupquote{.bashrc}} or \sphinxcode{\sphinxupquote{.bash\_profile}}. if your system does not have either, created one of them (e.g. vi \textasciitilde{}/.bash\_profile).
\end{sphinxadmonition}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Open the \sphinxcode{\sphinxupquote{.bash\_profile}} (or \sphinxcode{\sphinxupquote{.bashrc}}) with your favourite text editor. In this case, we use the \sphinxstylestrong{vi} editor:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} vi \PYGZti{}/.bash\PYGZus{}profile
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
To edit with \sphinxstylestrong{vi} press \sphinxcode{\sphinxupquote{i}} on your keyboard.
\end{sphinxadmonition}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Add the path to the previously created Neo4j database to the file:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nv}{NEO4J\PYGZus{}HOME}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}/Users/username/Library/Application Support/Neo4j Desktop/Application/neo4jDatabases/database\PYGZhy{}identifier/installation\PYGZhy{}3.X.X/\PYGZdq{}}
\PYG{n+nb}{export} NEO4J\PYGZus{}HOME
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
To save and close a file with \sphinxstylestrong{vi} editor, press \sphinxcode{\sphinxupquote{Esc}} followed by \sphinxcode{\sphinxupquote{:wq}}.
\end{sphinxadmonition}

\begin{sphinxadmonition}{warning}{Warning:}
Depending on your system, the path may vary. To check the path to the database go to \sphinxcode{\sphinxupquote{Logs}} in the Neo4j Desktop interface.
\end{sphinxadmonition}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Reload the \sphinxcode{\sphinxupquote{.bashrc}} (or \sphinxcode{\sphinxupquote{.bash\_profile}})  file:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{source} \PYGZti{}/.bashrc
\end{sphinxVerbatim}

or

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{source} \PYGZti{}/.bash\PYGZus{}profile
\end{sphinxVerbatim}


\section{Getting Started with the CKG Build}
\label{\detokenize{intro/getting-started-with-build:getting-started-with-the-ckg-build}}\label{\detokenize{intro/getting-started-with-build::doc}}
Setting up the Clinical Knowledge Graph is straightforward.
Assuming you have \sphinxstylestrong{Python 3.6} already installed and added to \sphinxcode{\sphinxupquote{PATH}}, you can choose to create a virtual environment where all the packages with the specific versions will be installed. To do so, use Virtualenv.

To check which Python version is currently installed:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} python3.6 \PYGZhy{}\PYGZhy{}version
\end{sphinxVerbatim}

And where this Python version is:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} which python3.6
\end{sphinxVerbatim}

If this does not correspond to the correct Python version you want to run, you can create a shell alias in the bash file:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Open the bash file:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} vi \PYGZti{}/.bash\PYGZus{}profile
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Add at the end of the file:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nb}{alias} python3.6\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}/path/to/correct/python3.6\PYGZdq{}}
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Save and close the bash file

\item {} 
Make the alias available in the current session:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{source} \PYGZti{}/.bash\PYGZus{}profile
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
If you don’t have \sphinxstylestrong{Python 3.6} installed, \sphinxhref{https://www.python.org/}{download} the Python 3.6 version appropriate for your machine, and run the installer package. Python should be installed in \sphinxcode{\sphinxupquote{/Library/Frameworks/Python.framework/Versions/3.6/bin/python3.6}} and also found in \sphinxcode{\sphinxupquote{/usr/local/bin/python3.6}}.
\end{sphinxadmonition}


\subsection{Create a virtual environment}
\label{\detokenize{intro/getting-started-with-build:create-a-virtual-environment}}
Virtualenv is not installed by default on Macbook machines. To install it, run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} python3 \PYGZhy{}m pip install virtualenv
\end{sphinxVerbatim}

To create a new virtual environment using a costum version of Python, follow the steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Take note of the full path to the Python version you would like to use inside the virtual environment.

\item {} 
Navigate to the directory where you would like your virtual environment to be (e.g. user’s root).

\item {} 
Create the virtual environment at the same time you specify the version of Python you wish to use. \sphinxcode{\sphinxupquote{env\_name}} is the name of the virtual environment and can be set to anything you like.

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} virtualenv \PYGZhy{}p /path/to/python env\PYGZus{}name
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Activate the virtual environment by running:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{source} path/to/env\PYGZus{}name/bin/activate
\end{sphinxVerbatim}

After this, the name of the virtual environment will now appear on the left of the prompt:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{(}env\PYGZus{}name\PYG{o}{)} username\PYGZdl{}
\end{sphinxVerbatim}

If you are finished working in the virtual environment for the moment, you can deactivate it by running:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} deactivate
\end{sphinxVerbatim}


\subsection{Setting up the Clinical Knowledge Graph}
\label{\detokenize{intro/getting-started-with-build:setting-up-the-clinical-knowledge-graph}}
The first step in setting up the CKG, is to obtain the complete code by clone the GitHub repository:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git clone https://github.com/MannLabs/CKG.git
\end{sphinxVerbatim}

Once this is finished, you can find all the Python modules necessary to run the Clinical Knowledge Graph in \sphinxcode{\sphinxupquote{requirements.txt}}.
To install all the packages required, simply run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} CKG/
\PYGZdl{} pip install \PYGZhy{}r requirements.txt
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
Make sure the virtual environment previously created is active before installing \sphinxcode{\sphinxupquote{requirements.txt}}.
\end{sphinxadmonition}

Now that all the packages are correctly installed, you will have to create the appropriate directory architecture within the local copy of the cloned repository:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} python setup\PYGZus{}CKG.py
\PYGZdl{} python setup\PYGZus{}config\PYGZus{}files.py
\end{sphinxVerbatim}

This will automatically create the \sphinxcode{\sphinxupquote{data}} folder and all subfolders, as well as setup the configuration for the log files where all errors and warnings related to the code will be written to.


\subsection{Add CKG to \sphinxstyleemphasis{.bashrc}}
\label{\detokenize{intro/getting-started-with-build:add-ckg-to-bashrc}}
In order run the the Clinical Knowledge Graph, add the path to the code to your \sphinxcode{\sphinxupquote{.bashrc}} (or \sphinxcode{\sphinxupquote{.bash\_profile}}):
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Open the .bashrc file.

\item {} 
Add the following lines to the file and save it:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nv}{PYTHONPATH}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZdl{}\PYGZob{}}\PYG{n+nv}{PYTHONPATH}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{:/path/to/folder/CKG/src/}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n+nb}{export} PYTHONPATH
\end{sphinxVerbatim}

Notice that the path should always finish with “/CKG/src/”.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
To reload the bash file, first deactivate the virtual environment, reload \textasciitilde{}/.bashrc, and activate the virtual environment again:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} deactivate
\PYGZdl{} \PYG{n+nb}{source} \PYGZti{}/.bashrc
\PYGZdl{} \PYG{n+nb}{source} path/to/env\PYGZus{}name/bin/activate
\end{sphinxVerbatim}
\begin{wrapfigure}{r}{240\sphinxpxdimen}
\centering
\noindent\sphinxincludegraphics[width=240\sphinxpxdimen]{{snomed_folder}.png}
\caption{SNOMED\sphinxhyphen{}CT ontology folder.}\label{\detokenize{intro/getting-started-with-build:id2}}\end{wrapfigure}


\subsection{Build Neo4j graph database}
\label{\detokenize{intro/getting-started-with-build:build-neo4j-graph-database}}\label{\detokenize{intro/getting-started-with-build:id1}}
The building of the CKG database is thoroughly automated. Most of the biomedical databases and ontology files will automatically be downloaded during building of the database. However, the following licensed databases have to be downloaded manually.
\begin{wrapfigure}{r}{240\sphinxpxdimen}
\centering
\noindent\sphinxincludegraphics[width=240\sphinxpxdimen]{{drugbank_folder}.png}
\caption{DrugBank database folder.}\label{\detokenize{intro/getting-started-with-build:id3}}\end{wrapfigure}
\begin{itemize}
\item {} 
\sphinxhref{https://www.phosphosite.org/staticDownloads}{PhosphoSitePlus}: \sphinxstyleemphasis{Acetylation\_site\_dataset.gz}, \sphinxstyleemphasis{Disease\sphinxhyphen{}associated\_sites.gz}, \sphinxstyleemphasis{Kinase\_Substrate\_Dataset.gz}, \sphinxstyleemphasis{Methylation\_site\_dataset.gz}, \sphinxstyleemphasis{O\sphinxhyphen{}GalNAc\_site\_dataset.gz}, \sphinxstyleemphasis{O\sphinxhyphen{}GlcNAc\_site\_dataset.gz}, \sphinxstyleemphasis{Phosphorylation\_site\_dataset.gz}, \sphinxstyleemphasis{Regulatory\_sites.gz}, \sphinxstyleemphasis{Sumoylation\_site\_dataset.gz} and \sphinxstyleemphasis{Ubiquitination\_site\_dataset.gz}.

\item {} 
\sphinxhref{https://www.drugbank.ca/releases/latest}{DrugBank}: \sphinxstyleemphasis{All drugs} (under \sphinxstyleemphasis{COMPLETE DATABASE}) and \sphinxstyleemphasis{DrugBank Vocabulary} (under \sphinxstyleemphasis{OPEN DATA}).

\end{itemize}
\begin{wrapfigure}{r}{240\sphinxpxdimen}
\centering
\noindent\sphinxincludegraphics[width=240\sphinxpxdimen]{{psp_folder}.png}
\caption{PhosphoSitePlus database folder.}\label{\detokenize{intro/getting-started-with-build:id4}}\end{wrapfigure}
\begin{itemize}
\item {} 
\sphinxhref{https://www.nlm.nih.gov/healthit/snomedct/international.html}{SNOMED\sphinxhyphen{}CT}: \sphinxstyleemphasis{Download RF2 Files Now!}.

\end{itemize}

After download, move the files to their respective folders:
\begin{itemize}
\item {} 
PhosphoSitePlus: \sphinxcode{\sphinxupquote{CKG/data/databases/PhosphoSitePlus}}

\item {} 
DrugBank: \sphinxcode{\sphinxupquote{CKG/data/databases/DrugBank}}

\item {} 
SNOMED\sphinxhyphen{}CT: \sphinxcode{\sphinxupquote{CKG/data/ontologies/SNOMED\sphinxhyphen{}CT}}

\end{itemize}

In the case of SNOMED\sphinxhyphen{}CT, unzip the downloaded file and copy all the subfolders and files to the \sphinxcode{\sphinxupquote{SNOMED\sphinxhyphen{}CT}} folder.

\begin{sphinxadmonition}{warning}{Warning:}
These three databases require login and authentication. To sign up go to \sphinxhref{https://www.phosphosite.org/signUpAction}{PSP Sign up}, \sphinxhref{https://www.drugbank.ca/public\_users/sign\_up}{DrugBank Sign up} and \sphinxhref{https://uts.nlm.nih.gov/license.html}{SNOMED\sphinxhyphen{}CT Sign up}. In the case of SNOMED\sphinxhyphen{}CT, the UMLS license can take several business days.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
If the respective database folder is not created, please do it manually.
\end{sphinxadmonition}

The last step is to build the database, which can be done using the \sphinxcode{\sphinxupquote{builder.py}} module or a \sphinxcode{\sphinxupquote{dump file}}.


\subsubsection{From builder.py}
\label{\detokenize{intro/getting-started-with-build:from-builder-py}}
To build the graph database, run \sphinxcode{\sphinxupquote{builder.py}}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} src/graphdb\PYGZus{}builder/builder
\PYGZdl{} python builder.py \PYGZhy{}b full \PYGZhy{}u neo4j
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
Before running \sphinxcode{\sphinxupquote{builder.py}}, please make sure your Neo4j graph is running. The builder will fail otherwise.
\end{sphinxadmonition}

This action will take aproximately 6 hours but depending on a multitude of factors, it can take up to 10 hours.


\subsubsection{From a backup dump file}
\label{\detokenize{intro/getting-started-with-build:from-a-backup-dump-file}}
A dump file of the database is also made available in \sphinxtitleref{add link to the dump file!!} and alternatively, you can use it to load the graph database contained in it:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} /path/to/neo4jDatabases/database\PYGZhy{}identifier/installation\PYGZhy{}x.x.x/
\PYGZdl{} mkdir backups
\PYGZdl{} mkdir backups/graph.db
\PYGZdl{} cp \PYG{l+m}{2019}\PYGZhy{}11\PYGZhy{}04.dump backups/graph.db/.
\end{sphinxVerbatim}

After copying the dump file to backups/graph.db/, make sure the graph database is shutdown and run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} bin/neo4j\PYGZhy{}admin load \PYGZhy{}\PYGZhy{}from\PYG{o}{=}backups/graph.db/2019\PYGZhy{}11\PYGZhy{}04.dump \PYGZhy{}\PYGZhy{}database\PYG{o}{=}graph.db \PYGZhy{}\PYGZhy{}force
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
Make sure the dump file naming in the command above, matches the one provided to you.
\end{sphinxadmonition}

In some systems you might have to run this as root:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} sudo bin/neo4j\PYGZhy{}admin load \PYGZhy{}\PYGZhy{}from\PYG{o}{=}backups/graph.db/2019\PYGZhy{}11\PYGZhy{}04.dump \PYGZhy{}\PYGZhy{}database\PYG{o}{=}graph.db \PYGZhy{}\PYGZhy{}force
\PYGZdl{} sudo chown \PYGZhy{}R username data/databases/graph.db/
\end{sphinxVerbatim}

Once you are done, start the database and you will have a functional graph database.

However, be aware the database contained in the dump file \sphinxstylestrong{does NOT} include the licensed databases (\sphinxstylestrong{PhosphoSitePlus}, \sphinxstylestrong{DrugBank} and \sphinxstylestrong{SNOMED\sphinxhyphen{}CT}).

To add the missing ontology and databases, as well as their dependencies (relationships to other nodes), please manually download the files as explained in {\hyperref[\detokenize{intro/getting-started-with-build:build-neo4j-graph-database}]{\sphinxcrossref{\DUrole{std,std-ref}{Build Neo4j graph database}}}}, and run the following commands:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} CKG/src/graphdb\PYGZus{}builder/builder
\PYGZdl{} python builder.py \PYGZhy{}b import \PYGZhy{}i ontologies \PYGZhy{}d Clinical\PYGZus{}variable \PYGZhy{}w False \PYGZhy{}u username
\PYGZdl{} python builder.py \PYGZhy{}b import \PYGZhy{}i databases \PYGZhy{}d phosphositeplus drugbank \PYGZhy{}n \PYG{l+m}{2} \PYGZhy{}w False \PYGZhy{}u username
\PYGZdl{} python builder.py \PYGZhy{}b load \PYGZhy{}l ontologies \PYGZhy{}s SNOMED\PYGZus{}CT \PYGZhy{}u username
\PYGZdl{} python builder.py \PYGZhy{}b load \PYGZhy{}l modified\PYGZus{}proteins drugs mentions side effects pathway project experiment \PYGZhy{}u username
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
Remember of replace the \sphinxcode{\sphinxupquote{username}} in each command, with your own neo4j username.
\end{sphinxadmonition}


\subsubsection{More on the dump file}
\label{\detokenize{intro/getting-started-with-build:more-on-the-dump-file}}
Another great use for the dump file, is to generate backups of the database (e.g. different versions of the imported biomedical databases).
To generate a dump file of a specific Neo4j database, simply run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} /path/to/neo4jDatabases/database\PYGZhy{}identifier/installation\PYGZhy{}x.x.x/
\PYGZdl{} bin/neo4j\PYGZhy{}admin dump \PYGZhy{}\PYGZhy{}database\PYG{o}{=}neo4j \PYGZhy{}\PYGZhy{}to\PYG{o}{=}backups/graph.db/name\PYGZus{}of\PYGZus{}the\PYGZus{}file.dump
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
Remember to replace \sphinxcode{\sphinxupquote{name\_of\_the\_file}} with the name of the dump file you want to create.
\end{sphinxadmonition}


\section{Getting started with Windows}
\label{\detokenize{intro/getting-started-with-windows:getting-started-with-windows}}\label{\detokenize{intro/getting-started-with-windows:windows-installation}}\label{\detokenize{intro/getting-started-with-windows::doc}}
In this section we describe how to install all the neccessary requirements and set up the Clinical Knowledge Graph on a Windows operating system.


\subsection{Java}
\label{\detokenize{intro/getting-started-with-windows:java}}
Similarly to MacOS and Linux, Windows will also need a \sphinxstylestrong{Java} installation.

Be aware that different versions of a Neo4j database can have different requirements. For example, Neo4j 3.5 versions require Oracle Java 8, while Neo4j 4.0 versions already require Oracle Java 11.
When using a new version of Neo4j, always remember to read the respective Operations Manual, and check for the software requirements.

By default Java SE Development Kit should be installed on the Windows 10. If this is not your case, please follow this \sphinxhref{https://docs.oracle.com/javase/8/docs/technotes/guides/install/windows\_jdk\_install.html\#A1097936}{tutorial} to install it.


\subsection{R}
\label{\detokenize{intro/getting-started-with-windows:r}}
Another essential package for the functioning of the Clinical Knowledge Graph is R.

You can check if an \sphinxstylestrong{R version \textgreater{}= 3.5.2} is already installed by running:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} where R
\end{sphinxVerbatim}

If R is not installed in your machine, please follow \sphinxhref{https://rstudio-education.github.io/hopr/starting.html}{these tutorial}.

In order to simplify calling R from the command prompt, you can choose to add it to \sphinxcode{\sphinxupquote{PATH}} and to the environmental variables. To do so, follow the steps bellow:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Go to the Windows menu, right\sphinxhyphen{}click on \sphinxguilabel{Computer} and click on \sphinxguilabel{Properties}.

\item {} 
From the computer properties dialog, select \sphinxcode{\sphinxupquote{Advanced system settings}} on the left panel. And from there, click on \sphinxguilabel{Environment variables} button.

\item {} 
In the Environment variables dialog, click the \sphinxguilabel{New} button in the top half of the dialog, to make a new user variable.

\item {} 
Give the variable name as \sphinxcode{\sphinxupquote{R}} and the value is the path to the R executable, which is usually \sphinxcode{\sphinxupquote{C:\textbackslash{}Program Files\textbackslash{}R\textbackslash{}R\sphinxhyphen{}4.0.0\textbackslash{}bin\textbackslash{}R.exe}}.

\item {} 
In the bottom half of the Environment variables dialog, find the variable \sphinxcode{\sphinxupquote{Path}}, select it and click \sphinxguilabel{Edit}.

\item {} 
In the edit dialog window, add \sphinxcode{\sphinxupquote{;}} to the end of the variable value followed by the R path used when creating the previous environmental variable.

\item {} 
Click \sphinxguilabel{OK} to save, click \sphinxguilabel{OK} and \sphinxguilabel{OK} again to save the new variable and edit to \sphinxcode{\sphinxupquote{Path}}.

\end{enumerate}

To confirm that the environment variable is correctly set in command line type:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} \PYG{n+nb}{echo} \PYGZpc{}R\PYGZpc{}
\end{sphinxVerbatim}

This will print the path you used as value (e.g. \sphinxcode{\sphinxupquote{C:\textbackslash{}Program Files\textbackslash{}R\textbackslash{}R\sphinxhyphen{}4.0.0\textbackslash{}bin\textbackslash{}R.exe}}).

To run R from the command prompt, run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} R
\end{sphinxVerbatim}

All R packages can be installed by simply initiating R (command prompt or R shell) and running:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{install}\PYG{o}{.}\PYG{n}{packages}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{BiocManager}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{BiocManager}\PYG{p}{:}\PYG{p}{:}\PYG{n}{install}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{BiocManager}\PYG{p}{:}\PYG{p}{:}\PYG{n}{install}\PYG{p}{(}\PYG{n}{c}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{AnnotationDbi}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{GO.db}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{preprocessCore}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{impute}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{install}\PYG{o}{.}\PYG{n}{packages}\PYG{p}{(}\PYG{n}{c}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{flashClust}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{WGCNA}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{samr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{,} \PYG{n}{dependencies}\PYG{o}{=}\PYG{n}{TRUE}\PYG{p}{,} \PYG{n}{repos}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{http://cran.rstudio.com/}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
If the install does not work (cannot write to library), run a new command prompt as administrator:
\end{sphinxadmonition}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Go to the Windows menu, right\sphinxhyphen{}click on \sphinxguilabel{Command Prompt} and select \sphinxcode{\sphinxupquote{Run as administrator}}.

\end{enumerate}

In this new prompt, launch R and run the previous R install packages.


\subsection{Neo4j}
\label{\detokenize{intro/getting-started-with-windows:neo4j}}
The installation of Neo4j on Windows follows the same steps as {\hyperref[\detokenize{intro/getting-started-with-neo4j:getting-started-with-neo4j}]{\sphinxcrossref{\DUrole{std,std-ref}{Getting Started with Neo4j}}}}:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Download a copy of the Neo4j desktop version from the \sphinxhref{https://neo4j.com/download/}{Neo4j download page}.

\item {} 
Install Neo4j by following the instructions automatically opened in the browser.

\item {} 
Open the Neo4j Desktop App and create a database by clicking \sphinxguilabel{Add graph}, followed by \sphinxguilabel{Create a Local Graph}, using the password “neo4j”.

\item {} 
Click \sphinxguilabel{Manage} and then \sphinxguilabel{Plugins}. Install “\sphinxstylestrong{APOC}” and “\sphinxstylestrong{GRAPH ALGORITHMS}”.

\item {} 
Click the tab \sphinxguilabel{Settings}, and comment the option \sphinxcode{\sphinxupquote{dbms.directories.import=import}} by adding \sphinxcode{\sphinxupquote{\#}} at the beginning of the line.

\item {} 
Click \sphinxguilabel{Apply} at the bottom of the window.

\item {} 
Start the Graph by clicking the play sign, at the top of the window.

\end{enumerate}

To check for errors, please go to tab \sphinxguilabel{Logs}.


\subsubsection{Add Neo4j graph database to environmental variables}
\label{\detokenize{intro/getting-started-with-windows:add-neo4j-graph-database-to-environmental-variables}}\label{\detokenize{intro/getting-started-with-windows:id1}}
To add the Neo4j database you just created to your Windows environment variables:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Go to the Windows menu, right\sphinxhyphen{}click on \sphinxguilabel{Computer} and click on \sphinxguilabel{Properties}.

\item {} 
From the computer properties dialog, select \sphinxcode{\sphinxupquote{Advanced system settings}} on the left panel. And from there, click on \sphinxguilabel{Environment variables} button.

\item {} 
In the Environment variables dialog, click the \sphinxguilabel{New} button in the top half of the dialog, to make a new user variable

\item {} 
Give the variable name as \sphinxcode{\sphinxupquote{NEO4J\_HOME}} and the value is the path to the previously created Neo4j database to the file, for example \sphinxcode{\sphinxupquote{C:\textbackslash{}Neo4J\textbackslash{}neo4jDatabases\textbackslash{}database\sphinxhyphen{}bab515f2\sphinxhyphen{}ffe7\sphinxhyphen{}4282\sphinxhyphen{}9bb5\sphinxhyphen{}648a53b8b566\textbackslash{}installation\sphinxhyphen{}3.5.2\textbackslash{}}}

\item {} 
Click \sphinxguilabel{OK} and \sphinxguilabel{OK} again to save this variable.

\end{enumerate}

To confirm that the environment variable is correctly set in command line type:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} \PYG{n+nb}{echo} \PYGZpc{}NEO4J\PYGZus{}HOME\PYGZpc{}
\end{sphinxVerbatim}

This will print the path you used as value (e.g. \sphinxcode{\sphinxupquote{C:\textbackslash{}Neo4J\textbackslash{}neo4jDatabases\textbackslash{}database\sphinxhyphen{}bab515f2\sphinxhyphen{}ffe7\sphinxhyphen{}4282\sphinxhyphen{}9bb5\sphinxhyphen{}648a53b8b566\textbackslash{}installation\sphinxhyphen{}3.5.2\textbackslash{}}}).

\begin{sphinxadmonition}{warning}{Warning:}
Depending on your system, the path may vary. To check the path to the database go to \sphinxcode{\sphinxupquote{Logs}} in the Neo4j Desktop interface.
\end{sphinxadmonition}


\subsection{Getting Started with the CKG Build}
\label{\detokenize{intro/getting-started-with-windows:getting-started-with-the-ckg-build}}
Setting up the Clinical Knowledge Graph is thoroughly described here.
Assuming you have \sphinxstylestrong{Python 3.6} already installed, you can choose to create a virtual environment where all the packges with the specific versions will installed.

To check which Python version is currently installed, run in the command prompt (cmd.exe):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} python \PYGZhy{}\PYGZhy{}version
\end{sphinxVerbatim}

And to find your this Python version is installed:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} where python
\end{sphinxVerbatim}

If you don’t have \sphinxstylestrong{Python 3.6} installed in Windows, we recommend installing and using the Anaconda distribution of Python. Download at \sphinxcode{\sphinxupquote{https://www.anaconda.com/products/individual\#Downloads}}, and follow the instructions from \sphinxcode{\sphinxupquote{https://docs.anaconda.com/anaconda/install/windows/\#}}.


\subsubsection{Create a virtual environment}
\label{\detokenize{intro/getting-started-with-windows:create-a-virtual-environment}}
To create a new Python virtual environment, please use Anaconda conda installation.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
In the command prompt (cmd.exe) type:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} conda create \PYGZhy{}n env\PYGZus{}name \PYG{n+nv}{python}\PYG{o}{=}\PYG{l+m}{3}.6
\end{sphinxVerbatim}

The key word \sphinxcode{\sphinxupquote{python}} specifies which version of Python the virtual environment is requested to use, while \sphinxcode{\sphinxupquote{env\_name}} is the name of the virtual environment and can be set to anything you like.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{1}
\item {} 
Activate the virtual environment by running:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} conda activate env\PYGZus{}name
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{2}
\item {} 
After this, the name of the virtual environment will now appear on the left of the prompt:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{(}env\PYGZus{}name\PYG{o}{)} C:\PYG{l+s+se}{\PYGZbs{}\PYGZgt{}}
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{3}
\item {} 
If you are finished working in the virtual environment for the moment, you can deactivate it by running:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} conda deactivate
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
Remember, everytime you are working with the CKG, the virtual environment needs to be activated first.
\end{sphinxadmonition}


\subsubsection{Setting up the CKG}
\label{\detokenize{intro/getting-started-with-windows:setting-up-the-ckg}}
Once you have cloned the master branch of the CKG GitHub repository, all the Python packages necessary to run the Clinical Knowledge Graph can be found in \sphinxcode{\sphinxupquote{requirements.txt}}.

Unfortunately, due to incompatibilities of the current Anaconda version of the \sphinxcode{\sphinxupquote{rpy2}} package, this package needs to be removed from \sphinxcode{\sphinxupquote{requirements.txt}} before installing all other packages.

To do so, open the mentioned file in your preferred text editor tool (e.g. Notepad) and remove the line \sphinxcode{\sphinxupquote{rpy2==3.0.5}}. Save and close the file, making sure it is saved as a plain text file.

\begin{sphinxadmonition}{warning}{Warning:}
Part of the CKG functionality includes interfacing Python and R, and seemingly use R functions for data analysis. The python package \sphinxcode{\sphinxupquote{rpy2}} is used as this interface and unfortunately, the current release of this package for Windows is not compatible with CKG. Installation of the CKG on Windows machines, will therefore \sphinxstylestrong{not} allow the usage of R packages (SAMR and WGCNA) within the CKG.
\end{sphinxadmonition}

To install all the required packages, simply run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} \PYG{n+nb}{cd} CKG/
\PYGZgt{} conda install \PYGZhy{}\PYGZhy{}file requirements.txt
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
Make sure the virtual environment previously created is active before installing \sphinxcode{\sphinxupquote{requirements.txt}}.
\end{sphinxadmonition}

Now that all the packages are correctly installed, you will have to create the appropriate directory architecture within the local copy of the cloned repository:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} python setup\PYGZus{}CKG.py
\PYGZgt{} python setup\PYGZus{}config\PYGZus{}files.py
\end{sphinxVerbatim}

This will automatically create the \sphinxcode{\sphinxupquote{data}} folder and all subfolders, as well as setup the configuration for the log files where all errors and warnings related to the code will be written to.


\subsubsection{Add CKG to environmental variables}
\label{\detokenize{intro/getting-started-with-windows:add-ckg-to-environmental-variables}}
Similarly to {\hyperref[\detokenize{intro/getting-started-with-windows:add-neo4j-graph-database-to-environmental-variables}]{\sphinxcrossref{\DUrole{std,std-ref}{Add Neo4j graph database to environmental variables}}}}, CKG also needs to be added to the environmental variables.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Go to the Windows menu, right\sphinxhyphen{}click on \sphinxguilabel{Computer} and click on \sphinxguilabel{Properties}.

\item {} 
From the computer properties dialog, select \sphinxcode{\sphinxupquote{Advanced system settings}} on the left panel. And from there, click on \sphinxguilabel{Environment variables}.

\item {} 
In the Environment variables dialog, click \sphinxguilabel{New} in the top half of the dialog, to make a new user variable

\item {} 
Give the variable name as \sphinxcode{\sphinxupquote{PYTHONPATH}} and the value is the path to the CKG code directory, for example \sphinxcode{\sphinxupquote{C:\textbackslash{}CKG\textbackslash{}src}}. Notice that the path should always finish with \sphinxcode{\sphinxupquote{\textbackslash{}CKG\textbackslash{}src}}.

\item {} 
Click \sphinxguilabel{OK} and \sphinxguilabel{OK} again to save this variable.

\end{enumerate}

To confirm that the environment variable is correctly set in command line type:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZgt{} \PYG{n+nb}{echo} \PYGZpc{}PYTHONPATH\PYGZpc{}
\end{sphinxVerbatim}

This will print the path you used as value (e.g. \sphinxcode{\sphinxupquote{C:\textbackslash{}CKG\textbackslash{}src}}).


\subsubsection{Build Neo4j graph database (Windows)}
\label{\detokenize{intro/getting-started-with-windows:build-neo4j-graph-database-windows}}
Building the CKG database in Windows follows the same steps as in MacOS and Linux so, from here on, please follow the tutorial {\hyperref[\detokenize{intro/getting-started-with-build:build-neo4j-graph-database}]{\sphinxcrossref{\DUrole{std,std-ref}{Build Neo4j graph database}}}}.


\chapter{Getting started}
\label{\detokenize{index:getting-started}}\begin{itemize}
\item {} 
\sphinxstylestrong{Connecting to the CKG}:
{\hyperref[\detokenize{getting_started/connect-to-ckg::doc}]{\sphinxcrossref{\DUrole{doc}{Connect to DB}}}}

\item {} 
\sphinxstylestrong{Create a new user in the graph database}:
{\hyperref[\detokenize{getting_started/create-new-user::doc}]{\sphinxcrossref{\DUrole{doc}{Create new user}}}}

\item {} 
\sphinxstylestrong{Create a project in the database}:
{\hyperref[\detokenize{getting_started/create-new-project::doc}]{\sphinxcrossref{\DUrole{doc}{Project Creation}}}}

\item {} 
\sphinxstylestrong{Upload experimental data}:
{\hyperref[\detokenize{getting_started/upload-data::doc}]{\sphinxcrossref{\DUrole{doc}{Data Upload}}}}

\item {} 
\sphinxstylestrong{Define data analysis settings}:
{\hyperref[\detokenize{getting_started/data-analysis-config::doc}]{\sphinxcrossref{\DUrole{doc}{Configuration}}}}

\item {} 
\sphinxstylestrong{Access the analysis report}:
{\hyperref[\detokenize{getting_started/access-report::doc}]{\sphinxcrossref{\DUrole{doc}{Access report}}}}

\item {} 
\sphinxstylestrong{Report notification}:
{\hyperref[\detokenize{getting_started/notifications::doc}]{\sphinxcrossref{\DUrole{doc}{Notifications}}}}

\end{itemize}


\section{Connecting to the Clinical Knowledge Graph database}
\label{\detokenize{getting_started/connect-to-ckg:connecting-to-the-clinical-knowledge-graph-database}}\label{\detokenize{getting_started/connect-to-ckg::doc}}
In order to make use of the CKG database you just built, we need to connect to it and be able to query for data.
This connection is established via one of Neo4j’s Python drivers \sphinxcode{\sphinxupquote{Py2neo}}, a library and comprehensive toolkit developed to enable working with Neo4j from within Python applications, and should already be installed in your virtual environment.

Another essential tool when working with Neo4j databases, is the \sphinxhref{https://neo4j.com/developer/cypher-query-language/}{Cypher query language}. We recommend becoming familiar with it, to understand the queries used in the different analyses.


\subsection{Py2neo connector}
\label{\detokenize{getting_started/connect-to-ckg:py2neo-connector}}
Within the CKG package, the \sphinxcode{\sphinxupquote{graph\_connector}} module was created to connect the different parts of the Python code, to the Neo4j database and allow their interaction.

In this module, the \sphinxstyleemphasis{Graph} class from \sphinxcode{\sphinxupquote{py2neo}} is used to represent the graph data storage space within the Neo4j database, and a YAML configuration file is parsed to retrieve the connection details. The configuration file \sphinxcode{\sphinxupquote{connector\_config.yml}} contains the database server host name, server port, user to authenticate as, and password to use for authentication.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{graphdb\PYGZus{}connector} \PYG{k+kn}{import} \PYG{n}{connector}
\PYG{n}{driver} \PYG{o}{=} \PYG{n}{connector}\PYG{o}{.}\PYG{n}{getGraphDatabaseConnectionConfiguration}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

Once the connection is established, we can start querying the database. For example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{example\PYGZus{}query} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{MATCH (p:Project)\PYGZhy{}[:HAS\PYGZus{}ENROLLED]\PYGZhy{}(s:Subject) RETURN p.id as project\PYGZus{}id, COUNT(s) as n\PYGZus{}subjects}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{results} \PYG{o}{=} \PYG{n}{connector}\PYG{o}{.}\PYG{n}{getCursorData}\PYG{p}{(}\PYG{n}{driver}\PYG{o}{=}\PYG{n}{driver}\PYG{p}{,} \PYG{n}{query}\PYG{o}{=}\PYG{n}{example\PYGZus{}query}\PYG{p}{,} \PYG{n}{parameters}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

This query searches the database for all the available projects and counts how many subjects have been enrolled in each one, returning a pandas DataFrame with “project\_id” and “n\_subjects” as columns.


\section{Create a new user in the graph database}
\label{\detokenize{getting_started/create-new-user:create-a-new-user-in-the-graph-database}}\label{\detokenize{getting_started/create-new-user::doc}}
The creation of a new user includes two steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
The user who is currently logged\sphinxhyphen{}in in the database and invoking the commands, has to add the new user to the system and attribute it a role, by default \sphinxcode{\sphinxupquote{reader}}.

\item {} 
Each user is added in the graph database as a new \sphinxcode{\sphinxupquote{User}} node, with attributes: id, username, name, acronym, email, secondary email, phone number, affiliation, rolename and expiration date.

\end{enumerate}

There are multiple ways to create a new user:

\sphinxstylestrong{From the command line:} \sphinxstyleemphasis{(one user at a time)}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} src/graphdb\PYGZus{}builder/builder
\PYGZdl{} python create\PYGZus{}user.py \PYGZhy{}u username \PYGZhy{}d password \PYGZhy{}n name \PYGZhy{}e email \PYGZhy{}s second\PYGZus{}email \PYGZhy{}p phone\PYGZus{}number \PYGZhy{}a affiliation
\end{sphinxVerbatim}

\sphinxstylestrong{From an excel file:} \sphinxstyleemphasis{(multiple users)}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} src/graphdb\PYGZus{}builder/builder
\PYGZdl{} python create\PYGZus{}user.py \PYGZhy{}f path/to/excel/file
\end{sphinxVerbatim}

For help on how to use \sphinxcode{\sphinxupquote{create\_user.py}}, run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} python create\PYGZus{}user.py \PYGZhy{}h
\end{sphinxVerbatim}

For each new user, an access password is set be the same the the \sphinxcode{\sphinxupquote{username}}. All users are advised
to modify the default password as soon as they are granted access.


\section{Create a new project in the database}
\label{\detokenize{getting_started/create-new-project:create-a-new-project-in-the-database}}\label{\detokenize{getting_started/create-new-project::doc}}
The project creation app in the Clinical Knowledge Graph was designed to make the process straightforward and user\sphinxhyphen{}friendly.
To create a project, please follow the steps below.

\sphinxstylestrong{Neo4j}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Open neo4j desktop

\item {} 
Start the database

\end{enumerate}

\sphinxstylestrong{Terminal}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
In one terminal window:

\end{enumerate}
\begin{itemize}
\item {} 
Activate the virtual environment (if created beforehand)

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{source} /path/to/virtualenvironment/bin/activate\PYG{p}{;}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
Start a redis\sphinxhyphen{}server:

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} redis\PYGZhy{}server
\end{sphinxVerbatim}

\begin{sphinxadmonition}{warning}{Warning:}
If redis\sphinxhyphen{}server is not found, install with \sphinxcode{\sphinxupquote{brew install redis}} (Mac) or \sphinxcode{\sphinxupquote{sudo apt\sphinxhyphen{}get install redis\sphinxhyphen{}server}} (Linux).
\end{sphinxadmonition}

\begin{sphinxadmonition}{warning}{Warning:}
On Windows, redis\sphinxhyphen{}server should be installed by default (to check, go to start menu \textgreater{} services.msc \textgreater{} Redis). If that is not the case, go to \sphinxcode{\sphinxupquote{https://github.com/microsoftarchive/redis/releases}}, download the latest release installer (.msi file), and follow the installation instructions. Let all options as default but remember to select \sphinxstylestrong{Add the Redis installation folder to the PATH environment variable.}. To start a redis\sphinxhyphen{}server, make sure it is \sphinxstylestrong{not running in \textasciigrave{}\textasciigrave{}services.msc\textasciigrave{}\textasciigrave{}}, and run in the command prompt: C:”Program Files”Redisredis\sphinxhyphen{}server
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
\sphinxcode{\sphinxupquote{C:\textbackslash{}"Program Files"}} can be replaced with the correct path where you installed Redis from the installer (.msi file).
\end{sphinxadmonition}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{1}
\item {} 
In two separate terminal windows:

\end{enumerate}
\begin{itemize}
\item {} 
Navigate to \sphinxcode{\sphinxupquote{report\_manager}} in both of them

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} CKG/src/report\PYGZus{}manager
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
Start a celery queue from the report\_manager directory, in each window:

\end{itemize}

\sphinxstylestrong{Default queue}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} celery \PYGZhy{}A worker worker \PYGZhy{}l debug
\end{sphinxVerbatim}

\sphinxstylestrong{Compute queue \sphinxhyphen{} Report generation}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} celery \PYGZhy{}A worker worker \PYGZhy{}\PYGZhy{}loglevel\PYG{o}{=}DEBUG \PYGZhy{}\PYGZhy{}concurrency\PYG{o}{=}\PYG{l+m}{3} \PYGZhy{}E \PYGZhy{}Q compute
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{2}
\item {} 
In a fourth terminal window:

\end{enumerate}
\begin{itemize}
\item {} 
Run the report manager index app:

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} CKG/src/report\PYGZus{}manager
\PYGZdl{} python index.py
\end{sphinxVerbatim}

This will print some warnings, which should be okay.

\begin{sphinxadmonition}{warning}{Warning:}
Make sure that your virtual environment is always activated in each terminal window, before running any other command.
\end{sphinxadmonition}

\noindent{\hspace*{\fill}\sphinxincludegraphics[width=0.320\linewidth]{{homepage_app}.png}}

\sphinxstylestrong{Browser}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Copy the url \sphinxcode{\sphinxupquote{http://localhost:5000/}} into a web browser and you will be directed to a login page.

\item {} 
Enter your username and password

\end{enumerate}

This action will redirect you to the CKG home page app. From here, you can navigate to different applications, including the “Project Creation” app.

\begin{sphinxadmonition}{note}{Note:}
Username and password will be authenticated in the CKG database. For this reason, you should have been created as a new user in the database before this step.
\end{sphinxadmonition}


\subsection{Project creation}
\label{\detokenize{getting_started/create-new-project:project-creation}}\label{\detokenize{getting_started/create-new-project:id1}}
From the CKG app home page, you can navigate to the project creation app by clicking \sphinxcode{\sphinxupquote{PROJECT CREATION}} or pasting the url \sphinxcode{\sphinxupquote{http://localhost:5000/apps/projectCreationApp}} in the browser.
\begin{wrapfigure}{r}{240\sphinxpxdimen}
\centering
\noindent\sphinxincludegraphics[width=240\sphinxpxdimen]{{project_creation_app}.png}
\caption{Project Creation App}\label{\detokenize{getting_started/create-new-project:id2}}\end{wrapfigure}

Once you have been redirected, please fill in all the information needed to create a project. This includes all the fields marked with \sphinxcode{\sphinxupquote{*}} (mandatory). \sphinxstylestrong{(1)}
After all fields are filled in, please revise all the information and press \sphinxcode{\sphinxupquote{Create Project}}. \sphinxstylestrong{(2)}
The page will refresh and once finished, the project identifier will be depicted in front of the \sphinxcode{\sphinxupquote{Project information}} header. \sphinxstylestrong{(3)} Use this identifier to search for data related to your project.

At this stage, and if your project has been successfully created in the database, a new button will appear and the message will instruct you to download a compressed file with the experimental design and clinical data template files. To do so, please press the button “Download Clinical Data template”. \sphinxstylestrong{(4)}

\begin{sphinxadmonition}{note}{Note:}
Each field, with the exception of \sphinxcode{\sphinxupquote{Project name}}, \sphinxcode{\sphinxupquote{Project Acronym}},\textasciigrave{}\textasciigrave{}Number of subjects\textasciigrave{}\textasciigrave{},  \sphinxcode{\sphinxupquote{Project Description}}, \sphinxcode{\sphinxupquote{Starting Date}} and \sphinxcode{\sphinxupquote{Ending Date}}, can take multiple values. Select the most appropriate ones for your specific project.
\end{sphinxadmonition}
\begin{wrapfigure}{r}{240\sphinxpxdimen}
\centering
\noindent\sphinxincludegraphics[width=240\sphinxpxdimen]{{design_file}.png}
\caption{Experimental Design file example}\label{\detokenize{getting_started/create-new-project:id3}}\end{wrapfigure}

Fill in the \sphinxcode{\sphinxupquote{ExperimentalDesign\_Pxxxxxxx.xlsx}} file with your subject, biological sample and analytical sample identifiers. Please double\sphinxhyphen{}check they are correct, this information is essential to map the results correctly in the database.

The \sphinxcode{\sphinxupquote{ClinicalData\_Pxxxxxxx.xlsx}} file needs to be filled in with all the relevant clinical data and sample information. For more instructions on how to fill in the file, please see {\hyperref[\detokenize{getting_started/upload-data:upload-data}]{\sphinxcrossref{\DUrole{std,std-ref}{Upload project experimental data}}}}.

To check your project in the neo4j database interface:
\begin{itemize}
\item {} 
Open the Neo4j desktop app

\item {} 
Find the graph database in use and click \sphinxguilabel{Manage}, followed by \sphinxguilabel{Open Browser} (opens a new window).

\item {} 
In the new Neo4j window, click on the database symbol (top left corner) and, under \sphinxguilabel{Node Labels}, click \sphinxguilabel{Project}

\end{itemize}

At this point, you should be able to see all the nodes corresponding to projects loaded in the database.
To expand your project information, click on your project node and in the bottom of the window press the \sphinxcode{\sphinxupquote{\textless{}}} symbol. Here you will find all the attributes of the project, including the project identifier (typically “P000000xx”).


\section{Upload project experimental data}
\label{\detokenize{getting_started/upload-data:upload-project-experimental-data}}\label{\detokenize{getting_started/upload-data:upload-data}}\label{\detokenize{getting_started/upload-data::doc}}

\subsection{Prepare data for upload}
\label{\detokenize{getting_started/upload-data:prepare-data-for-upload}}

\subsubsection{Experimental Design}
\label{\detokenize{getting_started/upload-data:experimental-design}}\label{\detokenize{getting_started/upload-data:id1}}
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.800\linewidth]{{experimental_design}.png}
\caption{Experimental Design file example}\label{\detokenize{getting_started/upload-data:id3}}\end{figure}

Open the Experimental Design excel file, automatically downloaded when the project was created, and fill in the columns for \sphinxtitleref{subject external\_id}, \sphinxtitleref{biological\_sample external\_id} and \sphinxtitleref{analytical\_sample external\_id}.
The identifiers provided in this file \sphinxstylestrong{must} correspond to the identifiers used in the \sphinxstyleemphasis{Clinical Data} file, and to the column names in the \sphinxstyleemphasis{Proteomics} files (see below).

\begin{sphinxadmonition}{warning}{Warning:}
Make sure, within each column, the identifiers are unique. This means, if you have a subject “KO1”, no other subject can have the same identifier, but you can have a biological sample and/or analytical sample “KO1”.
\end{sphinxadmonition}


\subsubsection{Clinical Data}
\label{\detokenize{getting_started/upload-data:clinical-data}}
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.800\linewidth]{{clinical_file}.png}
\caption{Clinical Data file example}\label{\detokenize{getting_started/upload-data:id4}}\end{figure}

Open the Clinical Data excel file, automatically downloaded when the project was created, and fill in as much information as you can.
Be aware that the following columns are mandatory to fill in:
\begin{itemize}
\item {} 
\sphinxstylestrong{subject external\_id}: This is the identifier your subject has in your study so far (same identifiers as used in {\hyperref[\detokenize{getting_started/upload-data:experimental-design}]{\sphinxcrossref{\DUrole{std,std-ref}{Experimental Design}}}}, \sphinxstylestrong{subject external\_id}).

\item {} 
\sphinxstylestrong{tissue}: This is the name of the tissue each sample came from. Make sure it is also one of the tissues selected during Project creation.

\item {} 
\sphinxstylestrong{disease}: This should match the disease(s) you selected from the drop\sphinxhyphen{}down menu in the {\hyperref[\detokenize{getting_started/create-new-project:project-creation}]{\sphinxcrossref{\DUrole{std,std-ref}{Project creation}}}}.

\item {} 
\sphinxstylestrong{biological\_sample external\_id}: This is the identifier of the sample taken from your subject, if you have both blood and urine for every subject, you should correspondingly have two biological sample identifiers for each subject identifier (same identifiers as used in {\hyperref[\detokenize{getting_started/upload-data:experimental-design}]{\sphinxcrossref{\DUrole{std,std-ref}{Experimental Design}}}}, \sphinxstylestrong{biological\_sample external\_id}).

\item {} 
\sphinxstylestrong{biological\_sample quantity}: Amount of biological sample.

\item {} 
\sphinxstylestrong{biological\_sample quantity\_units}: Unit.

\item {} 
\sphinxstylestrong{analytical\_sample external\_id}: If multiple analyses were performed on the same biological sample, eg. proteomics and transcriptomics, there should be multiple analytical sample identifiers for every biological sample (same identifiers as used in {\hyperref[\detokenize{getting_started/upload-data:experimental-design}]{\sphinxcrossref{\DUrole{std,std-ref}{Experimental Design}}}}, \sphinxstylestrong{analytical\_sample external\_id}).

\item {} 
\sphinxstylestrong{analytical\_sample quantity}: Amount of sample used in the experiment.

\item {} 
\sphinxstylestrong{analytical\_sample quantity\_units}: Unit.

\item {} 
\sphinxstylestrong{grouping1}: Annotated grouping of each sample.

\item {} 
\sphinxstylestrong{grouping2}: If there are more than one grouping (two independent variables) use this column to add a second level.

\end{itemize}

Additional clinical information about your study subjects can be added in the subsequent columns (i.e. columns after “grouping2”).
Please use SNOWMED terms as headers for every new column you add. This will be used to gather existing information about the type of data you have.
To find an adequate SNOMED term for your clinical variables, please visit the \sphinxhref{https://browser.ihtsdotools.org/?perspective=full\&conceptId1=734000001\&edition=MAIN/2019-07-31\&release=\&languages=en}{SNOMED browser}.

\begin{sphinxadmonition}{note}{Note:}
Be aware, the two\sphinxhyphen{}independent\sphinxhyphen{}variable statistics is not yet implemented in the default analysis pipeline.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
To add a column with “Age” search for “age” in the SNOMED browser. This gives multiple matches, with the first one being: “Age (qualifier value), SCTID:397669002”. Please enter this information as your clinical variable column header with the SCTID in brackets: Age (qualifier value) (397669002)
\end{sphinxadmonition}

\begin{sphinxadmonition}{warning}{Warning:}
If an adequate SNOMED term is not available, please write an e\sphinxhyphen{}mail to \sphinxhref{mailto:annelaura.bach@cpr.ku.dk}{annelaura.bach@cpr.ku.dk} with the subject “Header Creation, CKG”. In the email please provide your “missing” header and a description of what it is. Do this before uploading the Clinical Data.
\end{sphinxadmonition}

\sphinxstylestrong{Additional columns:}
\begin{itemize}
\item {} 
\sphinxstylestrong{timpeoint}: To be used in the case of a longitudinal study. This is a relative measure within your samples timepoints. For example, if your timepoints are years 2015, 2016, 2017, 2018 and 2019, you would use “0”, “1”, “2”, “3” and “4” as values in this column.

\item {} 
\sphinxstylestrong{timepoint units}: Unit in which your \sphinxstylestrong{timepoint} is measured (e.g. “hours”, “days”, “years”).

\item {} 
\sphinxstylestrong{had\_intervention}: If a subject has been subjected to a determined medical intervention. For now, select only drugs that have been given to the subject (e.g. “327032007”). Use an appropriate SNOMED SCTID value.

\item {} 
\sphinxstylestrong{had\_intervention\_type}: This is the type of intervention applied to a subject. “drug treatment” is the only value available for now.

\item {} 
\sphinxstylestrong{had\_intervention\_in\_combination}: Boolean. If True, requires more than one value in \sphinxstylestrong{had\_intervention}.

\item {} 
\sphinxstylestrong{had\_intervention\_response}: “positive” or “negative”.

\item {} 
\sphinxstylestrong{studies\_intervention}: A medical intervention under study in the project. For example, study subjects before and after stomach bypass (SCTID:442338001). Use an appropriate SNOMED SCTID value.

\end{itemize}


\subsubsection{Proteomics data}
\label{\detokenize{getting_started/upload-data:proteomics-data}}\begin{itemize}
\item {} 
\sphinxstylestrong{MaxQuant}: Use “proteinGroups.txt”, “peptides.txt” and “Oxidation (M)Sites.txt” files, and any other relevant MaxQuant output files.

\item {} 
\sphinxstylestrong{Spectronaut}: Use “proteinGroupsReport.xlsx”. When exporting the results table from Spectronaut, please select “PG.ProteinAccessions” and “PG.Qvalue” under \sphinxtitleref{Row Labels}, and under \sphinxtitleref{Cell Values} select “PG.Quantity”, “PG.NrOfStrippedSequencesMeasured”, “PG.NrOfStrippedSequencesIdentified”, “PG.NrOfPrecursorsIdentified”, “PG.IsSingleHit”, “PG.NrOfStrippedSequencesUsedForQuantification”, “PG.NrOfModifiedSequencesUsedForQuantification”, “PG.NrOfPrecursorsUsedForQuantification”, “PG.MS1Quantity” and “PG.MS2Quantity”.

\end{itemize}

It is very important that all your column names have the following format: “LFQ intensity TechnicalReplicateNumber\_AnalyticalSampleIdentifier” or “TechnicalReplicateNumber\_AnalyticalSampleIdentifier.PG.Quantity”. Where “TechnicalReplicateNumber\_AnalyticalSampleIdentifier” should be replaced as shown in the example table below:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|T|}
\hline
\sphinxstyletheadfamily 
Technical
replicate
&\sphinxstyletheadfamily 
Analytical
sample id
&\sphinxstyletheadfamily 
Timepoint
&\sphinxstyletheadfamily 
Result
\\
\hline
1
&
KO\_plate1
&&
1\_KO\_plate1
\\
\hline
1
&
KO2\_plate1
&
0
&
1\_KO\_plate1\_0
\\
\hline
1
&
KO3\_plate1
&
30
&
1\_KO\_plate1\_30
\\
\hline
1
&
KO4\_plate2
&&
1\_KO4\_plate2
\\
\hline
2
&
KO4\_plate2
&&
2\_KO4\_plate2
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}

As shown in the example table, if your experimental design is a timecourse experiment, you should add “\_” followed by the timepoint, right after the analytical sample identifier. Otherwise, you can omit it.

Do not perform any post\sphinxhyphen{}processing filtering, imputations or similar on your data before uploading it. This will be carried out by the CKG. In the case of Spectronaut outputs, the missing values are automatically replaced by the keyword “Filtered”.

You can proceed to {\hyperref[\detokenize{getting_started/upload-data:upload}]{\sphinxcrossref{\DUrole{std,std-ref}{Upload Data}}}} when you have prepared your experimental design file, clinical and proteomics data.


\subsection{Upload Data}
\label{\detokenize{getting_started/upload-data:upload}}\label{\detokenize{getting_started/upload-data:id2}}
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.700\linewidth]{{data_upload_app}.png}
\caption{Data Upload App}\label{\detokenize{getting_started/upload-data:id5}}\end{figure}

In order to make data uploading simple, we created an app that takes care of this in only a few steps:

Go to \sphinxhref{http://localhost:5000/apps/dataUploadApp/}{dataUploadApp} or use the \sphinxcode{\sphinxupquote{Data Upload}} button in the \sphinxhref{http://localhost:5000/}{homepage app}, and follow the steps.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Fill in \sphinxcode{\sphinxupquote{Project identifier}} with your project external identifier from {\hyperref[\detokenize{getting_started/create-new-project:project-creation}]{\sphinxcrossref{\DUrole{std,std-ref}{Project creation}}}} and press \sphinxstyleemphasis{Enter}. \sphinxstylestrong{(1)} If the project identifier does not exist in the database, you will get and error. Otherwise, the menus below will unlock.

\item {} 
Select the type of data you will upload first. \sphinxstylestrong{(2)}
\begin{itemize}
\item {} 
If \sphinxcode{\sphinxupquote{proteomics}} or \sphinxcode{\sphinxupquote{longitudinal\_proteomics}} is selected, please also select the processing tool used (\sphinxcode{\sphinxupquote{MaxQuant}} or \sphinxcode{\sphinxupquote{Spectronaut}}). \sphinxstylestrong{(2a)}

\end{itemize}

\item {} 
Drag and drop or select the files to upload to the selected data type. \sphinxstylestrong{(3)}
\begin{itemize}
\item {} 
Multiple files can be selected at once. This is specially important in the case of proteomics files, please make sure you select all of the relevant MS files at once.

\end{itemize}

\item {} 
Select another data type to upload \sphinxstylestrong{(2)}, and drag and drop or select the files to upload \sphinxstylestrong{(3)}.

\item {} 
When you have uploaded all the relevant files, click \sphinxcode{\sphinxupquote{UPLOAD DATA TO CKG}}. After this button is clicked, it will deactivate all the menus. To restore its function, insert the project identifier and go through the previous steps again. \sphinxstylestrong{(4)}

\item {} 
Once the data is uploaded, click \sphinxcode{\sphinxupquote{Download Uploaded Files (.zip)}} to download all the upload files in a compressed format. \sphinxstylestrong{(5)}

\end{enumerate}

\begin{sphinxadmonition}{note}{Note:}
When the files are uploaded, the filenames are shown under \sphinxcode{\sphinxupquote{Uploaded Files:}}
To replace the files uploaded, just select the correct data type and processing tool, and reselect the files again.
\end{sphinxadmonition}

\begin{sphinxadmonition}{warning}{Warning:}
It is very important that you select \sphinxstylestrong{all} the files that should be uploaded to a certain data type \sphinxstylestrong{at once}.
\end{sphinxadmonition}


\section{Define data analysis parameters}
\label{\detokenize{getting_started/data-analysis-config:define-data-analysis-parameters}}\label{\detokenize{getting_started/data-analysis-config::doc}}
A multitude of different analysis methods and visualization plots have been implemented within the \sphinxcode{\sphinxupquote{analytics core}} of the Clinical Knowledge Graph.
The default workflow makes use of these resources and runs, for each data type, the analysis pipeline defined in a configuration file. In the CKG, we have
default analysis defined for Clinical data, Proteomics, and Multiomics. All the analysis configuration files can be modified to fit your project or data.

To check how each configuration file looks like and how to modify them, please follow the links below.


\subsection{Clinical data analysis parameters}
\label{\detokenize{getting_started/data_settings/clinical-data:clinical-data-analysis-parameters}}\label{\detokenize{getting_started/data_settings/clinical-data:clinical-data-conf-file}}\label{\detokenize{getting_started/data_settings/clinical-data::doc}}\begin{wrapfigure}{r}{350\sphinxpxdimen}
\centering
\noindent\sphinxincludegraphics[width=350\sphinxpxdimen]{{clinical_config}.png}
\caption{Clinical Data configuration file}\label{\detokenize{getting_started/data_settings/clinical-data:id1}}\end{wrapfigure}

The Clinical data configuration file contains two sections: \sphinxcode{\sphinxupquote{args}} and \sphinxcode{\sphinxupquote{overview}}.
The first section contains the parameters used for the processing of the raw clinical data. To obtain the raw clinical data, we query the CKG database for all the clinical variables connected to biological samples in a specific project. This results in a Pandas dataframe with all the relevant information. To process the raw data, a number of parameters are defined in the \sphinxcode{\sphinxupquote{args}} section of the configuration file:
\begin{itemize}
\item {} 
\sphinxstylestrong{subject\_id}: column label containing subject identifiers.

\item {} 
\sphinxstylestrong{sample\_id}: column label containing biological sample identifiers.

\item {} 
\sphinxstylestrong{group\_id}: column label containing group identifiers.

\item {} 
\sphinxstylestrong{imputation\_method}: method for missing values imputation (“KNN”, “distribuition”, or “mixed”).

\item {} 
\sphinxstylestrong{columns}: list of column names whose unique values will become the new column names

\item {} 
\sphinxstylestrong{values}: column label containing clinical variable values.

\item {} 
\sphinxstylestrong{extra}: additional column labels to be kept as columns

\end{itemize}

The result is another Pandas dataframe, stored as “processed”, where columns are the clinical variables and biological samples are rows, group and subject identifier are kept as columns as well.

\begin{sphinxadmonition}{note}{Note:}
We advise to change only \sphinxstylestrong{imputation\_method}, if needed.
\end{sphinxadmonition}

The second section (\sphinxcode{\sphinxupquote{overview}}) depicts the analysis performed for the clinical data, and the parameters used to do it.
Among the analysis is:
\begin{itemize}
\item {} 
Summary table (\sphinxstylestrong{clinical variables})

\item {} 
Stratification plot (\sphinxstylestrong{stratification})

\item {} 
Clinical variables per group (\sphinxstylestrong{measurement matrix})

\item {} 
Hypothesis test (\sphinxstylestrong{regulation})

\item {} 
Correlation network (\sphinxstylestrong{correlation})

\end{itemize}

Within each analysis, specific parameters are defined:
\begin{itemize}
\item {} 
\sphinxstylestrong{description}: Definition of the analysis used.

\item {} 
\sphinxstylestrong{data}: defines on which dataset dataframe the analysis will be ran (e.g. “clinical variables”, “original”, “processed”).

\item {} 
\sphinxstylestrong{analyses}: which statistical analysis to run on the \sphinxcode{\sphinxupquote{data}}. These functions are called from the mocdule \sphinxcode{\sphinxupquote{analytics\_factory.py}}.

\item {} 
\sphinxstylestrong{plots}: which plot to use to show the results of \sphinxstylestrong{analyses}. Functions also called from the mocdule \sphinxcode{\sphinxupquote{analytics\_factory.py}}.

\item {} 
\sphinxstylestrong{store\_analysis}: boolean. True if the dataframe resulting from \sphinxstylestrong{analyses} is to be stored.

\item {} 
\sphinxstylestrong{args}: all arguments necessary for \sphinxstylestrong{analyses} and \sphinxstylestrong{plots}.

\end{itemize}

You can modify the analysis parameters just by changing the respective parameters within the configuration file. Remember to consult the modules \sphinxcode{\sphinxupquote{analytics.py}} and \sphinxcode{\sphinxupquote{viz.py}}, to learn more about the arguments of each function.
If you would like to add a specific analysis step to the default pipeline, remember to add a call to the function in \sphinxcode{\sphinxupquote{analytics\_factory.py}}.


\subsection{Proteomics data analysis parameters}
\label{\detokenize{getting_started/data_settings/proteomics:proteomics-data-analysis-parameters}}\label{\detokenize{getting_started/data_settings/proteomics:proteomics-data-conf-file}}\label{\detokenize{getting_started/data_settings/proteomics::doc}}
Similarly to the {\hyperref[\detokenize{getting_started/data_settings/clinical-data:clinical-data-conf-file}]{\sphinxcrossref{\DUrole{std,std-ref}{Clinical data analysis parameters}}}}, the Proteomics configuration file is divided into sections: \sphinxcode{\sphinxupquote{args}}, \sphinxcode{\sphinxupquote{overview}}, \sphinxcode{\sphinxupquote{data exploration}}, \sphinxcode{\sphinxupquote{data associations}} and \sphinxcode{\sphinxupquote{enrichment}}.
\begin{wrapfigure}{r}{350\sphinxpxdimen}
\centering
\noindent\sphinxincludegraphics[width=350\sphinxpxdimen]{{proteomics_config_1}.png}
\caption{Proteomics configuration file}\label{\detokenize{getting_started/data_settings/proteomics:id1}}\end{wrapfigure}

The \sphinxcode{\sphinxupquote{args}} section contains the parameters used to process the proteomics data extracted from the CKG database, which includes the filtering of proteins according to the determined threshold, as well as the imputation of all missing values. These parameters include:
\begin{itemize}
\item {} 
\sphinxstylestrong{imputation}: boolean. Set to \sphinxstyleemphasis{True} if missing values shall be imputed.

\item {} 
\sphinxstylestrong{imputation\_method}: method for missing values imputation (“KNN”, “distribuition”, or “mixed”).

\item {} 
\sphinxstylestrong{missing\_method}:

\item {} 
\sphinxstylestrong{missing\_per\_group}: boolean. If \sphinxstyleemphasis{True}, proteins are filtered based on valid values per group; if \sphinxstyleemphasis{False} filter across all samples.

\item {} 
\sphinxstylestrong{missing\_max}: maximum ratio of missing/valid values to be filtered. (e.g. \sphinxstylestrong{0.3} filters all proteins with more than 30\% missing values).

\item {} 
\sphinxstylestrong{min\_valid}: minimum number of required valid values to keep a protein.

\item {} 
\sphinxstylestrong{value\_col}: column label containing expression values.

\item {} 
\sphinxstylestrong{index}: column labels to be be kept as index identifiers.

\end{itemize}

The result is a Pandas dataframe, stored as “processed”, where columns are protein identifiers (UniprotID\textasciitilde{}GeneName) and analytical samples are rows, group and subject identifier are kept as columns as well.

\begin{sphinxadmonition}{note}{Note:}
We advise to change only \sphinxstylestrong{imputation}, \sphinxstylestrong{imputation\_method}, \sphinxstylestrong{missing\_method}, \sphinxstylestrong{missing\_per\_group}, and \sphinxstylestrong{missing\_max} or \sphinxstylestrong{min\_valid}.
\end{sphinxadmonition}
\begin{wrapfigure}{r}{350\sphinxpxdimen}
\centering
\noindent\sphinxincludegraphics[width=350\sphinxpxdimen]{{proteomics_config_2}.png}
\caption{Proteomics configuration file (\sphinxstyleemphasis{continuation})}\label{\detokenize{getting_started/data_settings/proteomics:id2}}\end{wrapfigure}

The second section (\sphinxcode{\sphinxupquote{overview}}) corresponds to basic statistics and includes:
\begin{itemize}
\item {} 
Data summary statistics (\sphinxstylestrong{overview statistics})

\item {} 
Number of peptides per sample (\sphinxstylestrong{peptides})

\item {} 
Number of proteins per sample (\sphinxstylestrong{proteins})

\item {} 
Number of modifications per sample (\sphinxstylestrong{modifications})

\item {} 
Dynamic range per group (\sphinxstylestrong{ranking})

\item {} 
Coefficient of variation per group (\sphinxstylestrong{coefficient\_variation})

\end{itemize}

In the \sphinxcode{\sphinxupquote{data exploration}} section, we look at the sample stratification (\sphinxstylestrong{stratification}), differentially expressed proteins (\sphinxstylestrong{regulation}), and protein\sphinxhyphen{}protein correlation network (\sphinxstylestrong{correlation}). In this section, you can choose to modify parameters like \sphinxcode{\sphinxupquote{alpha}} (FDR value), \sphinxcode{\sphinxupquote{s0}} (artificial within groups variance) or \sphinxcode{\sphinxupquote{fc}} (fold\sphinxhyphen{}change), to better fit your data and experimental design. Likewise, the correlation network is, by default, set to show correlations above 50\% (\sphinxcode{\sphinxupquote{cutoff: 0.5}}), this too can be changed.

The \sphinxcode{\sphinxupquote{data associations}} section includes analyses that correlated differentially expressed proteins to proteins (\sphinxstylestrong{interaction\_network}), drugs (\sphinxstylestrong{drug\_associations}), diseases (\sphinxstylestrong{disease\_associations}) and publications (\sphinxstylestrong{literature\_associations}). All these associations are directly queried from the CKG database.

The last section in the default analysis pipeline corresponds to the \sphinxcode{\sphinxupquote{enrichment}} analysis, and includes both Gene Ontology (\sphinxstylestrong{go\_enrichment}) and Pathway (\sphinxstylestrong{pathway\_enrichment}) enrichment analyses, using Fisher’s exact test (\sphinxcode{\sphinxupquote{method: fisher}})

Likewise in the {\hyperref[\detokenize{getting_started/data_settings/clinical-data:clinical-data-conf-file}]{\sphinxcrossref{\DUrole{std,std-ref}{Clinical data analysis parameters}}}}, within each analysis, specific parameters are defined:
\begin{itemize}
\item {} 
\sphinxstylestrong{description}: Definition of the analysis used.

\item {} 
\sphinxstylestrong{data}: defines on which dataset dataframe the analysis will be ran (e.g. “clinical variables”, “original”, “processed”).

\item {} 
\sphinxstylestrong{analyses}: which statistical analysis to run on the \sphinxcode{\sphinxupquote{data}}. These functions are called from the mocdule \sphinxcode{\sphinxupquote{analytics\_factory.py}}.

\item {} 
\sphinxstylestrong{plots}: which plot to use to show the results of \sphinxstylestrong{analyses}. Functions also called from the mocdule \sphinxcode{\sphinxupquote{analytics\_factory.py}}.

\item {} 
\sphinxstylestrong{store\_analysis}: boolean. True if the dataframe resulting from \sphinxstylestrong{analyses} is to be stored.

\item {} 
\sphinxstylestrong{args}: all arguments necessary for \sphinxstylestrong{analyses} and \sphinxstylestrong{plots}.

\end{itemize}

You can modify the analysis parameters just by changing the respective parameters within the configuration file. Remember to consult the modules \sphinxcode{\sphinxupquote{analytics.py}} and \sphinxcode{\sphinxupquote{viz.py}}, to learn more about the arguments of each function.
If you would like to add a specific analysis step to the default pipeline, remember to add a call to the function in \sphinxcode{\sphinxupquote{analytics\_factory.py}}.


\subsection{Multiomics data analysis parameters}
\label{\detokenize{getting_started/data_settings/multiomics:multiomics-data-analysis-parameters}}\label{\detokenize{getting_started/data_settings/multiomics:multiomics-data-conf-file}}\label{\detokenize{getting_started/data_settings/multiomics::doc}}
The Multiomics configuration file allows you to integrate different data types and, as a default we integrate clinical and proteomics data.
In this context, the configuration file includes a multi\sphinxhyphen{}correlation section where all clinical and proteomics data are correlated and depicted as a network, and a Weighted Gene Co\sphinxhyphen{}expression Network Analysis (WGCNA) section, where features (proteins) are clustered in co\sphinxhyphen{}expression modules and further correlated to the clinical variables.
\begin{wrapfigure}{r}{350\sphinxpxdimen}
\centering
\noindent\sphinxincludegraphics[width=350\sphinxpxdimen]{{multiomics_config}.png}
\caption{Multiomics configuration file}\label{\detokenize{getting_started/data_settings/multiomics:id1}}\end{wrapfigure}


\section{Accessing the analysis report}
\label{\detokenize{getting_started/access-report:accessing-the-analysis-report}}\label{\detokenize{getting_started/access-report::doc}}
The standard report provides initial evaluation of the quality of the generated data, highlight relevant hits, and contextualize these hits in relation to different biomedical components in the graph.

There are two ways to acces the report, on the browser using a Dash App, or in a Jupyter notebook. For more information on each, please follow the links below.


\subsection{Report: Dash web app}
\label{\detokenize{getting_started/report_access/dash-app:report-dash-web-app}}\label{\detokenize{getting_started/report_access/dash-app::doc}}

\subsection{Report: Jupyter\sphinxhyphen{}notebook}
\label{\detokenize{getting_started/report_access/jupyter-notebook:report-jupyter-notebook}}\label{\detokenize{getting_started/report_access/jupyter-notebook::doc}}\begin{itemize}
\item {} 
Easy access to all the experimental data generated in the lab

\item {} 
All the results from statistical analysis

\item {} 
Visualization of reports

\item {} 
All the python functionality at hand

\item {} 
Additional analysis of project (e.g. treatment candidate selection, PTMs, predictions)

\end{itemize}


\section{Report notifications}
\label{\detokenize{getting_started/notifications:report-notifications}}\label{\detokenize{getting_started/notifications::doc}}
One of the biggest advantages of the standard analysis workflow of the Clinical Knowledge Graph is its speed. However, and depending on your experimental design, number od samples, quantified proteins, etc, this might take several minutes.
To make it easier on the user’s, and avoid waiting time in front of the screen,


\subsection{Slack notifications}
\label{\detokenize{getting_started/notifications:slack-notifications}}

\subsection{E\sphinxhyphen{}mail notifications}
\label{\detokenize{getting_started/notifications:e-mail-notifications}}

\chapter{The project report}
\label{\detokenize{index:the-project-report}}\begin{itemize}
\item {} 
\sphinxstylestrong{Generate a project}:
{\hyperref[\detokenize{project_report/project-report::doc}]{\sphinxcrossref{\DUrole{doc}{Project}}}}

\item {} 
\sphinxstylestrong{The Tabs}:
{\hyperref[\detokenize{project_report/project-tabs::doc}]{\sphinxcrossref{\DUrole{doc}{Project tabs}}}}

\end{itemize}


\section{Generating a project report}
\label{\detokenize{project_report/project-report:generating-a-project-report}}\label{\detokenize{project_report/project-report::doc}}

\section{Project report tabs}
\label{\detokenize{project_report/project-tabs:project-report-tabs}}\label{\detokenize{project_report/project-tabs::doc}}

\chapter{CKG Builder}
\label{\detokenize{index:ckg-builder}}\begin{itemize}
\item {} 
\sphinxstylestrong{Ontology sources and parsers}:
{\hyperref[\detokenize{ckg_builder/ontologies::doc}]{\sphinxcrossref{\DUrole{doc}{Ontologies}}}}

\item {} 
\sphinxstylestrong{Biomedical databases and resources}:
{\hyperref[\detokenize{ckg_builder/databases::doc}]{\sphinxcrossref{\DUrole{doc}{Databases}}}}

\item {} 
\sphinxstylestrong{Parsing experimental data}:
{\hyperref[\detokenize{ckg_builder/experiments::doc}]{\sphinxcrossref{\DUrole{doc}{Experiments}}}}

\item {} 
\sphinxstylestrong{Building the graph database from one module}:
{\hyperref[\detokenize{ckg_builder/graphdb-builder::doc}]{\sphinxcrossref{\DUrole{doc}{Builder}}}}

\end{itemize}


\section{Ontology sources and raw file parsers}
\label{\detokenize{ckg_builder/ontologies:ontology-sources-and-raw-file-parsers}}\label{\detokenize{ckg_builder/ontologies::doc}}

\section{Biomedical databases and resources}
\label{\detokenize{ckg_builder/databases:biomedical-databases-and-resources}}\label{\detokenize{ckg_builder/databases::doc}}

\section{Parsing experimental data}
\label{\detokenize{ckg_builder/experiments:parsing-experimental-data}}\label{\detokenize{ckg_builder/experiments::doc}}

\subsection{Clinical data}
\label{\detokenize{ckg_builder/experiments:clinical-data}}

\subsection{Proteomics}
\label{\detokenize{ckg_builder/experiments:proteomics}}\begin{itemize}
\item {} 
MaxQuant

\item {} 
Biognosys

\end{itemize}


\subsection{Phosphoproteomics}
\label{\detokenize{ckg_builder/experiments:phosphoproteomics}}

\subsection{WES}
\label{\detokenize{ckg_builder/experiments:wes}}

\section{Building the graph database from one Python module}
\label{\detokenize{ckg_builder/graphdb-builder:building-the-graph-database-from-one-python-module}}\label{\detokenize{ckg_builder/graphdb-builder::doc}}

\chapter{Advanced featues}
\label{\detokenize{index:advanced-featues}}\begin{itemize}
\item {} 
\sphinxstylestrong{CKG Statistics}:
{\hyperref[\detokenize{advanced_features/import-statistics::doc}]{\sphinxcrossref{\DUrole{doc}{Imports stats}}}} |
{\hyperref[\detokenize{advanced_features/graphdb-statistics::doc}]{\sphinxcrossref{\DUrole{doc}{Graph database stats}}}}

\item {} 
\sphinxstylestrong{Jupyter notebooks}:
{\hyperref[\detokenize{advanced_features/ckg-notebooks::doc}]{\sphinxcrossref{\DUrole{doc}{Notebooks}}}}

\item {} 
\sphinxstylestrong{Retrieving data from the CKG}:
{\hyperref[\detokenize{advanced_features/ckg-queries::doc}]{\sphinxcrossref{\DUrole{doc}{DB Querying}}}}  \textgreater{}\textgreater{}\textgreater{}\textgreater{}\textgreater{} query\_utils.find\_queries\_involving\_nodes + query\_utils.read\_knowledge\_queries()

\item {} 
\sphinxstylestrong{Data Analysis}:
{\hyperref[\detokenize{advanced_features/standard-analysis::doc}]{\sphinxcrossref{\DUrole{doc}{Analysis}}}}

\item {} 
\sphinxstylestrong{Visualization}:
{\hyperref[\detokenize{advanced_features/visualization-plots::doc}]{\sphinxcrossref{\DUrole{doc}{Plots}}}}

\item {} 
\sphinxstylestrong{R interface}:
{\hyperref[\detokenize{advanced_features/R-interface::doc}]{\sphinxcrossref{\DUrole{doc}{R wrapper}}}}

\end{itemize}


\section{Clinical Knowledge Graph Statistics: Imports}
\label{\detokenize{advanced_features/import-statistics:clinical-knowledge-graph-statistics-imports}}\label{\detokenize{advanced_features/import-statistics::doc}}

\section{Clinical Knowledge Graph Statistics: Database}
\label{\detokenize{advanced_features/graphdb-statistics:clinical-knowledge-graph-statistics-database}}\label{\detokenize{advanced_features/graphdb-statistics::doc}}

\section{Using Jupyter Notebooks with the Clinical Knowledge Graph}
\label{\detokenize{advanced_features/ckg-notebooks:using-jupyter-notebooks-with-the-clinical-knowledge-graph}}\label{\detokenize{advanced_features/ckg-notebooks::doc}}
The Jupyter Notebook is used to interact with the notebooks provided in the Clinical Knowledge Graph.
This open\sphinxhyphen{}source application allows you to create and share code, vizualize outputs and integrated multiple big data tools.

In order to get started, make sure you have Python installed (3.3 or greater) as well as Jupyter Notebook. The latter can be installed using pip (see below).For more detailed instructions, visit the \sphinxhref{https://jupyter.org/install.html}{offical guide}.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} python3 \PYGZhy{}m pip install \PYGZhy{}\PYGZhy{}upgrade pip
\PYGZdl{} python3 \PYGZhy{}m pip install jupyter
\end{sphinxVerbatim}

Congratulations! Now you can run the notebook, by typing the following command in the Terminal (Mac/Linux):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} jupyter notebook
\end{sphinxVerbatim}

Or,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} jupyter\PYGZhy{}notebook
\end{sphinxVerbatim}

As part of the Clinical Knowledge Graph package, we provide a series of Jupyter notebooks to facilitate the analysis of data, database querying and the use of multiple visualization tools. These notebooks can be found in \sphinxcode{\sphinxupquote{src/notebooks}}, under \sphinxcode{\sphinxupquote{reporting}} or \sphinxcode{\sphinxupquote{development}}.

\begin{sphinxadmonition}{note}{Note:}
If you would like to use two instances of the same notebook, just duplicate inplace and modify the name accordingly.
\end{sphinxadmonition}

\begin{sphinxadmonition}{warning}{Warning:}
If the Clinical Knowledge Graph is deployed in a server, please set up a Jupyter Hub in order to allow access to the Jupyter Notebook.
\end{sphinxadmonition}


\subsection{Reporting notebooks}
\label{\detokenize{advanced_features/ckg-notebooks:reporting-notebooks}}
Reporting notebooks referes to Jupyter notebooks that have been finished and properly tested by the devleopers, and are ready to be used by the community.
\begin{itemize}
\item {} 
\sphinxstylestrong{project\_reporting.ipynb}

\end{itemize}

Easy access to all the projects in the graph database. Loads all the data from a specific project (e.g. “P0000001”) and shows the report in the notebook.
This notebook enables the visualization of all the plots and tables that constitue a report, as well as all the dataframes used and produced during its generation.
By accessing the data directly, you can use the python functionality to further the analysis or visualization.
\begin{itemize}
\item {} 
\sphinxstylestrong{working\_with\_R.ipynb}

\end{itemize}

Notebook entirely written in R. One of the many advantages of using Jupyter notebooks is the possibility of writing in different programming languages.
In this notebook, we demonstrate how R can be used to, similarly to \sphinxstyleemphasis{project\_reporting.ipynb}, load a project and explore the analysis and plots.

In the beginning of the notebook, we create costum functions to load a project, read the report, read a dataset, and plot and network from a json file.
Other R functions like these can be developed by the users according to their needs.
\begin{itemize}
\item {} 
\sphinxstylestrong{Parallel plots.ipynb}

\end{itemize}

An example of a new interactive visualization method, not currently implemented in the Clinical Knowledge Graph, but using data from a stored project. We start by loading all the data and the report of a specific project (e.g. “P0000001”), and accessing diferent dataframes within the proteomics dataset, as well as, the correlation network. This plot is then converted into a Pandas DataFrame and used as input for the interactive Parallel plot.

The function is created and made interactive with Jupyter Widgets \sphinxcode{\sphinxupquote{interact}} function (\sphinxcode{\sphinxupquote{ipywidgets.interact}}), which automatically creates user interface (UI) controlos within the notebook. In this case, the user can select different clusters of proteins (from the correlation network) and observe their variation across groups.
\begin{itemize}
\item {} 
\sphinxstylestrong{Urachal Carcinoma Case Study.ipynb}

\end{itemize}

Jupyter notebook depicting the use of the Clinical Knowledge Graph database and the analytics core as a decision support tool, proposing a drug candidate in a specific subject case.

The project is analysed with the standard analytics workflow and a list of significantly differentially expressed proteins is returned. To further this analysis we first filter for regulated proteins that have been associated to the disease in study (lung cancer); we then search the database for known inhibitory drugs for these proteins; and to narrow down the list, we can query the database for each drug’s side effects. The treatment regimens are also available in the database and their side effects can be used to rank the proposed drugs. We can prioritize drugs with side effects dissimilar to the ones that caused an adverse reaction in the patient, and identify papers where these drugs have already been associated to the studied disease, further reducing the list of potential drugs candidates.


\subsection{Development notebooks}
\label{\detokenize{advanced_features/ckg-notebooks:development-notebooks}}
In \sphinxcode{\sphinxupquote{src/report\_manager/development}} we gathered Jupyter notebooks with analysis and workflows we believe are of interest for the users but are still under development.

When a notebook in this folder is functional and successfully benchmarked, the notebook is moved to the reporting directory.


\section{Retrieving data from the Clinical Knowledge Graph database}
\label{\detokenize{advanced_features/ckg-queries:retrieving-data-from-the-clinical-knowledge-graph-database}}\label{\detokenize{advanced_features/ckg-queries::doc}}

\section{Standardizing the data analysis}
\label{\detokenize{advanced_features/standard-analysis:standardizing-the-data-analysis}}\label{\detokenize{advanced_features/standard-analysis::doc}}

\section{Visualization plots}
\label{\detokenize{advanced_features/visualization-plots:visualization-plots}}\label{\detokenize{advanced_features/visualization-plots::doc}}

\section{Python \sphinxhyphen{} R interface}
\label{\detokenize{advanced_features/R-interface:python-r-interface}}\label{\detokenize{advanced_features/R-interface::doc}}

\chapter{System Requirements}
\label{\detokenize{index:system-requirements}}

\section{System Requirements}
\label{\detokenize{system_requirements:system-requirements}}\label{\detokenize{system_requirements::doc}}
The Clinical Knowledge Graph was conceived as a multi\sphinxhyphen{}user platform and therefore requires installation in a server\sphinxhyphen{}like setup and data systems administration knowledge. However, individual users can have local instances of the CKG, making sure data, software and hardware requirements are fulfilled.


\subsection{Data}
\label{\detokenize{system_requirements:data}}
Licensed databases used by the CKG package require login and authentication in order to download their data. This is the case of \sphinxstylestrong{SNOMED\sphinxhyphen{}CT}, \sphinxstylestrong{DrugBank} and \sphinxstylestrong{PhosphoSitePlus}.
Make sure you sign up to these three databases well in advance as the licensing process can take several days to conclude.

To sign up go to \sphinxhref{https://www.phosphosite.org/signUpAction}{PSP Sign up}, \sphinxhref{https://www.drugbank.ca/public\_users/sign\_up}{DrugBank Sign up} and \sphinxhref{https://uts.nlm.nih.gov/license.html}{SNOMED\sphinxhyphen{}CT Sign up}, and follow the instructions.

Once you have been given authorization to access the data, please download the files as follows:
\begin{itemize}
\item {} 
\sphinxhref{https://www.phosphosite.org/staticDownloads}{PhosphoSitePlus}: \sphinxstyleemphasis{Acetylation\_site\_dataset.gz}, \sphinxstyleemphasis{Disease\sphinxhyphen{}associated\_sites.gz}, \sphinxstyleemphasis{Kinase\_Substrate\_Dataset.gz}, \sphinxstyleemphasis{Methylation\_site\_dataset.gz}, \sphinxstyleemphasis{O\sphinxhyphen{}GalNAc\_site\_dataset.gz}, \sphinxstyleemphasis{O\sphinxhyphen{}GlcNAc\_site\_dataset.gz}, \sphinxstyleemphasis{Phosphorylation\_site\_dataset.gz}, \sphinxstyleemphasis{Regulatory\_sites.gz}, \sphinxstyleemphasis{Sumoylation\_site\_dataset.gz} and \sphinxstyleemphasis{Ubiquitination\_site\_dataset.gz}.

\item {} 
\sphinxhref{https://www.drugbank.ca/releases/latest}{DrugBank}: \sphinxstyleemphasis{All drugs} (under \sphinxstyleemphasis{COMPLETE DATABASE}) and \sphinxstyleemphasis{DrugBank Vocabulary} (under \sphinxstyleemphasis{OPEN DATA}).

\item {} 
\sphinxhref{https://www.nlm.nih.gov/healthit/snomedct/international.html}{SNOMED\sphinxhyphen{}CT}: \sphinxstyleemphasis{Download RF2 Files Now!}.

\end{itemize}

These files will be later used in {\hyperref[\detokenize{intro/getting-started-with-build:build-neo4j-graph-database}]{\sphinxcrossref{\DUrole{std,std-ref}{Build Neo4j graph database}}}}.


\subsection{Software}
\label{\detokenize{system_requirements:software}}\begin{itemize}
\item {} 
Java, version compatible with chosen Neo4j version (check Neo4j requirements)

\item {} 
R \textgreater{}= 3.5.2

\item {} 
Python 3.6

\item {} 
Redis server

\item {} 
Neo4j =\textless{} 3.15

\end{itemize}


\subsection{Hardware}
\label{\detokenize{system_requirements:hardware}}\begin{itemize}
\item {} 
Memory: 16Gb

\item {} 
Disk space: \textgreater{}= 80Gb

\item {} 
Stable internet connection

\end{itemize}


\chapter{API Reference}
\label{\detokenize{index:api-reference}}

\section{CKG package API Reference}
\label{\detokenize{src:ckg-package-api-reference}}\label{\detokenize{src::doc}}

\begin{savenotes}\sphinxatlongtablestart\begin{longtable}[c]{\X{1}{2}\X{1}{2}}
\hline

\endfirsthead

\multicolumn{2}{c}%
{\makebox[0pt]{\sphinxtablecontinued{\tablename\ \thetable{} \textendash{} continued from previous page}}}\\
\hline

\endhead

\hline
\multicolumn{2}{r}{\makebox[0pt][r]{\sphinxtablecontinued{Continued on next page}}}\\
\endfoot

\endlastfoot

\sphinxcode{\sphinxupquote{graphdb\_connector}}
&

\\
\hline
\sphinxcode{\sphinxupquote{graphdb\_builder}}
&

\\
\hline
\sphinxcode{\sphinxupquote{report\_manager}}
&

\\
\hline
\sphinxcode{\sphinxupquote{analytics\_core}}
&

\\
\hline
\sphinxcode{\sphinxupquote{notebooks}}
&

\\
\hline
\end{longtable}\sphinxatlongtableend\end{savenotes}


\subsection{Graph Database Connector}
\label{\detokenize{_autosummary/graphdb_connector:graph-database-connector}}\label{\detokenize{_autosummary/graphdb_connector::doc}}

\subsubsection{connector.py}
\label{\detokenize{_autosummary/graphdb_connector:module-graphdb_connector.connector}}\label{\detokenize{_autosummary/graphdb_connector:connector-py}}\index{graphdb\_connector.connector (module)@\spxentry{graphdb\_connector.connector}\spxextra{module}}\index{getGraphDatabaseConnectionConfiguration() (in module graphdb\_connector.connector)@\spxentry{getGraphDatabaseConnectionConfiguration()}\spxextra{in module graphdb\_connector.connector}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.connector.getGraphDatabaseConnectionConfiguration}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.connector.}}\sphinxbfcode{\sphinxupquote{getGraphDatabaseConnectionConfiguration}}}{\emph{configuration=None}, \emph{database=None}}{}
\end{fulllineitems}

\index{connectToDB() (in module graphdb\_connector.connector)@\spxentry{connectToDB()}\spxextra{in module graphdb\_connector.connector}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.connector.connectToDB}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.connector.}}\sphinxbfcode{\sphinxupquote{connectToDB}}}{\emph{host=\textquotesingle{}localhost\textquotesingle{}}, \emph{port=7687}, \emph{user=\textquotesingle{}neo4j\textquotesingle{}}, \emph{password=\textquotesingle{}password\textquotesingle{}}}{}
\end{fulllineitems}

\index{removeRelationshipDB() (in module graphdb\_connector.connector)@\spxentry{removeRelationshipDB()}\spxextra{in module graphdb\_connector.connector}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.connector.removeRelationshipDB}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.connector.}}\sphinxbfcode{\sphinxupquote{removeRelationshipDB}}}{\emph{entity1}, \emph{entity2}, \emph{relationship}}{}
\end{fulllineitems}

\index{modifyEntityProperty() (in module graphdb\_connector.connector)@\spxentry{modifyEntityProperty()}\spxextra{in module graphdb\_connector.connector}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.connector.modifyEntityProperty}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.connector.}}\sphinxbfcode{\sphinxupquote{modifyEntityProperty}}}{\emph{parameters}}{}
parameters: tuple with entity name, entity id, property name to modify, and value

\end{fulllineitems}

\index{sendQuery() (in module graphdb\_connector.connector)@\spxentry{sendQuery()}\spxextra{in module graphdb\_connector.connector}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.connector.sendQuery}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.connector.}}\sphinxbfcode{\sphinxupquote{sendQuery}}}{\emph{driver}, \emph{query}, \emph{parameters=\{\}}}{}
\end{fulllineitems}

\index{getCursorData() (in module graphdb\_connector.connector)@\spxentry{getCursorData()}\spxextra{in module graphdb\_connector.connector}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.connector.getCursorData}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.connector.}}\sphinxbfcode{\sphinxupquote{getCursorData}}}{\emph{driver}, \emph{query}, \emph{parameters=\{\}}}{}
\end{fulllineitems}

\index{create\_node() (in module graphdb\_connector.connector)@\spxentry{create\_node()}\spxextra{in module graphdb\_connector.connector}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.connector.create_node}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.connector.}}\sphinxbfcode{\sphinxupquote{create\_node}}}{\emph{driver}, \emph{node\_type}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{find\_node() (in module graphdb\_connector.connector)@\spxentry{find\_node()}\spxextra{in module graphdb\_connector.connector}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.connector.find_node}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.connector.}}\sphinxbfcode{\sphinxupquote{find\_node}}}{\emph{driver}, \emph{node\_type}, \emph{**kwargs}}{}
\end{fulllineitems}



\subsubsection{query\_utils.py}
\label{\detokenize{_autosummary/graphdb_connector:module-graphdb_connector.query_utils}}\label{\detokenize{_autosummary/graphdb_connector:query-utils-py}}\index{graphdb\_connector.query\_utils (module)@\spxentry{graphdb\_connector.query\_utils}\spxextra{module}}\index{read\_knowledge\_queries() (in module graphdb\_connector.query\_utils)@\spxentry{read\_knowledge\_queries()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.read_knowledge_queries}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{read\_knowledge\_queries}}}{\emph{dataset\_type=\textquotesingle{}proteomics\textquotesingle{}}}{}
\end{fulllineitems}

\index{read\_queries() (in module graphdb\_connector.query\_utils)@\spxentry{read\_queries()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.read_queries}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{read\_queries}}}{\emph{queries\_file}}{}
\end{fulllineitems}

\index{list\_queries() (in module graphdb\_connector.query\_utils)@\spxentry{list\_queries()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.list_queries}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{list\_queries}}}{\emph{queries}}{}
\end{fulllineitems}

\index{find\_queries\_involving\_nodes() (in module graphdb\_connector.query\_utils)@\spxentry{find\_queries\_involving\_nodes()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.find_queries_involving_nodes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{find\_queries\_involving\_nodes}}}{\emph{queries}, \emph{nodes}, \emph{print\_pretty=False}}{}
\end{fulllineitems}

\index{find\_queries\_involving\_relationships() (in module graphdb\_connector.query\_utils)@\spxentry{find\_queries\_involving\_relationships()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.find_queries_involving_relationships}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{find\_queries\_involving\_relationships}}}{\emph{queries}, \emph{rels}}{}
\end{fulllineitems}

\index{get\_query() (in module graphdb\_connector.query\_utils)@\spxentry{get\_query()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.get_query}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{get\_query}}}{\emph{queries}, \emph{query\_id}}{}
\end{fulllineitems}

\index{get\_description() (in module graphdb\_connector.query\_utils)@\spxentry{get\_description()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.get_description}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{get\_description}}}{\emph{query}}{}
\end{fulllineitems}

\index{get\_nodes() (in module graphdb\_connector.query\_utils)@\spxentry{get\_nodes()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.get_nodes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{get\_nodes}}}{\emph{query}}{}
\end{fulllineitems}

\index{get\_relationships() (in module graphdb\_connector.query\_utils)@\spxentry{get\_relationships()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.get_relationships}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{get\_relationships}}}{\emph{query}}{}
\end{fulllineitems}

\index{map\_node\_name\_to\_id() (in module graphdb\_connector.query\_utils)@\spxentry{map\_node\_name\_to\_id()}\spxextra{in module graphdb\_connector.query\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_connector:graphdb_connector.query_utils.map_node_name_to_id}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_connector.query\_utils.}}\sphinxbfcode{\sphinxupquote{map\_node\_name\_to\_id}}}{\emph{driver}, \emph{node}, \emph{value}}{}
\end{fulllineitems}



\subsection{Graph Database Builder}
\label{\detokenize{_autosummary/graphdb_builder:graph-database-builder}}\label{\detokenize{_autosummary/graphdb_builder::doc}}

\subsubsection{Ontology Databases}
\label{\detokenize{_autosummary/graphdb_builder.ontologies:ontology-databases}}\label{\detokenize{_autosummary/graphdb_builder.ontologies::doc}}

\paragraph{Ontologies Parsers}
\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:ontologies-parsers}}\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers::doc}}

\subparagraph{icdParser.py}
\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:module-graphdb_builder.ontologies.parsers.icdParser}}\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:icdparser-py}}\index{graphdb\_builder.ontologies.parsers.icdParser (module)@\spxentry{graphdb\_builder.ontologies.parsers.icdParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.ontologies.parsers.icdParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.ontologies.parsers.icdParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:graphdb_builder.ontologies.parsers.icdParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.ontologies.parsers.icdParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{ICDfile}}{}
Parses and extracts relevant data from ICD\sphinxhyphen{}10 files (Classification of Diseases).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{ICDfile}} \textendash{} list of files downloaded from the ontology database and used to generate nodes and relationships to the graph database.

\item[{Returns}] \leavevmode

Three nested dictionaries: terms, relationships between terms, and definitions of the terms.
\begin{itemize}
\item {} 
terms: Dictionary where each key is an ontology identifier (\sphinxstyleemphasis{str}) and the values are lists of names and synonyms (\sphinxstyleemphasis{list{[}str{]}}).

\item {} 
relationships: Dictionary of tuples (\sphinxstyleemphasis{str}). Each tuple contains two ontology identifiers (source and target) and                         the relationship type between them.

\item {} 
definitions: Dictionary with ontology identifiers as keys (\sphinxstyleemphasis{str}), and definition of the terms as values (\sphinxstyleemphasis{str}).

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}



\subparagraph{oboParser.py}
\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:module-graphdb_builder.ontologies.parsers.oboParser}}\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:oboparser-py}}\index{graphdb\_builder.ontologies.parsers.oboParser (module)@\spxentry{graphdb\_builder.ontologies.parsers.oboParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.ontologies.parsers.oboParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.ontologies.parsers.oboParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:graphdb_builder.ontologies.parsers.oboParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.ontologies.parsers.oboParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{ontology}, \emph{files}}{}
Multiple ontology database parser.
This function parses and extracts relevant data from: Disease Ontology, Tissues, Human Phenotype Ontology,     HUPO\sphinxhyphen{}PSI and Gene Ontology databases.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ontology}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the ontology to be imported (‘Disease’, ‘Tissue’, ‘Phenotype’, ‘Experiment’,                         ‘Modification’, ‘Molecular\_interactions’, ‘Gene\_ontology’)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{files}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of files downloaded from an ontology and used to generate nodes and relationships in the graph database.

\end{itemize}

\item[{Returns}] \leavevmode

Three nested dictionaries: terms, relationships between terms, and definitions of the terms.
\begin{itemize}
\item {} 
terms: Dictionary where each key is an ontology identifier (\sphinxstyleemphasis{str}) and the values are lists of names and synonyms (\sphinxstyleemphasis{list{[}str{]}}).

\item {} 
relationships: Dictionary of tuples (\sphinxstyleemphasis{str}). Each tuple contains two ontology identifiers (source and target) and                         the relationship type between them.

\item {} 
definitions: Dictionary with ontology identifiers as keys (\sphinxstyleemphasis{str}), and definition of the terms as values (\sphinxstyleemphasis{str}).

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}



\subparagraph{reflectParser.py}
\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:module-graphdb_builder.ontologies.parsers.reflectParser}}\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:reflectparser-py}}\index{graphdb\_builder.ontologies.parsers.reflectParser (module)@\spxentry{graphdb\_builder.ontologies.parsers.reflectParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.ontologies.parsers.reflectParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.ontologies.parsers.reflectParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:graphdb_builder.ontologies.parsers.reflectParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.ontologies.parsers.reflectParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{files}, \emph{filters}, \emph{qtype=None}}{}
Parses and extracts relevant data from REFLECT ontologies: Disease Ontology, Tissues, STITCH and     Gene Ontology databases.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{files}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of files downloaded from an ontology and used to generate nodes and relationships in the graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{filters}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of ontology identifiers to be ignored.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{qtype}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} ontology type code.

\end{itemize}

\item[{Returns}] \leavevmode

Three nested dictionaries: terms, relationships between terms, and definitions of the terms.
\begin{itemize}
\item {} 
terms: Dictionary where each key is an ontology dentifier (\sphinxstyleemphasis{str}) and the values are lists of names and synonyms (\sphinxstyleemphasis{list{[}str{]}}).

\item {} 
relationships: Dictionary of tuples (\sphinxstyleemphasis{str}). Each tuple contains two ontology identifiers (source and target) and                         the relationship type between them.

\item {} 
definitions: Dictionary with ontology dentifiers as keys (\sphinxstyleemphasis{str}), and definition of the terms as values (\sphinxstyleemphasis{str}).

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}



\subparagraph{snomedParser.py}
\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:module-graphdb_builder.ontologies.parsers.snomedParser}}\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:snomedparser-py}}\index{graphdb\_builder.ontologies.parsers.snomedParser (module)@\spxentry{graphdb\_builder.ontologies.parsers.snomedParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.ontologies.parsers.snomedParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.ontologies.parsers.snomedParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:graphdb_builder.ontologies.parsers.snomedParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.ontologies.parsers.snomedParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{files}, \emph{filters}}{}
Parses and extracts relevant data from SNOMED CT database files.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{files}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of files downloaded from SNOMED CT and used to generate nodes and relationships in the graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{filters}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of SNOMED CT Identifiers to be ignored.

\end{itemize}

\item[{Returns}] \leavevmode

Three nested dictionaries: terms, relationships between terms, and definitions of the terms.
\begin{itemize}
\item {} 
terms: Dictionary where each key is a SNOMED CT Identifier (\sphinxstyleemphasis{str}) and the values are lists of names and synonyms (\sphinxstyleemphasis{list{[}str{]}}).

\item {} 
relationships: Dictionary of tuples (\sphinxstyleemphasis{str}). Each tuple contains two SNOMED CT Identifiers (source and target) and                         the relationship type between them.

\item {} 
definitions: Dictionary with SNOMED CT Identifiers as keys (\sphinxstyleemphasis{str}), and definition of the terms as values (\sphinxstyleemphasis{str}).

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\index{get\_inactive\_terms() (in module graphdb\_builder.ontologies.parsers.snomedParser)@\spxentry{get\_inactive\_terms()}\spxextra{in module graphdb\_builder.ontologies.parsers.snomedParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.ontologies.parsers:graphdb_builder.ontologies.parsers.snomedParser.get_inactive_terms}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.ontologies.parsers.snomedParser.}}\sphinxbfcode{\sphinxupquote{get\_inactive\_terms}}}{\emph{concept\_file}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{concept\_file}} \textendash{} 

\item[{Return set inactive\_terms}] \leavevmode
inactive terms

\end{description}\end{quote}

\end{fulllineitems}



\paragraph{Ontologies Controller}
\label{\detokenize{_autosummary/graphdb_builder.ontologies:module-graphdb_builder.ontologies.ontologies_controller}}\label{\detokenize{_autosummary/graphdb_builder.ontologies:ontologies-controller}}\index{graphdb\_builder.ontologies.ontologies\_controller (module)@\spxentry{graphdb\_builder.ontologies.ontologies\_controller}\spxextra{module}}\index{entries\_to\_remove() (in module graphdb\_builder.ontologies.ontologies\_controller)@\spxentry{entries\_to\_remove()}\spxextra{in module graphdb\_builder.ontologies.ontologies\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.ontologies:graphdb_builder.ontologies.ontologies_controller.entries_to_remove}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.ontologies.ontologies\_controller.}}\sphinxbfcode{\sphinxupquote{entries\_to\_remove}}}{\emph{entries}, \emph{the\_dict}}{}
This function removes pairs from a given dictionary, based on a list of provided keys.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{entries}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of keys to be deleted from dictionary.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{the\_dict}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary.

\end{itemize}

\item[{Returns}] \leavevmode
The original dictionary minus the key,value pairs from the provided entries list.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_extra\_entities\_rels() (in module graphdb\_builder.ontologies.ontologies\_controller)@\spxentry{get\_extra\_entities\_rels()}\spxextra{in module graphdb\_builder.ontologies.ontologies\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.ontologies:graphdb_builder.ontologies.ontologies_controller.get_extra_entities_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.ontologies.ontologies\_controller.}}\sphinxbfcode{\sphinxupquote{get\_extra\_entities\_rels}}}{\emph{ontology\_directory}}{}
\end{fulllineitems}

\index{parse\_ontology() (in module graphdb\_builder.ontologies.ontologies\_controller)@\spxentry{parse\_ontology()}\spxextra{in module graphdb\_builder.ontologies.ontologies\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.ontologies:graphdb_builder.ontologies.ontologies_controller.parse_ontology}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.ontologies.ontologies\_controller.}}\sphinxbfcode{\sphinxupquote{parse\_ontology}}}{\emph{ontology}, \emph{download=True}}{}
Parses and extracts data from a given ontology file(s), and returns a tuple with multiple dictionaries.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ontology}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} acronym of the ontology to be parsed (e.g. Disease Ontology:’DO’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{download}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} wether database is to be downloaded.

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with three nested dictionaries: terms, relationships between terms, and definitions of the terms.            For more information on the returned dictionaries, see the documentation for any ontology parser.

\end{description}\end{quote}

\end{fulllineitems}

\index{generate\_graphFiles() (in module graphdb\_builder.ontologies.ontologies\_controller)@\spxentry{generate\_graphFiles()}\spxextra{in module graphdb\_builder.ontologies.ontologies\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.ontologies:graphdb_builder.ontologies.ontologies_controller.generate_graphFiles}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.ontologies.ontologies\_controller.}}\sphinxbfcode{\sphinxupquote{generate\_graphFiles}}}{\emph{import\_directory}, \emph{ontologies=None}, \emph{download=True}}{}
This function parses and extracts data from a given list of ontologies. If no ontologies are provided,     all availables ontologies are used. Terms, relationships and definitions are saved as .tsv files to be loaded into     the graph database.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{import\_directory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} relative path from current python module to ‘imports’ directory.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ontologies}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{ or }}\sphinxhref{https://docs.python.org/3.6/library/constants.html\#None}{\sphinxstyleliteralemphasis{\sphinxupquote{None}}}) \textendash{} list of ontologies to be imported. If None, all available ontologies are imported.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{download}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} wether database is to be downloaded.

\end{itemize}

\item[{Returns}] \leavevmode
Dictionary of tuples. Each tuple corresponds to a unique label/relationship type, date, time,             database, and number of nodes and relationships.

\end{description}\end{quote}

\end{fulllineitems}



\subsubsection{Biomedical Databases}
\label{\detokenize{_autosummary/graphdb_builder.databases:biomedical-databases}}\label{\detokenize{_autosummary/graphdb_builder.databases::doc}}

\paragraph{Biomedical Databases Parsers}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:biomedical-databases-parsers}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers::doc}}

\subparagraph{cancerGenomeInterpreterParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.cancerGenomeInterpreterParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:cancergenomeinterpreterparser-py}}\index{graphdb\_builder.databases.parsers.cancerGenomeInterpreterParser (module)@\spxentry{graphdb\_builder.databases.parsers.cancerGenomeInterpreterParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.cancerGenomeInterpreterParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.cancerGenomeInterpreterParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.cancerGenomeInterpreterParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.cancerGenomeInterpreterParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{corumParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.corumParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:corumparser-py}}\index{graphdb\_builder.databases.parsers.corumParser (module)@\spxentry{graphdb\_builder.databases.parsers.corumParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.corumParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.corumParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.corumParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.corumParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{disgenetParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.disgenetParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:disgenetparser-py}}\index{graphdb\_builder.databases.parsers.disgenetParser (module)@\spxentry{graphdb\_builder.databases.parsers.disgenetParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.disgenetParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.disgenetParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.disgenetParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.disgenetParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{readDisGeNetProteinMapping() (in module graphdb\_builder.databases.parsers.disgenetParser)@\spxentry{readDisGeNetProteinMapping()}\spxextra{in module graphdb\_builder.databases.parsers.disgenetParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.disgenetParser.readDisGeNetProteinMapping}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.disgenetParser.}}\sphinxbfcode{\sphinxupquote{readDisGeNetProteinMapping}}}{\emph{config}, \emph{directory}}{}
\end{fulllineitems}

\index{readDisGeNetDiseaseMapping() (in module graphdb\_builder.databases.parsers.disgenetParser)@\spxentry{readDisGeNetDiseaseMapping()}\spxextra{in module graphdb\_builder.databases.parsers.disgenetParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.disgenetParser.readDisGeNetDiseaseMapping}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.disgenetParser.}}\sphinxbfcode{\sphinxupquote{readDisGeNetDiseaseMapping}}}{\emph{config}, \emph{directory}}{}
\end{fulllineitems}



\subparagraph{drugBankParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.drugBankParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:drugbankparser-py}}\index{graphdb\_builder.databases.parsers.drugBankParser (module)@\spxentry{graphdb\_builder.databases.parsers.drugBankParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.drugBankParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.drugBankParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.drugBankParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.drugBankParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}}{}
\end{fulllineitems}

\index{extract\_drugs() (in module graphdb\_builder.databases.parsers.drugBankParser)@\spxentry{extract\_drugs()}\spxextra{in module graphdb\_builder.databases.parsers.drugBankParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.drugBankParser.extract_drugs}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.drugBankParser.}}\sphinxbfcode{\sphinxupquote{extract\_drugs}}}{\emph{config}, \emph{directory}}{}
\end{fulllineitems}

\index{parseDrugBankVocabulary() (in module graphdb\_builder.databases.parsers.drugBankParser)@\spxentry{parseDrugBankVocabulary()}\spxextra{in module graphdb\_builder.databases.parsers.drugBankParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.drugBankParser.parseDrugBankVocabulary}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.drugBankParser.}}\sphinxbfcode{\sphinxupquote{parseDrugBankVocabulary}}}{\emph{config}, \emph{directory}}{}
\end{fulllineitems}

\index{build\_relationships\_from\_DrugBank() (in module graphdb\_builder.databases.parsers.drugBankParser)@\spxentry{build\_relationships\_from\_DrugBank()}\spxextra{in module graphdb\_builder.databases.parsers.drugBankParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.drugBankParser.build_relationships_from_DrugBank}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.drugBankParser.}}\sphinxbfcode{\sphinxupquote{build\_relationships\_from\_DrugBank}}}{\emph{config}, \emph{drugs}}{}
\end{fulllineitems}

\index{build\_drug\_entity() (in module graphdb\_builder.databases.parsers.drugBankParser)@\spxentry{build\_drug\_entity()}\spxextra{in module graphdb\_builder.databases.parsers.drugBankParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.drugBankParser.build_drug_entity}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.drugBankParser.}}\sphinxbfcode{\sphinxupquote{build\_drug\_entity}}}{\emph{config}, \emph{drugs}}{}
\end{fulllineitems}

\index{build\_DrugBank\_dictionary() (in module graphdb\_builder.databases.parsers.drugBankParser)@\spxentry{build\_DrugBank\_dictionary()}\spxextra{in module graphdb\_builder.databases.parsers.drugBankParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.drugBankParser.build_DrugBank_dictionary}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.drugBankParser.}}\sphinxbfcode{\sphinxupquote{build\_DrugBank\_dictionary}}}{\emph{config}, \emph{directory}, \emph{drugs}}{}
\end{fulllineitems}



\subparagraph{drugGeneInteractionDBParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.drugGeneInteractionDBParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:druggeneinteractiondbparser-py}}\index{graphdb\_builder.databases.parsers.drugGeneInteractionDBParser (module)@\spxentry{graphdb\_builder.databases.parsers.drugGeneInteractionDBParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.drugGeneInteractionDBParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.drugGeneInteractionDBParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.drugGeneInteractionDBParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.drugGeneInteractionDBParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{exposomeParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.exposomeParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:exposomeparser-py}}\index{graphdb\_builder.databases.parsers.exposomeParser (module)@\spxentry{graphdb\_builder.databases.parsers.exposomeParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.exposomeParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.exposomeParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.exposomeParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.exposomeParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{parseBiomarkersFile() (in module graphdb\_builder.databases.parsers.exposomeParser)@\spxentry{parseBiomarkersFile()}\spxextra{in module graphdb\_builder.databases.parsers.exposomeParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.exposomeParser.parseBiomarkersFile}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.exposomeParser.}}\sphinxbfcode{\sphinxupquote{parseBiomarkersFile}}}{\emph{fhandler}, \emph{file\_name}}{}
\end{fulllineitems}

\index{parseCorrelationsFile() (in module graphdb\_builder.databases.parsers.exposomeParser)@\spxentry{parseCorrelationsFile()}\spxextra{in module graphdb\_builder.databases.parsers.exposomeParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.exposomeParser.parseCorrelationsFile}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.exposomeParser.}}\sphinxbfcode{\sphinxupquote{parseCorrelationsFile}}}{\emph{fhandler}, \emph{file\_name}, \emph{biomarkers}, \emph{mapping}}{}
\end{fulllineitems}



\subparagraph{foodbParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.foodbParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:foodbparser-py}}\index{graphdb\_builder.databases.parsers.foodbParser (module)@\spxentry{graphdb\_builder.databases.parsers.foodbParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.foodbParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.foodbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.foodbParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.foodbParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{parseContents() (in module graphdb\_builder.databases.parsers.foodbParser)@\spxentry{parseContents()}\spxextra{in module graphdb\_builder.databases.parsers.foodbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.foodbParser.parseContents}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.foodbParser.}}\sphinxbfcode{\sphinxupquote{parseContents}}}{\emph{fhandler}}{}
\end{fulllineitems}

\index{parseFood() (in module graphdb\_builder.databases.parsers.foodbParser)@\spxentry{parseFood()}\spxextra{in module graphdb\_builder.databases.parsers.foodbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.foodbParser.parseFood}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.foodbParser.}}\sphinxbfcode{\sphinxupquote{parseFood}}}{\emph{fhandler}}{}
\end{fulllineitems}

\index{parseCompounds() (in module graphdb\_builder.databases.parsers.foodbParser)@\spxentry{parseCompounds()}\spxextra{in module graphdb\_builder.databases.parsers.foodbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.foodbParser.parseCompounds}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.foodbParser.}}\sphinxbfcode{\sphinxupquote{parseCompounds}}}{\emph{fhandler}}{}
\end{fulllineitems}



\subparagraph{goaParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.goaParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:goaparser-py}}\index{graphdb\_builder.databases.parsers.goaParser (module)@\spxentry{graphdb\_builder.databases.parsers.goaParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.goaParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.goaParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.goaParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.goaParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_dir}, \emph{download=True}}{}
\end{fulllineitems}

\index{parse\_annotations\_with\_pandas() (in module graphdb\_builder.databases.parsers.goaParser)@\spxentry{parse\_annotations\_with\_pandas()}\spxextra{in module graphdb\_builder.databases.parsers.goaParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.goaParser.parse_annotations_with_pandas}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.goaParser.}}\sphinxbfcode{\sphinxupquote{parse\_annotations\_with\_pandas}}}{\emph{annotation\_file}, \emph{valid\_proteins=None}}{}
\end{fulllineitems}



\subparagraph{gwasCatalogParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.gwasCatalogParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:gwascatalogparser-py}}\index{graphdb\_builder.databases.parsers.gwasCatalogParser (module)@\spxentry{graphdb\_builder.databases.parsers.gwasCatalogParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.gwasCatalogParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.gwasCatalogParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.gwasCatalogParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.gwasCatalogParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{hgncParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.hgncParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:hgncparser-py}}\index{graphdb\_builder.databases.parsers.hgncParser (module)@\spxentry{graphdb\_builder.databases.parsers.hgncParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.hgncParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.hgncParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.hgncParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.hgncParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{hmdbParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.hmdbParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:hmdbparser-py}}\index{graphdb\_builder.databases.parsers.hmdbParser (module)@\spxentry{graphdb\_builder.databases.parsers.hmdbParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.hmdbParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.hmdbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.hmdbParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.hmdbParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{extract\_metabolites() (in module graphdb\_builder.databases.parsers.hmdbParser)@\spxentry{extract\_metabolites()}\spxextra{in module graphdb\_builder.databases.parsers.hmdbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.hmdbParser.extract_metabolites}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.hmdbParser.}}\sphinxbfcode{\sphinxupquote{extract\_metabolites}}}{\emph{config}, \emph{directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{build\_metabolite\_entity() (in module graphdb\_builder.databases.parsers.hmdbParser)@\spxentry{build\_metabolite\_entity()}\spxextra{in module graphdb\_builder.databases.parsers.hmdbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.hmdbParser.build_metabolite_entity}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.hmdbParser.}}\sphinxbfcode{\sphinxupquote{build\_metabolite\_entity}}}{\emph{config}, \emph{directory}, \emph{metabolites}}{}
\end{fulllineitems}

\index{build\_relationships\_from\_HMDB() (in module graphdb\_builder.databases.parsers.hmdbParser)@\spxentry{build\_relationships\_from\_HMDB()}\spxextra{in module graphdb\_builder.databases.parsers.hmdbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.hmdbParser.build_relationships_from_HMDB}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.hmdbParser.}}\sphinxbfcode{\sphinxupquote{build\_relationships\_from\_HMDB}}}{\emph{config}, \emph{metabolites}, \emph{mapping}}{}
\end{fulllineitems}

\index{build\_HMDB\_dictionary() (in module graphdb\_builder.databases.parsers.hmdbParser)@\spxentry{build\_HMDB\_dictionary()}\spxextra{in module graphdb\_builder.databases.parsers.hmdbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.hmdbParser.build_HMDB_dictionary}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.hmdbParser.}}\sphinxbfcode{\sphinxupquote{build\_HMDB\_dictionary}}}{\emph{directory}, \emph{metabolites}}{}
\end{fulllineitems}



\subparagraph{hpaParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.hpaParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:hpaparser-py}}\index{graphdb\_builder.databases.parsers.hpaParser (module)@\spxentry{graphdb\_builder.databases.parsers.hpaParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.hpaParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.hpaParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.hpaParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.hpaParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{parsePathologyFile() (in module graphdb\_builder.databases.parsers.hpaParser)@\spxentry{parsePathologyFile()}\spxextra{in module graphdb\_builder.databases.parsers.hpaParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.hpaParser.parsePathologyFile}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.hpaParser.}}\sphinxbfcode{\sphinxupquote{parsePathologyFile}}}{\emph{config}, \emph{fhandler}, \emph{file\_name}, \emph{protein\_mapping}, \emph{disease\_mapping}}{}
\end{fulllineitems}



\subparagraph{intactParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.intactParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:intactparser-py}}\index{graphdb\_builder.databases.parsers.intactParser (module)@\spxentry{graphdb\_builder.databases.parsers.intactParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.intactParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.intactParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.intactParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.intactParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{internalDBsParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.jensenlabParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:internaldbsparser-py}}\index{graphdb\_builder.databases.parsers.jensenlabParser (module)@\spxentry{graphdb\_builder.databases.parsers.jensenlabParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.jensenlabParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.jensenlabParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.jensenlabParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.jensenlabParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{parsePairs() (in module graphdb\_builder.databases.parsers.jensenlabParser)@\spxentry{parsePairs()}\spxextra{in module graphdb\_builder.databases.parsers.jensenlabParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.jensenlabParser.parsePairs}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.jensenlabParser.}}\sphinxbfcode{\sphinxupquote{parsePairs}}}{\emph{config}, \emph{databases\_directory}, \emph{qtype}, \emph{mapping}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{mutationDsParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.mutationDsParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:mutationdsparser-py}}\index{graphdb\_builder.databases.parsers.mutationDsParser (module)@\spxentry{graphdb\_builder.databases.parsers.mutationDsParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.mutationDsParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.mutationDsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.mutationDsParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.mutationDsParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{oncokbParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.oncokbParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:oncokbparser-py}}\index{graphdb\_builder.databases.parsers.oncokbParser (module)@\spxentry{graphdb\_builder.databases.parsers.oncokbParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.oncokbParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.oncokbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.oncokbParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.oncokbParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{pathwayCommonsParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.pathwayCommonsParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:pathwaycommonsparser-py}}\index{graphdb\_builder.databases.parsers.pathwayCommonsParser (module)@\spxentry{graphdb\_builder.databases.parsers.pathwayCommonsParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.pathwayCommonsParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.pathwayCommonsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.pathwayCommonsParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.pathwayCommonsParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{pfamParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.pfamParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:pfamparser-py}}\index{graphdb\_builder.databases.parsers.pfamParser (module)@\spxentry{graphdb\_builder.databases.parsers.pfamParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.pfamParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.pfamParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.pfamParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.pfamParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{import\_directory}, \emph{download=True}, \emph{updated\_on=None}}{}
\end{fulllineitems}

\index{print\_files() (in module graphdb\_builder.databases.parsers.pfamParser)@\spxentry{print\_files()}\spxextra{in module graphdb\_builder.databases.parsers.pfamParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.pfamParser.print_files}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.pfamParser.}}\sphinxbfcode{\sphinxupquote{print\_files}}}{\emph{data}, \emph{header}, \emph{outputfile}, \emph{is\_first}, \emph{filter\_for=None}}{}
\end{fulllineitems}



\subparagraph{pspParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.pspParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:pspparser-py}}\index{graphdb\_builder.databases.parsers.pspParser (module)@\spxentry{graphdb\_builder.databases.parsers.pspParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.pspParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.pspParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.pspParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.pspParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}}{}
\end{fulllineitems}

\index{parseSites() (in module graphdb\_builder.databases.parsers.pspParser)@\spxentry{parseSites()}\spxextra{in module graphdb\_builder.databases.parsers.pspParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.pspParser.parseSites}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.pspParser.}}\sphinxbfcode{\sphinxupquote{parseSites}}}{\emph{fhandler}, \emph{modifications}}{}
\end{fulllineitems}

\index{parseKinaseSubstrates() (in module graphdb\_builder.databases.parsers.pspParser)@\spxentry{parseKinaseSubstrates()}\spxextra{in module graphdb\_builder.databases.parsers.pspParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.pspParser.parseKinaseSubstrates}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.pspParser.}}\sphinxbfcode{\sphinxupquote{parseKinaseSubstrates}}}{\emph{fhandler}, \emph{modifications}}{}
\end{fulllineitems}

\index{parseRegulationAnnotations() (in module graphdb\_builder.databases.parsers.pspParser)@\spxentry{parseRegulationAnnotations()}\spxextra{in module graphdb\_builder.databases.parsers.pspParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.pspParser.parseRegulationAnnotations}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.pspParser.}}\sphinxbfcode{\sphinxupquote{parseRegulationAnnotations}}}{\emph{fhandler}, \emph{modifications}, \emph{mapping}}{}
\end{fulllineitems}

\index{parseDiseaseAnnotations() (in module graphdb\_builder.databases.parsers.pspParser)@\spxentry{parseDiseaseAnnotations()}\spxextra{in module graphdb\_builder.databases.parsers.pspParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.pspParser.parseDiseaseAnnotations}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.pspParser.}}\sphinxbfcode{\sphinxupquote{parseDiseaseAnnotations}}}{\emph{fhandler}, \emph{modifications}, \emph{mapping}}{}
\end{fulllineitems}



\subparagraph{reactomeParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.reactomeParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:reactomeparser-py}}\index{graphdb\_builder.databases.parsers.reactomeParser (module)@\spxentry{graphdb\_builder.databases.parsers.reactomeParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.reactomeParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.reactomeParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.reactomeParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.reactomeParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{parsePathways() (in module graphdb\_builder.databases.parsers.reactomeParser)@\spxentry{parsePathways()}\spxextra{in module graphdb\_builder.databases.parsers.reactomeParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.reactomeParser.parsePathways}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.reactomeParser.}}\sphinxbfcode{\sphinxupquote{parsePathways}}}{\emph{config}, \emph{databases\_directory}, \emph{fhandler}}{}
\end{fulllineitems}

\index{parsePathwayHierarchy() (in module graphdb\_builder.databases.parsers.reactomeParser)@\spxentry{parsePathwayHierarchy()}\spxextra{in module graphdb\_builder.databases.parsers.reactomeParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.reactomeParser.parsePathwayHierarchy}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.reactomeParser.}}\sphinxbfcode{\sphinxupquote{parsePathwayHierarchy}}}{\emph{fhandler}}{}
\end{fulllineitems}

\index{parsePathwayRelationships() (in module graphdb\_builder.databases.parsers.reactomeParser)@\spxentry{parsePathwayRelationships()}\spxextra{in module graphdb\_builder.databases.parsers.reactomeParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.reactomeParser.parsePathwayRelationships}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.reactomeParser.}}\sphinxbfcode{\sphinxupquote{parsePathwayRelationships}}}{\emph{config}, \emph{fhandler}, \emph{mapping=None}}{}
\end{fulllineitems}



\subparagraph{refseqParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.refseqParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:refseqparser-py}}\index{graphdb\_builder.databases.parsers.refseqParser (module)@\spxentry{graphdb\_builder.databases.parsers.refseqParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.refseqParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.refseqParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.refseqParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.refseqParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{siderParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.siderParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:siderparser-py}}\index{graphdb\_builder.databases.parsers.siderParser (module)@\spxentry{graphdb\_builder.databases.parsers.siderParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.siderParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.siderParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.siderParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.siderParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{drug\_source}, \emph{download=True}}{}
\end{fulllineitems}

\index{parserIndications() (in module graphdb\_builder.databases.parsers.siderParser)@\spxentry{parserIndications()}\spxextra{in module graphdb\_builder.databases.parsers.siderParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.siderParser.parserIndications}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.siderParser.}}\sphinxbfcode{\sphinxupquote{parserIndications}}}{\emph{databases\_directory}, \emph{drugMapping}, \emph{phenotypeMapping}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{smpdbParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.smpdbParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:smpdbparser-py}}\index{graphdb\_builder.databases.parsers.smpdbParser (module)@\spxentry{graphdb\_builder.databases.parsers.smpdbParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.smpdbParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.smpdbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.smpdbParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.smpdbParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{parsePathways() (in module graphdb\_builder.databases.parsers.smpdbParser)@\spxentry{parsePathways()}\spxextra{in module graphdb\_builder.databases.parsers.smpdbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.smpdbParser.parsePathways}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.smpdbParser.}}\sphinxbfcode{\sphinxupquote{parsePathways}}}{\emph{config}, \emph{fhandler}}{}
\end{fulllineitems}

\index{parsePathwayProteinRelationships() (in module graphdb\_builder.databases.parsers.smpdbParser)@\spxentry{parsePathwayProteinRelationships()}\spxextra{in module graphdb\_builder.databases.parsers.smpdbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.smpdbParser.parsePathwayProteinRelationships}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.smpdbParser.}}\sphinxbfcode{\sphinxupquote{parsePathwayProteinRelationships}}}{\emph{fhandler}}{}
\end{fulllineitems}

\index{parsePathwayMetaboliteDrugRelationships() (in module graphdb\_builder.databases.parsers.smpdbParser)@\spxentry{parsePathwayMetaboliteDrugRelationships()}\spxextra{in module graphdb\_builder.databases.parsers.smpdbParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.smpdbParser.parsePathwayMetaboliteDrugRelationships}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.smpdbParser.}}\sphinxbfcode{\sphinxupquote{parsePathwayMetaboliteDrugRelationships}}}{\emph{fhandler}}{}
\end{fulllineitems}



\subparagraph{stringParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.stringParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:stringparser-py}}\index{graphdb\_builder.databases.parsers.stringParser (module)@\spxentry{graphdb\_builder.databases.parsers.stringParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.stringParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.stringParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.stringParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.stringParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{importDirectory}, \emph{drug\_source=None}, \emph{download=True}, \emph{db=\textquotesingle{}STRING\textquotesingle{}}}{}
\end{fulllineitems}

\index{parseActions() (in module graphdb\_builder.databases.parsers.stringParser)@\spxentry{parseActions()}\spxextra{in module graphdb\_builder.databases.parsers.stringParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.stringParser.parseActions}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.stringParser.}}\sphinxbfcode{\sphinxupquote{parseActions}}}{\emph{databases\_directory}, \emph{importDirectory}, \emph{proteinMapping}, \emph{drugMapping=None}, \emph{download=True}, \emph{db=\textquotesingle{}STRING\textquotesingle{}}}{}
\end{fulllineitems}



\subparagraph{textminingParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.textminingParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:textminingparser-py}}\index{graphdb\_builder.databases.parsers.textminingParser (module)@\spxentry{graphdb\_builder.databases.parsers.textminingParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.textminingParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.textminingParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.textminingParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.textminingParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{importDirectory}, \emph{download=True}}{}
\end{fulllineitems}

\index{read\_valid\_pubs() (in module graphdb\_builder.databases.parsers.textminingParser)@\spxentry{read\_valid\_pubs()}\spxextra{in module graphdb\_builder.databases.parsers.textminingParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.textminingParser.read_valid_pubs}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.textminingParser.}}\sphinxbfcode{\sphinxupquote{read\_valid\_pubs}}}{\emph{organisms}, \emph{organisms\_file}}{}
\end{fulllineitems}

\index{parse\_PMC\_list() (in module graphdb\_builder.databases.parsers.textminingParser)@\spxentry{parse\_PMC\_list()}\spxextra{in module graphdb\_builder.databases.parsers.textminingParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.textminingParser.parse_PMC_list}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.textminingParser.}}\sphinxbfcode{\sphinxupquote{parse\_PMC\_list}}}{\emph{config}, \emph{directory}, \emph{download=True}, \emph{valid\_pubs=None}}{}
\end{fulllineitems}

\index{parse\_mentions() (in module graphdb\_builder.databases.parsers.textminingParser)@\spxentry{parse\_mentions()}\spxextra{in module graphdb\_builder.databases.parsers.textminingParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.textminingParser.parse_mentions}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.textminingParser.}}\sphinxbfcode{\sphinxupquote{parse\_mentions}}}{\emph{config}, \emph{directory}, \emph{qtype}, \emph{importDirectory}, \emph{download=True}}{}
\end{fulllineitems}



\subparagraph{uniprotParser.py}
\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:module-graphdb_builder.databases.parsers.uniprotParser}}\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:uniprotparser-py}}\index{graphdb\_builder.databases.parsers.uniprotParser (module)@\spxentry{graphdb\_builder.databases.parsers.uniprotParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{databases\_directory}, \emph{import\_directory}, \emph{download=True}, \emph{updated\_on=None}}{}
\end{fulllineitems}

\index{parse\_release\_notes() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{parse\_release\_notes()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.parse_release_notes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{parse\_release\_notes}}}{\emph{databases\_directory}, \emph{config}, \emph{download=True}}{}
\end{fulllineitems}

\index{parse\_fasta() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{parse\_fasta()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.parse_fasta}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{parse\_fasta}}}{\emph{databases\_directory}, \emph{config}, \emph{import\_directory}, \emph{download=True}, \emph{updated\_on=None}}{}
\end{fulllineitems}

\index{parse\_idmapping\_file() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{parse\_idmapping\_file()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.parse_idmapping_file}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{parse\_idmapping\_file}}}{\emph{databases\_directory}, \emph{config}, \emph{import\_directory}, \emph{download=True}, \emph{updated\_on=None}}{}
\end{fulllineitems}

\index{format\_output() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{format\_output()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.format_output}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{format\_output}}}{\emph{proteins}}{}
\end{fulllineitems}

\index{print\_single\_file() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{print\_single\_file()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.print_single_file}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{print\_single\_file}}}{\emph{data}, \emph{header}, \emph{output\_file}, \emph{data\_type}, \emph{data\_object}, \emph{is\_first}, \emph{updated\_on}}{}
\end{fulllineitems}

\index{print\_multiple\_relationships\_files() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{print\_multiple\_relationships\_files()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.print_multiple_relationships_files}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{print\_multiple\_relationships\_files}}}{\emph{data}, \emph{header}, \emph{output\_dir}, \emph{is\_first}, \emph{updated\_on}}{}
\end{fulllineitems}

\index{addUniProtTexts() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{addUniProtTexts()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.addUniProtTexts}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{addUniProtTexts}}}{\emph{textsFile}, \emph{proteins}}{}
\end{fulllineitems}

\index{parseUniProtVariants() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{parseUniProtVariants()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.parseUniProtVariants}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{parseUniProtVariants}}}{\emph{config}, \emph{databases\_directory}, \emph{import\_directory}, \emph{download=True}, \emph{updated\_on=None}}{}
\end{fulllineitems}

\index{parseUniProtAnnotations() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{parseUniProtAnnotations()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.parseUniProtAnnotations}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{parseUniProtAnnotations}}}{\emph{config}, \emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}

\index{parseUniProtPeptides() (in module graphdb\_builder.databases.parsers.uniprotParser)@\spxentry{parseUniProtPeptides()}\spxextra{in module graphdb\_builder.databases.parsers.uniprotParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases.parsers:graphdb_builder.databases.parsers.uniprotParser.parseUniProtPeptides}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.parsers.uniprotParser.}}\sphinxbfcode{\sphinxupquote{parseUniProtPeptides}}}{\emph{config}, \emph{databases\_directory}, \emph{download=True}}{}
\end{fulllineitems}



\paragraph{Databases Controller}
\label{\detokenize{_autosummary/graphdb_builder.databases:module-graphdb_builder.databases.databases_controller}}\label{\detokenize{_autosummary/graphdb_builder.databases:databases-controller}}\index{graphdb\_builder.databases.databases\_controller (module)@\spxentry{graphdb\_builder.databases.databases\_controller}\spxextra{module}}\index{parseDatabase() (in module graphdb\_builder.databases.databases\_controller)@\spxentry{parseDatabase()}\spxextra{in module graphdb\_builder.databases.databases\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases:graphdb_builder.databases.databases_controller.parseDatabase}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.databases\_controller.}}\sphinxbfcode{\sphinxupquote{parseDatabase}}}{\emph{importDirectory}, \emph{database}, \emph{download=True}}{}
\end{fulllineitems}

\index{generateGraphFiles() (in module graphdb\_builder.databases.databases\_controller)@\spxentry{generateGraphFiles()}\spxextra{in module graphdb\_builder.databases.databases\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.databases:graphdb_builder.databases.databases_controller.generateGraphFiles}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.databases.databases\_controller.}}\sphinxbfcode{\sphinxupquote{generateGraphFiles}}}{\emph{importDirectory}, \emph{databases=None}, \emph{download=True}, \emph{n\_jobs=4}}{}
\end{fulllineitems}



\subsubsection{Experimental Data}
\label{\detokenize{_autosummary/graphdb_builder.experiments:experimental-data}}\label{\detokenize{_autosummary/graphdb_builder.experiments::doc}}

\paragraph{Experiments Parsers}
\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:experiments-parsers}}\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers::doc}}

\subparagraph{Clinical parser}
\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:module-graphdb_builder.experiments.parsers.clinicalParser}}\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:clinical-parser}}\index{graphdb\_builder.experiments.parsers.clinicalParser (module)@\spxentry{graphdb\_builder.experiments.parsers.clinicalParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{projectId}}{}
\end{fulllineitems}

\index{project\_parser() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{project\_parser()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.project_parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{project\_parser}}}{\emph{projectId}, \emph{config}, \emph{directory}, \emph{separator}}{}
\end{fulllineitems}

\index{experimental\_design\_parser() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{experimental\_design\_parser()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.experimental_design_parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{experimental\_design\_parser}}}{\emph{projectId}, \emph{config}, \emph{directory}}{}
\end{fulllineitems}

\index{clinical\_parser() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{clinical\_parser()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.clinical_parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{clinical\_parser}}}{\emph{projectId}, \emph{config}, \emph{directory}, \emph{separator}}{}
\end{fulllineitems}

\index{parse\_dataset() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{parse\_dataset()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.parse_dataset}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{parse\_dataset}}}{\emph{projectId}, \emph{configuration}, \emph{dataDir}, \emph{key=\textquotesingle{}project\textquotesingle{}}}{}
This function parses clinical data from subjects in the project
Input: uri of the clinical data file. Format: Subjects as rows, clinical variables as columns
Output: pandas DataFrame with the same input format but the clinical variables mapped to the
right ontology (defined in config), i.e. type = \sphinxhyphen{}40 \sphinxhyphen{}\textgreater{} SNOMED CT

\end{fulllineitems}

\index{extract\_project\_info() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_project\_info()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_project_info}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_project\_info}}}{\emph{project\_data}}{}
\end{fulllineitems}

\index{extract\_responsible\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_responsible\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_responsible_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_responsible\_rels}}}{\emph{project\_data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
\end{fulllineitems}

\index{extract\_participant\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_participant\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_participant_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_participant\_rels}}}{\emph{project\_data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
\end{fulllineitems}

\index{extract\_project\_tissue\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_project\_tissue\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_project_tissue_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_project\_tissue\_rels}}}{\emph{project\_data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
\end{fulllineitems}

\index{extract\_project\_disease\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_project\_disease\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_project_disease_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_project\_disease\_rels}}}{\emph{project\_data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
\end{fulllineitems}

\index{extract\_project\_intervention\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_project\_intervention\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_project_intervention_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_project\_intervention\_rels}}}{\emph{project\_data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
\end{fulllineitems}

\index{extract\_project\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_project\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_project_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_project\_rels}}}{\emph{project\_data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
\end{fulllineitems}

\index{extract\_timepoints() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_timepoints()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_timepoints}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_timepoints}}}{\emph{project\_data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
\end{fulllineitems}

\index{extract\_project\_subject\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_project\_subject\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_project_subject_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_project\_subject\_rels}}}{\emph{projectId}, \emph{design\_data}}{}
\end{fulllineitems}

\index{extract\_subject\_identifiers() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_subject\_identifiers()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_subject_identifiers}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_subject\_identifiers}}}{\emph{design\_data}}{}
\end{fulllineitems}

\index{extract\_biosample\_identifiers() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_biosample\_identifiers()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_biosample_identifiers}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_biosample\_identifiers}}}{\emph{design\_data}}{}
\end{fulllineitems}

\index{extract\_analytical\_sample\_identifiers() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_analytical\_sample\_identifiers()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_analytical_sample_identifiers}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_analytical\_sample\_identifiers}}}{\emph{design\_data}}{}
\end{fulllineitems}

\index{extract\_biological\_sample\_subject\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_biological\_sample\_subject\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_biological_sample_subject_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_biological\_sample\_subject\_rels}}}{\emph{design\_data}}{}
\end{fulllineitems}

\index{extract\_biological\_sample\_analytical\_sample\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_biological\_sample\_analytical\_sample\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_biological_sample_analytical_sample_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_biological\_sample\_analytical\_sample\_rels}}}{\emph{design\_data}}{}
\end{fulllineitems}

\index{extract\_biological\_samples\_info() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_biological\_samples\_info()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_biological_samples_info}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_biological\_samples\_info}}}{\emph{clinical\_data}}{}
\end{fulllineitems}

\index{extract\_analytical\_samples\_info() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_analytical\_samples\_info()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_analytical_samples_info}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_analytical\_samples\_info}}}{\emph{clinical\_data}}{}
\end{fulllineitems}

\index{extract\_biosample\_analytical\_sample\_relationship\_attributes() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_biosample\_analytical\_sample\_relationship\_attributes()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_biosample_analytical_sample_relationship_attributes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_biosample\_analytical\_sample\_relationship\_attributes}}}{\emph{clinical\_data}}{}
\end{fulllineitems}

\index{extract\_biological\_sample\_timepoint\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_biological\_sample\_timepoint\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_biological_sample_timepoint_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_biological\_sample\_timepoint\_rels}}}{\emph{clinical\_data}}{}
\end{fulllineitems}

\index{extract\_biological\_sample\_tissue\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_biological\_sample\_tissue\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_biological_sample_tissue_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_biological\_sample\_tissue\_rels}}}{\emph{clinical\_data}}{}
\end{fulllineitems}

\index{extract\_subject\_disease\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_subject\_disease\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_subject_disease_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_subject\_disease\_rels}}}{\emph{clinical\_data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
\end{fulllineitems}

\index{extract\_subject\_intervention\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_subject\_intervention\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_subject_intervention_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_subject\_intervention\_rels}}}{\emph{clinical\_data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
\end{fulllineitems}

\index{extract\_biological\_sample\_group\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_biological\_sample\_group\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_biological_sample_group_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_biological\_sample\_group\_rels}}}{\emph{clinical\_data}}{}
\end{fulllineitems}

\index{extract\_biological\_sample\_clinical\_variables\_rels() (in module graphdb\_builder.experiments.parsers.clinicalParser)@\spxentry{extract\_biological\_sample\_clinical\_variables\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.clinicalParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.clinicalParser.extract_biological_sample_clinical_variables_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.clinicalParser.}}\sphinxbfcode{\sphinxupquote{extract\_biological\_sample\_clinical\_variables\_rels}}}{\emph{clinical\_data}}{}
\end{fulllineitems}



\subparagraph{Proteomics parser}
\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:module-graphdb_builder.experiments.parsers.proteomicsParser}}\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:proteomics-parser}}\index{graphdb\_builder.experiments.parsers.proteomicsParser (module)@\spxentry{graphdb\_builder.experiments.parsers.proteomicsParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{projectId}, \emph{directory=None}}{}
\end{fulllineitems}

\index{parse\_from\_directory() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{parse\_from\_directory()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.parse_from_directory}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{parse\_from\_directory}}}{\emph{projectId}, \emph{directory}, \emph{configuration=None}}{}
\end{fulllineitems}

\index{parser\_from\_file() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{parser\_from\_file()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.parser_from_file}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{parser\_from\_file}}}{\emph{file\_path}, \emph{configuration}, \emph{data\_type}, \emph{is\_standard=True}}{}
\end{fulllineitems}

\index{update\_configuration() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{update\_configuration()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.update_configuration}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{update\_configuration}}}{\emph{data\_type}, \emph{processing\_tool}, \emph{value\_col=\textquotesingle{}LFQ intensity\textquotesingle{}}, \emph{columns={[}{]}}}{}
\end{fulllineitems}

\index{parse\_dataset() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{parse\_dataset()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.parse_dataset}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{parse\_dataset}}}{\emph{filepath}, \emph{configuration}}{}
\end{fulllineitems}

\index{parse\_standard\_dataset() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{parse\_standard\_dataset()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.parse_standard_dataset}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{parse\_standard\_dataset}}}{\emph{file\_path}, \emph{configuration}}{}
\end{fulllineitems}

\index{check\_columns() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{check\_columns()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.check_columns}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{check\_columns}}}{\emph{data}, \emph{req\_columns}, \emph{generated\_columns}}{}
\end{fulllineitems}

\index{check\_minimum\_configuration() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{check\_minimum\_configuration()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.check_minimum_configuration}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{check\_minimum\_configuration}}}{\emph{configuration}}{}
\end{fulllineitems}

\index{load\_dataset() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{load\_dataset()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.load_dataset}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{load\_dataset}}}{\emph{uri}, \emph{configuration}}{}
This function gets the molecular data from a proteomics experiment.
Input: uri of the processed file resulting from MQ
Output: pandas DataFrame with the columns and filters defined in config.py

\end{fulllineitems}

\index{remove\_contaminant\_tag() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{remove\_contaminant\_tag()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.remove_contaminant_tag}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{remove\_contaminant\_tag}}}{\emph{column}, \emph{tag=\textquotesingle{}CON\_\_\textquotesingle{}}}{}
\end{fulllineitems}

\index{expand\_groups() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{expand\_groups()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.expand_groups}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{expand\_groups}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_modification\_protein\_rels() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_modification\_protein\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_modification_protein_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_modification\_protein\_rels}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_protein\_modification\_subject\_rels() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_protein\_modification\_subject\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_protein_modification_subject_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_protein\_modification\_subject\_rels}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_protein\_protein\_modification\_rels() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_protein\_protein\_modification\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_protein_protein_modification_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_protein\_protein\_modification\_rels}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_peptide\_protein\_modification\_rels() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_peptide\_protein\_modification\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_peptide_protein_modification_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_peptide\_protein\_modification\_rels}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_protein\_modifications\_rels() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_protein\_modifications\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_protein_modifications_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_protein\_modifications\_rels}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_protein\_modifications\_modification\_rels() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_protein\_modifications\_modification\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_protein_modifications_modification_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_protein\_modifications\_modification\_rels}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_peptides() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_peptides()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_peptides}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_peptides}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_peptide\_subject\_rels() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_peptide\_subject\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_peptide_subject_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_peptide\_subject\_rels}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_peptide\_protein\_rels() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_peptide\_protein\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_peptide_protein_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_peptide\_protein\_rels}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_protein\_subject\_rels() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_protein\_subject\_rels()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_protein_subject_rels}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_protein\_subject\_rels}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{get\_value\_cols() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{get\_value\_cols()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.get_value_cols}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{get\_value\_cols}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}

\index{extract\_subject\_replicates\_from\_regex() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_subject\_replicates\_from\_regex()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_subject_replicates_from_regex}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_subject\_replicates\_from\_regex}}}{\emph{data}, \emph{regex}}{}
\end{fulllineitems}

\index{extract\_subject\_replicates() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_subject\_replicates()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_subject_replicates}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_subject\_replicates}}}{\emph{data}, \emph{value\_cols}}{}
\end{fulllineitems}

\index{extract\_attributes() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{extract\_attributes()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.extract_attributes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{extract\_attributes}}}{\emph{data}, \emph{attributes}}{}
\end{fulllineitems}

\index{merge\_regex\_attributes() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{merge\_regex\_attributes()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.merge_regex_attributes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{merge\_regex\_attributes}}}{\emph{data}, \emph{attributes}, \emph{index}, \emph{regexCols}}{}
\end{fulllineitems}

\index{merge\_col\_attributes() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{merge\_col\_attributes()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.merge_col_attributes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{merge\_col\_attributes}}}{\emph{data}, \emph{attributes}, \emph{index}}{}
\end{fulllineitems}

\index{calculate\_median\_replicates() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{calculate\_median\_replicates()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.calculate_median_replicates}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{calculate\_median\_replicates}}}{\emph{data}, \emph{log=\textquotesingle{}log2\textquotesingle{}}}{}
\end{fulllineitems}

\index{update\_groups() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{update\_groups()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.update_groups}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{update\_groups}}}{\emph{data}, \emph{groups}}{}
\end{fulllineitems}

\index{get\_dataset\_configuration() (in module graphdb\_builder.experiments.parsers.proteomicsParser)@\spxentry{get\_dataset\_configuration()}\spxextra{in module graphdb\_builder.experiments.parsers.proteomicsParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.proteomicsParser.get_dataset_configuration}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.proteomicsParser.}}\sphinxbfcode{\sphinxupquote{get\_dataset\_configuration}}}{\emph{processing\_format}, \emph{data\_type}}{}
\end{fulllineitems}



\subparagraph{WES parser}
\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:module-graphdb_builder.experiments.parsers.wesParser}}\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:wes-parser}}\index{graphdb\_builder.experiments.parsers.wesParser (module)@\spxentry{graphdb\_builder.experiments.parsers.wesParser}\spxextra{module}}\index{parser() (in module graphdb\_builder.experiments.parsers.wesParser)@\spxentry{parser()}\spxextra{in module graphdb\_builder.experiments.parsers.wesParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.wesParser.parser}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.wesParser.}}\sphinxbfcode{\sphinxupquote{parser}}}{\emph{projectId}}{}
\end{fulllineitems}

\index{parseWESDataset() (in module graphdb\_builder.experiments.parsers.wesParser)@\spxentry{parseWESDataset()}\spxextra{in module graphdb\_builder.experiments.parsers.wesParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.wesParser.parseWESDataset}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.wesParser.}}\sphinxbfcode{\sphinxupquote{parseWESDataset}}}{\emph{projectId}, \emph{configuration}, \emph{dataDir}}{}
\end{fulllineitems}

\index{loadWESDataset() (in module graphdb\_builder.experiments.parsers.wesParser)@\spxentry{loadWESDataset()}\spxextra{in module graphdb\_builder.experiments.parsers.wesParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.wesParser.loadWESDataset}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.wesParser.}}\sphinxbfcode{\sphinxupquote{loadWESDataset}}}{\emph{uri}, \emph{configuration}}{}
This function gets the molecular data from a Whole Exome Sequencing experiment.
Input: uri of the processed file resulting from the WES analysis pipeline. The resulting
Annovar annotated VCF file from Mutect (sampleID\_mutect\_annovar.vcf)
Output: pandas DataFrame with the columns and filters defined in config.py

\end{fulllineitems}

\index{extractWESRelationships() (in module graphdb\_builder.experiments.parsers.wesParser)@\spxentry{extractWESRelationships()}\spxextra{in module graphdb\_builder.experiments.parsers.wesParser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments.parsers:graphdb_builder.experiments.parsers.wesParser.extractWESRelationships}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.parsers.wesParser.}}\sphinxbfcode{\sphinxupquote{extractWESRelationships}}}{\emph{data}, \emph{configuration}}{}
\end{fulllineitems}



\paragraph{Experiments Controller}
\label{\detokenize{_autosummary/graphdb_builder.experiments:module-graphdb_builder.experiments.experiments_controller}}\label{\detokenize{_autosummary/graphdb_builder.experiments:experiments-controller}}\index{graphdb\_builder.experiments.experiments\_controller (module)@\spxentry{graphdb\_builder.experiments.experiments\_controller}\spxextra{module}}\index{generate\_dataset\_imports() (in module graphdb\_builder.experiments.experiments\_controller)@\spxentry{generate\_dataset\_imports()}\spxextra{in module graphdb\_builder.experiments.experiments\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments:graphdb_builder.experiments.experiments_controller.generate_dataset_imports}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.experiments\_controller.}}\sphinxbfcode{\sphinxupquote{generate\_dataset\_imports}}}{\emph{projectId}, \emph{dataType}, \emph{dataset\_import\_dir}}{}
\end{fulllineitems}

\index{generate\_graph\_files() (in module graphdb\_builder.experiments.experiments\_controller)@\spxentry{generate\_graph\_files()}\spxextra{in module graphdb\_builder.experiments.experiments\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.experiments:graphdb_builder.experiments.experiments_controller.generate_graph_files}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.experiments.experiments\_controller.}}\sphinxbfcode{\sphinxupquote{generate\_graph\_files}}}{\emph{data}, \emph{dataType}, \emph{projectId}, \emph{stats}, \emph{ot=\textquotesingle{}w\textquotesingle{}}, \emph{dataset\_import\_dir=\textquotesingle{}experiments\textquotesingle{}}}{}
\end{fulllineitems}



\subsubsection{User Creation}
\label{\detokenize{_autosummary/graphdb_builder.users:user-creation}}\label{\detokenize{_autosummary/graphdb_builder.users::doc}}

\paragraph{Users Controller}
\label{\detokenize{_autosummary/graphdb_builder.users:module-graphdb_builder.users.users_controller}}\label{\detokenize{_autosummary/graphdb_builder.users:users-controller}}\index{graphdb\_builder.users.users\_controller (module)@\spxentry{graphdb\_builder.users.users\_controller}\spxextra{module}}\index{parseUsersFile() (in module graphdb\_builder.users.users\_controller)@\spxentry{parseUsersFile()}\spxextra{in module graphdb\_builder.users.users\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.users:graphdb_builder.users.users_controller.parseUsersFile}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.users.users\_controller.}}\sphinxbfcode{\sphinxupquote{parseUsersFile}}}{\emph{importDirectory}, \emph{expiration=365}}{}
Creates new user in the graph database and corresponding node, through the following steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Generates new user identifier

\item {} 
Checks if a user with given properties already exists in the database. If not:

\item {} 
Creates new local user (access to graph database)

\item {} 
Saves data to tab\sphinxhyphen{}delimited file.

\end{enumerate}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{importDirectory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to the directory where all the import files are generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{expiration}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of days a user is given access.

\end{itemize}

\item[{Returns}] \leavevmode
Writes relevant .tsv file for the users in the provided file.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_user\_creation\_queries() (in module graphdb\_builder.users.users\_controller)@\spxentry{get\_user\_creation\_queries()}\spxextra{in module graphdb\_builder.users.users\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.users:graphdb_builder.users.users_controller.get_user_creation_queries}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.users.users\_controller.}}\sphinxbfcode{\sphinxupquote{get\_user\_creation\_queries}}}{}{}
Reads the YAML file containing the queries relevant to user creation, parses the given stream and     returns a Python object (dict{[}dict{]}).

\end{fulllineitems}

\index{get\_new\_user\_identifier() (in module graphdb\_builder.users.users\_controller)@\spxentry{get\_new\_user\_identifier()}\spxextra{in module graphdb\_builder.users.users\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.users:graphdb_builder.users.users_controller.get_new_user_identifier}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.users.users\_controller.}}\sphinxbfcode{\sphinxupquote{get\_new\_user\_identifier}}}{\emph{driver}}{}
Queries the database for the last user identifier and returns a new sequential identifier.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item[{Returns}] \leavevmode
User identifier.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{str}

\end{description}\end{quote}

\end{fulllineitems}

\index{check\_if\_node\_exists() (in module graphdb\_builder.users.users\_controller)@\spxentry{check\_if\_node\_exists()}\spxextra{in module graphdb\_builder.users.users\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.users:graphdb_builder.users.users_controller.check_if_node_exists}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.users.users\_controller.}}\sphinxbfcode{\sphinxupquote{check\_if\_node\_exists}}}{\emph{driver}, \emph{node\_property}, \emph{value}}{}
Queries the graph database and checks if a node with a specific property and property value already exists.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{node\_property}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} property of the node.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{value}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}\sphinxstyleliteralemphasis{\sphinxupquote{ or }}\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} property value.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with user identifier if User with node\_property and value already exists,             if User does not exist, returns and empty dataframe.

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_db\_user() (in module graphdb\_builder.users.users\_controller)@\spxentry{create\_db\_user()}\spxextra{in module graphdb\_builder.users.users\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.users:graphdb_builder.users.users_controller.create_db_user}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.users.users\_controller.}}\sphinxbfcode{\sphinxupquote{create\_db\_user}}}{\emph{driver}, \emph{data}}{}
Creates and assigns role to new graph database user, if user not in list of local users.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} (\sphinxstyleliteralemphasis{\sphinxupquote{Series}}) \textendash{} pandas Series with required user information (see set\_arguments()).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GenerateGraphFiles() (in module graphdb\_builder.users.users\_controller)@\spxentry{GenerateGraphFiles()}\spxextra{in module graphdb\_builder.users.users\_controller}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.users:graphdb_builder.users.users_controller.GenerateGraphFiles}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.users.users\_controller.}}\sphinxbfcode{\sphinxupquote{GenerateGraphFiles}}}{\emph{data}, \emph{output\_file}}{}
Saves pandas dataframe to users.tsv.
If file already exists, appends new lines.     Else, creates file and writes dataframe to it.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe to be written to .tsv file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{output\_file}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to output csv file.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}



\subsubsection{CKG Builder}
\label{\detokenize{_autosummary/graphdb_builder.builder:ckg-builder}}\label{\detokenize{_autosummary/graphdb_builder.builder::doc}}

\paragraph{User Creation Module}
\label{\detokenize{_autosummary/graphdb_builder.builder:module-graphdb_builder.builder.create_user}}\label{\detokenize{_autosummary/graphdb_builder.builder:user-creation-module}}\index{graphdb\_builder.builder.create\_user (module)@\spxentry{graphdb\_builder.builder.create\_user}\spxextra{module}}\index{create\_user\_node() (in module graphdb\_builder.builder.create\_user)@\spxentry{create\_user\_node()}\spxextra{in module graphdb\_builder.builder.create\_user}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.create_user.create_user_node}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.create\_user.}}\sphinxbfcode{\sphinxupquote{create\_user\_node}}}{\emph{driver}, \emph{data}}{}
Creates graph database node for new user and adds respective properties to node.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} (\sphinxstyleliteralemphasis{\sphinxupquote{Series}}) \textendash{} pandas Series with new user identifier and required user information (see set\_arguments()).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_user\_from\_command\_line() (in module graphdb\_builder.builder.create\_user)@\spxentry{create\_user\_from\_command\_line()}\spxextra{in module graphdb\_builder.builder.create\_user}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.create_user.create_user_from_command_line}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.create\_user.}}\sphinxbfcode{\sphinxupquote{create\_user\_from\_command\_line}}}{\emph{args}, \emph{expiration}}{}
Creates new user in the graph database and corresponding node, from a terminal window (command line),     and adds the new user information to the users excel and import files. Arguments as in set\_arguments().
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxstyleliteralemphasis{\sphinxupquote{any object with \_\_dict\_\_ attribute}}) \textendash{} object. Contains all the parameters neccessary to create a user (‘username’, ‘name’, ‘email’,                 ‘secondary\_email’, ‘phone\_number’ and ‘affiliation’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{expiration}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of days users is given access.

\end{itemize}

\end{description}\end{quote}

\begin{sphinxadmonition}{note}{Note:}
This function can be used directly with \sphinxstyleemphasis{python create\_user\_from\_command\_line.py \sphinxhyphen{}u username                 \sphinxhyphen{}n user\_name \sphinxhyphen{}e email \sphinxhyphen{}s secondary\_email \sphinxhyphen{}p phone\_number \sphinxhyphen{}a affiliation} .
\end{sphinxadmonition}

\end{fulllineitems}

\index{create\_user\_from\_file() (in module graphdb\_builder.builder.create\_user)@\spxentry{create\_user\_from\_file()}\spxextra{in module graphdb\_builder.builder.create\_user}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.create_user.create_user_from_file}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.create\_user.}}\sphinxbfcode{\sphinxupquote{create\_user\_from\_file}}}{\emph{filepath}, \emph{expiration}}{}
Creates new user in the graph database and corresponding node, from an excel file.     Rows in the file must be users, and columns must follow set\_arguments() fields.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{filepath}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} filepath and filename containing users information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{output\_file}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to output csv file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{expiration}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of days users is given access.

\end{itemize}

\end{description}\end{quote}

\begin{sphinxadmonition}{note}{Note:}
This function can be used directly with \sphinxstyleemphasis{python create\_user\_from\_file.py \sphinxhyphen{}f path\_to\_file} .
\end{sphinxadmonition}

\end{fulllineitems}

\index{create\_user() (in module graphdb\_builder.builder.create\_user)@\spxentry{create\_user()}\spxextra{in module graphdb\_builder.builder.create\_user}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.create_user.create_user}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.create\_user.}}\sphinxbfcode{\sphinxupquote{create\_user}}}{\emph{data}, \emph{output\_file}, \emph{expiration=365}}{}
Creates new user in the graph database and corresponding node, through the following steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Checks if a user with given properties already exists in the database. If not:

\item {} 
Generates new user identifier

\item {} 
Creates new local user (access to graph database)

\item {} 
Creates new user node

\item {} 
Saves data to users.tsv

\end{enumerate}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with users as rows and arguments and columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{output\_file}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to output csv file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{expiration}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of days users is given access.

\end{itemize}

\item[{Returns}] \leavevmode
Writes relevant .tsv file for the users in data.

\end{description}\end{quote}

\end{fulllineitems}

\index{set\_arguments() (in module graphdb\_builder.builder.create\_user)@\spxentry{set\_arguments()}\spxextra{in module graphdb\_builder.builder.create\_user}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.create_user.set_arguments}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.create\_user.}}\sphinxbfcode{\sphinxupquote{set\_arguments}}}{}{}
This function sets the arguments to be used as input for \sphinxstylestrong{create\_user.py} in the command line.

\end{fulllineitems}



\paragraph{Importer Module}
\label{\detokenize{_autosummary/graphdb_builder.builder:module-graphdb_builder.builder.importer}}\label{\detokenize{_autosummary/graphdb_builder.builder:importer-module}}\index{graphdb\_builder.builder.importer (module)@\spxentry{graphdb\_builder.builder.importer}\spxextra{module}}
Generates all the import files: Ontologies, Databases and Experiments.
The module is reponsible for generating all the csv files that will
be loaded into the Graph database and also updates a stats object
(hdf table) with the number of entities and relationships from each
dataset imported. A new stats object is created the first time a
full import is run.
\index{ontologiesImport() (in module graphdb\_builder.builder.importer)@\spxentry{ontologiesImport()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.ontologiesImport}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{ontologiesImport}}}{\emph{importDirectory}, \emph{ontologies=None}, \emph{download=True}, \emph{import\_type=\textquotesingle{}partial\textquotesingle{}}}{}
Generates all the entities and relationships from the provided ontologies. If the ontologies list is    not provided, then all the ontologies listed in the configuration will be imported (full\_import).     This function also updates the stats object with numbers from the imported ontologies.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{importDirectory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path of the import directory where files will be created.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ontologies}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} a list of ontology names to be imported.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{download}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} wether database is to be downloaded.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{import\_type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} type of import (´full´ or ´partial´).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{databasesImport() (in module graphdb\_builder.builder.importer)@\spxentry{databasesImport()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.databasesImport}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{databasesImport}}}{\emph{importDirectory}, \emph{databases=None}, \emph{n\_jobs=1}, \emph{download=True}, \emph{import\_type=\textquotesingle{}partial\textquotesingle{}}}{}
Generates all the entities and relationships from the provided databases. If the databases list is    not provided, then all the databases listed in the configuration will be imported (full\_import).    This function also updates the stats object with numbers from the imported databases.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{importDirectory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path of the import directory where files will be created.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{databases}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} a list of database names to be imported.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n\_jobs}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of jobs to run in parallel. 1 by default when updating one database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{import\_type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} type of import (´full´ or ´partial´).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{experimentsImport() (in module graphdb\_builder.builder.importer)@\spxentry{experimentsImport()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.experimentsImport}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{experimentsImport}}}{\emph{projects=None}, \emph{n\_jobs=1}, \emph{import\_type=\textquotesingle{}partial\textquotesingle{}}}{}
Generates all the entities and relationships from the specified Projects. If the projects list is    not provided, then all the projects the experiments directory will be imported (full\_import).     Calls function experimentImport.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{projects}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of project identifiers to be imported.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n\_jobs}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of jobs to run in parallel. 1 by default when updating one project.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{import\_type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} type of import (´full´ or ´partial´).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{experimentImport() (in module graphdb\_builder.builder.importer)@\spxentry{experimentImport()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.experimentImport}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{experimentImport}}}{\emph{importDirectory}, \emph{experimentsDirectory}, \emph{project}}{}
Generates all the entities and relationships from the specified Project. Called from function experimentsImport.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{importDirectory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to the directory where all the import files are generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{experimentDirectory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to the directory where all the experiments are located.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{project}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of the project to be imported.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{usersImport() (in module graphdb\_builder.builder.importer)@\spxentry{usersImport()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.usersImport}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{usersImport}}}{\emph{importDirectory}, \emph{import\_type=\textquotesingle{}partial\textquotesingle{}}}{}
Generates User entities from excel file and grants access of new users to the database.
This function also writes the relevant information to a tab\sphinxhyphen{}delimited file in the import     directory.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{importDirectory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to the directory where all the import files are generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{import\_type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} type of import (´full´ or ´partial).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{fullImport() (in module graphdb\_builder.builder.importer)@\spxentry{fullImport()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.fullImport}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{fullImport}}}{\emph{download=True}, \emph{n\_jobs=4}}{}
Calls the different importer functions: Ontologies, databases,     experiments. The first step is to check if the stats object exists     and create it otherwise. Calls setupStats.

\end{fulllineitems}

\index{generateStatsDataFrame() (in module graphdb\_builder.builder.importer)@\spxentry{generateStatsDataFrame()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.generateStatsDataFrame}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{generateStatsDataFrame}}}{\emph{stats}}{}
Generates a dataframe with the stats from each import.
:param list stats: a list with statistics collected from each importer function.
:return: Pandas dataframe with the collected statistics.

\end{fulllineitems}

\index{setupStats() (in module graphdb\_builder.builder.importer)@\spxentry{setupStats()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.setupStats}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{setupStats}}}{\emph{import\_type}}{}
Creates a stats object that will collect all the statistics collected from each import.

\end{fulllineitems}

\index{createEmptyStats() (in module graphdb\_builder.builder.importer)@\spxentry{createEmptyStats()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.createEmptyStats}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{createEmptyStats}}}{\emph{statsCols}, \emph{statsFile}, \emph{statsName}}{}
Creates a HDFStore object with a empty dataframe with the collected stats columns.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{statsCols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} a list of columns with the fields collected from the import statistics.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{statsFile}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path where the object should be stored.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{statsName}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name if the file containing the stats object.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{writeStats() (in module graphdb\_builder.builder.importer)@\spxentry{writeStats()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.writeStats}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{writeStats}}}{\emph{statsDf}, \emph{import\_type}, \emph{stats\_name=None}}{}
Appends the new collected statistics to the existing stats object.
:param statsDf: a pandas dataframe with the new statistics from the importing.
:param str statsName: If the statistics should be stored with a specific name.

\end{fulllineitems}

\index{getStatsName() (in module graphdb\_builder.builder.importer)@\spxentry{getStatsName()}\spxextra{in module graphdb\_builder.builder.importer}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.importer.getStatsName}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.importer.}}\sphinxbfcode{\sphinxupquote{getStatsName}}}{\emph{import\_type}}{}
Generates the stats object name where to store the importing statistics from the CKG version,     which is defined in the configuration.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
statsName: key used to store in the stats object.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{str}

\end{description}\end{quote}

\end{fulllineitems}



\paragraph{Loader Module}
\label{\detokenize{_autosummary/graphdb_builder.builder:module-graphdb_builder.builder.loader}}\label{\detokenize{_autosummary/graphdb_builder.builder:loader-module}}\index{graphdb\_builder.builder.loader (module)@\spxentry{graphdb\_builder.builder.loader}\spxextra{module}}
Populates the graph database with all the files generated by the importer.py module: 
Ontologies, Databases and Experiments.
The module loads all the entities and relationships defined in the importer files. It
calls Cypher queries defined in the cypher.py module. Further, it generates an hdf object
with the number of enities and relationships loaded for each Database, Ontology and Experiment.
This module also generates a compressed backup file of all the loaded files.

There are two types of updates:
\begin{itemize}
\item {} 
Full: all the entities and relationships in the graph database are populated

\item {} 
Partial: only the specified entities and relationships are loaded

\end{itemize}

The compressed files for each type of update are named accordingly and saved in the archive/ folder
in data/.
\index{load\_into\_database() (in module graphdb\_builder.builder.loader)@\spxentry{load\_into\_database()}\spxextra{in module graphdb\_builder.builder.loader}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.loader.load_into_database}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.loader.}}\sphinxbfcode{\sphinxupquote{load\_into\_database}}}{\emph{driver}, \emph{queries}, \emph{requester}}{}
This function runs the queries provided in the graph database using a py2neo driver.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{queries}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{{[}}}\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}\sphinxstyleliteralemphasis{\sphinxupquote{{]}}}) \textendash{} list of queries to be passed to the database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{requester}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of the query.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{updateDB() (in module graphdb\_builder.builder.loader)@\spxentry{updateDB()}\spxextra{in module graphdb\_builder.builder.loader}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.loader.updateDB}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.loader.}}\sphinxbfcode{\sphinxupquote{updateDB}}}{\emph{driver}, \emph{imports=None}, \emph{specific={[}{]}}}{}
Populates the graph database with information for each Database, Ontology or Experiment     specified in imports. If imports is not defined, the function populates the entire graph     database based on the graph variable defined in the grapher\_config.py module.     This function also updates the graph stats object with numbers from the loaded entities and     relationships.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{imports}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} a list of entities to be loaded into the graph.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{fullUpdate() (in module graphdb\_builder.builder.loader)@\spxentry{fullUpdate()}\spxextra{in module graphdb\_builder.builder.loader}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.loader.fullUpdate}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.loader.}}\sphinxbfcode{\sphinxupquote{fullUpdate}}}{}{}
Main method that controls the population of the graph database. Firstly, it gets a connection     to the database (driver) and then initiates the update of the entire database getting     all the graph entities to update from configuration. Once the graph database has been     populated, the imports folder in data/ is compressed and archived in the archive/ folder     so that a backup of the imports files is kept (full).

\end{fulllineitems}

\index{partialUpdate() (in module graphdb\_builder.builder.loader)@\spxentry{partialUpdate()}\spxextra{in module graphdb\_builder.builder.loader}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.loader.partialUpdate}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.loader.}}\sphinxbfcode{\sphinxupquote{partialUpdate}}}{\emph{imports}, \emph{specific={[}{]}}}{}
Method that controls the update of the graph database with the specified entities and     relationships. Firstly, it gets a connection to the database (driver) and then initiates     the update of the specified graph entities.     Once the graph database has been populated, the data files uploaded to the graph are compressed     and archived in the archive/ folder (partial).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{imports}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of entities to update

\end{description}\end{quote}

\end{fulllineitems}

\index{archiveImportDirectory() (in module graphdb\_builder.builder.loader)@\spxentry{archiveImportDirectory()}\spxextra{in module graphdb\_builder.builder.loader}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.loader.archiveImportDirectory}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.loader.}}\sphinxbfcode{\sphinxupquote{archiveImportDirectory}}}{\emph{archive\_type=\textquotesingle{}full\textquotesingle{}}}{}
This function creates the compressed backup imports folder with either the whole folder     (full update) or with only the files uploaded (partial update). The folder or files are     compressed into a gzipped tarball file and stored in the archive/ folder defined in the     configuration.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{archive\_type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} whether it is a full update or a partial update.

\end{description}\end{quote}

\end{fulllineitems}



\paragraph{Builder Module}
\label{\detokenize{_autosummary/graphdb_builder.builder:module-graphdb_builder.builder.builder}}\label{\detokenize{_autosummary/graphdb_builder.builder:builder-module}}\index{graphdb\_builder.builder.builder (module)@\spxentry{graphdb\_builder.builder.builder}\spxextra{module}}
Builds the database in two main steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{)}%
\item {} 
Imports all the data from ontologies, databases and experiments

\item {} 
Loads these data into the database

\end{enumerate}

The module can perform full updates, executing both steps for all the ontologies,
databases and experiments or a partial update. Partial updates can execute step 1 or
step 2 for specific data.
\index{set\_arguments() (in module graphdb\_builder.builder.builder)@\spxentry{set\_arguments()}\spxextra{in module graphdb\_builder.builder.builder}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder.builder:graphdb_builder.builder.builder.set_arguments}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder.builder.}}\sphinxbfcode{\sphinxupquote{set\_arguments}}}{}{}
This function sets the arguments to be used as input for \sphinxstylestrong{builder.py} in the command line.

\end{fulllineitems}



\subsubsection{builder\_utils.py}
\label{\detokenize{_autosummary/graphdb_builder:module-graphdb_builder.builder_utils}}\label{\detokenize{_autosummary/graphdb_builder:builder-utils-py}}\index{graphdb\_builder.builder\_utils (module)@\spxentry{graphdb\_builder.builder\_utils}\spxextra{module}}\index{readDataset() (in module graphdb\_builder.builder\_utils)@\spxentry{readDataset()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.readDataset}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{readDataset}}}{\emph{uri}}{}
\end{fulllineitems}

\index{readDataFromCSV() (in module graphdb\_builder.builder\_utils)@\spxentry{readDataFromCSV()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.readDataFromCSV}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{readDataFromCSV}}}{\emph{uri}}{}
Read the data from csv file

\end{fulllineitems}

\index{readDataFromTXT() (in module graphdb\_builder.builder\_utils)@\spxentry{readDataFromTXT()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.readDataFromTXT}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{readDataFromTXT}}}{\emph{uri}}{}
Read the data from tsv or txt file

\end{fulllineitems}

\index{readDataFromExcel() (in module graphdb\_builder.builder\_utils)@\spxentry{readDataFromExcel()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.readDataFromExcel}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{readDataFromExcel}}}{\emph{uri}}{}
Read the data from Excel file

\end{fulllineitems}

\index{get\_files\_by\_pattern() (in module graphdb\_builder.builder\_utils)@\spxentry{get\_files\_by\_pattern()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.get_files_by_pattern}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{get\_files\_by\_pattern}}}{\emph{regex\_path}}{}
\end{fulllineitems}

\index{get\_extra\_pairs() (in module graphdb\_builder.builder\_utils)@\spxentry{get\_extra\_pairs()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.get_extra_pairs}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{get\_extra\_pairs}}}{\emph{directory}, \emph{extra\_file}}{}
\end{fulllineitems}

\index{parse\_contents() (in module graphdb\_builder.builder\_utils)@\spxentry{parse\_contents()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.parse_contents}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{parse\_contents}}}{\emph{contents}, \emph{filename}}{}
Reads binary string files and returns a Pandas DataFrame.

\end{fulllineitems}

\index{export\_contents() (in module graphdb\_builder.builder\_utils)@\spxentry{export\_contents()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.export_contents}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{export\_contents}}}{\emph{data}, \emph{dataDir}, \emph{filename}}{}
Export Pandas DataFrame to file, with UTF\sphinxhyphen{}8 endocing.

\end{fulllineitems}

\index{write\_relationships() (in module graphdb\_builder.builder\_utils)@\spxentry{write\_relationships()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.write_relationships}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{write\_relationships}}}{\emph{relationships}, \emph{header}, \emph{outputfile}}{}
Reads a set of relationships and saves them to a file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{relationships}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#set}{\sphinxstyleliteralemphasis{\sphinxupquote{set}}}) \textendash{} set of tuples with relationship data: source node, target node,                                 relationship type, source and other attributes.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{header}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of column names.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{outputfile}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to file to be saved (including filename and extention).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{write\_entities() (in module graphdb\_builder.builder\_utils)@\spxentry{write\_entities()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.write_entities}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{write\_entities}}}{\emph{entities}, \emph{header}, \emph{outputfile}}{}
Reads a set of entities and saves them to a file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{entities}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#set}{\sphinxstyleliteralemphasis{\sphinxupquote{set}}}) \textendash{} set of tuples with entities data: identifier, label, name                        and other attributes.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{header}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of column names.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{outputfile}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to file to be saved (including filename and extention).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_config() (in module graphdb\_builder.builder\_utils)@\spxentry{get\_config()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.get_config}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{get\_config}}}{\emph{config\_name}, \emph{data\_type=\textquotesingle{}databases\textquotesingle{}}}{}
Reads YAML configuration file and converts it into a Python dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{config\_name}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the configuration YAML file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data\_type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} configuration type (‘databases’ or ‘ontologies’).

\end{itemize}

\item[{Returns}] \leavevmode
Dictionary.

\end{description}\end{quote}

\begin{sphinxadmonition}{note}{Note:}
Use this function to obtain configuration for individual database/ontology parsers.
\end{sphinxadmonition}

\end{fulllineitems}

\index{expand\_cols() (in module graphdb\_builder.builder\_utils)@\spxentry{expand\_cols()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.expand_cols}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{expand\_cols}}}{\emph{data}, \emph{col}, \emph{sep=\textquotesingle{};\textquotesingle{}}}{}
Expands the rows of a dataframe by splitting the specified column
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} dataframe to be expanded

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column that contains string to be expanded (i.e. ‘P02788;E7EQB2;E7ER44;P02788\sphinxhyphen{}2;C9JCF5’)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sep}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} separator (i.e. ‘;’)

\end{itemize}

\item[{Returns}] \leavevmode
expanded pandas dataframe

\end{description}\end{quote}

\end{fulllineitems}

\index{setup\_config() (in module graphdb\_builder.builder\_utils)@\spxentry{setup\_config()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.setup_config}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{setup\_config}}}{\emph{data\_type=\textquotesingle{}databases\textquotesingle{}}}{}
Reads YAML configuration file and converts it into a Python dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{data\_type}} \textendash{} configuration type (‘databases’, ‘ontologies’, ‘experiments’ or ‘builder’).

\item[{Returns}] \leavevmode
Dictionary.

\end{description}\end{quote}

\begin{sphinxadmonition}{note}{Note:}
This function should be used to obtain the configuration for databases\_controller.py,                 ontologies\_controller.py, experiments\_controller.py and builder.py.
\end{sphinxadmonition}

\end{fulllineitems}

\index{get\_full\_path\_directories() (in module graphdb\_builder.builder\_utils)@\spxentry{get\_full\_path\_directories()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.get_full_path_directories}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{get\_full\_path\_directories}}}{}{}
Reads Builder YAML configuration file and returns the full path of all directories.
:return: Dictionary.

\end{fulllineitems}

\index{list\_ftp\_directory() (in module graphdb\_builder.builder\_utils)@\spxentry{list\_ftp\_directory()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.list_ftp_directory}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{list\_ftp\_directory}}}{\emph{ftp\_url}, \emph{user=\textquotesingle{}\textquotesingle{}}, \emph{password=\textquotesingle{}\textquotesingle{}}}{}
Lists all files present in folder from FTP server.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ftp\_url}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} link to access ftp server.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{user}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} username to access ftp server if required.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{password}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} password to access ftp server if required.

\end{itemize}

\item[{Returns}] \leavevmode
List of files contained in ftp server folder provided with ftp\_url.

\end{description}\end{quote}

\end{fulllineitems}

\index{setup\_logging() (in module graphdb\_builder.builder\_utils)@\spxentry{setup\_logging()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.setup_logging}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{setup\_logging}}}{\emph{path=\textquotesingle{}log.config\textquotesingle{}}, \emph{key=None}}{}
Setup logging configuration.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{path}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to file containing configuration for logging file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{key}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the logger.

\end{itemize}

\item[{Returns}] \leavevmode
Logger with the specified name from ‘key’. If key is \sphinxstyleemphasis{None}, returns a logger which is                 the root logger of the hierarchy.

\end{description}\end{quote}

\end{fulllineitems}

\index{downloadDB() (in module graphdb\_builder.builder\_utils)@\spxentry{downloadDB()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.downloadDB}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{downloadDB}}}{\emph{databaseURL}, \emph{directory=None}, \emph{file\_name=None}, \emph{user=\textquotesingle{}\textquotesingle{}}, \emph{password=\textquotesingle{}\textquotesingle{}}}{}
This function downloads the raw files from a biomedical database server when a link is provided.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{databaseURL}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} link to access biomedical database server.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{directory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}\sphinxstyleliteralemphasis{\sphinxupquote{ or }}\sphinxhref{https://docs.python.org/3.6/library/constants.html\#None}{\sphinxstyleliteralemphasis{\sphinxupquote{None}}}) \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{file\_name}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}\sphinxstyleliteralemphasis{\sphinxupquote{ or }}\sphinxhref{https://docs.python.org/3.6/library/constants.html\#None}{\sphinxstyleliteralemphasis{\sphinxupquote{None}}}) \textendash{} name of the file to dowload. If None, ‘databaseURL’ must contain                         filename after the last ‘/’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{user}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} username to access biomedical database server if required.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{password}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} password to access biomedical database server if required.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{searchPubmed() (in module graphdb\_builder.builder\_utils)@\spxentry{searchPubmed()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.searchPubmed}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{searchPubmed}}}{\emph{searchFields}, \emph{sortby=\textquotesingle{}relevance\textquotesingle{}}, \emph{num=\textquotesingle{}10\textquotesingle{}}, \emph{resultsFormat=\textquotesingle{}json\textquotesingle{}}}{}
Searches PubMed database for MeSH terms and other additional fields (‘searchFields’), sorts them by relevance and     returns the top ‘num’.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{searchFields}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of search fields to query for.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sortby}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} parameter to use for sorting.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{num}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} number of PubMed identifiers to return.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{resultsFormat}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} format of the PubMed result.

\end{itemize}

\item[{Returns}] \leavevmode
Dictionary with total number of PubMed ids, and top ‘num’ ids.

\end{description}\end{quote}

\end{fulllineitems}

\index{is\_number() (in module graphdb\_builder.builder\_utils)@\spxentry{is\_number()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.is_number}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{is\_number}}}{\emph{s}}{}
This function checks if given input is a float and returns True if so, and False if it is not.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{s}} \textendash{} input

\item[{Returns}] \leavevmode
Boolean.

\end{description}\end{quote}

\end{fulllineitems}

\index{getMedlineAbstracts() (in module graphdb\_builder.builder\_utils)@\spxentry{getMedlineAbstracts()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.getMedlineAbstracts}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{getMedlineAbstracts}}}{\emph{idList}}{}
This function accesses NCBI over the WWWW and returns Medline data as a handle object,     which is parsed and converted to a Pandas DataFrame.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{idList}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}\sphinxstyleliteralemphasis{\sphinxupquote{ or }}\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} single identifier or comma\sphinxhyphen{}delimited list of identifiers. All the identifiers                     must be from the database PubMed.

\item[{Returns}] \leavevmode
Pandas DataFrame with columns: ‘title’, ‘authors’, ‘journal’, ‘keywords’, ‘abstract’, ‘PMID’ and ‘url’.

\end{description}\end{quote}

\end{fulllineitems}

\index{remove\_directory() (in module graphdb\_builder.builder\_utils)@\spxentry{remove\_directory()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.remove_directory}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{remove\_directory}}}{\emph{directory}}{}
\end{fulllineitems}

\index{listDirectoryFiles() (in module graphdb\_builder.builder\_utils)@\spxentry{listDirectoryFiles()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.listDirectoryFiles}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{listDirectoryFiles}}}{\emph{directory}}{}
Lists all files in a specified directory.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{directory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to folder.

\item[{Returns}] \leavevmode
List of file names.

\end{description}\end{quote}

\end{fulllineitems}

\index{listDirectoryFolders() (in module graphdb\_builder.builder\_utils)@\spxentry{listDirectoryFolders()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.listDirectoryFolders}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{listDirectoryFolders}}}{\emph{directory}}{}
Lists all directories in a specified directory.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{directory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to folder.

\item[{Returns}] \leavevmode
List of folder names.

\end{description}\end{quote}

\end{fulllineitems}

\index{listDirectoryFoldersNotEmpty() (in module graphdb\_builder.builder\_utils)@\spxentry{listDirectoryFoldersNotEmpty()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.listDirectoryFoldersNotEmpty}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{listDirectoryFoldersNotEmpty}}}{\emph{directory}}{}
Lists all directories in a specified directory.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{directory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to folder.

\item[{Returns}] \leavevmode
List of folder names.

\end{description}\end{quote}

\end{fulllineitems}

\index{checkDirectory() (in module graphdb\_builder.builder\_utils)@\spxentry{checkDirectory()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.checkDirectory}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{checkDirectory}}}{\emph{directory}}{}
Checks if given directory exists and if not, creates it.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{directory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to folder.

\end{description}\end{quote}

\end{fulllineitems}

\index{flatten() (in module graphdb\_builder.builder\_utils)@\spxentry{flatten()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.flatten}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{flatten}}}{\emph{t}}{}
Code from: \sphinxurl{https://gist.github.com/shaxbee/0ada767debf9eefbdb6e}
Acknowledgements: Zbigniew Mandziejewicz (shaxbee)
Generator flattening the structure

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{p}{(}\PYG{l+m+mi}{4}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{7}\PYG{p}{]}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{6}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{6}\PYG{p}{]}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{]}\PYG{p}{]}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{)}\PYG{p}{]}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}
\PYG{g+go}{[2, 2, 4, 5, 7, 2, 6, 2, 6, 6, 4, 6]}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{pretty\_print() (in module graphdb\_builder.builder\_utils)@\spxentry{pretty\_print()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.pretty_print}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{pretty\_print}}}{\emph{data}}{}
This function provides a capability to “pretty\sphinxhyphen{}print” arbitrary Python data structures in a forma that can be     used as input to the interpreter. For more information visit \sphinxurl{https://docs.python.org/2/library/pprint.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} python object.

\end{description}\end{quote}

\end{fulllineitems}

\index{convertOBOtoNet() (in module graphdb\_builder.builder\_utils)@\spxentry{convertOBOtoNet()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.convertOBOtoNet}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{convertOBOtoNet}}}{\emph{ontologyFile}}{}
Takes an .obo file and returns a NetworkX graph representation of the ontology, that holds multiple     edges between two nodes.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{ontologyFile}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to ontology file.

\item[{Returns}] \leavevmode
NetworkX graph.

\end{description}\end{quote}

\end{fulllineitems}

\index{getCurrentTime() (in module graphdb\_builder.builder\_utils)@\spxentry{getCurrentTime()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.getCurrentTime}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{getCurrentTime}}}{}{}
Returns current date (Year\sphinxhyphen{}Month\sphinxhyphen{}Day) and time (Hour\sphinxhyphen{}Minute\sphinxhyphen{}Second).
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
Two strings: date and time.

\end{description}\end{quote}

\end{fulllineitems}

\index{convert\_bytes() (in module graphdb\_builder.builder\_utils)@\spxentry{convert\_bytes()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.convert_bytes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{convert\_bytes}}}{\emph{num}}{}
This function will convert bytes to MB…. GB… etc.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{num}} \textendash{} float, integer or pandas.Series.

\end{description}\end{quote}

\end{fulllineitems}

\index{copytree() (in module graphdb\_builder.builder\_utils)@\spxentry{copytree()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.copytree}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{copytree}}}{\emph{src}, \emph{dst}, \emph{symlinks=False}, \emph{ignore=None}}{}
\end{fulllineitems}

\index{file\_size() (in module graphdb\_builder.builder\_utils)@\spxentry{file\_size()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.file_size}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{file\_size}}}{\emph{file\_path}}{}
This function returns the file size.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{file\_path}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to file.

\item[{Returns}] \leavevmode
Size in bytes of a plain file.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{str}

\end{description}\end{quote}

\end{fulllineitems}

\index{buildStats() (in module graphdb\_builder.builder\_utils)@\spxentry{buildStats()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.buildStats}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{buildStats}}}{\emph{count}, \emph{otype}, \emph{name}, \emph{dataset}, \emph{filename}, \emph{updated\_on=None}}{}
Returns a tuple with all the information needed to build a stats file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{count}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of entities/relationships.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{otype}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} ‘entity’ or ‘relationsgips’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} entity/relationship label.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dataset}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} database/ontology.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{filename}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to file where entities/relationships are stored.

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with date, time, database name, file where entities/relationships are stored,     file size, number of entities/relationships imported, type and label.

\end{description}\end{quote}

\end{fulllineitems}

\index{compress\_directory() (in module graphdb\_builder.builder\_utils)@\spxentry{compress\_directory()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.compress_directory}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{compress\_directory}}}{\emph{folder\_to\_backup}, \emph{dest\_folder}, \emph{file\_name}}{}
Compresses folder to .tar.gz to create data backup archive file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{folder\_to\_backup}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to folder to compress and backup.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dest\_folder}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path where to save compressed folder.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{file\_name}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the compressed file.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{read\_gzipped\_file() (in module graphdb\_builder.builder\_utils)@\spxentry{read\_gzipped\_file()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.read_gzipped_file}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{read\_gzipped\_file}}}{\emph{filepath}}{}
Opens an underlying process to access a gzip file through the creation of a new pipe to the child.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{filepath}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to gzip file.

\item[{Returns}] \leavevmode
A bytes sequence that specifies the standard output.

\end{description}\end{quote}

\end{fulllineitems}

\index{parse\_fasta() (in module graphdb\_builder.builder\_utils)@\spxentry{parse\_fasta()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.parse_fasta}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{parse\_fasta}}}{\emph{file\_handler}}{}
Using BioPython to read fasta file as SeqIO objects
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{file\_handler}} (\sphinxstyleliteralemphasis{\sphinxupquote{file\_handler}}) \textendash{} opened fasta file

\item[{Return iterator records}] \leavevmode
iterator of sequence objects

\end{description}\end{quote}

\end{fulllineitems}

\index{batch\_iterator() (in module graphdb\_builder.builder\_utils)@\spxentry{batch\_iterator()}\spxextra{in module graphdb\_builder.builder\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.builder_utils.batch_iterator}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.builder\_utils.}}\sphinxbfcode{\sphinxupquote{batch\_iterator}}}{\emph{iterator}, \emph{batch\_size}}{}
Returns lists of length batch\_size.

This can be used on any iterator, for example to batch up
SeqRecord objects from Bio.SeqIO.parse(…), or to batch
Alignment objects from Bio.AlignIO.parse(…), or simply
lines from a file handle.

This is a generator function, and it returns lists of the
entries from the supplied iterator.  Each list will have
batch\_size entries, although the final list may be shorter.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{iterator}} (\sphinxstyleliteralemphasis{\sphinxupquote{iterator}}) \textendash{} batch to be extracted

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{batch\_size}} (\sphinxstyleliteralemphasis{\sphinxupquote{integer}}) \textendash{} size of the batch

\end{itemize}

\item[{Return list batch}] \leavevmode
list with the batch elements of size batch\_size

\end{description}\end{quote}

source: \sphinxurl{https://biopython.org/wiki/Split\_large\_file}

\end{fulllineitems}



\subsubsection{mapping.py}
\label{\detokenize{_autosummary/graphdb_builder:module-graphdb_builder.mapping}}\label{\detokenize{_autosummary/graphdb_builder:mapping-py}}\index{graphdb\_builder.mapping (module)@\spxentry{graphdb\_builder.mapping}\spxextra{module}}\index{reset\_mapping() (in module graphdb\_builder.mapping)@\spxentry{reset\_mapping()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.reset_mapping}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{reset\_mapping}}}{\emph{entity}}{}
Checks if mapping.tsv file exists and removes it.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{entity}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} entity label as defined in databases\_config.yml

\end{description}\end{quote}

\end{fulllineitems}

\index{mark\_complete\_mapping() (in module graphdb\_builder.mapping)@\spxentry{mark\_complete\_mapping()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.mark_complete_mapping}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{mark\_complete\_mapping}}}{\emph{entity}}{}
Checks if mapping.tsv file exists and renames it to complete\_mapping.tsv.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{entity}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} entity label as defined in databases\_config.yml

\end{description}\end{quote}

\end{fulllineitems}

\index{getMappingFromOntology() (in module graphdb\_builder.mapping)@\spxentry{getMappingFromOntology()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.getMappingFromOntology}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{getMappingFromOntology}}}{\emph{ontology}, \emph{source=None}}{}
Converts .tsv file with complete list of ontology identifiers and aliases,
to dictionary with aliases as keys and ontology identifiers as values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ontology}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} ontology label as defined in ontologies\_config.yml.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}\sphinxstyleliteralemphasis{\sphinxupquote{ or }}\sphinxhref{https://docs.python.org/3.6/library/constants.html\#None}{\sphinxstyleliteralemphasis{\sphinxupquote{None}}}) \textendash{} name of the source database for selecting aliases.

\end{itemize}

\item[{Returns}] \leavevmode
Dictionary of aliases (keys) and ontology identifiers (values).

\end{description}\end{quote}

\end{fulllineitems}

\index{getMappingFromDatabase() (in module graphdb\_builder.mapping)@\spxentry{getMappingFromDatabase()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.getMappingFromDatabase}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{getMappingFromDatabase}}}{\emph{id\_list}, \emph{node}, \emph{attribute\_from=\textquotesingle{}id\textquotesingle{}}, \emph{attribute\_to=\textquotesingle{}name\textquotesingle{}}}{}
\end{fulllineitems}

\index{getMappingForEntity() (in module graphdb\_builder.mapping)@\spxentry{getMappingForEntity()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.getMappingForEntity}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{getMappingForEntity}}}{\emph{entity}}{}
Converts .tsv file with complete list of entity identifiers and aliases,     to dictionary with aliases as keys and entity identifiers as values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{entity}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} entity label as defined in databases\_config.yml.

\item[{Returns}] \leavevmode
Dictionary of aliases (keys) and entity identifiers (value).

\end{description}\end{quote}

\end{fulllineitems}

\index{getMultipleMappingForEntity() (in module graphdb\_builder.mapping)@\spxentry{getMultipleMappingForEntity()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.getMultipleMappingForEntity}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{getMultipleMappingForEntity}}}{\emph{entity}}{}
Converts .tsv file with complete list of entity identifiers and aliases,     to dictionary with aliases to other databases as keys and entity identifiers as values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{entity}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} entity label as defined in databases\_config.yml.

\item[{Returns}] \leavevmode
Dictionary of aliases (keys) and set of unique entity identifiers (values).

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_STRING\_mapping\_url() (in module graphdb\_builder.mapping)@\spxentry{get\_STRING\_mapping\_url()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.get_STRING_mapping_url}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{get\_STRING\_mapping\_url}}}{\emph{db=\textquotesingle{}STRING\textquotesingle{}}}{}
Get the url for downloading the mapping file from either STRING or STITCH
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{db}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} Which database to get the url from: STRING or STITCH

\item[{Returns}] \leavevmode
url where to download the mapping file

\end{description}\end{quote}

\end{fulllineitems}

\index{getSTRINGMapping() (in module graphdb\_builder.mapping)@\spxentry{getSTRINGMapping()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.getSTRINGMapping}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{getSTRINGMapping}}}{\emph{source=\textquotesingle{}BLAST\_UniProt\_AC\textquotesingle{}}, \emph{download=True}, \emph{db=\textquotesingle{}STRING\textquotesingle{}}}{}
Parses database (db) and extracts relationships between identifiers to order databases (source).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{url}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} link to download database raw file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the source database for selecting aliases.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{download}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} wether to download the file or not.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{db}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the database to be parsed.

\end{itemize}

\item[{Returns}] \leavevmode
Dictionary of database identifers (keys) and set of unique aliases to other databases (values).

\end{description}\end{quote}

\end{fulllineitems}

\index{buildMappingFromOBO() (in module graphdb\_builder.mapping)@\spxentry{buildMappingFromOBO()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.buildMappingFromOBO}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{buildMappingFromOBO}}}{\emph{oboFile}, \emph{ontology}}{}
Parses and extracts ontology idnetifiers, names and synonyms from raw file, and writes all the information     to a .tsv file.
:param str oboFile: path to ontology raw file.
:param str ontology: ontology database acronym as defined in ontologies\_config.yml.

\end{fulllineitems}

\index{map\_experiment\_files() (in module graphdb\_builder.mapping)@\spxentry{map\_experiment\_files()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.map_experiment_files}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{map\_experiment\_files}}}{\emph{project\_id}, \emph{datasetPath}, \emph{mapping}}{}
\end{fulllineitems}

\index{map\_experimental\_data() (in module graphdb\_builder.mapping)@\spxentry{map\_experimental\_data()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.map_experimental_data}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{map\_experimental\_data}}}{\emph{data}, \emph{mapping}}{}
\end{fulllineitems}

\index{get\_mapping\_analytical\_samples() (in module graphdb\_builder.mapping)@\spxentry{get\_mapping\_analytical\_samples()}\spxextra{in module graphdb\_builder.mapping}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/graphdb_builder:graphdb_builder.mapping.get_mapping_analytical_samples}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{graphdb\_builder.mapping.}}\sphinxbfcode{\sphinxupquote{get\_mapping\_analytical\_samples}}}{\emph{project\_id}}{}
\end{fulllineitems}



\subsection{Report Manager}
\label{\detokenize{_autosummary/report_manager:report-manager}}\label{\detokenize{_autosummary/report_manager::doc}}

\subsubsection{Report Dash Apps}
\label{\detokenize{_autosummary/report_manager.apps:report-dash-apps}}\label{\detokenize{_autosummary/report_manager.apps::doc}}

\paragraph{Basic App}
\label{\detokenize{_autosummary/report_manager.apps:basic-app}}

\paragraph{Data Upload App}
\label{\detokenize{_autosummary/report_manager.apps:data-upload-app}}

\paragraph{Data Upload}
\label{\detokenize{_autosummary/report_manager.apps:module-report_manager.apps.dataUpload}}\label{\detokenize{_autosummary/report_manager.apps:data-upload}}\index{report\_manager.apps.dataUpload (module)@\spxentry{report\_manager.apps.dataUpload}\spxextra{module}}\index{get\_data\_upload\_queries() (in module report\_manager.apps.dataUpload)@\spxentry{get\_data\_upload\_queries()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.get_data_upload_queries}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{get\_data\_upload\_queries}}}{}{}
Reads the YAML file containing the queries relevant to parsing of clinical data and     returns a Python object (dict{[}dict{]}).
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
Nested dictionary.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_new\_subject\_identifier() (in module report\_manager.apps.dataUpload)@\spxentry{get\_new\_subject\_identifier()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.get_new_subject_identifier}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{get\_new\_subject\_identifier}}}{\emph{driver}}{}
Queries the database for the last subject identifier and returns a new sequential identifier.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{projectId}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} external project identifier (from the graph database).

\end{itemize}

\item[{Returns}] \leavevmode
Subject identifier.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{str}

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_new\_biosample\_identifier() (in module report\_manager.apps.dataUpload)@\spxentry{get\_new\_biosample\_identifier()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.get_new_biosample_identifier}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{get\_new\_biosample\_identifier}}}{\emph{driver}}{}
Queries the database for the last biological sample internal identifier and returns a new sequential identifier.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{driver}} \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item[{Returns}] \leavevmode
Biological sample identifier.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_new\_analytical\_sample\_identifier() (in module report\_manager.apps.dataUpload)@\spxentry{get\_new\_analytical\_sample\_identifier()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.get_new_analytical_sample_identifier}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{get\_new\_analytical\_sample\_identifier}}}{\emph{driver}}{}
Queries the database for the last analytical sample internal identifier and returns a new sequential identifier.
:param driver: py2neo driver, which provides the connection to the neo4j graph database.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
Analytical sample identifier.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_subjects\_enrolled\_in\_project() (in module report\_manager.apps.dataUpload)@\spxentry{get\_subjects\_enrolled\_in\_project()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.get_subjects_enrolled_in_project}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{get\_subjects\_enrolled\_in\_project}}}{\emph{driver}, \emph{projectId}}{}
Extracts the number of subjects included in a given project.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{projectId}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} external project identifier (from the graph database).

\end{itemize}

\item[{Returns}] \leavevmode
Number of subjects.

\item[{Return type}] \leavevmode
Numpy ndarray

\end{description}\end{quote}

\end{fulllineitems}

\index{check\_samples\_in\_project() (in module report\_manager.apps.dataUpload)@\spxentry{check\_samples\_in\_project()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.check_samples_in_project}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{check\_samples\_in\_project}}}{\emph{driver}, \emph{projectId}}{}
\end{fulllineitems}

\index{check\_external\_ids\_in\_db() (in module report\_manager.apps.dataUpload)@\spxentry{check\_external\_ids\_in\_db()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.check_external_ids_in_db}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{check\_external\_ids\_in\_db}}}{\emph{driver}, \emph{projectId}}{}
\end{fulllineitems}

\index{remove\_samples\_nodes\_db() (in module report\_manager.apps.dataUpload)@\spxentry{remove\_samples\_nodes\_db()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.remove_samples_nodes_db}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{remove\_samples\_nodes\_db}}}{\emph{driver}, \emph{projectId}}{}
\end{fulllineitems}

\index{create\_new\_subjects() (in module report\_manager.apps.dataUpload)@\spxentry{create\_new\_subjects()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.create_new_subjects}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{create\_new\_subjects}}}{\emph{driver}, \emph{data}, \emph{projectId}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas Dataframe with clinical data as columns and samples as rows.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{projectId}} (\sphinxstyleliteralemphasis{\sphinxupquote{string}}) \textendash{} project identifier.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas DataFrame where new biological sample internal identifiers have been added.

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_new\_biosamples() (in module report\_manager.apps.dataUpload)@\spxentry{create\_new\_biosamples()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.create_new_biosamples}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{create\_new\_biosamples}}}{\emph{driver}, \emph{data}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas Dataframe with clinical data as columns and samples as rows.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas DataFrame where new biological sample internal identifiers have been added.

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_new\_ansamples() (in module report\_manager.apps.dataUpload)@\spxentry{create\_new\_ansamples()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.create_new_ansamples}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{create\_new\_ansamples}}}{\emph{driver}, \emph{data}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas Dataframe with clinical data as columns and samples as rows.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas DataFrame where new analytical sample internal identifiers have been added.

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_experiment\_internal\_identifiers() (in module report\_manager.apps.dataUpload)@\spxentry{create\_experiment\_internal\_identifiers()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.create_experiment_internal_identifiers}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{create\_experiment\_internal\_identifiers}}}{\emph{driver}, \emph{projectId}, \emph{data}, \emph{directory}, \emph{filename}}{}
\end{fulllineitems}

\index{create\_mapping\_cols\_clinical() (in module report\_manager.apps.dataUpload)@\spxentry{create\_mapping\_cols\_clinical()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.create_mapping_cols_clinical}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{create\_mapping\_cols\_clinical}}}{\emph{driver}, \emph{data}, \emph{directory}, \emph{filename}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas Dataframe with clinical data as columns and samples as rows.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{separator}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} character used to separate multiple entries in an attribute.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas Dataframe with all clinical data and graph database internal identifiers.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_project\_information() (in module report\_manager.apps.dataUpload)@\spxentry{get\_project\_information()}\spxextra{in module report\_manager.apps.dataUpload}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.dataUpload.get_project_information}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.dataUpload.}}\sphinxbfcode{\sphinxupquote{get\_project\_information}}}{\emph{driver}, \emph{project\_id}}{}
\end{fulllineitems}



\paragraph{HomePage Stats App}
\label{\detokenize{_autosummary/report_manager.apps:homepage-stats-app}}

\paragraph{HomePage Stats}
\label{\detokenize{_autosummary/report_manager.apps:module-report_manager.apps.homepageStats}}\label{\detokenize{_autosummary/report_manager.apps:homepage-stats}}\index{report\_manager.apps.homepageStats (module)@\spxentry{report\_manager.apps.homepageStats}\spxextra{module}}\index{size\_converter() (in module report\_manager.apps.homepageStats)@\spxentry{size\_converter()}\spxextra{in module report\_manager.apps.homepageStats}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.homepageStats.size_converter}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.homepageStats.}}\sphinxbfcode{\sphinxupquote{size\_converter}}}{\emph{value}}{}
Converts a given value to the highest possible unit, maintaining two decimals.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{or float value}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} 

\item[{Returns}] \leavevmode
String with converted value and units.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_query() (in module report\_manager.apps.homepageStats)@\spxentry{get\_query()}\spxextra{in module report\_manager.apps.homepageStats}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.homepageStats.get_query}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.homepageStats.}}\sphinxbfcode{\sphinxupquote{get\_query}}}{}{}~\begin{quote}

Reads the YAML file containing the queries relevant for graph database stats, parses the given stream and        returns a Python object (dict{[}dict{]}).
\end{quote}
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
Nested dictionary.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_db\_schema() (in module report\_manager.apps.homepageStats)@\spxentry{get\_db\_schema()}\spxextra{in module report\_manager.apps.homepageStats}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.homepageStats.get_db_schema}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.homepageStats.}}\sphinxbfcode{\sphinxupquote{get\_db\_schema}}}{}{}
Retrieves the database schema
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
network with all the database nodes and how they are related

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_db\_stats\_data() (in module report\_manager.apps.homepageStats)@\spxentry{get\_db\_stats\_data()}\spxextra{in module report\_manager.apps.homepageStats}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.homepageStats.get_db_stats_data}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.homepageStats.}}\sphinxbfcode{\sphinxupquote{get\_db\_stats\_data}}}{}{}
Retrieves all the stats data from the graph database and returns them as a dictionary.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
Dictionary of dataframes.

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_store\_size\_components() (in module report\_manager.apps.homepageStats)@\spxentry{plot\_store\_size\_components()}\spxextra{in module report\_manager.apps.homepageStats}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.homepageStats.plot_store_size_components}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.homepageStats.}}\sphinxbfcode{\sphinxupquote{plot\_store\_size\_components}}}{\emph{dfs}, \emph{title}, \emph{args}}{}
Plots the store size of different components of the graph database, as a Pie Chart.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dfs}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary of json objects.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} title of the Dash div where plot is located.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{valueCol} (str) \textendash{} name of the column with the values to be plotted.

\item {} 
\sphinxstylestrong{textCol} (str) \textendash{} name of the column containing information for the hoverinfo parameter.

\item {} 
\sphinxstylestrong{height} (str) \textendash{} height of the plot.

\item {} 
\sphinxstylestrong{width} (str) \textendash{} width of the plot.

\end{itemize}

\item[{Returns}] \leavevmode
New Dash div containing title and pie chart.

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_node\_rel\_per\_label() (in module report\_manager.apps.homepageStats)@\spxentry{plot\_node\_rel\_per\_label()}\spxextra{in module report\_manager.apps.homepageStats}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.homepageStats.plot_node_rel_per_label}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.homepageStats.}}\sphinxbfcode{\sphinxupquote{plot\_node\_rel\_per\_label}}}{\emph{dfs}, \emph{title}, \emph{args}, \emph{focus=\textquotesingle{}nodes\textquotesingle{}}}{}
Plots the number of nodes or relationships (depending on ‘focus’) per label, contained in the     grapha database.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dfs}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary of json objects.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} title of the Dash div where plot is located.

\end{itemize}

\item[{Paeam str focus}] \leavevmode
plot number of nodes per label (‘nodes’) or the number of relationships                                             per type (‘relationships’).

\item[{Returns}] \leavevmode
New Dash div containing title and barplot.

\end{description}\end{quote}

\end{fulllineitems}

\index{indicator() (in module report\_manager.apps.homepageStats)@\spxentry{indicator()}\spxextra{in module report\_manager.apps.homepageStats}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.homepageStats.indicator}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.homepageStats.}}\sphinxbfcode{\sphinxupquote{indicator}}}{\emph{color}, \emph{text}, \emph{id\_value}}{}
Builds a new Dash div styled as a container, with borders and background.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{color}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} background color of the container (RGB or Hex colors).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{text}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name to be plotted inside the container.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_value}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of the container.

\end{itemize}

\item[{Returns}] \leavevmode
Dash div containing title and an html.P element.

\end{description}\end{quote}

\end{fulllineitems}

\index{quick\_numbers\_panel() (in module report\_manager.apps.homepageStats)@\spxentry{quick\_numbers\_panel()}\spxextra{in module report\_manager.apps.homepageStats}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.homepageStats.quick_numbers_panel}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.homepageStats.}}\sphinxbfcode{\sphinxupquote{quick\_numbers\_panel}}}{}{}
Creates a panel of Dash containers where an overviem of the graph database numbers can be plotted.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
List of Dash components.

\end{description}\end{quote}

\end{fulllineitems}



\paragraph{DB Imports Stats App}
\label{\detokenize{_autosummary/report_manager.apps:db-imports-stats-app}}

\paragraph{DB Imports Stats}
\label{\detokenize{_autosummary/report_manager.apps:module-report_manager.apps.imports}}\label{\detokenize{_autosummary/report_manager.apps:db-imports-stats}}\index{report\_manager.apps.imports (module)@\spxentry{report\_manager.apps.imports}\spxextra{module}}\index{get\_stats\_data() (in module report\_manager.apps.imports)@\spxentry{get\_stats\_data()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.get_stats_data}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{get\_stats\_data}}}{\emph{filename}, \emph{n=3}}{}
Reads graph database stats file and filters for the last ‘n’ full and partial independent     imports, returning a Pandas DataFrame.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{filename}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} path to stats file (including filename and ‘.hdf’ extension).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of independent imports to plot.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas Dataframe with different entities and relationships as rows and columns:

\end{description}\end{quote}

\end{fulllineitems}

\index{select\_last\_n\_imports() (in module report\_manager.apps.imports)@\spxentry{select\_last\_n\_imports()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.select_last_n_imports}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{select\_last\_n\_imports}}}{\emph{stats\_file}, \emph{n=3}}{}
Selects which independent full and partial imports should be plotted based on n.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{stats\_file}} \textendash{} pandas DataFrame with stats data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of independent imports to select.

\end{itemize}

\item[{Returns}] \leavevmode
List of import ids to be plotted according to selection criterion.

\end{description}\end{quote}

\end{fulllineitems}

\index{remove\_legend\_duplicates() (in module report\_manager.apps.imports)@\spxentry{remove\_legend\_duplicates()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.remove_legend_duplicates}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{remove\_legend\_duplicates}}}{\emph{figure}}{}
Removes duplicated legend items.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{figure}} \textendash{} plotly graph object figure.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_databases\_entities\_relationships() (in module report\_manager.apps.imports)@\spxentry{get\_databases\_entities\_relationships()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.get_databases_entities_relationships}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{get\_databases\_entities\_relationships}}}{\emph{stats\_file}, \emph{key=\textquotesingle{}full\textquotesingle{}}, \emph{options=\textquotesingle{}databases\textquotesingle{}}}{}
Builds dictionary from stats file. Depending on ‘options’, keys and values can differ.     If \sphinxstyleemphasis{options} is set to ‘dates’, keys are dates of the imports and values are databases imported at each date;     if ‘databases’, keys are databases and values are entities and relationships created from each database;     if ‘entities’, keys are databases and values are entities created from each database;     if ‘relationships’, keys are databases and values are relationships created from each database.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{stats\_file}} \textendash{} pandas DataFrame with stats data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{key}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} use only full, partial or both kinds of imports (‘full’, ‘partial’, ‘all’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{options}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the variables to be used as keys in the output dictionary (‘dates’,                         ‘databases’, ‘entities’ or ‘relationships’).

\end{itemize}

\item[{Returns}] \leavevmode
Dictionary.

\end{description}\end{quote}

\end{fulllineitems}

\index{set\_colors() (in module report\_manager.apps.imports)@\spxentry{set\_colors()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.set_colors}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{set\_colors}}}{\emph{dictionary}}{}
This function takes the values in a dictionary and attributes them an RGB color.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{dictionary}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary with variables to be attributed a color, as values.

\item[{Returns}] \leavevmode
Dictionary where ‘dictionary’ values are keys and random RGB colors are the values.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_dropdown\_menu() (in module report\_manager.apps.imports)@\spxentry{get\_dropdown\_menu()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.get_dropdown_menu}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{get\_dropdown\_menu}}}{\emph{fig}, \emph{options\_dict}, \emph{add\_button=True}, \emph{equal\_traces=True}, \emph{number\_traces=2}}{}
Builds a list for the dropdown menu, based on a plotly figure traces and a dictionary with     the options to be used in the dropdown.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{fig}} \textendash{} plotly graph object figure.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{options\_dict}} \textendash{} dictionary where keys are used as dropdown options and values data points.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{add\_button}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} add option to display all dropdown options simultaneously.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{equal\_traces}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} defines if all dropdown options have the same number of traces each.                                 If True, define ‘number\_traces’ as well. If False, number of traces                                 will be the same as the number of values for each ‘options\_dict’ key.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{number\_traces}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of traces created for each ‘options\_dict’ key.

\end{itemize}

\item[{Returns}] \leavevmode
List of nested structures. Each dictionary within \sphinxstyleemphasis{updatemenus{[}0{]}{[}‘buttons’{]}{[}0{]}} corresponds             to one dropdown menu options and contains information on which traces are visible, label and method.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_totals\_per\_date() (in module report\_manager.apps.imports)@\spxentry{get\_totals\_per\_date()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.get_totals_per_date}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{get\_totals\_per\_date}}}{\emph{stats\_file}, \emph{key=\textquotesingle{}full\textquotesingle{}}, \emph{import\_types=False}}{}
Summarizes stats file to a Pandas DataFrame with import dates and total number of     imported entities and relationships.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{stats\_file}} \textendash{} pandas DataFrame with stats data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{key}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} use only full or partial imports (‘full’, ‘partial’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{import\_types}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} breakdown importing stats into entities or relationships related.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas DataFrame with independent import dates as rows and imported numbers as columns.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_imports\_per\_database\_date() (in module report\_manager.apps.imports)@\spxentry{get\_imports\_per\_database\_date()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.get_imports_per_database_date}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{get\_imports\_per\_database\_date}}}{\emph{stats\_file}}{}
Summarizes stats file to a Pandas DataFrame with import dates, databases and total number of     imported entities and relationships per database.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{stats\_file}} \textendash{} pandas DataFrame with stats data.

\item[{Returns}] \leavevmode
Pandas DataFrame with independent import dates and databases as rows and imported numbers as columns.

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_total\_number\_imported() (in module report\_manager.apps.imports)@\spxentry{plot\_total\_number\_imported()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.plot_total_number_imported}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{plot\_total\_number\_imported}}}{\emph{stats\_file}, \emph{plot\_title}}{}
Creates plot with overview of imports numbers per date.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{stats\_file}} \textendash{} pandas DataFrame with stats data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{plot\_title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} title of the plot.

\end{itemize}

\item[{Returns}] \leavevmode
Line plot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_total\_numbers\_per\_date() (in module report\_manager.apps.imports)@\spxentry{plot\_total\_numbers\_per\_date()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.plot_total_numbers_per_date}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{plot\_total\_numbers\_per\_date}}}{\emph{stats\_file}, \emph{plot\_title}}{}
Plots number of entities and relationships imported per date, with scaled markers reflecting numbers rations.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{stats\_file}} \textendash{} pandas DataFrame with stats data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{plot\_title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} title of the plot.

\end{itemize}

\item[{Returns}] \leavevmode
Scatter plot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}, with scalled markers.

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_databases\_numbers\_per\_date() (in module report\_manager.apps.imports)@\spxentry{plot\_databases\_numbers\_per\_date()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.plot_databases_numbers_per_date}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{plot\_databases\_numbers\_per\_date}}}{\emph{stats\_file}, \emph{plot\_title}, \emph{key=\textquotesingle{}full\textquotesingle{}}, \emph{dropdown=False}, \emph{dropdown\_options=\textquotesingle{}dates\textquotesingle{}}}{}
Grouped horizontal barplot showing the number of entities and relationships imported from each biomedical database.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{stats\_file}} \textendash{} pandas DataFrame with stats data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{plot\_title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} title of the plot.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{key}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} use only full or partial imports (‘full’, ‘partial’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dropdown}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} add dropdown menu to figure or not.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dropdown\_options}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the variables to be used as options in the dropdown menu (‘dates’,                         ‘databases’, ‘entities’ or ‘relationships’).

\end{itemize}

\item[{Returns}] \leavevmode
Horizontal barplot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_import\_numbers\_per\_database() (in module report\_manager.apps.imports)@\spxentry{plot\_import\_numbers\_per\_database()}\spxextra{in module report\_manager.apps.imports}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.imports.plot_import_numbers_per_database}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.imports.}}\sphinxbfcode{\sphinxupquote{plot\_import\_numbers\_per\_database}}}{\emph{stats\_file}, \emph{plot\_title}, \emph{key=\textquotesingle{}full\textquotesingle{}}, \emph{subplot\_titles=(\textquotesingle{}\textquotesingle{}}, \emph{\textquotesingle{}\textquotesingle{})}, \emph{colors=True}, \emph{plots\_1=\textquotesingle{}entities\textquotesingle{}}, \emph{plots\_2=\textquotesingle{}relationships\textquotesingle{}}, \emph{dropdown=True}, \emph{dropdown\_options=\textquotesingle{}databases\textquotesingle{}}}{}
Creates plotly multiplot figure with breakdown of imported numbers and size of the respective files, per database and     import type (entities or relationships).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{stats\_file}} \textendash{} pandas DataFrame with stats data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{plot\_title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} title of the plot.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{key}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} use only full or partial imports (‘full’, ‘partial’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subplot\_titles}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#tuple}{\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}}) \textendash{} title of the subplots (tuple of strings, one for each subplot).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{colors}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} define standard colors for entities and for relationships.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{plots\_1}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the variable plotted.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{plots\_2}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the variable plotted.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dropdown}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} add dropdown menu to figure or not.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dropdown\_options}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the variables to be used as options in the dropdown menu (‘dates’,                         ‘databases’, ‘entities’ or ‘relationships’).

\end{itemize}

\item[{Returns}] \leavevmode
Multi\sphinxhyphen{}scatterplot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

\end{fulllineitems}



\paragraph{Initial App}
\label{\detokenize{_autosummary/report_manager.apps:module-report_manager.apps.initialApp}}\label{\detokenize{_autosummary/report_manager.apps:initial-app}}\index{report\_manager.apps.initialApp (module)@\spxentry{report\_manager.apps.initialApp}\spxextra{module}}

\paragraph{Login App}
\label{\detokenize{_autosummary/report_manager.apps:login-app}}

\paragraph{Project App}
\label{\detokenize{_autosummary/report_manager.apps:project-app}}

\paragraph{Project Creation App}
\label{\detokenize{_autosummary/report_manager.apps:project-creation-app}}

\paragraph{Project Creation}
\label{\detokenize{_autosummary/report_manager.apps:module-report_manager.apps.projectCreation}}\label{\detokenize{_autosummary/report_manager.apps:project-creation}}\index{report\_manager.apps.projectCreation (module)@\spxentry{report\_manager.apps.projectCreation}\spxextra{module}}\index{get\_project\_creation\_queries() (in module report\_manager.apps.projectCreation)@\spxentry{get\_project\_creation\_queries()}\spxextra{in module report\_manager.apps.projectCreation}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.projectCreation.get_project_creation_queries}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.projectCreation.}}\sphinxbfcode{\sphinxupquote{get\_project\_creation\_queries}}}{}{}
Reads the YAML file containing the queries relevant to user creation, parses the given stream and     returns a Python object (dict{[}dict{]}).
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
Nested dictionary.

\end{description}\end{quote}

\end{fulllineitems}

\index{check\_if\_node\_exists() (in module report\_manager.apps.projectCreation)@\spxentry{check\_if\_node\_exists()}\spxextra{in module report\_manager.apps.projectCreation}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.projectCreation.check_if_node_exists}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.projectCreation.}}\sphinxbfcode{\sphinxupquote{check\_if\_node\_exists}}}{\emph{driver}, \emph{node}, \emph{node\_property}, \emph{value}}{}
Queries the graph database and checks if a node with a specific property and property value already exists.
:param driver: py2neo driver, which provides the connection to the neo4j graph database.
:type driver: py2neo driver
:param str node: node to be matched in the database.
:param str node\_property: property of the node.
:param value: property value.
:type value: str, int, float or bool
:return: Pandas dataframe with user identifier if User with node\_property and value already exists,             if User does not exist, returns and empty dataframe.

\end{fulllineitems}

\index{get\_new\_project\_identifier() (in module report\_manager.apps.projectCreation)@\spxentry{get\_new\_project\_identifier()}\spxextra{in module report\_manager.apps.projectCreation}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.projectCreation.get_new_project_identifier}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.projectCreation.}}\sphinxbfcode{\sphinxupquote{get\_new\_project\_identifier}}}{\emph{driver}, \emph{projectId}}{}
Queries the database for the last project external identifier and returns a new sequential identifier.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{projectId}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} internal project identifier (CPxxxxxxxxxxxx).

\end{itemize}

\item[{Returns}] \leavevmode
Project external identifier.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{str}

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_subject\_number\_in\_project() (in module report\_manager.apps.projectCreation)@\spxentry{get\_subject\_number\_in\_project()}\spxextra{in module report\_manager.apps.projectCreation}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.projectCreation.get_subject_number_in_project}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.projectCreation.}}\sphinxbfcode{\sphinxupquote{get\_subject\_number\_in\_project}}}{\emph{driver}, \emph{projectId}}{}
Extracts the number of subjects included in a given project.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{projectId}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} external project identifier (from the graph database).

\end{itemize}

\item[{Returns}] \leavevmode
Integer with the number of subjects.

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_new\_project() (in module report\_manager.apps.projectCreation)@\spxentry{create\_new\_project()}\spxextra{in module report\_manager.apps.projectCreation}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager.apps:report_manager.apps.projectCreation.create_new_project}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.apps.projectCreation.}}\sphinxbfcode{\sphinxupquote{create\_new\_project}}}{\emph{driver}, \emph{projectId}, \emph{data}, \emph{separator=\textquotesingle{}|\textquotesingle{}}}{}
Creates a new project in the graph database, following the steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Retrieves new project external identifier and creates project node and relationships in the graph database.

\item {} 
Creates subjects, timepoints and intervention nodes.

\item {} 
Saves all the entities and relationships to tab\sphinxhyphen{}delimited files.

\item {} 
Returns the number of projects created and the project external identifier.

\end{enumerate}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{driver}} (\sphinxstyleliteralemphasis{\sphinxupquote{py2neo driver}}) \textendash{} py2neo driver, which provides the connection to the neo4j graph database.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{projectId}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} internal project identifier (CPxxxxxxxxxxxx).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas Dataframe with project as row and other attributes as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{separator}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} character used to separate multiple entries in a project attribute.

\end{itemize}

\item[{Returns}] \leavevmode
Two strings: number of projects created and the project external identifier.

\end{description}\end{quote}

\end{fulllineitems}



\subsubsection{app.py}
\label{\detokenize{_autosummary/report_manager:app-py}}

\subsubsection{dataset.py}
\label{\detokenize{_autosummary/report_manager:module-report_manager.dataset}}\label{\detokenize{_autosummary/report_manager:dataset-py}}\index{report\_manager.dataset (module)@\spxentry{report\_manager.dataset}\spxextra{module}}\index{Dataset (class in report\_manager.dataset)@\spxentry{Dataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{Dataset}}}{\emph{identifier}, \emph{dataset\_type}, \emph{configuration=None}, \emph{data=\{\}}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: \sphinxhref{https://docs.python.org/3.6/library/functions.html\#object}{\sphinxcode{\sphinxupquote{object}}}
\index{identifier() (report\_manager.dataset.Dataset property)@\spxentry{identifier()}\spxextra{report\_manager.dataset.Dataset property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.identifier}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{identifier}}}
\end{fulllineitems}

\index{dataset\_type() (report\_manager.dataset.Dataset property)@\spxentry{dataset\_type()}\spxextra{report\_manager.dataset.Dataset property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.dataset_type}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{dataset\_type}}}
\end{fulllineitems}

\index{data() (report\_manager.dataset.Dataset property)@\spxentry{data()}\spxextra{report\_manager.dataset.Dataset property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.data}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{data}}}
\end{fulllineitems}

\index{analyses() (report\_manager.dataset.Dataset property)@\spxentry{analyses()}\spxextra{report\_manager.dataset.Dataset property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.analyses}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{analyses}}}
\end{fulllineitems}

\index{configuration() (report\_manager.dataset.Dataset property)@\spxentry{configuration()}\spxextra{report\_manager.dataset.Dataset property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.configuration}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{configuration}}}
\end{fulllineitems}

\index{analysis\_queries() (report\_manager.dataset.Dataset property)@\spxentry{analysis\_queries()}\spxextra{report\_manager.dataset.Dataset property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.analysis_queries}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{analysis\_queries}}}
\end{fulllineitems}

\index{report() (report\_manager.dataset.Dataset property)@\spxentry{report()}\spxextra{report\_manager.dataset.Dataset property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.report}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{report}}}
\end{fulllineitems}

\index{generate\_dataset() (report\_manager.dataset.Dataset method)@\spxentry{generate\_dataset()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.generate_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_dataset}}}{}{}
\end{fulllineitems}

\index{update\_report() (report\_manager.dataset.Dataset method)@\spxentry{update\_report()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.update_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_report}}}{\emph{report}}{}
\end{fulllineitems}

\index{update\_analysis\_queries() (report\_manager.dataset.Dataset method)@\spxentry{update\_analysis\_queries()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.update_analysis_queries}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_analysis\_queries}}}{\emph{query}}{}
\end{fulllineitems}

\index{get\_dataframe() (report\_manager.dataset.Dataset method)@\spxentry{get\_dataframe()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.get_dataframe}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dataframe}}}{\emph{dataset\_name}}{}
\end{fulllineitems}

\index{get\_dataframes() (report\_manager.dataset.Dataset method)@\spxentry{get\_dataframes()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.get_dataframes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dataframes}}}{\emph{dataset\_names}}{}
\end{fulllineitems}

\index{list\_dataframes() (report\_manager.dataset.Dataset method)@\spxentry{list\_dataframes()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.list_dataframes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{list\_dataframes}}}{}{}
\end{fulllineitems}

\index{get\_analysis() (report\_manager.dataset.Dataset method)@\spxentry{get\_analysis()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.get_analysis}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_analysis}}}{\emph{analysis}}{}
\end{fulllineitems}

\index{update\_data() (report\_manager.dataset.Dataset method)@\spxentry{update\_data()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.update_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_data}}}{\emph{new}}{}
\end{fulllineitems}

\index{update\_analyses() (report\_manager.dataset.Dataset method)@\spxentry{update\_analyses()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.update_analyses}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_analyses}}}{\emph{new}}{}
\end{fulllineitems}

\index{set\_configuration\_from\_file() (report\_manager.dataset.Dataset method)@\spxentry{set\_configuration\_from\_file()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.set_configuration_from_file}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_configuration\_from\_file}}}{\emph{configuration\_file}}{}
\end{fulllineitems}

\index{update\_configuration\_from\_file() (report\_manager.dataset.Dataset method)@\spxentry{update\_configuration\_from\_file()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.update_configuration_from_file}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_configuration\_from\_file}}}{\emph{configuration\_file}}{}
\end{fulllineitems}

\index{get\_dataset\_data\_directory() (report\_manager.dataset.Dataset method)@\spxentry{get\_dataset\_data\_directory()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.get_dataset_data_directory}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dataset\_data\_directory}}}{\emph{directory=\textquotesingle{}../../../data/reports\textquotesingle{}}}{}
\end{fulllineitems}

\index{query\_data() (report\_manager.dataset.Dataset method)@\spxentry{query\_data()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.query_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{query\_data}}}{}{}
\end{fulllineitems}

\index{send\_query() (report\_manager.dataset.Dataset method)@\spxentry{send\_query()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.send_query}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{send\_query}}}{\emph{query}}{}
\end{fulllineitems}

\index{extract\_configuration() (report\_manager.dataset.Dataset method)@\spxentry{extract\_configuration()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.extract_configuration}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{extract\_configuration}}}{\emph{configuration}}{}
\end{fulllineitems}

\index{add\_configuration\_to\_report() (report\_manager.dataset.Dataset method)@\spxentry{add\_configuration\_to\_report()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.add_configuration_to_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_configuration\_to\_report}}}{\emph{report\_pipeline}}{}
\end{fulllineitems}

\index{generate\_report() (report\_manager.dataset.Dataset method)@\spxentry{generate\_report()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.generate_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_report}}}{}{}
\end{fulllineitems}

\index{save\_dataset\_recursively() (report\_manager.dataset.Dataset method)@\spxentry{save\_dataset\_recursively()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.save_dataset_recursively}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_dataset\_recursively}}}{\emph{dset}, \emph{group}, \emph{dt}}{}
\end{fulllineitems}

\index{save\_dataset() (report\_manager.dataset.Dataset method)@\spxentry{save\_dataset()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.save_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_dataset}}}{\emph{dataset\_directory}}{}
\end{fulllineitems}

\index{save\_dataset\_recursively\_to\_file() (report\_manager.dataset.Dataset method)@\spxentry{save\_dataset\_recursively\_to\_file()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.save_dataset_recursively_to_file}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_dataset\_recursively\_to\_file}}}{\emph{dset}, \emph{dataset\_directory}, \emph{base\_name=\textquotesingle{}\textquotesingle{}}}{}
\end{fulllineitems}

\index{save\_dataset\_to\_file() (report\_manager.dataset.Dataset method)@\spxentry{save\_dataset\_to\_file()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.save_dataset_to_file}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_dataset\_to\_file}}}{\emph{dataset\_directory}}{}
\end{fulllineitems}

\index{save\_report() (report\_manager.dataset.Dataset method)@\spxentry{save\_report()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.save_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_report}}}{\emph{dataset\_directory}}{}
\end{fulllineitems}

\index{load\_dataset\_recursively() (report\_manager.dataset.Dataset method)@\spxentry{load\_dataset\_recursively()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.load_dataset_recursively}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_dataset\_recursively}}}{\emph{dset}, \emph{loaded\_dset=\{\}}}{}
\end{fulllineitems}

\index{load\_dataset() (report\_manager.dataset.Dataset method)@\spxentry{load\_dataset()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.load_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_dataset}}}{\emph{dataset\_directory}}{}
\end{fulllineitems}

\index{load\_dataset\_report() (report\_manager.dataset.Dataset method)@\spxentry{load\_dataset\_report()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.load_dataset_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_dataset\_report}}}{\emph{report\_dir}}{}
\end{fulllineitems}

\index{generate\_knowledge() (report\_manager.dataset.Dataset method)@\spxentry{generate\_knowledge()}\spxextra{report\_manager.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset.generate_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge}}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{MultiOmicsDataset (class in report\_manager.dataset)@\spxentry{MultiOmicsDataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.MultiOmicsDataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{MultiOmicsDataset}}}{\emph{identifier}, \emph{data}, \emph{configuration=None}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.dataset.Dataset}}}}}
\index{get\_dataframes() (report\_manager.dataset.MultiOmicsDataset method)@\spxentry{get\_dataframes()}\spxextra{report\_manager.dataset.MultiOmicsDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.MultiOmicsDataset.get_dataframes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dataframes}}}{\emph{datasets}}{}
\end{fulllineitems}

\index{generate\_knowledge() (report\_manager.dataset.MultiOmicsDataset method)@\spxentry{generate\_knowledge()}\spxextra{report\_manager.dataset.MultiOmicsDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.MultiOmicsDataset.generate_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge}}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{ProteomicsDataset (class in report\_manager.dataset)@\spxentry{ProteomicsDataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ProteomicsDataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{ProteomicsDataset}}}{\emph{identifier}, \emph{dataset\_type=\textquotesingle{}proteomics\textquotesingle{}}, \emph{data=\{\}}, \emph{configuration=None}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.dataset.Dataset}}}}}
\index{generate\_dataset() (report\_manager.dataset.ProteomicsDataset method)@\spxentry{generate\_dataset()}\spxextra{report\_manager.dataset.ProteomicsDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ProteomicsDataset.generate_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_dataset}}}{}{}
\end{fulllineitems}

\index{process\_dataset() (report\_manager.dataset.ProteomicsDataset method)@\spxentry{process\_dataset()}\spxextra{report\_manager.dataset.ProteomicsDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ProteomicsDataset.process_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{process\_dataset}}}{}{}
\end{fulllineitems}

\index{processing() (report\_manager.dataset.ProteomicsDataset method)@\spxentry{processing()}\spxextra{report\_manager.dataset.ProteomicsDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ProteomicsDataset.processing}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{processing}}}{}{}
\end{fulllineitems}

\index{generate\_knowledge() (report\_manager.dataset.ProteomicsDataset method)@\spxentry{generate\_knowledge()}\spxextra{report\_manager.dataset.ProteomicsDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ProteomicsDataset.generate_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge}}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{PTMDataset (class in report\_manager.dataset)@\spxentry{PTMDataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.PTMDataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{PTMDataset}}}{\emph{identifier}, \emph{data=\{\}}, \emph{configuration=None}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.dataset.ProteomicsDataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.dataset.ProteomicsDataset}}}}}

\end{fulllineitems}

\index{PhosphoproteomicsDataset (class in report\_manager.dataset)@\spxentry{PhosphoproteomicsDataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.PhosphoproteomicsDataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{PhosphoproteomicsDataset}}}{\emph{identifier}, \emph{data=\{\}}, \emph{configuration=None}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.dataset.PTMDataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.dataset.PTMDataset}}}}}

\end{fulllineitems}

\index{InteractomicsDataset (class in report\_manager.dataset)@\spxentry{InteractomicsDataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.InteractomicsDataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{InteractomicsDataset}}}{\emph{identifier}, \emph{data=\{\}}, \emph{configuration=None}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.dataset.ProteomicsDataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.dataset.ProteomicsDataset}}}}}

\end{fulllineitems}

\index{LongitudinalProteomicsDataset (class in report\_manager.dataset)@\spxentry{LongitudinalProteomicsDataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.LongitudinalProteomicsDataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{LongitudinalProteomicsDataset}}}{\emph{identifier}, \emph{data=\{\}}, \emph{configuration=None}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.dataset.ProteomicsDataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.dataset.ProteomicsDataset}}}}}

\end{fulllineitems}

\index{ClinicalDataset (class in report\_manager.dataset)@\spxentry{ClinicalDataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ClinicalDataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{ClinicalDataset}}}{\emph{identifier}, \emph{data=\{\}}, \emph{configuration=None}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.dataset.Dataset}}}}}
\index{generate\_dataset() (report\_manager.dataset.ClinicalDataset method)@\spxentry{generate\_dataset()}\spxextra{report\_manager.dataset.ClinicalDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ClinicalDataset.generate_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_dataset}}}{}{}
\end{fulllineitems}

\index{process\_dataset() (report\_manager.dataset.ClinicalDataset method)@\spxentry{process\_dataset()}\spxextra{report\_manager.dataset.ClinicalDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ClinicalDataset.process_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{process\_dataset}}}{}{}
\end{fulllineitems}

\index{widen\_original\_dataset() (report\_manager.dataset.ClinicalDataset method)@\spxentry{widen\_original\_dataset()}\spxextra{report\_manager.dataset.ClinicalDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ClinicalDataset.widen_original_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{widen\_original\_dataset}}}{}{}
\end{fulllineitems}

\index{processing() (report\_manager.dataset.ClinicalDataset method)@\spxentry{processing()}\spxextra{report\_manager.dataset.ClinicalDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ClinicalDataset.processing}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{processing}}}{}{}
\end{fulllineitems}

\index{generate\_knowledge() (report\_manager.dataset.ClinicalDataset method)@\spxentry{generate\_knowledge()}\spxextra{report\_manager.dataset.ClinicalDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.ClinicalDataset.generate_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge}}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{DNAseqDataset (class in report\_manager.dataset)@\spxentry{DNAseqDataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.DNAseqDataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{DNAseqDataset}}}{\emph{identifier}, \emph{dataset\_type}, \emph{data=\{\}}, \emph{configuration=None}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.dataset.Dataset}}}}}
\index{generate\_dataset() (report\_manager.dataset.DNAseqDataset method)@\spxentry{generate\_dataset()}\spxextra{report\_manager.dataset.DNAseqDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.DNAseqDataset.generate_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_dataset}}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{RNAseqDataset (class in report\_manager.dataset)@\spxentry{RNAseqDataset}\spxextra{class in report\_manager.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.RNAseqDataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.dataset.}}\sphinxbfcode{\sphinxupquote{RNAseqDataset}}}{\emph{identifier}, \emph{data=\{\}}, \emph{configuration=None}, \emph{analyses=\{\}}, \emph{analysis\_queries=\{\}}, \emph{report=None}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.dataset.Dataset}}}}}
\index{generate\_dataset() (report\_manager.dataset.RNAseqDataset method)@\spxentry{generate\_dataset()}\spxextra{report\_manager.dataset.RNAseqDataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.dataset.RNAseqDataset.generate_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_dataset}}}{}{}
\end{fulllineitems}


\end{fulllineitems}



\subsubsection{index.py}
\label{\detokenize{_autosummary/report_manager:index-py}}

\subsubsection{knowledge.py}
\label{\detokenize{_autosummary/report_manager:module-report_manager.knowledge}}\label{\detokenize{_autosummary/report_manager:knowledge-py}}\index{report\_manager.knowledge (module)@\spxentry{report\_manager.knowledge}\spxextra{module}}\index{Knowledge (class in report\_manager.knowledge)@\spxentry{Knowledge}\spxextra{class in report\_manager.knowledge}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.knowledge.}}\sphinxbfcode{\sphinxupquote{Knowledge}}}{\emph{identifier}, \emph{data}, \emph{nodes=\{\}}, \emph{relationships=\{\}}, \emph{queries\_file=None}, \emph{colors=\{\}}, \emph{graph=None}, \emph{report=\{\}}}{}
Bases: \sphinxhref{https://docs.python.org/3.6/library/functions.html\#object}{\sphinxcode{\sphinxupquote{object}}}
\index{identifier() (report\_manager.knowledge.Knowledge property)@\spxentry{identifier()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.identifier}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{identifier}}}
\end{fulllineitems}

\index{data() (report\_manager.knowledge.Knowledge property)@\spxentry{data()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.data}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{data}}}
\end{fulllineitems}

\index{entities() (report\_manager.knowledge.Knowledge property)@\spxentry{entities()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.entities}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{entities}}}
\end{fulllineitems}

\index{nodes() (report\_manager.knowledge.Knowledge property)@\spxentry{nodes()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.nodes}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{nodes}}}
\end{fulllineitems}

\index{relationships() (report\_manager.knowledge.Knowledge property)@\spxentry{relationships()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.relationships}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{relationships}}}
\end{fulllineitems}

\index{queries\_file() (report\_manager.knowledge.Knowledge property)@\spxentry{queries\_file()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.queries_file}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{queries\_file}}}
\end{fulllineitems}

\index{colors() (report\_manager.knowledge.Knowledge property)@\spxentry{colors()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.colors}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{colors}}}
\end{fulllineitems}

\index{default\_color() (report\_manager.knowledge.Knowledge property)@\spxentry{default\_color()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.default_color}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{default\_color}}}
\end{fulllineitems}

\index{report() (report\_manager.knowledge.Knowledge property)@\spxentry{report()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.report}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{report}}}
\end{fulllineitems}

\index{graph() (report\_manager.knowledge.Knowledge property)@\spxentry{graph()}\spxextra{report\_manager.knowledge.Knowledge property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.graph}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{graph}}}
\end{fulllineitems}

\index{generate\_knowledge\_from\_regulation() (report\_manager.knowledge.Knowledge method)@\spxentry{generate\_knowledge\_from\_regulation()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.generate_knowledge_from_regulation}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge\_from\_regulation}}}{\emph{entity}}{}
\end{fulllineitems}

\index{genreate\_knowledge\_from\_correlation() (report\_manager.knowledge.Knowledge method)@\spxentry{genreate\_knowledge\_from\_correlation()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.genreate_knowledge_from_correlation}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{genreate\_knowledge\_from\_correlation}}}{\emph{entity\_node1}, \emph{entity\_node2}, \emph{filter}, \emph{cutoff=0.5}}{}
\end{fulllineitems}

\index{generate\_knowledge\_from\_wgcna() (report\_manager.knowledge.Knowledge method)@\spxentry{generate\_knowledge\_from\_wgcna()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.generate_knowledge_from_wgcna}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge\_from\_wgcna}}}{\emph{data}, \emph{entity1}, \emph{entity2}, \emph{cutoff=0.2}}{}
\end{fulllineitems}

\index{generate\_knowledge\_from\_annotations() (report\_manager.knowledge.Knowledge method)@\spxentry{generate\_knowledge\_from\_annotations()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.generate_knowledge_from_annotations}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge\_from\_annotations}}}{\emph{entity1}, \emph{entity2}, \emph{filter=None}}{}
\end{fulllineitems}

\index{generate\_knowledge\_from\_similarity() (report\_manager.knowledge.Knowledge method)@\spxentry{generate\_knowledge\_from\_similarity()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.generate_knowledge_from_similarity}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge\_from\_similarity}}}{\emph{entity=\textquotesingle{}Project\textquotesingle{}}}{}
\end{fulllineitems}

\index{generate\_knowledge\_from\_queries() (report\_manager.knowledge.Knowledge method)@\spxentry{generate\_knowledge\_from\_queries()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.generate_knowledge_from_queries}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge\_from\_queries}}}{\emph{entity}, \emph{queries\_results}}{}
\end{fulllineitems}

\index{send\_query() (report\_manager.knowledge.Knowledge method)@\spxentry{send\_query()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.send_query}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{send\_query}}}{\emph{query}}{}
\end{fulllineitems}

\index{query\_data() (report\_manager.knowledge.Knowledge method)@\spxentry{query\_data()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.query_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{query\_data}}}{\emph{replace}}{}
\end{fulllineitems}

\index{generate\_cypher\_nodes\_list() (report\_manager.knowledge.Knowledge method)@\spxentry{generate\_cypher\_nodes\_list()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.generate_cypher_nodes_list}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_cypher\_nodes\_list}}}{}{}
\end{fulllineitems}

\index{generate\_knowledge\_graph() (report\_manager.knowledge.Knowledge method)@\spxentry{generate\_knowledge\_graph()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.generate_knowledge_graph}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge\_graph}}}{}{}
\end{fulllineitems}

\index{reduce\_to\_subgraph() (report\_manager.knowledge.Knowledge method)@\spxentry{reduce\_to\_subgraph()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.reduce_to_subgraph}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{reduce\_to\_subgraph}}}{\emph{nodes}}{}
\end{fulllineitems}

\index{get\_knowledge\_graph\_plot() (report\_manager.knowledge.Knowledge method)@\spxentry{get\_knowledge\_graph\_plot()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.get_knowledge_graph_plot}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_knowledge\_graph\_plot}}}{}{}
\end{fulllineitems}

\index{generate\_report() (report\_manager.knowledge.Knowledge method)@\spxentry{generate\_report()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.generate_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_report}}}{\emph{visualization=\textquotesingle{}sankey\textquotesingle{}}}{}
\end{fulllineitems}

\index{save\_report() (report\_manager.knowledge.Knowledge method)@\spxentry{save\_report()}\spxextra{report\_manager.knowledge.Knowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge.save_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_report}}}{\emph{directory}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{ProjectKnowledge (class in report\_manager.knowledge)@\spxentry{ProjectKnowledge}\spxextra{class in report\_manager.knowledge}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.ProjectKnowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.knowledge.}}\sphinxbfcode{\sphinxupquote{ProjectKnowledge}}}{\emph{identifier}, \emph{data}, \emph{nodes=\{\}}, \emph{relationships=\{\}}, \emph{colors=\{\}}, \emph{graph=None}, \emph{report=\{\}}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.knowledge.Knowledge}}}}}
\index{generate\_knowledge() (report\_manager.knowledge.ProjectKnowledge method)@\spxentry{generate\_knowledge()}\spxextra{report\_manager.knowledge.ProjectKnowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.ProjectKnowledge.generate_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge}}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{ProteomicsKnowledge (class in report\_manager.knowledge)@\spxentry{ProteomicsKnowledge}\spxextra{class in report\_manager.knowledge}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.ProteomicsKnowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.knowledge.}}\sphinxbfcode{\sphinxupquote{ProteomicsKnowledge}}}{\emph{identifier}, \emph{data}, \emph{nodes=\{\}}, \emph{relationships=\{\}}, \emph{colors=\{\}}, \emph{graph=None}, \emph{report=\{\}}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.knowledge.Knowledge}}}}}
\index{generate\_knowledge() (report\_manager.knowledge.ProteomicsKnowledge method)@\spxentry{generate\_knowledge()}\spxextra{report\_manager.knowledge.ProteomicsKnowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.ProteomicsKnowledge.generate_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge}}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{ClinicalKnowledge (class in report\_manager.knowledge)@\spxentry{ClinicalKnowledge}\spxextra{class in report\_manager.knowledge}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.ClinicalKnowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.knowledge.}}\sphinxbfcode{\sphinxupquote{ClinicalKnowledge}}}{\emph{identifier}, \emph{data}, \emph{nodes=\{\}}, \emph{relationships=\{\}}, \emph{colors=\{\}}, \emph{graph=None}, \emph{report=\{\}}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.knowledge.Knowledge}}}}}
\index{generate\_knowledge() (report\_manager.knowledge.ClinicalKnowledge method)@\spxentry{generate\_knowledge()}\spxextra{report\_manager.knowledge.ClinicalKnowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.ClinicalKnowledge.generate_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge}}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{MultiOmicsKnowledge (class in report\_manager.knowledge)@\spxentry{MultiOmicsKnowledge}\spxextra{class in report\_manager.knowledge}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.MultiOmicsKnowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.knowledge.}}\sphinxbfcode{\sphinxupquote{MultiOmicsKnowledge}}}{\emph{identifier}, \emph{data}, \emph{nodes=\{\}}, \emph{relationships=\{\}}, \emph{colors=\{\}}, \emph{graph=None}, \emph{report=\{\}}}{}
Bases: {\hyperref[\detokenize{_autosummary/report_manager:report_manager.knowledge.Knowledge}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{report\_manager.knowledge.Knowledge}}}}}
\index{generate\_knowledge() (report\_manager.knowledge.MultiOmicsKnowledge method)@\spxentry{generate\_knowledge()}\spxextra{report\_manager.knowledge.MultiOmicsKnowledge method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.knowledge.MultiOmicsKnowledge.generate_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge}}}{}{}
\end{fulllineitems}


\end{fulllineitems}



\subsubsection{project.py}
\label{\detokenize{_autosummary/report_manager:module-report_manager.project}}\label{\detokenize{_autosummary/report_manager:project-py}}\index{report\_manager.project (module)@\spxentry{report\_manager.project}\spxextra{module}}\index{Project (class in report\_manager.project)@\spxentry{Project}\spxextra{class in report\_manager.project}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.project.}}\sphinxbfcode{\sphinxupquote{Project}}}{\emph{identifier}, \emph{configuration\_files=\{\}}, \emph{datasets=\{\}}, \emph{knowledge=None}, \emph{report=\{\}}}{}
Bases: \sphinxhref{https://docs.python.org/3.6/library/functions.html\#object}{\sphinxcode{\sphinxupquote{object}}}

A project class that defines an experimental project.
A project can be of different types, contain several datasets and reports.

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{p} \PYG{o}{=} \PYG{n}{Project}\PYG{p}{(}\PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{P0000001}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{datasets}\PYG{o}{=}\PYG{k+kc}{None}\PYG{p}{,} \PYG{n}{report}\PYG{o}{=}\PYG{k+kc}{None}\PYG{p}{)}
\PYG{n}{p}\PYG{o}{.}\PYG{n}{show\PYGZus{}report}\PYG{p}{(}\PYG{n}{environment}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{notebook}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}
\index{identifier() (report\_manager.project.Project property)@\spxentry{identifier()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.identifier}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{identifier}}}
\end{fulllineitems}

\index{configuration\_files() (report\_manager.project.Project property)@\spxentry{configuration\_files()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.configuration_files}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{configuration\_files}}}
\end{fulllineitems}

\index{queries\_file() (report\_manager.project.Project property)@\spxentry{queries\_file()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.queries_file}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{queries\_file}}}
\end{fulllineitems}

\index{name() (report\_manager.project.Project property)@\spxentry{name()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.name}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{name}}}
\end{fulllineitems}

\index{acronym() (report\_manager.project.Project property)@\spxentry{acronym()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.acronym}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{acronym}}}
\end{fulllineitems}

\index{data\_types() (report\_manager.project.Project property)@\spxentry{data\_types()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.data_types}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{data\_types}}}
\end{fulllineitems}

\index{append\_data\_type() (report\_manager.project.Project method)@\spxentry{append\_data\_type()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.append_data_type}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{append\_data\_type}}}{\emph{data\_type}}{}
\end{fulllineitems}

\index{responsible() (report\_manager.project.Project property)@\spxentry{responsible()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.responsible}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{responsible}}}
\end{fulllineitems}

\index{description() (report\_manager.project.Project property)@\spxentry{description()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.description}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{description}}}
\end{fulllineitems}

\index{status() (report\_manager.project.Project property)@\spxentry{status()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.status}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{status}}}
\end{fulllineitems}

\index{num\_subjects() (report\_manager.project.Project property)@\spxentry{num\_subjects()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.num_subjects}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{num\_subjects}}}
\end{fulllineitems}

\index{datasets() (report\_manager.project.Project property)@\spxentry{datasets()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.datasets}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{datasets}}}
\end{fulllineitems}

\index{knowledge() (report\_manager.project.Project property)@\spxentry{knowledge()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.knowledge}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{knowledge}}}
\end{fulllineitems}

\index{report() (report\_manager.project.Project property)@\spxentry{report()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.report}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{report}}}
\end{fulllineitems}

\index{similar\_projects() (report\_manager.project.Project property)@\spxentry{similar\_projects()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.similar_projects}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{similar\_projects}}}
\end{fulllineitems}

\index{overlap() (report\_manager.project.Project property)@\spxentry{overlap()}\spxextra{report\_manager.project.Project property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.overlap}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{overlap}}}
\end{fulllineitems}

\index{get\_dataset() (report\_manager.project.Project method)@\spxentry{get\_dataset()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.get_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dataset}}}{\emph{dataset}}{}
\end{fulllineitems}

\index{update\_dataset() (report\_manager.project.Project method)@\spxentry{update\_dataset()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.update_dataset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_dataset}}}{\emph{dataset}}{}
\end{fulllineitems}

\index{update\_report() (report\_manager.project.Project method)@\spxentry{update\_report()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.update_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_report}}}{\emph{new}}{}
\end{fulllineitems}

\index{remove\_project() (report\_manager.project.Project method)@\spxentry{remove\_project()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.remove_project}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{remove\_project}}}{\emph{host=\textquotesingle{}localhost\textquotesingle{}}, \emph{port=7687}, \emph{user=\textquotesingle{}neo4j\textquotesingle{}}, \emph{password=\textquotesingle{}password\textquotesingle{}}}{}
\end{fulllineitems}

\index{get\_report\_directory() (report\_manager.project.Project method)@\spxentry{get\_report\_directory()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.get_report_directory}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_report\_directory}}}{}{}
\end{fulllineitems}

\index{get\_downloads\_directory() (report\_manager.project.Project method)@\spxentry{get\_downloads\_directory()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.get_downloads_directory}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_downloads\_directory}}}{}{}
\end{fulllineitems}

\index{set\_attributes() (report\_manager.project.Project method)@\spxentry{set\_attributes()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.set_attributes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_attributes}}}{\emph{project\_info}}{}
\end{fulllineitems}

\index{from\_dict() (report\_manager.project.Project method)@\spxentry{from\_dict()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.from_dict}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{from\_dict}}}{\emph{attributes}}{}
\end{fulllineitems}

\index{to\_dict() (report\_manager.project.Project method)@\spxentry{to\_dict()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.to_dict}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{to\_dict}}}{}{}
\end{fulllineitems}

\index{to\_dataframe() (report\_manager.project.Project method)@\spxentry{to\_dataframe()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.to_dataframe}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{to\_dataframe}}}{}{}
\end{fulllineitems}

\index{list\_datasets() (report\_manager.project.Project method)@\spxentry{list\_datasets()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.list_datasets}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{list\_datasets}}}{}{}
\end{fulllineitems}

\index{to\_json() (report\_manager.project.Project method)@\spxentry{to\_json()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.to_json}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{to\_json}}}{}{}
\end{fulllineitems}

\index{from\_json() (report\_manager.project.Project method)@\spxentry{from\_json()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.from_json}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{from\_json}}}{\emph{json\_str}}{}
\end{fulllineitems}

\index{query\_data() (report\_manager.project.Project method)@\spxentry{query\_data()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.query_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{query\_data}}}{}{}
\end{fulllineitems}

\index{check\_report\_exists() (report\_manager.project.Project method)@\spxentry{check\_report\_exists()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.check_report_exists}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_report\_exists}}}{}{}
\end{fulllineitems}

\index{load\_project\_report() (report\_manager.project.Project method)@\spxentry{load\_project\_report()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.load_project_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_project\_report}}}{}{}
\end{fulllineitems}

\index{load\_project() (report\_manager.project.Project method)@\spxentry{load\_project()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.load_project}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_project}}}{\emph{directory}}{}
\end{fulllineitems}

\index{load\_project\_data() (report\_manager.project.Project method)@\spxentry{load\_project\_data()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.load_project_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_project\_data}}}{}{}
\end{fulllineitems}

\index{build\_project() (report\_manager.project.Project method)@\spxentry{build\_project()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.build_project}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{build\_project}}}{\emph{force=False}}{}
\end{fulllineitems}

\index{get\_projects\_overlap() (report\_manager.project.Project method)@\spxentry{get\_projects\_overlap()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.get_projects_overlap}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_projects\_overlap}}}{\emph{project\_info}}{}
\end{fulllineitems}

\index{get\_similar\_projects() (report\_manager.project.Project method)@\spxentry{get\_similar\_projects()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.get_similar_projects}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_similar\_projects}}}{\emph{project\_info}}{}
\end{fulllineitems}

\index{generate\_project\_attributes\_plot() (report\_manager.project.Project method)@\spxentry{generate\_project\_attributes\_plot()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.generate_project_attributes_plot}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_project\_attributes\_plot}}}{}{}
\end{fulllineitems}

\index{generate\_project\_similarity\_plots() (report\_manager.project.Project method)@\spxentry{generate\_project\_similarity\_plots()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.generate_project_similarity_plots}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_project\_similarity\_plots}}}{}{}
\end{fulllineitems}

\index{generate\_overlap\_plots() (report\_manager.project.Project method)@\spxentry{generate\_overlap\_plots()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.generate_overlap_plots}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_overlap\_plots}}}{}{}
\end{fulllineitems}

\index{get\_similarity\_network\_style() (report\_manager.project.Project method)@\spxentry{get\_similarity\_network\_style()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.get_similarity_network_style}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_similarity\_network\_style}}}{}{}
\end{fulllineitems}

\index{get\_similarity\_network() (report\_manager.project.Project method)@\spxentry{get\_similarity\_network()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.get_similarity_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_similarity\_network}}}{}{}
\end{fulllineitems}

\index{generate\_knowledge() (report\_manager.project.Project method)@\spxentry{generate\_knowledge()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.generate_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_knowledge}}}{}{}
\end{fulllineitems}

\index{generate\_project\_info\_report() (report\_manager.project.Project method)@\spxentry{generate\_project\_info\_report()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.generate_project_info_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_project\_info\_report}}}{}{}
\end{fulllineitems}

\index{generate\_report() (report\_manager.project.Project method)@\spxentry{generate\_report()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.generate_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_report}}}{}{}
\end{fulllineitems}

\index{notify\_project\_ready() (report\_manager.project.Project method)@\spxentry{notify\_project\_ready()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.notify_project_ready}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{notify\_project\_ready}}}{\emph{message\_type=\textquotesingle{}slack\textquotesingle{}}}{}
\end{fulllineitems}

\index{empty\_report() (report\_manager.project.Project method)@\spxentry{empty\_report()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.empty_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{empty\_report}}}{}{}
\end{fulllineitems}

\index{save\_project\_report() (report\_manager.project.Project method)@\spxentry{save\_project\_report()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.save_project_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_project\_report}}}{}{}
\end{fulllineitems}

\index{save\_project\_datasets\_reports() (report\_manager.project.Project method)@\spxentry{save\_project\_datasets\_reports()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.save_project_datasets_reports}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_project\_datasets\_reports}}}{}{}
\end{fulllineitems}

\index{save\_project() (report\_manager.project.Project method)@\spxentry{save\_project()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.save_project}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_project}}}{}{}
\end{fulllineitems}

\index{save\_project\_datasets\_data() (report\_manager.project.Project method)@\spxentry{save\_project\_datasets\_data()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.save_project_datasets_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_project\_datasets\_data}}}{}{}
\end{fulllineitems}

\index{show\_report() (report\_manager.project.Project method)@\spxentry{show\_report()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.show_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{show\_report}}}{\emph{environment}}{}
\end{fulllineitems}

\index{download\_project() (report\_manager.project.Project method)@\spxentry{download\_project()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.download_project}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{download\_project}}}{}{}
\end{fulllineitems}

\index{download\_project\_report() (report\_manager.project.Project method)@\spxentry{download\_project\_report()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.download_project_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{download\_project\_report}}}{}{}
\end{fulllineitems}

\index{download\_knowledge() (report\_manager.project.Project method)@\spxentry{download\_knowledge()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.download_knowledge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{download\_knowledge}}}{\emph{directory}}{}
\end{fulllineitems}

\index{download\_project\_datasets() (report\_manager.project.Project method)@\spxentry{download\_project\_datasets()}\spxextra{report\_manager.project.Project method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.project.Project.download_project_datasets}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{download\_project\_datasets}}}{}{}
\end{fulllineitems}


\end{fulllineitems}



\subsubsection{report.py}
\label{\detokenize{_autosummary/report_manager:module-report_manager.report}}\label{\detokenize{_autosummary/report_manager:report-py}}\index{report\_manager.report (module)@\spxentry{report\_manager.report}\spxextra{module}}\index{Report (class in report\_manager.report)@\spxentry{Report}\spxextra{class in report\_manager.report}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.report.}}\sphinxbfcode{\sphinxupquote{Report}}}{\emph{identifier}, \emph{plots=\{\}}}{}
Bases: \sphinxhref{https://docs.python.org/3.6/library/functions.html\#object}{\sphinxcode{\sphinxupquote{object}}}
\index{identifier() (report\_manager.report.Report property)@\spxentry{identifier()}\spxextra{report\_manager.report.Report property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.identifier}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{identifier}}}
\end{fulllineitems}

\index{plots() (report\_manager.report.Report property)@\spxentry{plots()}\spxextra{report\_manager.report.Report property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.plots}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{plots}}}
\end{fulllineitems}

\index{get\_plot() (report\_manager.report.Report method)@\spxentry{get\_plot()}\spxextra{report\_manager.report.Report method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.get_plot}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_plot}}}{\emph{plot}}{}
\end{fulllineitems}

\index{update\_plots() (report\_manager.report.Report method)@\spxentry{update\_plots()}\spxextra{report\_manager.report.Report method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.update_plots}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_plots}}}{\emph{plot}}{}
\end{fulllineitems}

\index{list\_plots() (report\_manager.report.Report method)@\spxentry{list\_plots()}\spxextra{report\_manager.report.Report method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.list_plots}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{list\_plots}}}{}{}
\end{fulllineitems}

\index{print\_report() (report\_manager.report.Report method)@\spxentry{print\_report()}\spxextra{report\_manager.report.Report method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.print_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{print\_report}}}{\emph{directory}, \emph{plot\_format=\textquotesingle{}pdf\textquotesingle{}}}{}
\end{fulllineitems}

\index{save\_report() (report\_manager.report.Report method)@\spxentry{save\_report()}\spxextra{report\_manager.report.Report method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.save_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_report}}}{\emph{directory}}{}
\end{fulllineitems}

\index{read\_report() (report\_manager.report.Report method)@\spxentry{read\_report()}\spxextra{report\_manager.report.Report method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.read_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{read\_report}}}{\emph{directory}}{}
\end{fulllineitems}

\index{visualize\_report() (report\_manager.report.Report method)@\spxentry{visualize\_report()}\spxextra{report\_manager.report.Report method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.visualize_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{visualize\_report}}}{\emph{environment}}{}
\end{fulllineitems}

\index{visualize\_plot() (report\_manager.report.Report method)@\spxentry{visualize\_plot()}\spxextra{report\_manager.report.Report method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.visualize_plot}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{visualize\_plot}}}{\emph{environment}, \emph{plot\_type}}{}
\end{fulllineitems}

\index{download\_report() (report\_manager.report.Report method)@\spxentry{download\_report()}\spxextra{report\_manager.report.Report method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.report.Report.download_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{download\_report}}}{\emph{directory}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsubsection{user.py}
\label{\detokenize{_autosummary/report_manager:module-report_manager.user}}\label{\detokenize{_autosummary/report_manager:user-py}}\index{report\_manager.user (module)@\spxentry{report\_manager.user}\spxextra{module}}\index{User (class in report\_manager.user)@\spxentry{User}\spxextra{class in report\_manager.user}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.user.User}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.user.}}\sphinxbfcode{\sphinxupquote{User}}}{\emph{username}}{}
Bases: \sphinxhref{https://docs.python.org/3.6/library/functions.html\#object}{\sphinxcode{\sphinxupquote{object}}}
\index{find() (report\_manager.user.User method)@\spxentry{find()}\spxextra{report\_manager.user.User method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.user.User.find}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{find}}}{}{}
\end{fulllineitems}

\index{register() (report\_manager.user.User method)@\spxentry{register()}\spxextra{report\_manager.user.User method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.user.User.register}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{register}}}{\emph{password}}{}
\end{fulllineitems}

\index{verify\_password() (report\_manager.user.User method)@\spxentry{verify\_password()}\spxextra{report\_manager.user.User method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.user.User.verify_password}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{verify\_password}}}{\emph{password}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsubsection{utils.py}
\label{\detokenize{_autosummary/report_manager:module-report_manager.utils}}\label{\detokenize{_autosummary/report_manager:utils-py}}\index{report\_manager.utils (module)@\spxentry{report\_manager.utils}\spxextra{module}}\index{copy\_file\_to\_destination() (in module report\_manager.utils)@\spxentry{copy\_file\_to\_destination()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.copy_file_to_destination}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{copy\_file\_to\_destination}}}{\emph{cfile}, \emph{destination}}{}
\end{fulllineitems}

\index{send\_message\_to\_slack\_webhook() (in module report\_manager.utils)@\spxentry{send\_message\_to\_slack\_webhook()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.send_message_to_slack_webhook}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{send\_message\_to\_slack\_webhook}}}{\emph{message}, \emph{message\_to}, \emph{username=\textquotesingle{}albsantosdel\textquotesingle{}}}{}
\end{fulllineitems}

\index{send\_email() (in module report\_manager.utils)@\spxentry{send\_email()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.send_email}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{send\_email}}}{\emph{message}, \emph{subject}, \emph{message\_from}, \emph{message\_to}}{}
\end{fulllineitems}

\index{compress\_directory() (in module report\_manager.utils)@\spxentry{compress\_directory()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.compress_directory}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{compress\_directory}}}{\emph{name}, \emph{directory}, \emph{compression\_format=\textquotesingle{}zip\textquotesingle{}}}{}
\end{fulllineitems}

\index{get\_markdown\_date() (in module report\_manager.utils)@\spxentry{get\_markdown\_date()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.get_markdown_date}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{get\_markdown\_date}}}{\emph{extra\_text}}{}
\end{fulllineitems}

\index{convert\_html\_to\_dash() (in module report\_manager.utils)@\spxentry{convert\_html\_to\_dash()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.convert_html_to_dash}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{convert\_html\_to\_dash}}}{\emph{el}, \emph{style=None}}{}
\end{fulllineitems}

\index{extract\_style() (in module report\_manager.utils)@\spxentry{extract\_style()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.extract_style}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{extract\_style}}}{\emph{el}}{}
\end{fulllineitems}

\index{is\_jsonable() (in module report\_manager.utils)@\spxentry{is\_jsonable()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.is_jsonable}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{is\_jsonable}}}{\emph{x}}{}
\end{fulllineitems}

\index{convert\_dash\_to\_json() (in module report\_manager.utils)@\spxentry{convert\_dash\_to\_json()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.convert_dash_to_json}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{convert\_dash\_to\_json}}}{\emph{dash\_object}}{}
\end{fulllineitems}

\index{get\_image() (in module report\_manager.utils)@\spxentry{get\_image()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.get_image}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{get\_image}}}{\emph{figure}, \emph{width}, \emph{height}}{}
\end{fulllineitems}

\index{parse\_html() (in module report\_manager.utils)@\spxentry{parse\_html()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.parse_html}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{parse\_html}}}{\emph{html\_snippet}}{}
\end{fulllineitems}

\index{hex2rgb() (in module report\_manager.utils)@\spxentry{hex2rgb()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.hex2rgb}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{hex2rgb}}}{\emph{color}}{}
\end{fulllineitems}

\index{getNumberText() (in module report\_manager.utils)@\spxentry{getNumberText()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.getNumberText}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{getNumberText}}}{\emph{num}}{}
\end{fulllineitems}

\index{get\_rgb\_colors() (in module report\_manager.utils)@\spxentry{get\_rgb\_colors()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.get_rgb_colors}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{get\_rgb\_colors}}}{\emph{n}}{}
\end{fulllineitems}

\index{get\_hex\_colors() (in module report\_manager.utils)@\spxentry{get\_hex\_colors()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.get_hex_colors}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{get\_hex\_colors}}}{\emph{n}}{}
\end{fulllineitems}

\index{convert\_html\_to\_pdf() (in module report\_manager.utils)@\spxentry{convert\_html\_to\_pdf()}\spxextra{in module report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.convert_html_to_pdf}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{convert\_html\_to\_pdf}}}{\emph{source\_html}, \emph{output\_filename}}{}
\end{fulllineitems}

\index{NumpyEncoder (class in report\_manager.utils)@\spxentry{NumpyEncoder}\spxextra{class in report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.NumpyEncoder}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{NumpyEncoder}}}{\emph{*}, \emph{skipkeys=False}, \emph{ensure\_ascii=True}, \emph{check\_circular=True}, \emph{allow\_nan=True}, \emph{sort\_keys=False}, \emph{indent=None}, \emph{separators=None}, \emph{default=None}}{}
Bases: \sphinxcode{\sphinxupquote{json.encoder.JSONEncoder}}
\index{default() (report\_manager.utils.NumpyEncoder method)@\spxentry{default()}\spxextra{report\_manager.utils.NumpyEncoder method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.NumpyEncoder.default}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{default}}}{\emph{obj}}{}
Implement this method in a subclass such that it returns
a serializable object for \sphinxcode{\sphinxupquote{o}}, or calls the base implementation
(to raise a \sphinxcode{\sphinxupquote{TypeError}}).

For example, to support arbitrary iterators, you could
implement default like this:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{def} \PYG{n+nf}{default}\PYG{p}{(}\PYG{n+nb+bp}{self}\PYG{p}{,} \PYG{n}{o}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{try}\PYG{p}{:}
        \PYG{n}{iterable} \PYG{o}{=} \PYG{n+nb}{iter}\PYG{p}{(}\PYG{n}{o}\PYG{p}{)}
    \PYG{k}{except} \PYG{n+ne}{TypeError}\PYG{p}{:}
        \PYG{k}{pass}
    \PYG{k}{else}\PYG{p}{:}
        \PYG{k}{return} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{iterable}\PYG{p}{)}
    \PYG{c+c1}{\PYGZsh{} Let the base class default method raise the TypeError}
    \PYG{k}{return} \PYG{n}{JSONEncoder}\PYG{o}{.}\PYG{n}{default}\PYG{p}{(}\PYG{n+nb+bp}{self}\PYG{p}{,} \PYG{n}{o}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}


\end{fulllineitems}

\index{DictDFEncoder (class in report\_manager.utils)@\spxentry{DictDFEncoder}\spxextra{class in report\_manager.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.DictDFEncoder}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{report\_manager.utils.}}\sphinxbfcode{\sphinxupquote{DictDFEncoder}}}{\emph{*}, \emph{skipkeys=False}, \emph{ensure\_ascii=True}, \emph{check\_circular=True}, \emph{allow\_nan=True}, \emph{sort\_keys=False}, \emph{indent=None}, \emph{separators=None}, \emph{default=None}}{}
Bases: \sphinxcode{\sphinxupquote{json.encoder.JSONEncoder}}
\index{default() (report\_manager.utils.DictDFEncoder method)@\spxentry{default()}\spxextra{report\_manager.utils.DictDFEncoder method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/report_manager:report_manager.utils.DictDFEncoder.default}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{default}}}{\emph{obj}}{}
Implement this method in a subclass such that it returns
a serializable object for \sphinxcode{\sphinxupquote{o}}, or calls the base implementation
(to raise a \sphinxcode{\sphinxupquote{TypeError}}).

For example, to support arbitrary iterators, you could
implement default like this:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{def} \PYG{n+nf}{default}\PYG{p}{(}\PYG{n+nb+bp}{self}\PYG{p}{,} \PYG{n}{o}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{try}\PYG{p}{:}
        \PYG{n}{iterable} \PYG{o}{=} \PYG{n+nb}{iter}\PYG{p}{(}\PYG{n}{o}\PYG{p}{)}
    \PYG{k}{except} \PYG{n+ne}{TypeError}\PYG{p}{:}
        \PYG{k}{pass}
    \PYG{k}{else}\PYG{p}{:}
        \PYG{k}{return} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{iterable}\PYG{p}{)}
    \PYG{c+c1}{\PYGZsh{} Let the base class default method raise the TypeError}
    \PYG{k}{return} \PYG{n}{JSONEncoder}\PYG{o}{.}\PYG{n}{default}\PYG{p}{(}\PYG{n+nb+bp}{self}\PYG{p}{,} \PYG{n}{o}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}


\end{fulllineitems}



\subsubsection{worker.py}
\label{\detokenize{_autosummary/report_manager:module-report_manager.worker}}\label{\detokenize{_autosummary/report_manager:worker-py}}\index{report\_manager.worker (module)@\spxentry{report\_manager.worker}\spxextra{module}}

\subsection{Analytics Core}
\label{\detokenize{_autosummary/analytics_core:analytics-core}}\label{\detokenize{_autosummary/analytics_core::doc}}

\subsubsection{Analysis}
\label{\detokenize{_autosummary/analytics_core.analytics:analysis}}\label{\detokenize{_autosummary/analytics_core.analytics::doc}}

\paragraph{Analytics}
\label{\detokenize{_autosummary/analytics_core.analytics:module-analytics_core.analytics.analytics}}\label{\detokenize{_autosummary/analytics_core.analytics:analytics}}\index{analytics\_core.analytics.analytics (module)@\spxentry{analytics\_core.analytics.analytics}\spxextra{module}}\index{unit\_vector() (in module analytics\_core.analytics.analytics)@\spxentry{unit\_vector()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.unit_vector}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{unit\_vector}}}{\emph{vector}}{}
Returns the unit vector of the vector.
:param tuple vector: vector
:return tuple unit\_vector: unit vector

\end{fulllineitems}

\index{flatten() (in module analytics\_core.analytics.analytics)@\spxentry{flatten()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.flatten}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{flatten}}}{\emph{t}, \emph{my\_list={[}{]}}}{}
Code from: \sphinxurl{https://gist.github.com/shaxbee/0ada767debf9eefbdb6e}
Acknowledgements: Zbigniew Mandziejewicz (shaxbee)
Generator flattening the structure

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{p}{(}\PYG{l+m+mi}{4}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{7}\PYG{p}{]}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{6}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{,} \PYG{p}{[}\PYG{l+m+mi}{6}\PYG{p}{]}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{]}\PYG{p}{]}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{)}\PYG{p}{]}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}
\PYG{g+go}{[2, 2, 4, 5, 7, 2, 6, 2, 6, 6, 4, 6]}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{angle\_between() (in module analytics\_core.analytics.analytics)@\spxentry{angle\_between()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.angle_between}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{angle\_between}}}{\emph{v1}, \emph{v2}}{}
Returns the angle in radians between vectors ‘v1’ and ‘v2’
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{v1}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#tuple}{\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}}) \textendash{} vector 1

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{v2}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#tuple}{\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}}) \textendash{} vector 2

\end{itemize}

\item[{Return float angle}] \leavevmode
angle between two vectors in radians

\end{description}\end{quote}
\begin{description}
\item[{Example::}] \leavevmode
angle = angle\_between((1, 0, 0), (0, 1, 0))

\end{description}

\end{fulllineitems}

\index{transform\_into\_wide\_format() (in module analytics\_core.analytics.analytics)@\spxentry{transform\_into\_wide\_format()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.transform_into_wide_format}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{transform\_into\_wide\_format}}}{\emph{data}, \emph{index}, \emph{columns}, \emph{values}, \emph{extra={[}{]}}}{}
This function converts a Pandas DataFrame from long to wide format using 
pandas pivot\_table() function.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} long\sphinxhyphen{}format Pandas DataFrame

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} columns that will be converted into the index

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{columns}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column name whose unique values will become the new column names

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{values}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column to aggregate

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{extra}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} additional columns to be kept as columns

\end{itemize}

\item[{Returns}] \leavevmode
Wide\sphinxhyphen{}format pandas DataFrame

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{transform\PYGZus{}into\PYGZus{}wide\PYGZus{}format}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{index}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{index}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{columns}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{values}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{extra}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{transform\_into\_long\_format() (in module analytics\_core.analytics.analytics)@\spxentry{transform\_into\_long\_format()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.transform_into_long_format}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{transform\_into\_long\_format}}}{\emph{data, drop\_columns, group, columns={[}\textquotesingle{}name\textquotesingle{}, \textquotesingle{}y\textquotesingle{}{]}}}{}
Converts a Pandas DataDrame from wide to long format using pd.melt() 
function.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} wide\sphinxhyphen{}format Pandas DataFrame

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_columns}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} columns to be deleted

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}\sphinxstyleliteralemphasis{\sphinxupquote{ or }}\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column(s) to use as identifier variables

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{columns}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} names to use for the 1)variable column, and for the 2)value column

\end{itemize}

\item[{Returns}] \leavevmode
Long\sphinxhyphen{}format Pandas DataFrame.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{transform\PYGZus{}into\PYGZus{}long\PYGZus{}format}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{drop\PYGZus{}columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{name}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_ranking\_with\_markers() (in module analytics\_core.analytics.analytics)@\spxentry{get\_ranking\_with\_markers()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_ranking_with_markers}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_ranking\_with\_markers}}}{\emph{data}, \emph{drop\_columns}, \emph{group}, \emph{columns}, \emph{list\_markers}, \emph{annotation=\{\}}}{}
This function creates a long\sphinxhyphen{}format dataframe with features and values to be plotted together with disease biomarker annotations.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} wide\sphinxhyphen{}format Pandas DataFrame with samples as rows and features as columns

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_columns}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} columns to be deleted

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column to use as identifier variables

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{columns}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} names to use for the 1)variable column, and for the 2)value column

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{list\_markers}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of features from data, known to be markers associated to disease.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{annotation}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} markers, from list\_markers, and associated diseases.

\end{itemize}

\item[{Returns}] \leavevmode
Long\sphinxhyphen{}format pandas DataFrame with group identifiers as rows and columns: ‘name’ (identifier), ‘y’ (LFQ intensity), ‘symbol’ and ‘size’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}ranking\PYGZus{}with\PYGZus{}markers}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{drop\PYGZus{}columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{name}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{list\PYGZus{}markers}\PYG{p}{,} \PYG{n}{annotation}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{extract\_number\_missing() (in module analytics\_core.analytics.analytics)@\spxentry{extract\_number\_missing()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.extract_number_missing}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{extract\_number\_missing}}}{\emph{data, min\_valid, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}{]}, group=\textquotesingle{}group\textquotesingle{}}}{}
Counts how many valid values exist in each column and filters column labels with more valid values than the minimum threshold defined.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas DataFrame with group as rows and protein identifier as column.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers. If None, number of valid values is counted across all samples, otherwise is counted per unique group identifier.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{min\_valid}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} minimum number of valid values to be filtered.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_columns}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped.

\end{itemize}

\item[{Returns}] \leavevmode
List of column labels above the threshold.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{extract\PYGZus{}number\PYGZus{}missing}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{min\PYGZus{}valid}\PYG{o}{=}\PYG{l+m+mi}{3}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{extract\_percentage\_missing() (in module analytics\_core.analytics.analytics)@\spxentry{extract\_percentage\_missing()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.extract_percentage_missing}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{extract\_percentage\_missing}}}{\emph{data, missing\_max, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}{]}, group=\textquotesingle{}group\textquotesingle{}}}{}
Extracts ratio of missing/valid values in each column and filters column labels with lower ratio than the minimum threshold defined.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with group as rows and protein identifier as column.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers. If None, ratio is calculated across all samples, otherwise is calculated per unique group identifier.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{missing\_max}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} maximum ratio of missing/valid values to be filtered.

\end{itemize}

\item[{Returns}] \leavevmode
List of column labels below the threshold.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{extract\PYGZus{}percentage\PYGZus{}missing}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{missing\PYGZus{}max}\PYG{o}{=}\PYG{l+m+mf}{0.3}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{imputation\_KNN() (in module analytics\_core.analytics.analytics)@\spxentry{imputation\_KNN()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.imputation_KNN}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{imputation\_KNN}}}{\emph{data, drop\_cols={[}\textquotesingle{}group\textquotesingle{}, \textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, group=\textquotesingle{}group\textquotesingle{}, cutoff=0.6, alone=True}}{}
k\sphinxhyphen{}Nearest Neighbors imputation for pandas dataframes with missing data. For more information visit \sphinxurl{https://github.com/iskandr/fancyimpute/blob/master/fancyimpute/knn.py}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped. Final dataframe should only have gene/protein/etc identifiers as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cutoff}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} minimum ratio of missing/valid values required to impute in each column.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alone}} (\sphinxstyleliteralemphasis{\sphinxupquote{boolean}}) \textendash{} if True removes all columns with any missing values.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with samples as rows and protein identifiers as columns.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{imputation\PYGZus{}KNN}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{cutoff}\PYG{o}{=}\PYG{l+m+mf}{0.6}\PYG{p}{,} \PYG{n}{alone} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{imputation\_mixed\_norm\_KNN() (in module analytics\_core.analytics.analytics)@\spxentry{imputation\_mixed\_norm\_KNN()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.imputation_mixed_norm_KNN}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{imputation\_mixed\_norm\_KNN}}}{\emph{data, index\_cols={[}\textquotesingle{}group\textquotesingle{}, \textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, shift=1.8, nstd=0.3, group=\textquotesingle{}group\textquotesingle{}, cutoff=0.6}}{}
Missing values are replaced in two steps: 1) using k\sphinxhyphen{}Nearest Neighbors we impute protein columns with a higher ratio of missing/valid values than the defined cutoff,     2) the remaining missing values are replaced by random numbers that are drawn from a normal distribution.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of column labels to be set as dataframe index.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{shift}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} specifies the amount by which the distribution used for the random numbers is shifted downwards. This is in units of the                         standard deviation of the valid data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nstd}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} defines the width of the Gaussian distribution relative to the standard deviation of measured values.                         A value of 0.5 would mean that the width of the distribution used for drawing random numbers is half of the standard deviation of the data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cutoff}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} minimum ratio of missing/valid values required to impute in each column.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with samples as rows and protein identifiers as columns.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{imputation\PYGZus{}mixed\PYGZus{}norm\PYGZus{}KNN}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{index\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{shift} \PYG{o}{=} \PYG{l+m+mf}{1.8}\PYG{p}{,} \PYG{n}{nstd} \PYG{o}{=} \PYG{l+m+mf}{0.3}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{cutoff}\PYG{o}{=}\PYG{l+m+mf}{0.6}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{imputation\_normal\_distribution() (in module analytics\_core.analytics.analytics)@\spxentry{imputation\_normal\_distribution()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.imputation_normal_distribution}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{imputation\_normal\_distribution}}}{\emph{data, index\_cols={[}\textquotesingle{}group\textquotesingle{}, \textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, shift=1.8, nstd=0.3}}{}
Missing values will be replaced by random numbers that are drawn from a normal distribution. The imputation is done for each sample (across all proteins) separately.
For more information visit \sphinxurl{http://www.coxdocs.org/doku.php?id=perseus:user:activities:matrixprocessing:imputation:replacemissingfromgaussian}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of column labels to be set as dataframe index.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{shift}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} specifies the amount by which the distribution used for the random numbers is shifted downwards. This is in units of the standard deviation of the valid data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nstd}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} defines the width of the Gaussian distribution relative to the standard deviation of measured values. A value of 0.5 would mean that the width of the distribution used for drawing random numbers is half of the standard deviation of the data.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with samples as rows and protein identifiers as columns.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{imputation\PYGZus{}normal\PYGZus{}distribution}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{index\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{shift} \PYG{o}{=} \PYG{l+m+mf}{1.8}\PYG{p}{,} \PYG{n}{nstd} \PYG{o}{=} \PYG{l+m+mf}{0.3}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{normalize\_data() (in module analytics\_core.analytics.analytics)@\spxentry{normalize\_data()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.normalize_data}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{normalize\_data}}}{\emph{data}, \emph{method=\textquotesingle{}median\_polish\textquotesingle{}}}{}
This function normalizes the data using the selected method
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} DataFrame with the data to be normalized (samples x features)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxstyleliteralemphasis{\sphinxupquote{string}}) \textendash{} normalization method to choose among: median\_polish, median,
quantile, linear

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{normalize\PYGZus{}data}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{median\PYGZus{}polish}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{median\_normalization() (in module analytics\_core.analytics.analytics)@\spxentry{median\_normalization()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.median_normalization}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{median\_normalization}}}{\emph{data}}{}
This function normalizes each sample by using its median.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} 

\item[{Returns}] \leavevmode
Pandas dataframe.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{median\PYGZus{}normalization}\PYG{p}{(}\PYG{n}{data}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{zscore\_normalization() (in module analytics\_core.analytics.analytics)@\spxentry{zscore\_normalization()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.zscore_normalization}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{zscore\_normalization}}}{\emph{data}}{}
This function normalizes each sample by using its mean and standard deviation (mean=0, std=1).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} 

\item[{Returns}] \leavevmode
Pandas dataframe.

\end{description}\end{quote}
\begin{description}
\item[{Example::}] \leavevmode
data = pd.DataFrame(\{‘a’: {[}2,5,4,3,3{]}, ‘b’:{[}4,4,6,5,3{]}, ‘c’:{[}4,14,8,8,9{]}\})
result = zscore\_normalization(data)
result
\begin{quote}
\begin{quote}

a         b         c
\end{quote}

0 \sphinxhyphen{}1.154701  0.577350  0.577350
1 \sphinxhyphen{}0.484182 \sphinxhyphen{}0.665750  1.149932
2 \sphinxhyphen{}1.000000  0.000000  1.000000
3 \sphinxhyphen{}0.927173 \sphinxhyphen{}0.132453  1.059626
4 \sphinxhyphen{}0.577350 \sphinxhyphen{}0.577350  1.154701
\end{quote}

\end{description}

\end{fulllineitems}

\index{median\_polish\_normalization() (in module analytics\_core.analytics.analytics)@\spxentry{median\_polish\_normalization()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.median_polish_normalization}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{median\_polish\_normalization}}}{\emph{data}, \emph{max\_iter=250}}{}
This function iteratively normalizes each sample and each feature to its 
median until medians converge.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{max\_iter}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of maximum iterations to prevent infinite loop.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{median\PYGZus{}polish\PYGZus{}normalization}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{max\PYGZus{}iter} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{quantile\_normalization() (in module analytics\_core.analytics.analytics)@\spxentry{quantile\_normalization()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.quantile_normalization}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{quantile\_normalization}}}{\emph{data}}{}
Applies quantile normalization to each column in pandas dataframe.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with features as rows and samples as columns.

\item[{Returns}] \leavevmode
Pandas dataframe

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{quantile\PYGZus{}normalization}\PYG{p}{(}\PYG{n}{data}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{linear\_normalization() (in module analytics\_core.analytics.analytics)@\spxentry{linear\_normalization()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.linear_normalization}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{linear\_normalization}}}{\emph{data}, \emph{method=\textquotesingle{}l1\textquotesingle{}}, \emph{axis=0}}{}
This function scales input data to a unit norm. For more information visit \sphinxurl{https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.normalize.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} norm to use to normalize each non\sphinxhyphen{}zero sample or non\sphinxhyphen{}zero feature (depends on axis).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{axis}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} axis used to normalize the data along. If 1, independently normalize each sample, otherwise (if 0) normalize each feature.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{linear\PYGZus{}normalization}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{method} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{l1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{axis} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{remove\_group() (in module analytics\_core.analytics.analytics)@\spxentry{remove\_group()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.remove_group}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{remove\_group}}}{\emph{data}}{}
Removes column with label ‘group’.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with one column labelled ‘group’

\item[{Returns}] \leavevmode
Pandas dataframe

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{remove\PYGZus{}group}\PYG{p}{(}\PYG{n}{data}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_coefficient\_variation() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_coefficient\_variation()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_coefficient_variation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_coefficient\_variation}}}{\emph{values}}{}
Compute the coefficient of variation, the ratio of the biased standard 
deviation to the mean, in percentage. For more information 
visit \sphinxurl{https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.variation.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{values}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} numpy array

\item[{Returns}] \leavevmode
The calculated variation along rows.

\item[{Return type}] \leavevmode
ndarray

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}coefficient\PYGZus{}variation}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_coefficient\_variation() (in module analytics\_core.analytics.analytics)@\spxentry{get\_coefficient\_variation()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_coefficient_variation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_coefficient\_variation}}}{\emph{data, drop\_columns, group, columns={[}\textquotesingle{}name\textquotesingle{}, \textquotesingle{}y\textquotesingle{}{]}}}{}
Extracts the coefficients of variation in each group.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_columns}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped from the dataframe

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{columns}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} names to use for the variable column(s), and for the value column(s)

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns ‘name’ (protein identifier), ‘x’ (coefficient of variation), ‘y’ (mean) and ‘group’.

\end{description}\end{quote}

Exmaple:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}coefficient\PYGZus{}variation}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{drop\PYGZus{}columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_proteomics\_measurements\_ready() (in module analytics\_core.analytics.analytics)@\spxentry{get\_proteomics\_measurements\_ready()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_proteomics_measurements_ready}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_proteomics\_measurements\_ready}}}{\emph{df, index\_cols={[}\textquotesingle{}group\textquotesingle{}, \textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}{]}, group=\textquotesingle{}group\textquotesingle{}, identifier=\textquotesingle{}identifier\textquotesingle{}, extra\_identifier=\textquotesingle{}name\textquotesingle{}, imputation=True, method=\textquotesingle{}distribution\textquotesingle{}, missing\_method=\textquotesingle{}percentage\textquotesingle{}, missing\_per\_group=True, missing\_max=0.3, min\_valid=1, value\_col=\textquotesingle{}LFQ\_intensity\textquotesingle{}, shift=1.8, nstd=0.3, knn\_cutoff=0.6}}{}
Processes proteomics data extracted from the database: 1) filter proteins with high number of missing values (\textgreater{} missing\_max or min\_valid), 2) impute missing values.
For more information on imputation method visit \sphinxurl{http://www.coxdocs.org/doku.php?id=perseus:user:activities:matrixprocessing:filterrows:filtervalidvaluesrows}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} long\sphinxhyphen{}format pandas dataframe with columns ‘group’, ‘sample’, ‘subject’, ‘identifier’ (protein), ‘name’ (gene) and ‘LFQ\_intensity’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be be kept as index identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped from the dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing feature identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{extra\_identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing additional protein identifiers (e.g. gene names).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{imputation}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True performs imputation of missing values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method for missing values imputation (‘KNN’, ‘distribuition’, or ‘mixed’)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{missing\_method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} defines which expression rows are counted to determine if a column has enough valid values to survive the filtering process.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{missing\_per\_group}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True filter proteins based on valid values per group; if False filter across all samples.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{missing\_max}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} maximum ratio of missing/valid values to be filtered.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{min\_valid}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} minimum number of valid values to be filtered.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{value\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing expression values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{shift}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} when using distribution imputation, the down\sphinxhyphen{}shift

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nstd}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} when using distribution imputation, the width of the distribution

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{knn\_cutoff}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} when using KNN imputation, the minimum percentage of valid values for which to use KNN imputation (i.e. 0.6 \sphinxhyphen{}\textgreater{} if 60\% valid values use KNN, otherwise MinProb)

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with samples as rows and protein identifiers (UniprotID\textasciitilde{}GeneName) as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\end{description}\end{quote}

Example 1:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}proteomics\PYGZus{}measurements\PYGZus{}ready}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{index\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{identifier}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{extra\PYGZus{}identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{name}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{imputation} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{distribution}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{missing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{percentage}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{missing\PYGZus{}per\PYGZus{}group}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{missing\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mf}{0.3}\PYG{p}{,} \PYG{n}{value\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{LFQ\PYGZus{}intensity}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

Example 2:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}proteomics\PYGZus{}measurements\PYGZus{}ready}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{index\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{identifier}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{extra\PYGZus{}identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{name}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{imputation} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{mixed}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{missing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{at\PYGZus{}least\PYGZus{}x}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{missing\PYGZus{}per\PYGZus{}group}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{min\PYGZus{}valid}\PYG{o}{=}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{value\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{LFQ\PYGZus{}intensity}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_clinical\_measurements\_ready() (in module analytics\_core.analytics.analytics)@\spxentry{get\_clinical\_measurements\_ready()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_clinical_measurements_ready}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_clinical\_measurements\_ready}}}{\emph{df, subject\_id=\textquotesingle{}subject\textquotesingle{}, sample\_id=\textquotesingle{}biological\_sample\textquotesingle{}, group\_id=\textquotesingle{}group\textquotesingle{}, columns={[}\textquotesingle{}clinical\_variable\textquotesingle{}{]}, values=\textquotesingle{}values\textquotesingle{}, extra={[}\textquotesingle{}group\textquotesingle{}{]}, imputation=True, imputation\_method=\textquotesingle{}KNN\textquotesingle{}}}{}
Processes clinical data extracted from the database by converting dataframe to wide\sphinxhyphen{}format and imputing missing values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} long\sphinxhyphen{}format pandas dataframe with columns ‘group’, ‘biological\_sample’, ‘subject’, ‘clinical\_variable’, ‘value’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject\_id}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing subject identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sample\_id}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing biological sample identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group\_id}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{columns}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column name whose unique values will become the new column names

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{values}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing clinical variable values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{extra}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} additional column labels to be kept as columns

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{imputation}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True performs imputation of missing values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{imputation\_method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method for missing values imputation (‘KNN’, ‘distribuition’, or ‘mixed’).

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with samples as rows and clinical variables as columns (with additional columns ‘group’, ‘subject’ and ‘biological\_sample’).

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}clinical\PYGZus{}measurements\PYGZus{}ready}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{subject\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{sample\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{biological\PYGZus{}sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{clinical\PYGZus{}variable}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{values}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{extra}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{imputation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{imputation\PYGZus{}method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{KNN}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_summary\_data\_matrix() (in module analytics\_core.analytics.analytics)@\spxentry{get\_summary\_data\_matrix()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_summary_data_matrix}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_summary\_data\_matrix}}}{\emph{data}}{}
Returns some statistics on the data matrix provided.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe.

\item[{Returns}] \leavevmode
dictionary with the type of statistics as key and the statistic as value in the shape of a pandas data frame

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}summary\PYGZus{}data\PYGZus{}matrix}\PYG{p}{(}\PYG{n}{data}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{check\_equal\_variances() (in module analytics\_core.analytics.analytics)@\spxentry{check\_equal\_variances()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.check_equal_variances}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{check\_equal\_variances}}}{\emph{data, drop\_cols={[}\textquotesingle{}group\textquotesingle{}, \textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, group\_col=\textquotesingle{}group\textquotesingle{}, alpha=0.05}}{}
\end{fulllineitems}

\index{check\_normality() (in module analytics\_core.analytics.analytics)@\spxentry{check\_normality()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.check_normality}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{check\_normality}}}{\emph{data, drop\_cols={[}\textquotesingle{}group\textquotesingle{}, \textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, group\_col=\textquotesingle{}group\textquotesingle{}, alpha=0.05}}{}
\end{fulllineitems}

\index{run\_pca() (in module analytics\_core.analytics.analytics)@\spxentry{run\_pca()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_pca}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_pca}}}{\emph{data, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, group=\textquotesingle{}group\textquotesingle{}, components=2, dropna=True}}{}
Performs principal component analysis and returns the values of each component for each sample and each protein, and the loadings for each protein.     For information visit \sphinxurl{https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped from the dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{components}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of components to keep.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dropna}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True removes all columns with any missing values.

\end{itemize}

\item[{Returns}] \leavevmode
Two dictionaries: 1) two pandas dataframes (first one with components values, the second with the components vectors for each protein), 2) xaxis and yaxis titles with components loadings for plotly.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}pca}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{components}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{dropna}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{run\_tsne() (in module analytics\_core.analytics.analytics)@\spxentry{run\_tsne()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_tsne}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_tsne}}}{\emph{data, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, group=\textquotesingle{}group\textquotesingle{}, components=2, perplexity=40, n\_iter=1000, init=\textquotesingle{}pca\textquotesingle{}, dropna=True}}{}
Performs t\sphinxhyphen{}distributed Stochastic Neighbor Embedding analysis. For more information visit \sphinxurl{https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped from the dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{components}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} dimension of the embedded space.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{perplexity}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} related to the number of nearest neighbors that is used in other manifold learning algorithms. Consider selecting a value between 5 and 50.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n\_iter}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} maximum number of iterations for the optimization (at least 250).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{init}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} initialization of embedding (‘random’, ‘pca’ or numpy array of shape n\_samples x n\_components).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dropna}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True removes all columns with any missing values.

\end{itemize}

\item[{Returns}] \leavevmode
Two dictionaries: 1) pandas dataframe with embedding vectors, 2) xaxis and yaxis titles for plotly.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}tsne}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{components}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{perplexity}\PYG{o}{=}\PYG{l+m+mi}{40}\PYG{p}{,} \PYG{n}{n\PYGZus{}iter}\PYG{o}{=}\PYG{l+m+mi}{1000}\PYG{p}{,} \PYG{n}{init}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pca}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{dropna}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{run\_umap() (in module analytics\_core.analytics.analytics)@\spxentry{run\_umap()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_umap}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_umap}}}{\emph{data, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, group=\textquotesingle{}group\textquotesingle{}, n\_neighbors=10, min\_dist=0.3, metric=\textquotesingle{}cosine\textquotesingle{}, dropna=True}}{}
Performs Uniform Manifold Approximation and Projection. For more information vist \sphinxurl{https://umap-learn.readthedocs.io}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped from the dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n\_neighbors}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of neighboring points used in local approximations of manifold structure.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{min\_dist}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} controls how tightly the embedding is allowed compress points together.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{metric}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} metric used to measure distance in the input space.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dropna}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True removes all columns with any missing values.

\end{itemize}

\item[{Returns}] \leavevmode
Two dictionaries: 1) pandas dataframe with embedding of the training data in low\sphinxhyphen{}dimensional space, 2) xaxis and yaxis titles for plotly.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}umap}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{n\PYGZus{}neighbors}\PYG{o}{=}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{min\PYGZus{}dist}\PYG{o}{=}\PYG{l+m+mf}{0.3}\PYG{p}{,} \PYG{n}{metric}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{cosine}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{dropna}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_correlations() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_correlations()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_correlations}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_correlations}}}{\emph{x}, \emph{y}, \emph{method=\textquotesingle{}pearson\textquotesingle{}}}{}
Calculates a Spearman (nonparametric) or a Pearson (parametric) correlation coefficient and p\sphinxhyphen{}value to test for non\sphinxhyphen{}correlation.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{x}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} array 1

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{y}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} array 2

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} chooses which kind of correlation method to run

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with two floats, correlation coefficient and two\sphinxhyphen{}tailed p\sphinxhyphen{}value.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}correlations}\PYG{p}{(}\PYG{n}{x}\PYG{p}{,} \PYG{n}{y}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pearson}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{apply\_pvalue\_correction() (in module analytics\_core.analytics.analytics)@\spxentry{apply\_pvalue\_correction()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.apply_pvalue_correction}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{apply\_pvalue\_correction}}}{\emph{pvalues}, \emph{alpha=0.05}, \emph{method=\textquotesingle{}bonferroni\textquotesingle{}}}{}
Performs p\sphinxhyphen{}value correction using the specified method. For more information visit \sphinxurl{https://www.statsmodels.org/dev/generated/statsmodels.stats.multitest.multipletests.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pvalues}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} et of p\sphinxhyphen{}values of the individual tests.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method of p\sphinxhyphen{}value correction:
\sphinxhyphen{} bonferroni : one\sphinxhyphen{}step correction
\sphinxhyphen{} sidak : one\sphinxhyphen{}step correction
\sphinxhyphen{} holm\sphinxhyphen{}sidak : step down method using Sidak adjustments
\sphinxhyphen{} holm : step\sphinxhyphen{}down method using Bonferroni adjustments
\sphinxhyphen{} simes\sphinxhyphen{}hochberg : step\sphinxhyphen{}up method (independent)
\sphinxhyphen{} hommel : closed method based on Simes tests (non\sphinxhyphen{}negative)
\sphinxhyphen{} fdr\_bh : Benjamini/Hochberg (non\sphinxhyphen{}negative)
\sphinxhyphen{} fdr\_by : Benjamini/Yekutieli (negative)
\sphinxhyphen{} fdr\_tsbh : two stage fdr correction (non\sphinxhyphen{}negative)
\sphinxhyphen{} fdr\_tsbky : two stage fdr correction (non\sphinxhyphen{}negative)

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with two arrays, boolen for rejecting H0 hypothesis and float for adjusted p\sphinxhyphen{}value.

\end{description}\end{quote}

Exmaple:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{apply\PYGZus{}pvalue\PYGZus{}correction}\PYG{p}{(}\PYG{n}{pvalues}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bonferroni}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{apply\_pvalue\_fdrcorrection() (in module analytics\_core.analytics.analytics)@\spxentry{apply\_pvalue\_fdrcorrection()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.apply_pvalue_fdrcorrection}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{apply\_pvalue\_fdrcorrection}}}{\emph{pvalues}, \emph{alpha=0.05}, \emph{method=\textquotesingle{}indep\textquotesingle{}}}{}
Performs p\sphinxhyphen{}value correction for false discovery rate. For more information visit \sphinxurl{https://www.statsmodels.org/devel/generated/statsmodels.stats.multitest.fdrcorrection.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pvalues}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} et of p\sphinxhyphen{}values of the individual tests.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method of p\sphinxhyphen{}value correction (‘indep’, ‘negcorr’).

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with two arrays, boolen for rejecting H0 hypothesis and float for adjusted p\sphinxhyphen{}value.

\end{description}\end{quote}

Exmaple:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{apply\PYGZus{}pvalue\PYGZus{}fdrcorrection}\PYG{p}{(}\PYG{n}{pvalues}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{indep}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{apply\_pvalue\_twostage\_fdrcorrection() (in module analytics\_core.analytics.analytics)@\spxentry{apply\_pvalue\_twostage\_fdrcorrection()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.apply_pvalue_twostage_fdrcorrection}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{apply\_pvalue\_twostage\_fdrcorrection}}}{\emph{pvalues}, \emph{alpha=0.05}, \emph{method=\textquotesingle{}bh\textquotesingle{}}}{}
Iterated two stage linear step\sphinxhyphen{}up procedure with estimation of number of true hypotheses. For more information visit \sphinxurl{https://www.statsmodels.org/dev/generated/statsmodels.stats.multitest.fdrcorrection\_twostage.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pvalues}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} et of p\sphinxhyphen{}values of the individual tests.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method of p\sphinxhyphen{}value correction (‘bky’, ‘bh’).

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with two arrays, boolen for rejecting H0 hypothesis and float for adjusted p\sphinxhyphen{}value.

\end{description}\end{quote}

Exmaple:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{apply\PYGZus{}pvalue\PYGZus{}twostage\PYGZus{}fdrcorrection}\PYG{p}{(}\PYG{n}{pvalues}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bh}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{apply\_pvalue\_permutation\_fdrcorrection() (in module analytics\_core.analytics.analytics)@\spxentry{apply\_pvalue\_permutation\_fdrcorrection()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.apply_pvalue_permutation_fdrcorrection}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{apply\_pvalue\_permutation\_fdrcorrection}}}{\emph{df}, \emph{observed\_pvalues}, \emph{group}, \emph{alpha=0.05}, \emph{permutations=50}}{}
This function applies multiple hypothesis testing correction using a permutation\sphinxhyphen{}based false discovery rate approach.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{oberved\_pvalues}} \textendash{} pandas Series with p\sphinxhyphen{}values calculated on the originally measured data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate. Values velow alpha are considered significant.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{permutations}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of permutations to be applied.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with adjusted p\sphinxhyphen{}values and rejected columns.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{apply\PYGZus{}pvalue\PYGZus{}permutation\PYGZus{}fdrcorrection}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{observed\PYGZus{}pvalues}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{permutations}\PYG{o}{=}\PYG{l+m+mi}{50}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_counts\_permutation\_fdr() (in module analytics\_core.analytics.analytics)@\spxentry{get\_counts\_permutation\_fdr()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_counts_permutation_fdr}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_counts\_permutation\_fdr}}}{\emph{value}, \emph{random}, \emph{observed}, \emph{n}, \emph{alpha}}{}
Calculates local FDR values (q\sphinxhyphen{}values) by computing the fraction of accepted hits from the permuted data over accepted hits from the measured data normalized by the total number of permutations.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{value}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} computed p\sphinxhyphen{}value on measured data for a feature.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{random}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} p\sphinxhyphen{}values computed on the permuted data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{observed}} \textendash{} pandas Series with p\sphinxhyphen{}values calculated on the originally measured data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of permutations to be applied.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate. Values velow alpha are considered significant.

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with q\sphinxhyphen{}value and boolean for H0 rejected.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}counts\PYGZus{}permutation\PYGZus{}fdr}\PYG{p}{(}\PYG{n}{value}\PYG{p}{,} \PYG{n}{random}\PYG{p}{,} \PYG{n}{observed}\PYG{p}{,} \PYG{n}{n}\PYG{o}{=}\PYG{l+m+mi}{250}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{convertToEdgeList() (in module analytics\_core.analytics.analytics)@\spxentry{convertToEdgeList()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.convertToEdgeList}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{convertToEdgeList}}}{\emph{data}, \emph{cols}}{}
This function converts a pandas dataframe to an edge list where index becomes the source nodes and columns the target nodes.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} names for dataframe columns.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns cols.

\end{description}\end{quote}

\end{fulllineitems}

\index{run\_correlation() (in module analytics\_core.analytics.analytics)@\spxentry{run\_correlation()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_correlation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_correlation}}}{\emph{df}, \emph{alpha=0.05}, \emph{subject=\textquotesingle{}subject\textquotesingle{}}, \emph{group=\textquotesingle{}group\textquotesingle{}}, \emph{method=\textquotesingle{}pearson\textquotesingle{}}, \emph{correction=\textquotesingle{}fdr\_bh\textquotesingle{}}}{}
This function calculates pairwise correlations for columns in dataframe, and returns it in the shape of a edge list with ‘weight’ as correlation score, and the ajusted p\sphinxhyphen{}values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of column containing subject identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of column containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method to use for correlation calculation (‘pearson’, ‘spearman’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxstyleliteralemphasis{\sphinxupquote{floar}}) \textendash{} error rate. Values velow alpha are considered significant.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{correction}} (\sphinxstyleliteralemphasis{\sphinxupquote{string}}) \textendash{} type of correction see apply\_pvalue\_correction for methods

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns: ‘node1’, ‘node2’, ‘weight’, ‘padj’ and ‘rejected’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}correlation}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pearson}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{correction}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fdr\PYGZus{}bh}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{run\_multi\_correlation() (in module analytics\_core.analytics.analytics)@\spxentry{run\_multi\_correlation()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_multi_correlation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_multi\_correlation}}}{\emph{df\_dict, alpha=0.05, subject=\textquotesingle{}subject\textquotesingle{}, on={[}\textquotesingle{}subject\textquotesingle{}, \textquotesingle{}biological\_sample\textquotesingle{}{]}, group=\textquotesingle{}group\textquotesingle{}, method=\textquotesingle{}pearson\textquotesingle{}, correction=\textquotesingle{}fdr\_bh\textquotesingle{}}}{}
This function merges all input dataframes and calculates pairwise correlations for all columns.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df\_dict}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary of pandas dataframes with samples as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column containing subject identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{on}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column names to join dataframes on (must be found in all dataframes).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method to use for correlation calculation (‘pearson’, ‘spearman’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate. Values velow alpha are considered significant.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{correction}} (\sphinxstyleliteralemphasis{\sphinxupquote{string}}) \textendash{} type of correction see apply\_pvalue\_correction for methods

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns: ‘node1’, ‘node2’, ‘weight’, ‘padj’ and ‘rejected’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}multi\PYGZus{}correlation}\PYG{p}{(}\PYG{n}{df\PYGZus{}dict}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{on}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{biological\PYGZus{}sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pearson}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{correction}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fdr\PYGZus{}bh}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_rm\_correlation() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_rm\_correlation()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_rm_correlation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_rm\_correlation}}}{\emph{df}, \emph{x}, \emph{y}, \emph{subject}}{}
Computes correlation and p\sphinxhyphen{}values between two columns a and b in df.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with subjects as rows and two features and columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{x}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} feature a name.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{y}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} feature b name.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} \textendash{} column name containing the covariate variable.

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with values for: feature a, feature b, correlation, p\sphinxhyphen{}value and degrees of freedom.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}rm\PYGZus{}correlation}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{x}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{feature a}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{y}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{feature b}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{run\_rm\_correlation() (in module analytics\_core.analytics.analytics)@\spxentry{run\_rm\_correlation()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_rm_correlation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_rm\_correlation}}}{\emph{df}, \emph{alpha=0.05}, \emph{subject=\textquotesingle{}subject\textquotesingle{}}, \emph{correction=\textquotesingle{}fdr\_bh\textquotesingle{}}}{}
Computes pairwise repeated measurements correlations for all columns in dataframe, and returns results as an edge list with ‘weight’ as correlation score, p\sphinxhyphen{}values, degrees of freedom and ajusted p\sphinxhyphen{}values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of column containing subject identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate. Values velow alpha are considered significant.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{correction}} (\sphinxstyleliteralemphasis{\sphinxupquote{string}}) \textendash{} type of correction type see apply\_pvalue\_correction for methods

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns: ‘node1’, ‘node2’, ‘weight’, ‘pvalue’, ‘dof’, ‘padj’ and ‘rejected’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}rm\PYGZus{}correlation}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{correction}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fdr\PYGZus{}bh}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{run\_efficient\_correlation() (in module analytics\_core.analytics.analytics)@\spxentry{run\_efficient\_correlation()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_efficient_correlation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_efficient\_correlation}}}{\emph{data}, \emph{method=\textquotesingle{}pearson\textquotesingle{}}}{}
Calculates pairwise correlations and returns lower triangle of the matrix with correlation values and p\sphinxhyphen{}values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as index and features as columns (numeric data only).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method to use for correlation calculation (‘pearson’, ‘spearman’).

\end{itemize}

\item[{Returns}] \leavevmode
Two numpy arrays: correlation and p\sphinxhyphen{}values.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}efficient\PYGZus{}correlation}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pearson}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_ttest\_samr() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_ttest\_samr()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_ttest_samr}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_ttest\_samr}}}{\emph{df}, \emph{labels}, \emph{n=2}, \emph{s0=0}, \emph{paired=False}}{}
Calculates modified T\sphinxhyphen{}test using ‘samr’ R package.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with group as columns and protein identifier as rows

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{abels}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} integers reflecting the group each sample belongs to (e.g. group1 = 1, group2 = 2)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of samples

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{s0}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} exchangeability factor for denominator of test statistic

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{paired}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} True if samples are paired

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns ‘identifier’, ‘group1’, ‘group2’, ‘mean(group1)’, ‘mean(group1)’, ‘log2FC’, ‘FC’, ‘t\sphinxhyphen{}statistics’, ‘p\sphinxhyphen{}value’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}ttest\PYGZus{}samr}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{n}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{s0}\PYG{o}{=}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{n}{paired}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_ttest() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_ttest()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_ttest}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_ttest}}}{\emph{df}, \emph{condition1}, \emph{condition2}, \emph{paired=False}, \emph{is\_logged=True}, \emph{non\_par=False}, \emph{tail=\textquotesingle{}two\sphinxhyphen{}sided\textquotesingle{}}, \emph{correction=\textquotesingle{}auto\textquotesingle{}}, \emph{r=0.707}}{}
Calculates the t\sphinxhyphen{}test for the means of independent samples belonging to two different groups. For more information visit \sphinxurl{https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest\_ind.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with groups and subjects as rows and protein identifier as column.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition1}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of first group.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition2}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} ientifier of second group.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{is\_logged}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} data is logged transformed

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{non\_par}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, normality and variance equality assumptions are checked and non\sphinxhyphen{}parametric test Mann Whitney U test if not passed

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with t\sphinxhyphen{}statistics, two\sphinxhyphen{}tailed p\sphinxhyphen{}value, mean of first group, mean of second group and logfc.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}ttest}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_THSD() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_THSD()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_THSD}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_THSD}}}{\emph{df}, \emph{column}, \emph{group=\textquotesingle{}group\textquotesingle{}}, \emph{alpha=0.05}, \emph{is\_logged=True}}{}
Pairwise Tukey\sphinxhyphen{}HSD posthoc test using pingouin stats. For more information visit \sphinxurl{https://pingouin-stats.org/generated/pingouin.pairwise\_tukey.html}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with group and protein identifier as columns

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{column}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column containing the protein identifier

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing the between factor

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} significance level

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}THSD}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{column}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{HBG2\PYGZti{}P69892}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_pairwise\_ttest() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_pairwise\_ttest()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_pairwise_ttest}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_pairwise\_ttest}}}{\emph{df}, \emph{column}, \emph{subject=\textquotesingle{}subject\textquotesingle{}}, \emph{group=\textquotesingle{}group\textquotesingle{}}, \emph{correction=\textquotesingle{}none\textquotesingle{}}, \emph{is\_logged=True}}{}
Performs pairwise t\sphinxhyphen{}test using pingouin, as a posthoc test, and calculates fold\sphinxhyphen{}changes. For more information visit \sphinxurl{https://pingouin-stats.org/generated/pingouin.pairwise\_ttests.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with subject and group as rows and protein identifier as column.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{column}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing the dependant variable

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing subject identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing the between factor

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{correction}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method used for testing and adjustment of p\sphinxhyphen{}values.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with means, standard deviations, test\sphinxhyphen{}statistics, degrees of freedom and effect size columns.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}pairwise\PYGZus{}ttest}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{protein a}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{correction}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{none}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{complement\_posthoc() (in module analytics\_core.analytics.analytics)@\spxentry{complement\_posthoc()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.complement_posthoc}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{complement\_posthoc}}}{\emph{posthoc}, \emph{identifier}, \emph{is\_logged}}{}
Calculates fold\sphinxhyphen{}changes after posthoc test.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{posthoc}} \textendash{} pandas dataframe from posthoc test. Should have at least columns ‘mean(group1)’ and ‘mean(group2)’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} feature identifier.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with additional columns ‘identifier’, ‘log2FC’ and ‘FC’.

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_dabest() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_dabest()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_dabest}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_dabest}}}{\emph{df}, \emph{idx}, \emph{x}, \emph{y}, \emph{paired=False}, \emph{id\_col=None}, \emph{test=\textquotesingle{}mean\_diff\textquotesingle{}}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{idx}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{x}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{y}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{paired}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_col}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{test}} \textendash{} 

\end{itemize}

\item[{Returns}] \leavevmode


\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_anova\_samr() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_anova\_samr()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_anova_samr}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_anova\_samr}}}{\emph{df}, \emph{labels}, \emph{s0=0}}{}
Calculates modified one\sphinxhyphen{}way ANOVA using ‘samr’ R package.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with group as columns and protein identifier as rows

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{labels}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} integers reflecting the group each sample belongs to (e.g. group1 = 1, group2 = 2, group3 = 3)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{s0}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} exchangeability factor for denominator of test statistic

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with protein identifiers and F\sphinxhyphen{}statistics.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}anova\PYGZus{}samr}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{s0}\PYG{o}{=}\PYG{l+m+mf}{0.1}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_anova() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_anova()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_anova}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_anova}}}{\emph{df}, \emph{column}, \emph{group=\textquotesingle{}group\textquotesingle{}}}{}
Calculates one\sphinxhyphen{}way ANOVA using pingouin.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with group as rows and protein identifier as column

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{column}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column in df to run ANOVA on

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with group identifiers

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with t\sphinxhyphen{}statistics and p\sphinxhyphen{}value.

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_repeated\_measures\_anova() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_repeated\_measures\_anova()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_repeated_measures_anova}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_repeated\_measures\_anova}}}{\emph{df}, \emph{column}, \emph{subject=\textquotesingle{}subject\textquotesingle{}}, \emph{group=\textquotesingle{}group\textquotesingle{}}}{}
One\sphinxhyphen{}way and two\sphinxhyphen{}way repeated measures ANOVA using pingouin stats.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifier as column. Data must be in long\sphinxhyphen{}format for two\sphinxhyphen{}way repeated measures.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{column}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing the dependant variable

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing subject identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing the within factor

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with protein identifier, t\sphinxhyphen{}statistics and p\sphinxhyphen{}value.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}repeated\PYGZus{}measures\PYGZus{}anova}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{protein a}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_max\_permutations() (in module analytics\_core.analytics.analytics)@\spxentry{get\_max\_permutations()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_max_permutations}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_max\_permutations}}}{\emph{df}, \emph{group=\textquotesingle{}group\textquotesingle{}}}{}
Get maximum number of permutations according to number of samples.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with group identifiers

\end{itemize}

\item[{Returns}] \leavevmode
Maximum number of permutations.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{int}

\end{description}\end{quote}

\end{fulllineitems}

\index{check\_is\_paired() (in module analytics\_core.analytics.analytics)@\spxentry{check\_is\_paired()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.check_is_paired}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{check\_is\_paired}}}{\emph{df}, \emph{subject}, \emph{group}}{}
Check if samples are paired.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with subject identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with group identifiers

\end{itemize}

\item[{Returns}] \leavevmode
True if paired samples.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{bool}

\end{description}\end{quote}

\end{fulllineitems}

\index{run\_dabest() (in module analytics\_core.analytics.analytics)@\spxentry{run\_dabest()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_dabest}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_dabest}}}{\emph{df, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}{]}, subject=\textquotesingle{}subject\textquotesingle{}, group=\textquotesingle{}group\textquotesingle{}, test=\textquotesingle{}mean\_diff\textquotesingle{}}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{test}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} 

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe

\end{description}\end{quote}

\end{fulllineitems}

\index{run\_anova() (in module analytics\_core.analytics.analytics)@\spxentry{run\_anova()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_anova}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_anova}}}{\emph{df, alpha=0.05, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}, \textquotesingle{}subject\textquotesingle{}{]}, subject=\textquotesingle{}subject\textquotesingle{}, group=\textquotesingle{}group\textquotesingle{}, permutations=0, correction=\textquotesingle{}fdr\_bh\textquotesingle{}, is\_logged=True, non\_par=False}}{}
Performs statistical test for each protein in a dataset.
Checks what type of data is the input (paired, unpaired or repeated measurements) and performs posthoc tests for multiclass data.
Multiple hypothesis correction uses permutation\sphinxhyphen{}based if permutations\textgreater{}0 and Benjamini/Hochberg if permutations=0.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with subject identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with group identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped from the dataframe

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate for multiple hypothesis correction

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{permutations}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of permutations used to estimate false discovery rates.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{non\_par}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, normality and variance equality assumptions are checked and non\sphinxhyphen{}parametric test Mann Whitney U test if not passed

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns ‘identifier’, ‘group1’, ‘group2’, ‘mean(group1)’, ‘mean(group2)’, ‘Log2FC’, ‘std\_error’, ‘tail’, ‘t\sphinxhyphen{}statistics’, ‘posthoc pvalue’, ‘effsize’, ‘efftype’, ‘FC’, ‘rejected’, ‘F\sphinxhyphen{}statistics’, ‘p\sphinxhyphen{}value’, ‘correction’, ‘\sphinxhyphen{}log10 p\sphinxhyphen{}value’, and ‘method’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}anova}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{sample}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{permutations}\PYG{o}{=}\PYG{l+m+mi}{50}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{correct\_pairwise\_ttest() (in module analytics\_core.analytics.analytics)@\spxentry{correct\_pairwise\_ttest()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.correct_pairwise_ttest}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{correct\_pairwise\_ttest}}}{\emph{df}, \emph{alpha}, \emph{correction=\textquotesingle{}fdr\_bh\textquotesingle{}}}{}
\end{fulllineitems}

\index{run\_repeated\_measurements\_anova() (in module analytics\_core.analytics.analytics)@\spxentry{run\_repeated\_measurements\_anova()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_repeated_measurements_anova}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_repeated\_measurements\_anova}}}{\emph{df, alpha=0.05, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}{]}, subject=\textquotesingle{}subject\textquotesingle{}, group=\textquotesingle{}group\textquotesingle{}, permutations=50, correction=\textquotesingle{}fdr\_bh\textquotesingle{}, is\_logged=True}}{}
Performs repeated measurements anova and pairwise posthoc tests for each protein in dataframe.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with subject identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxstyleliteralemphasis{\sphinxupquote{srt}}) \textendash{} column with group identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped from the dataframe

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate for multiple hypothesis correction

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{permutations}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of permutations used to estimate false discovery rates

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}repeated\PYGZus{}measurements\PYGZus{}anova}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{permutations}\PYG{o}{=}\PYG{l+m+mi}{50}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{format\_anova\_table() (in module analytics\_core.analytics.analytics)@\spxentry{format\_anova\_table()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.format_anova_table}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{format\_anova\_table}}}{\emph{df}, \emph{aov\_results}, \emph{pairwise\_results}, \emph{pairwise\_cols}, \emph{group}, \emph{permutations}, \emph{alpha}, \emph{correction}}{}
Performs p\sphinxhyphen{}value correction (permutation\sphinxhyphen{}based and FDR) and converts pandas dataframe into final format.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{aov\_results}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{{[}}}\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#tuple}{\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}}\sphinxstyleliteralemphasis{\sphinxupquote{{]}}}) \textendash{} list of tuples with anova results (one tuple per feature).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pairwise\_results}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{{[}}}\sphinxstyleliteralemphasis{\sphinxupquote{dataframes}}\sphinxstyleliteralemphasis{\sphinxupquote{{]}}}) \textendash{} list of pandas dataframes with posthoc tests results

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with group identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate for multiple hypothesis correction

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{permutations}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of permutations used to estimate false discovery rates

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe

\end{description}\end{quote}

\end{fulllineitems}

\index{run\_ttest() (in module analytics\_core.analytics.analytics)@\spxentry{run\_ttest()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_ttest}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_ttest}}}{\emph{df, condition1, condition2, alpha=0.05, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}{]}, subject=\textquotesingle{}subject\textquotesingle{}, group=\textquotesingle{}group\textquotesingle{}, paired=False, correction=\textquotesingle{}fdr\_bh\textquotesingle{}, permutations=50, is\_logged=True, non\_par=False}}{}
Runs t\sphinxhyphen{}test (paired/unpaired) for each protein in dataset and performs permutation\sphinxhyphen{}based (if permutations\textgreater{}0) or Benjamini/Hochberg (if permutations=0) multiple hypothesis correction.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition1}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} first of two conditions of the independent variable

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition2}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} second of two conditions of the independent variable

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with subject identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with group identifiers (independent variable)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be dropped from the dataframe

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{paired}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} paired or unpaired samples

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{correction}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method of pvalue correction see apply\_pvalue\_correction for methods

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate for multiple hypothesis correction

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{permutations}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of permutations used to estimate false discovery rates.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{is\_logged}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} data is log\sphinxhyphen{}transformed

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{non\_par}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, normality and variance equality assumptions are checked and non\sphinxhyphen{}parametric test Mann Whitney U test if not passed

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns ‘identifier’, ‘group1’, ‘group2’, ‘mean(group1)’, ‘mean(group2)’, ‘std(group1)’, ‘std(group2)’, ‘Log2FC’, ‘FC’, ‘rejected’, ‘T\sphinxhyphen{}statistics’, ‘p\sphinxhyphen{}value’, ‘correction’, ‘\sphinxhyphen{}log10 p\sphinxhyphen{}value’, and ‘method’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}ttest}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{condition1}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{condition2}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=} \PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{paired}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{correction}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fdr\PYGZus{}bh}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{permutations}\PYG{o}{=}\PYG{l+m+mi}{50}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{define\_samr\_method() (in module analytics\_core.analytics.analytics)@\spxentry{define\_samr\_method()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.define_samr_method}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{define\_samr\_method}}}{\emph{df}, \emph{subject}, \emph{group}, \emph{drop\_cols}}{}
Method to identify the correct problem type to run with SAMR
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with subject identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with group identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{droop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} columns to be dropped

\end{itemize}

\item[{Returns}] \leavevmode
tuple with the method to be used (One Class, Two class paired, Two class unpaired or Multiclass) and the labels (conditions)

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{method}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{define\PYGZus{}samr\PYGZus{}method}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{subject}\PYG{p}{,} \PYG{n}{group}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_pvalue\_from\_tstats() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_pvalue\_from\_tstats()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_pvalue_from_tstats}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_pvalue\_from\_tstats}}}{\emph{tstat}, \emph{dfn}, \emph{dfk}}{}
Calculate two\sphinxhyphen{}tailed p\sphinxhyphen{}values from T\sphinxhyphen{} or F\sphinxhyphen{}statistics.

tstat: T/F distribution
dfn: degrees of freedrom \sphinxstyleemphasis{n} (values) per protein (keys), i.e. number of obervations \sphinxhyphen{} number of groups (dict)
dfk: degrees of freedrom \sphinxstyleemphasis{n} (values) per protein (keys), i.e. number of groups \sphinxhyphen{} 1 (dict)

\end{fulllineitems}

\index{run\_samr() (in module analytics\_core.analytics.analytics)@\spxentry{run\_samr()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_samr}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_samr}}}{\emph{df, subject=\textquotesingle{}subject\textquotesingle{}, group=\textquotesingle{}group\textquotesingle{}, drop\_cols={[}\textquotesingle{}subject\textquotesingle{}, \textquotesingle{}sample\textquotesingle{}{]}, alpha=0.05, s0=\textquotesingle{}null\textquotesingle{}, permutations=250, fc=0, is\_logged=True, localfdr=False}}{}
Python adaptation of the ‘samr’ R package for statistical tests with permutation\sphinxhyphen{}based correction and s0 parameter.
For more information visit \sphinxurl{https://cran.r-project.org/web/packages/samr/samr.pdf}.
The method only runs if R is installed and permutations is higher than 0, otherwise ANOVA.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns (with additional columns ‘group’, ‘sample’ and ‘subject’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with subject identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column with group identifiers

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} columnlabels to be dropped from the dataframe

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{alpha}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} error rate for multiple hypothesis correction

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{s0}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} exchangeability factor for denominator of test statistic

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{permutations}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of permutations used to estimate false discovery rates. If number of permutations is equal to zero, the function will run anova with FDR Benjamini/Hochberg correction.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{fc}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} minimum fold change to define practical significance (needed when computing delta table)

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns ‘identifier’, ‘group1’, ‘group2’, ‘mean(group1)’, ‘mean(group2)’, ‘Log2FC’, ‘FC’, ‘T\sphinxhyphen{}statistics’, ‘p\sphinxhyphen{}value’, ‘padj’, ‘correction’, ‘\sphinxhyphen{}log10 p\sphinxhyphen{}value’, ‘rejected’ and ‘method’

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}samr}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{s0}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{permutations}\PYG{o}{=}\PYG{l+m+mi}{250}\PYG{p}{,} \PYG{n}{fc}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_discriminant\_lines() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_discriminant\_lines()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_discriminant_lines}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_discriminant\_lines}}}{\emph{result}}{}
\end{fulllineitems}

\index{run\_fisher() (in module analytics\_core.analytics.analytics)@\spxentry{run\_fisher()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_fisher}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_fisher}}}{\emph{group1}, \emph{group2}, \emph{alternative=\textquotesingle{}two\sphinxhyphen{}sided\textquotesingle{}}}{}
annotated   not\sphinxhyphen{}annotated
group1      a               b
group2      c               d
————————————

group1 = {[}a, b{]}
group2 = {[}c, d{]}

odds, pvalue = stats.fisher\_exact({[}{[}a, b{]}, {[}c, d{]}{]})

\end{fulllineitems}

\index{run\_site\_regulation\_enrichment() (in module analytics\_core.analytics.analytics)@\spxentry{run\_site\_regulation\_enrichment()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_site_regulation_enrichment}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_site\_regulation\_enrichment}}}{\emph{regulation\_data, annotation, identifier=\textquotesingle{}identifier\textquotesingle{}, groups={[}\textquotesingle{}group1\textquotesingle{}, \textquotesingle{}group2\textquotesingle{}{]}, annotation\_col=\textquotesingle{}annotation\textquotesingle{}, reject\_col=\textquotesingle{}rejected\textquotesingle{}, group\_col=\textquotesingle{}group\textquotesingle{}, method=\textquotesingle{}fisher\textquotesingle{}, regex=\textquotesingle{}(\textbackslash{}\textbackslash{}w+\textasciitilde{}.+)\_\textbackslash{}\textbackslash{}w\textbackslash{}\textbackslash{}d+\textbackslash{}\textbackslash{}\sphinxhyphen{}\textbackslash{}\textbackslash{}w+\textquotesingle{}, correction=\textquotesingle{}fdr\_bh\textquotesingle{}}}{}
This function runs a simple enrichment analysis for significantly regulated protein sites in a dataset.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{regulation\_data}} \textendash{} pandas dataframe resulting from differential regulation analysis.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{annotation}} \textendash{} pandas dataframe with annotations for features (columns: ‘annotation’, ‘identifier’ (feature identifiers), and ‘source’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column from annotation containing feature identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{groups}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column names from regulation\_data containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{annotation\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column from annotation containing annotation terms.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{reject\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column from regulatio\_data containing boolean for rejected null hypothesis.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column name for new column in annotation dataframe determining if feature belongs to foreground or background.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method used to compute enrichment (only ‘fisher’ is supported currently).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{regex}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} how to extract the annotated identifier from the site identifier

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns: ‘terms’, ‘identifiers’, ‘foreground’, ‘background’, ‘pvalue’, ‘padj’ and ‘rejected’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}site\PYGZus{}regulation\PYGZus{}enrichment}\PYG{p}{(}\PYG{n}{regulation\PYGZus{}data}\PYG{p}{,} \PYG{n}{annotation}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{identifier}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{groups}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{annotation\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{annotation}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{reject\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{rejected}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fisher}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{match}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{(}\PYG{l+s+s2}{\PYGZbs{}}\PYG{l+s+s2}{w+\PYGZti{}.+)\PYGZus{}}\PYG{l+s+s2}{\PYGZbs{}}\PYG{l+s+s2}{w}\PYG{l+s+s2}{\PYGZbs{}}\PYG{l+s+s2}{d+}\PYG{l+s+s2}{\PYGZbs{}}\PYG{l+s+s2}{\PYGZhy{}}\PYG{l+s+s2}{\PYGZbs{}}\PYG{l+s+s2}{w+}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{run\_regulation\_enrichment() (in module analytics\_core.analytics.analytics)@\spxentry{run\_regulation\_enrichment()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_regulation_enrichment}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_regulation\_enrichment}}}{\emph{regulation\_data, annotation, identifier=\textquotesingle{}identifier\textquotesingle{}, groups={[}\textquotesingle{}group1\textquotesingle{}, \textquotesingle{}group2\textquotesingle{}{]}, annotation\_col=\textquotesingle{}annotation\textquotesingle{}, reject\_col=\textquotesingle{}rejected\textquotesingle{}, group\_col=\textquotesingle{}group\textquotesingle{}, method=\textquotesingle{}fisher\textquotesingle{}, correction=\textquotesingle{}fdr\_bh\textquotesingle{}}}{}
This function runs a simple enrichment analysis for significantly regulated features in a dataset.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{regulation\_data}} \textendash{} pandas dataframe resulting from differential regulation analysis.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{annotation}} \textendash{} pandas dataframe with annotations for features (columns: ‘annotation’, ‘identifier’ (feature identifiers), and ‘source’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column from annotation containing feature identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{groups}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column names from regulation\_data containing group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{annotation\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column from annotation containing annotation terms.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{reject\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column from regulatio\_data containing boolean for rejected null hypothesis.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column name for new column in annotation dataframe determining if feature belongs to foreground or background.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method used to compute enrichment (only ‘fisher’ is supported currently).

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with columns: ‘terms’, ‘identifiers’, ‘foreground’, ‘background’, ‘pvalue’, ‘padj’ and ‘rejected’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}regulation\PYGZus{}enrichment}\PYG{p}{(}\PYG{n}{regulation\PYGZus{}data}\PYG{p}{,} \PYG{n}{annotation}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{identifier}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{groups}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{annotation\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{annotation}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{reject\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{rejected}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fisher}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{run\_enrichment() (in module analytics\_core.analytics.analytics)@\spxentry{run\_enrichment()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_enrichment}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_enrichment}}}{\emph{data}, \emph{foreground\_id}, \emph{background\_id}, \emph{annotation\_col=\textquotesingle{}annotation\textquotesingle{}}, \emph{group\_col=\textquotesingle{}group\textquotesingle{}}, \emph{identifier\_col=\textquotesingle{}identifier\textquotesingle{}}, \emph{method=\textquotesingle{}fisher\textquotesingle{}}, \emph{correction=\textquotesingle{}fdr\_bh\textquotesingle{}}}{}
Computes enrichment of the foreground relative to a given backgroung, using Fisher’s exact test, and corrects for multiple hypothesis testing.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with annotations for dataset features (columns: ‘annotation’, ‘identifier’, ‘source’, ‘group’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{foreground\_id}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} group identifier of features that belong to the foreground.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{background\_id}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} group identifier of features that belong to the background.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{annotation\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the column containing annotation terms.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of column containing the group identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of column containing dependent variables identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method used to compute enrichment (only ‘fisher’ is supported currently).

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with annotation terms, features, number of foregroung/background features in each term, p\sphinxhyphen{}values and corrected p\sphinxhyphen{}values (columns: ‘terms’, ‘identifiers’, ‘foreground’, ‘background’, ‘pvalue’, ‘padj’ and ‘rejected’).

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}enrichment}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{foreground}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{foreground}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{background}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{background}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{foreground\PYGZus{}pop}\PYG{o}{=}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{foreground\PYGZus{}list}\PYG{p}{)}\PYG{p}{,} \PYG{n}{background\PYGZus{}pop}\PYG{o}{=}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{background\PYGZus{}list}\PYG{p}{)}\PYG{p}{,} \PYG{n}{annotation\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{annotation}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{identifier\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{identifier}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{method}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fisher}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{calculate\_fold\_change() (in module analytics\_core.analytics.analytics)@\spxentry{calculate\_fold\_change()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.calculate_fold_change}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{calculate\_fold\_change}}}{\emph{df}, \emph{condition1}, \emph{condition2}}{}
Calculates fold\sphinxhyphen{}changes between two groups for all proteins in a dataframe.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition1}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of first group.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition2}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of second group.

\end{itemize}

\item[{Returns}] \leavevmode
Numpy array.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}fold\PYGZus{}change}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{cohen\_d() (in module analytics\_core.analytics.analytics)@\spxentry{cohen\_d()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.cohen_d}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{cohen\_d}}}{\emph{df}, \emph{condition1}, \emph{condition2}, \emph{ddof=0}}{}
Calculates Cohen’s d effect size based on the distance between two means, measured in standard deviations.
For more information visit \sphinxurl{https://docs.scipy.org/doc/numpy/reference/generated/numpy.nanstd.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition1}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of first group.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition2}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of second group.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ddof}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} means Delta Degrees of Freedom.

\end{itemize}

\item[{Returns}] \leavevmode
Numpy array.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{cohen\PYGZus{}d}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{ddof}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{hedges\_g() (in module analytics\_core.analytics.analytics)@\spxentry{hedges\_g()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.hedges_g}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{hedges\_g}}}{\emph{df}, \emph{condition1}, \emph{condition2}, \emph{ddof=0}}{}
Calculates Hedges’ g effect size (more accurate for sample sizes below 20 than Cohen’s d).
For more information visit \sphinxurl{https://docs.scipy.org/doc/numpy/reference/generated/numpy.nanstd.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and protein identifiers as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition1}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of first group.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{condition2}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of second group.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ddof}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} means Delta Degrees of Freedom.

\end{itemize}

\item[{Returns}] \leavevmode
Numpy array.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{hedges\PYGZus{}g}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{ddof}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{run\_mapper() (in module analytics\_core.analytics.analytics)@\spxentry{run\_mapper()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_mapper}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_mapper}}}{\emph{data, lenses={[}\textquotesingle{}l2norm\textquotesingle{}{]}, n\_cubes=15, overlap=0.5, n\_clusters=3, linkage=\textquotesingle{}complete\textquotesingle{}, affinity=\textquotesingle{}correlation\textquotesingle{}}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lenses}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n\_cubes}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{overlap}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{n\_clusters}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{linkage}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{affinity}} \textendash{} 

\end{itemize}

\item[{Returns}] \leavevmode


\end{description}\end{quote}

\end{fulllineitems}

\index{run\_WGCNA() (in module analytics\_core.analytics.analytics)@\spxentry{run\_WGCNA()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_WGCNA}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_WGCNA}}}{\emph{data}, \emph{drop\_cols\_exp}, \emph{drop\_cols\_cli}, \emph{RsquaredCut=0.8}, \emph{networkType=\textquotesingle{}unsigned\textquotesingle{}}, \emph{minModuleSize=30}, \emph{deepSplit=2}, \emph{pamRespectsDendro=False}, \emph{merge\_modules=True}, \emph{MEDissThres=0.25}, \emph{verbose=0}, \emph{sd\_cutoff=0}}{}
Runs an automated weighted gene co\sphinxhyphen{}expression network analysis (WGCNA), using input proteomics/transcriptomics/genomics and clinical variables data.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary of pandas dataframes with processed clinical and experimental datasets

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols\_exp}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column names to be removed from the experimental dataset.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols\_cli}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column names to be removed from the clinical dataset.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{RsquaredCut}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} desired minimum scale free topology fitting index R\textasciicircum{}2.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{networkType}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} network type (‘unsigned’, ‘signed’, ‘signed hybrid’, ‘distance’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{minModuleSize}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} minimum module size.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{deepSplit}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} provides a rough control over sensitivity to cluster splitting, the higher the value (with ‘hybrid’ method) or if True (with ‘tree’ method), the more and smaller modules.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pamRespectsDendro}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} only used for method ‘hybrid’. Objects and small modules will only be assigned to modules that belong to the same branch in the dendrogram structure.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{merge\_modules}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, very similar modules are merged.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{MEDissThres}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} maximum dissimilarity (i.e., 1\sphinxhyphen{}correlation) that qualifies modules for merging.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{verbose}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} integer level of verbosity. Zero means silent, higher values make the output progressively more and more verbose.

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with multiple pandas dataframes.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}WGCNA}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols\PYGZus{}exp}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{index}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols\PYGZus{}cli}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{biological\PYGZus{}sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{index}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{RsquaredCut}\PYG{o}{=}\PYG{l+m+mf}{0.8}\PYG{p}{,} \PYG{n}{networkType}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{unsigned}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minModuleSize}\PYG{o}{=}\PYG{l+m+mi}{30}\PYG{p}{,} \PYG{n}{deepSplit}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{pamRespectsDendro}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{merge\PYGZus{}modules}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{MEDissThres}\PYG{o}{=}\PYG{l+m+mf}{0.25}\PYG{p}{,} \PYG{n}{verbose}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{most\_central\_edge() (in module analytics\_core.analytics.analytics)@\spxentry{most\_central\_edge()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.most_central_edge}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{most\_central\_edge}}}{\emph{G}}{}
Compute the eigenvector centrality for the graph G, and finds the highest value.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{G}} (\sphinxstyleliteralemphasis{\sphinxupquote{graph}}) \textendash{} networkx graph

\item[{Returns}] \leavevmode
Highest eigenvector centrality value.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{float}

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_louvain\_partitions() (in module analytics\_core.analytics.analytics)@\spxentry{get\_louvain\_partitions()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_louvain_partitions}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_louvain\_partitions}}}{\emph{G}, \emph{weight}}{}
Computes the partition of the graph nodes which maximises the modularity (or try..) using the Louvain heuristices. For more information visit \sphinxurl{https://python-louvain.readthedocs.io/en/latest/api.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{G}} (\sphinxstyleliteralemphasis{\sphinxupquote{graph}}) \textendash{} networkx graph which is decomposed.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{weight}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} the key in graph to use as weight.

\end{itemize}

\item[{Returns}] \leavevmode
The partition, with communities numbered from 0 to number of communities.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{dict}

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_network\_communities() (in module analytics\_core.analytics.analytics)@\spxentry{get\_network\_communities()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_network_communities}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_network\_communities}}}{\emph{graph}, \emph{args}}{}
Finds communities in a graph using different methods. For more information on the methods visit:
\begin{itemize}
\item {} 
\sphinxurl{https://networkx.github.io/documentation/latest/reference/algorithms/generated/networkx.algorithms.community.modularity\_max.greedy\_modularity\_communities.html}

\item {} 
\sphinxurl{https://networkx.github.io/documentation/networkx-2.0/reference/algorithms/generated/networkx.algorithms.community.asyn\_lpa.asyn\_lpa\_communities.html}

\item {} 
\sphinxurl{https://networkx.github.io/documentation/latest/reference/algorithms/generated/networkx.algorithms.community.centrality.girvan\_newman.html}

\item {} 
\sphinxurl{https://networkx.github.io/documentation/latest/reference/generated/networkx.convert\_matrix.to\_pandas\_adjacency.html}

\end{itemize}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{graph}}) \textendash{} networkx graph

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} config file arguments

\end{itemize}

\item[{Returns}] \leavevmode
Dictionary of nodes and which community they belong to (from 0 to number of communities).

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_publications\_abstracts() (in module analytics\_core.analytics.analytics)@\spxentry{get\_publications\_abstracts()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.get_publications_abstracts}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{get\_publications\_abstracts}}}{\emph{data, publication\_col=\textquotesingle{}publication\textquotesingle{}, join\_by={[}\textquotesingle{}publication\textquotesingle{}, \textquotesingle{}Proteins\textquotesingle{}, \textquotesingle{}Diseases\textquotesingle{}{]}, index=\textquotesingle{}PMID\textquotesingle{}}}{}
Accesses NCBI PubMed over the WWW and retrieves the abstracts corresponding to a list of one or more PubMed IDs.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe of diseases and publications linked to a list of proteins (columns: ‘Diseases’, ‘Proteins’, ‘linkout’ and ‘publication’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{publication\_col}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing PubMed ids.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{join\_by}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column labels to be kept from the input dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column label containing PubMed ids from the NCBI retrieved data.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with publication information and columns ‘PMID’, ‘abstract’, ‘authors’, ‘date’, ‘journal’, ‘keywords’, ‘title’, ‘url’, ‘Proteins’ and ‘Diseases’.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}publications\PYGZus{}abstracts}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{publication\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{publication}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{join\PYGZus{}by}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{publication}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Proteins}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Diseases}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{index}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{PMID}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{eta\_squared() (in module analytics\_core.analytics.analytics)@\spxentry{eta\_squared()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.eta_squared}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{eta\_squared}}}{\emph{aov}}{}
Calculates the effect size using Eta\sphinxhyphen{}squared.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{aov}} \textendash{} pandas dataframe with anova results from statsmodels.

\item[{Returns}] \leavevmode
Pandas dataframe with additional Eta\sphinxhyphen{}squared column.

\end{description}\end{quote}

\end{fulllineitems}

\index{omega\_squared() (in module analytics\_core.analytics.analytics)@\spxentry{omega\_squared()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.omega_squared}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{omega\_squared}}}{\emph{aov}}{}
Calculates the effect size using Omega\sphinxhyphen{}squared.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{aov}} \textendash{} pandas dataframe with anova results from statsmodels.

\item[{Returns}] \leavevmode
Pandas dataframe with additional Omega\sphinxhyphen{}squared column.

\end{description}\end{quote}

\end{fulllineitems}

\index{run\_two\_way\_anova() (in module analytics\_core.analytics.analytics)@\spxentry{run\_two\_way\_anova()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_two_way_anova}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_two\_way\_anova}}}{\emph{df, drop\_cols={[}\textquotesingle{}sample\textquotesingle{}{]}, subject=\textquotesingle{}subject\textquotesingle{}, group={[}\textquotesingle{}group\textquotesingle{}, \textquotesingle{}secondary\_group\textquotesingle{}{]}}}{}
Run a 2\sphinxhyphen{}way ANOVA when data{[}‘secondary\_group’{]} is not empty
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} processed pandas dataframe with samples as rows, and proteins and groups as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column names to drop from dataframe

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subject}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} column name containing subject identifiers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} column names corresponding to independent variable groups

\end{itemize}

\item[{Returns}] \leavevmode
Two dataframes, anova results and residuals.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}two\PYGZus{}way\PYGZus{}anova}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{drop\PYGZus{}cols}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{subject}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{secondary\PYGZus{}group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{merge\_for\_polar() (in module analytics\_core.analytics.analytics)@\spxentry{merge\_for\_polar()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.merge_for_polar}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{merge\_for\_polar}}}{\emph{regulation\_data}, \emph{regulators}, \emph{identifier\_col=\textquotesingle{}identifier\textquotesingle{}}, \emph{group\_col=\textquotesingle{}group\textquotesingle{}}, \emph{theta\_col=\textquotesingle{}modifier\textquotesingle{}}, \emph{aggr\_func=\textquotesingle{}mean\textquotesingle{}}, \emph{normalize=True}}{}
\end{fulllineitems}

\index{run\_qc\_markers\_analysis() (in module analytics\_core.analytics.analytics)@\spxentry{run\_qc\_markers\_analysis()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_qc_markers_analysis}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_qc\_markers\_analysis}}}{\emph{data, qc\_markers, sample\_col=\textquotesingle{}sample\textquotesingle{}, group\_col=\textquotesingle{}group\textquotesingle{}, drop\_cols={[}\textquotesingle{}subject\textquotesingle{}{]}, identifier\_col=\textquotesingle{}identifier\textquotesingle{}, qcidentifier\_col=\textquotesingle{}identifier\textquotesingle{}, qcclass\_col=\textquotesingle{}class\textquotesingle{}}}{}
\end{fulllineitems}

\index{run\_snf() (in module analytics\_core.analytics.analytics)@\spxentry{run\_snf()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_snf}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_snf}}}{\emph{df\_dict}, \emph{clusters}, \emph{distance\_metric}, \emph{K\_affinity}, \emph{mu\_affinity}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df\_dict}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{clusters}} \textendash{} 

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{run\_km() (in module analytics\_core.analytics.analytics)@\spxentry{run\_km()}\spxextra{in module analytics\_core.analytics.analytics}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.analytics.run_km}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.analytics.}}\sphinxbfcode{\sphinxupquote{run\_km}}}{\emph{data}, \emph{time\_col}, \emph{event\_col}, \emph{group\_col}, \emph{args=\{\}}}{}
\end{fulllineitems}



\paragraph{WGCNA analytics}
\label{\detokenize{_autosummary/analytics_core.analytics:module-analytics_core.analytics.wgcnaAnalysis}}\label{\detokenize{_autosummary/analytics_core.analytics:wgcna-analytics}}\index{analytics\_core.analytics.wgcnaAnalysis (module)@\spxentry{analytics\_core.analytics.wgcnaAnalysis}\spxextra{module}}\index{get\_data() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{get\_data()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.get_data}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{get\_data}}}{\emph{data, drop\_cols\_exp={[}\textquotesingle{}subject\textquotesingle{}, \textquotesingle{}group\textquotesingle{}, \textquotesingle{}sample\textquotesingle{}, \textquotesingle{}index\textquotesingle{}{]}, drop\_cols\_cli={[}\textquotesingle{}subject\textquotesingle{}, \textquotesingle{}group\textquotesingle{}, \textquotesingle{}biological\_sample\textquotesingle{}, \textquotesingle{}index\textquotesingle{}{]}, sd\_cutoff=0}}{}
This function cleanes up and formats experimental and clinical data into similarly shaped dataframes.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary with processed clinical and proteomics datasets.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols\_exp}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of columns to drop from processed experimental (protemics/rna\sphinxhyphen{}seq/dna\sphinxhyphen{}seq) dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{drop\_cols\_cli}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of columns to drop from processed clinical dataframe.

\end{itemize}

\item[{Returns}] \leavevmode
Dictionary with experimental and clinical dataframes (keys are the same as in the input dictionary).

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_dendrogram() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{get\_dendrogram()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.get_dendrogram}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{get\_dendrogram}}}{\emph{df}, \emph{labels}, \emph{distfun=\textquotesingle{}euclidean\textquotesingle{}}, \emph{linkagefun=\textquotesingle{}ward\textquotesingle{}}, \emph{div\_clusters=False}, \emph{fcluster\_method=\textquotesingle{}distance\textquotesingle{}}, \emph{fcluster\_cutoff=15}}{}
This function calculates the distance matrix and performs hierarchical cluster analysis on a set of dissimilarities and methods for analyzing it.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples/subjects as index and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{labels}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} labels for the leaves of the tree.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{distfun}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} distance measure to be used (‘euclidean’, ‘maximum’, ‘manhattan’, ‘canberra’, ‘binary’, ‘minkowski’ or ‘jaccard’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{linkagefun}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} hierarchical/agglomeration method to be used (‘single’, ‘complete’, ‘average’, ‘weighted’, ‘centroid’, ‘median’ or ‘ward’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{div\_clusters}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} dividing dendrogram leaves into clusters (True or False).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{fcluster\_method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} criterion to use in forming flat clusters.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{fcluster\_cutoff}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} maximum cophenetic distance between observations in each cluster.

\end{itemize}

\item[{Returns}] \leavevmode
Dictionary of data structures computed to render the dendrogram. Keys: ‘icoords’, ‘dcoords’, ‘ivl’ and ‘leaves’. If div\_clusters is used, it will also return a dictionary of each cluster and respective leaves.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_clusters\_elements() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{get\_clusters\_elements()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.get_clusters_elements}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{get\_clusters\_elements}}}{\emph{linkage\_matrix}, \emph{fcluster\_method}, \emph{fcluster\_cutoff}, \emph{labels}}{}
This function implements the generation of flat clusters from an hierarchical clustering with the same interface as scipy.cluster.hierarchy.fcluster.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{linkage\_matrix}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} hierarchical clustering encoded with a linkage matrix.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{fcluster\_method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} criterion to use in forming flat clusters (‘inconsistent’, ‘distance’, ‘maxclust’, ‘monocrit’, ‘maxclust\_monocrit’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{fcluster\_cutoff}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} maximum cophenetic distance between observations in each cluster.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{labels}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} labels for the leaves of the dendrogram.

\end{itemize}

\item[{Returns}] \leavevmode
A dictionary where keys are the cluster numbers and values are the dendrogram leaves.

\end{description}\end{quote}

\end{fulllineitems}

\index{filter\_df\_by\_cluster() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{filter\_df\_by\_cluster()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.filter_df_by_cluster}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{filter\_df\_by\_cluster}}}{\emph{df}, \emph{clusters}, \emph{number}}{}
Select only the members of a defined cluster.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples/subjects as index and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{clusters}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} clusters dictionary from get\_dendrogram function if div\_clusters option was True.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{number}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} cluster number (key).

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with all the features (columns) and samples/subjects belonging to the defined cluster (index).

\end{description}\end{quote}

\end{fulllineitems}

\index{df\_sort\_by\_dendrogram() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{df\_sort\_by\_dendrogram()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.df_sort_by_dendrogram}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{df\_sort\_by\_dendrogram}}}{\emph{df}, \emph{Z\_dendrogram}}{}
Reorders pandas dataframe by index and according to the dendrogram list of leaf nodes labels.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with the labels to be reordered as index.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{Z\_dendrogram}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary of data structures computed to render the dendrogram. Keys: ‘icoords’, ‘dcoords’, ‘ivl’ and ‘leaves’.

\end{itemize}

\item[{Returns}] \leavevmode
Reordered pandas dataframe.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_percentiles\_heatmap() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{get\_percentiles\_heatmap()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.get_percentiles_heatmap}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{get\_percentiles\_heatmap}}}{\emph{df}, \emph{Z\_dendrogram}, \emph{bydendro=True}, \emph{bycols=False}}{}
This function transforms the absolute values in each row or column (option ‘bycols’) into relative values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples/subjects as index and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{Z\_dendrogram}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary of data structures computed to render the dendrogram. Keys: ‘icoords’, ‘dcoords’, ‘ivl’ and ‘leaves’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{bydendro}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if labels should be ordered according to dendrogram list of leaf nodes labels set to True, otherwise set to False.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{bycols}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} relative values calculated across rows (samples) then set to False. Calculation performed across columns (features) set to True.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_miss\_values\_df() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{get\_miss\_values\_df()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.get_miss_values_df}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{get\_miss\_values\_df}}}{\emph{data}}{}
Proccesses pandas dataframe so missing values can be plotted in heatmap with specific color.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe.

\item[{Returns}] \leavevmode
Pandas dataframe with missing values as integer 1, and originally valid values as NaN.

\end{description}\end{quote}

\end{fulllineitems}

\index{paste\_matrices() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{paste\_matrices()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.paste_matrices}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{paste\_matrices}}}{\emph{matrix1}, \emph{matrix2}, \emph{rows}, \emph{cols}}{}
Takes two matrices with analog shapes and concatenates each value in matrix 1 with corresponding one in matrix 2, returning a single pandas dataframe.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{matrix1}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} input 1

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{matrix2}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} input 2

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe.

\end{description}\end{quote}

\end{fulllineitems}

\index{cutreeDynamic() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{cutreeDynamic()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.cutreeDynamic}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{cutreeDynamic}}}{\emph{distmatrix}, \emph{linkagefun=\textquotesingle{}average\textquotesingle{}}, \emph{minModuleSize=50}, \emph{method=\textquotesingle{}hybrid\textquotesingle{}}, \emph{deepSplit=2}, \emph{pamRespectsDendro=False}, \emph{distfun=None}}{}
This function implements the R cutreeDynamic wrapper in Python, provinding an access point for methods of adaptive branh pruning of hierarchical clustering dendrograms.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{distfun}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} distance measure to be used (‘euclidean’, ‘maximum’, ‘manhattan’, ‘canberra’, ‘binary’, ‘minkowski’ or ‘jaccard’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{linkagefun}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} hierarchical/agglomeration method to be used (‘single’, ‘complete’, ‘average’, ‘weighted’, ‘centroid’, ‘median’ or ‘ward’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{minModuleSize}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} minimum module size.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method to use (‘hybrid’ or ‘tree’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{deepSplit}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} provides a rough control over sensitivity to cluster splitting, the higher the value (with ‘hybrid’ method) or if True (with ‘tree’ method), the more and smaller modules.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pamRespectsDendro}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} only used for method ‘hybrid’. Objects and small modules will only be assigned to modules that belong to the same branch in the dendrogram structure.

\end{itemize}

\item[{Returns}] \leavevmode
Numpy array of numerical labels giving assignment of objects to modules. Unassigned objects are labeled 0, the largest module has label 1, next largest 2 etc.

\end{description}\end{quote}

\end{fulllineitems}

\index{build\_network() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{build\_network()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.build_network}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{build\_network}}}{\emph{data}, \emph{softPower=6}, \emph{networkType=\textquotesingle{}unsigned\textquotesingle{}}, \emph{linkagefun=\textquotesingle{}average\textquotesingle{}}, \emph{method=\textquotesingle{}hybrid\textquotesingle{}}, \emph{minModuleSize=50}, \emph{deepSplit=2}, \emph{pamRespectsDendro=False}, \emph{merge\_modules=True}, \emph{MEDissThres=0.4}, \emph{verbose=0}}{}
Weighted gene network construction and module detection. Calculates co\sphinxhyphen{}expression similarity and adjacency, topological overlap matrix (TOM) and clusters features in modules.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{softPower}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} soft\sphinxhyphen{}thresholding power.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{networkType}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} network type (‘unsigned’, ‘signed’, ‘signed hybrid’, ‘distance’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{linkagefun}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} hierarchical/agglomeration method to be used (‘single’, ‘complete’, ‘average’, ‘weighted’, ‘centroid’, ‘median’ or ‘ward’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} method to use (‘hybrid’ or ‘tree’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{minModuleSize}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} minimum module size.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pamRespectsDendro}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} only used for method ‘hybrid’. Objects and small modules will only be assigned to modules that belong to the same branch in the dendrogram structure.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{merge\_modules}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, very similar modules are merged.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{MEDissThres}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} maximum dissimilarity (i.e., 1\sphinxhyphen{}correlation) that qualifies modules for merging.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{verbose}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} integer level of verbosity. Zero means silent, higher values make the output progressively more and more verbose.

\end{itemize}

\item[{Paran int deepSplit}] \leavevmode
provides a rough control over sensitivity to cluster splitting, the higher the value (with ‘hybrid’ method) or if True (with ‘tree’ method), the more and smaller modules.

\item[{Returns}] \leavevmode
Tuple with TOM dissimilarity pandas dataframe, numpy array with module colors per experimental feature.

\end{description}\end{quote}

\end{fulllineitems}

\index{pick\_softThreshold() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{pick\_softThreshold()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.pick_softThreshold}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{pick\_softThreshold}}}{\emph{data}, \emph{RsquaredCut=0.8}, \emph{networkType=\textquotesingle{}unsigned\textquotesingle{}}, \emph{verbose=0}}{}
Analysis of scale free topology for multiple soft thresholding powers. Aids the user in choosing a proper soft\sphinxhyphen{}thresholding power for network construction.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{RsquaredCut}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} desired minimum scale free topology fitting index R\textasciicircum{}2.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{networkType}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} network type (‘unsigned’, ‘signed’, ‘signed hybrid’, ‘distance’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{verbose}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} integer level of verbosity. Zero means silent, higher values make the output progressively more and more verbose.

\end{itemize}

\item[{Returns}] \leavevmode
Estimated appropriate soft\sphinxhyphen{}thresholding power: the lowest power for which the scale free topology fit R\textasciicircum{}2 exceeds RsquaredCut.

\item[{Return type}] \leavevmode
\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{int}

\end{description}\end{quote}

\end{fulllineitems}

\index{identify\_module\_colors() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{identify\_module\_colors()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.identify_module_colors}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{identify\_module\_colors}}}{\emph{matrix}, \emph{linkagefun=\textquotesingle{}average\textquotesingle{}}, \emph{method=\textquotesingle{}hybrid\textquotesingle{}}, \emph{minModuleSize=30}, \emph{deepSplit=2}, \emph{pamRespectsDendro=False}}{}
Identifies co\sphinxhyphen{}expression modules and converts the numeric labels into colors.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{matrix}} \textendash{} dissimilarity structure as produced by R.stats dist.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{minModuleSize}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} minimum module size.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{deepSplit}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} provides a rough control over sensitivity to cluster splitting, the higher the value (with ‘hybrid’ method) or if True (with ‘tree’ method), the more and smaller modules.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pamRespectsDendro}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} only used for method ‘hybrid’. Objects and small modules will only be assigned to modules that belong to the same branch in the dendrogram structure.

\end{itemize}

\item[{Returns}] \leavevmode
Numpy array of strings with module color of each experimental feature.

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_module\_eigengenes() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{calculate\_module\_eigengenes()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.calculate_module_eigengenes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{calculate\_module\_eigengenes}}}{\emph{data}, \emph{modColors}, \emph{softPower=6}, \emph{dissimilarity=True}}{}
Calculates modules eigengenes to quantify co\sphinxhyphen{}expression similarity of entire modules.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{modColors}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} array (numeric, character or a factor) attributing module colors to each feature in the experimental dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{softPower}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} soft\sphinxhyphen{}thresholding power.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dissimilarity}} \textendash{} calculates dissimilarity of module eigengenes.

\end{itemize}

\item[{Returns}] \leavevmode
Pandas dataframe with calculated module eigengenes. If dissimilarity is set to True, returns a tuple with two pandas dataframes, the first with the module eigengenes and the second with the eigengenes dissimilarity.

\end{description}\end{quote}

\end{fulllineitems}

\index{merge\_similar\_modules() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{merge\_similar\_modules()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.merge_similar_modules}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{merge\_similar\_modules}}}{\emph{data}, \emph{modColors}, \emph{MEDissThres=0.4}, \emph{verbose=0}}{}
Merges modules in co\sphinxhyphen{}expression network that are too close as measured by the correlation of their eigengenes.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{modColors}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} array (numeric, character or a factor) attributing module colors to each feature in the experimental dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{verbose}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} integer level of verbosity. Zero means silent, higher values make the output progressively more and more verbose.

\end{itemize}

\item[{Para, float MEDissThres}] \leavevmode
maximum dissimilarity (i.e., 1\sphinxhyphen{}correlation) that qualifies modules for merging.

\item[{Returns}] \leavevmode
Tuple containing pandas dataframe with eigengenes of the new merged modules, and array with module colors of each expeirmental feature.

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_ModuleTrait\_correlation() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{calculate\_ModuleTrait\_correlation()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.calculate_ModuleTrait_correlation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{calculate\_ModuleTrait\_correlation}}}{\emph{df\_exp}, \emph{df\_traits}, \emph{MEs}}{}
Correlates eigengenes with external traits in order to identify the most significant module\sphinxhyphen{}trait associations.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df\_exp}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df\_traits}} \textendash{} pandas dataframe containing clinical data, with samples/subjects as rows and clinical traits as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{MEs}} \textendash{} pandas dataframe with module eigengenes.

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with two pandas datafames, first the correlation between all module eigengenes and all clinical traits, second a dataframe with concatenated correlation and p\sphinxhyphen{}value used for heatmap annotation.

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_ModuleMembership() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{calculate\_ModuleMembership()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.calculate_ModuleMembership}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{calculate\_ModuleMembership}}}{\emph{data}, \emph{MEs}}{}
For each module, calculates the correlation of the module eigengene and the feature expression profile (quantitative measure of module membership (MM)).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{MEs}} \textendash{} pandas dataframe with module eigengenes.

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with two pandas dataframes, one with module membership correlations and another with p\sphinxhyphen{}values.

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_FeatureTraitSignificance() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{calculate\_FeatureTraitSignificance()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.calculate_FeatureTraitSignificance}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{calculate\_FeatureTraitSignificance}}}{\emph{df\_exp}, \emph{df\_traits}}{}
Quantifies associations of individual experimental features with the measured clinical traits, by defining Feature Significance (FS) as the absolute value of the correlation between the feature and the trait.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df\_exp}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df\_traits}} \textendash{} pandas dataframe containing clinical data, with samples/subjects as rows and clinical traits as columns.

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with two pandas dataframes, one with feature significance correlations and another with p\sphinxhyphen{}values.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_FeaturesPerModule() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{get\_FeaturesPerModule()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.get_FeaturesPerModule}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{get\_FeaturesPerModule}}}{\emph{data}, \emph{modColors}, \emph{mode=\textquotesingle{}dictionary\textquotesingle{}}}{}
Groups all experimental features by the co\sphinxhyphen{}expression module they belong to.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{modColors}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} array (numeric, character or a factor) attributing module colors to each feature in the experimental dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{mode}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} type of the value returned by the function (‘dictionary’ or ‘dataframe’).

\end{itemize}

\item[{Returns}] \leavevmode
Depending on selected mode, returns a dictionary or dataframe with module color per experimental feature.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_ModuleFeatures() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{get\_ModuleFeatures()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.get_ModuleFeatures}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{get\_ModuleFeatures}}}{\emph{data}, \emph{modColors}, \emph{modules={[}{]}}}{}
Groups and returns a list of the experimental features clustered in specific co\sphinxhyphen{}expression modules.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{modColors}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} array (numeric, character or a factor) attributing module colors to each feature in the experimental dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{modules}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of module colors of interest.

\end{itemize}

\item[{Returns}] \leavevmode
List of lists with experimental features in each selected module.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_EigengenesTrait\_correlation() (in module analytics\_core.analytics.wgcnaAnalysis)@\spxentry{get\_EigengenesTrait\_correlation()}\spxextra{in module analytics\_core.analytics.wgcnaAnalysis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.analytics:analytics_core.analytics.wgcnaAnalysis.get_EigengenesTrait_correlation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.analytics.wgcnaAnalysis.}}\sphinxbfcode{\sphinxupquote{get\_EigengenesTrait\_correlation}}}{\emph{MEs}, \emph{data}}{}
Eigengenes are used as representative profiles of the co\sphinxhyphen{}expression modules, and correlation between them is used to quantify module similarity.
Clinical traits are added to the eigengenes to see how the traits fir into the eigengen network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{MEs}} \textendash{} pandas dataframe with module eigengenes.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe containing clinical data, with samples/subjects as rows and clinical traits as columns.

\end{itemize}

\item[{Returns}] \leavevmode
Tuple with two pandas dataframes, one with features and traits recalculates module eigengenes dissimilarity, and another with all the overall correlations.

\end{description}\end{quote}

\end{fulllineitems}



\subsubsection{Vizualization}
\label{\detokenize{_autosummary/analytics_core.viz:vizualization}}\label{\detokenize{_autosummary/analytics_core.viz::doc}}

\paragraph{Viz module}
\label{\detokenize{_autosummary/analytics_core.viz:module-analytics_core.viz.viz}}\label{\detokenize{_autosummary/analytics_core.viz:viz-module}}\index{analytics\_core.viz.viz (module)@\spxentry{analytics\_core.viz.viz}\spxextra{module}}\index{getPlotTraces() (in module analytics\_core.viz.viz)@\spxentry{getPlotTraces()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.getPlotTraces}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{getPlotTraces}}}{\emph{data}, \emph{key=\textquotesingle{}full\textquotesingle{}}, \emph{type=\textquotesingle{}lines\textquotesingle{}}, \emph{div\_factor=10010.0}, \emph{horizontal=False}}{}
This function returns traces for different kinds of plots.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} Pandas DataFrame with one variable as data.index (i.e. ‘x’) and all others as columns (i.e. ‘y’).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} ‘lines’, ‘scaled markers’, ‘bars’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{div\_factor}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} relative size of the markers.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{horizontal}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} bar orientation.

\end{itemize}

\item[{Returns}] \leavevmode
list of traces.

\end{description}\end{quote}

Exmaple 1:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{getPlotTraces}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{key}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{full}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n+nb}{type} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{lines}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{horizontal}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\end{sphinxVerbatim}

Example 2:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{getPlotTraces}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{key}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{full}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n+nb}{type} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{scaled markers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{div\PYGZus{}factor}\PYG{o}{=}\PYG{n+nb}{float}\PYG{p}{(}\PYG{l+m+mi}{10}\PYG{o}{\PYGZca{}}\PYG{l+m+mi}{3000}\PYG{p}{)}\PYG{p}{,} \PYG{n}{horizontal}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_markdown() (in module analytics\_core.viz.viz)@\spxentry{get\_markdown()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_markdown}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_markdown}}}{\emph{text}, \emph{args=\{\}}}{}
Converts a given text into a Dash Markdown component. It includes a syntax for things like bold text and italics, links, inline code snippets, lists, quotes, and more.
For more information visit \sphinxurl{https://dash.plot.ly/dash-core-components/markdown}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{text}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} markdown string (or array of strings) that adhreres to the CommonMark spec.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary with items from \sphinxurl{https://dash.plot.ly/dash-core-components/markdown}.

\end{itemize}

\item[{Returns}] \leavevmode
dash Markdown component.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_pieplot() (in module analytics\_core.viz.viz)@\spxentry{get\_pieplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_pieplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_pieplot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function plots a simple Pie plot.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas DataFrame with values to plot as columns and labels as index.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{valueCol} (str) \textendash{} name of the column with the values to be plotted.

\item {} 
\sphinxstylestrong{textCol} (str) \textendash{} name of the column containing information for the hoverinfo parameter.

\item {} 
\sphinxstylestrong{height} (str) \textendash{} height of the plot.

\item {} 
\sphinxstylestrong{width} (str) \textendash{} width of the plot.

\end{itemize}

\item[{Returns}] \leavevmode
Pieplot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_distplot() (in module analytics\_core.viz.viz)@\spxentry{get\_distplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_distplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_distplot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}~\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} 

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{group} (str) \textendash{} name of the column containing the group.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_boxplot\_grid() (in module analytics\_core.viz.viz)@\spxentry{get\_boxplot\_grid()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_boxplot_grid}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_boxplot\_grid}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function plots a boxplot in a grid based on column values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas DataFrame with columns: ‘x’ values and ‘y’ values to plot, ‘color’ and ‘facet’ (color and facet can be the same).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\item {} 
\sphinxstylestrong{x} (str) \textendash{} name of column with x values.

\item {} 
\sphinxstylestrong{y} (str) \textendash{} name of column with y values.

\item {} 
\sphinxstylestrong{color} (str) \textendash{} name of column with colors

\item {} 
\sphinxstylestrong{facet} (str) \textendash{} name of column specifying grouping

\item {} 
\sphinxstylestrong{height} (str) \textendash{} plot height.

\item {} 
\sphinxstylestrong{width} (str) \textendash{} plot width.

\end{itemize}

\item[{Returns}] \leavevmode
boxplot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}boxplot\PYGZus{}grid}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Boxplot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Title}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Boxplot}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{identifier}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{color}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{facet}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{qc\PYGZus{}class}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{cols}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_barplot() (in module analytics\_core.viz.viz)@\spxentry{get\_barplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_barplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_barplot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function plots a simple barplot.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas DataFrame with three columns: ‘name’ of the bars, ‘x’ values and ‘y’ values to plot.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\item {} 
\sphinxstylestrong{x\_title} (str) \textendash{} plot x axis title.

\item {} 
\sphinxstylestrong{y\_title} (str) \textendash{} plot y axis title.

\item {} 
\sphinxstylestrong{height} (str) \textendash{} plot height.

\item {} 
\sphinxstylestrong{width} (str) \textendash{} plot width.

\end{itemize}

\item[{Returns}] \leavevmode
barplot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}barplot}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{barplot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Figure with Barplot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_facet\_grid\_plot() (in module analytics\_core.viz.viz)@\spxentry{get\_facet\_grid\_plot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_facet_grid_plot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_facet\_grid\_plot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function plots a scatterplot matrix where we can plot one variable against another to form a regular scatter plot, and we can pick a third faceting variable
to form panels along the columns to segment the data even further, forming a bunch of vertical panels. For more information visit \sphinxurl{https://plot.ly/python/facet-trellis/}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with format: ‘group’, ‘name’, ‘type’, and ‘x’ and ‘y’ values to be plotted.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxstyleliteralemphasis{\sphinxupquote{ditc}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{x} (str) \textendash{} name of the column containing values to plot in the x axis.

\item {} 
\sphinxstylestrong{y} (str) \textendash{} name of the column containing values to plot in the y axis.

\item {} 
\sphinxstylestrong{group} (str) \textendash{} name of the column containing the group.

\item {} 
\sphinxstylestrong{class} (str) \textendash{} name of the column to be used as ‘facet’ column.

\item {} 
\sphinxstylestrong{plot\_type} (str) \textendash{} decides the type of plot to appear in the facet grid. The options are ‘scatter’, ‘scattergl’, ‘histogram’, ‘bar’, and ‘box’.

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\end{itemize}

\item[{Returns}] \leavevmode
facet grid figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}facet\PYGZus{}grid\PYGZus{}plot}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{facet\PYGZus{}grid}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{a}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{b}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{class}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{type}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{plot\PYGZus{}type}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bar}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Facet Grid Plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_ranking\_plot() (in module analytics\_core.viz.viz)@\spxentry{get\_ranking\_plot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_ranking_plot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_ranking\_plot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
Creates abundance multiplots (one per sample group).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} long\sphinxhyphen{}format pandas dataframe with group as index, ‘name’ (protein identifiers) and ‘y’ (LFQ intensities) as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{group} (str) \textendash{} name of the column containing the group.

\item {} 
\sphinxstylestrong{index} (bool) \textendash{} set to True when multi samples per group. Calculates the mean intensity for each protein in each group.

\item {} 
\sphinxstylestrong{x\_title} (str) \textendash{} title of plot x axis.

\item {} 
\sphinxstylestrong{y\_title} (str) \textendash{} title of plot y axis.

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\item {} 
\sphinxstylestrong{width} (int) \textendash{} plot width.

\item {} 
\sphinxstylestrong{height} (int) \textendash{} plot height.

\item {} 
\sphinxstylestrong{annotations} (dict, optional) \textendash{} dictionary where data points names are the keys and descriptions are the values.

\end{itemize}

\item[{Returns}] \leavevmode
multi abundance plot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}ranking\PYGZus{}plot}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ranking}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{index}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}                                     \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Ranking Plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{width}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{100}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{height}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{150}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{annotations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{GPT\PYGZti{}P24298}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{liver disease}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CP\PYGZti{}P00450}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Wilson disease}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_scatterplot\_matrix() (in module analytics\_core.viz.viz)@\spxentry{get\_scatterplot\_matrix()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_scatterplot_matrix}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_scatterplot\_matrix}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function pltos a multi scatterplot (one for each unique element in args{[}‘group’{]}).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with four columns: ‘name’ of the data points, ‘x’ and ‘y’ values to plot, and ‘group’ they belong to.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{group} (str) \textendash{} name of the column containing the group.

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\item {} 
\sphinxstylestrong{x\_title} (str) \textendash{} plot x axis title.

\item {} 
\sphinxstylestrong{y\_title} (str) \textendash{} plot y axis title.

\item {} 
\sphinxstylestrong{height} (int) \textendash{} plot height.

\item {} 
\sphinxstylestrong{width} (int) \textendash{} plot width.

\item {} 
\sphinxstylestrong{annotations} (dict, optional) \textendash{} dictionary where data points names are the keys and descriptions are the values.

\end{itemize}

\item[{Returns}] \leavevmode
multi scatterplot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}scatterplot\PYGZus{}matrix}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{scatter matrix}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Scatter Plot Matrix}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}                                         \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{height}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{100}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{width}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{100}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{annotations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{GPT\PYGZti{}P24298}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{liver disease}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CP\PYGZti{}P00450}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Wilson disease}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_simple\_scatterplot() (in module analytics\_core.viz.viz)@\spxentry{get\_simple\_scatterplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_simple_scatterplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_simple\_scatterplot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
Plots a simple scatterplot with the possibility of including in\sphinxhyphen{}plot annotations of data points.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} long\sphinxhyphen{}format pandas dataframe with columns: ‘x’ (ranking position), ‘group’ (original dataframe position),                     ‘name’ (protein identifier), ‘y’ (LFQ intensity), ‘symbol’ (data point shape) and ‘size’ (data point size).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{annotations} (dict) \textendash{} dictionary where data points names are the keys and descriptions are the values.

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\item {} 
\sphinxstylestrong{x\_title} (str) \textendash{} plot x axis title.

\item {} 
\sphinxstylestrong{y\_title} (str) \textendash{} plot y axis title.

\item {} 
\sphinxstylestrong{height} (int) \textendash{} plot height.

\item {} 
\sphinxstylestrong{width} (int) \textendash{} plot width.

\end{itemize}

\item[{Returns}] \leavevmode
annotated scatterplot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}scatterplot\PYGZus{}matrix}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{scatter plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{annotations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{GPT\PYGZti{}P24298}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{liver disease}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CP\PYGZti{}P00450}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Wilson disease}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{,                                         }\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{Scatter} \PYG{n}{Plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{, }\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{x\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{x\PYGZus{}axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{, }\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{y\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{y\PYGZus{}axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{, }\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{height}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{:100, }\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{width}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{:100\PYGZcb{})}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_scatterplot() (in module analytics\_core.viz.viz)@\spxentry{get\_scatterplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_scatterplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_scatterplot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function plots a simple Scatterplot.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} is a Pandas DataFrame with four columns: “name”, x values and y values (provided as variables) to plot.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} is the id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{title} (str) \textendash{} title of the figure.

\item {} 
\sphinxstylestrong{x\_title} (str) \textendash{} plot x axis title.

\item {} 
\sphinxstylestrong{y\_title} (str) \textendash{} plot y axis title.

\item {} 
\sphinxstylestrong{height} (int) \textendash{} plot height.

\item {} 
\sphinxstylestrong{width} (int) \textendash{} plot width.

\end{itemize}

\item[{Returns}] \leavevmode
scatterplot figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}scatteplot}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{scatter plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Scatter Plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{height}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{100}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{width}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{100}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_volcanoplot() (in module analytics\_core.viz.viz)@\spxentry{get\_volcanoplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_volcanoplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_volcanoplot}}}{\emph{results}, \emph{args}}{}
This function plots volcano plots for each internal dictionary in a nested dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{results}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}\sphinxstyleliteralemphasis{\sphinxupquote{{[}}}\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}\sphinxstyleliteralemphasis{\sphinxupquote{{]}}}) \textendash{} nested dictionary with pairwise group comparisons as keys and internal dictionaries containing ‘x’ (log2FC values),                                 ‘y’ (\sphinxhyphen{}log10 p\sphinxhyphen{}values), ‘text’, ‘color’, ‘pvalue’ and ‘annotations’ (number of hits to be highlighted).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{fc} (float) \textendash{} fold change threshold.

\item {} 
\sphinxstylestrong{range\_x} (list) \textendash{} list with minimum and maximum values for x axis.

\item {} 
\sphinxstylestrong{range\_y} (list) \textendash{} list with minimum and maximum values for y axis.

\item {} 
\sphinxstylestrong{x\_title} (str) \textendash{} plot x axis title.

\item {} 
\sphinxstylestrong{y\_title} (str) \textendash{} plot y axis title.

\item {} 
\sphinxstylestrong{colorscale} (str) \textendash{} string for predefined plotly colorscales or dict containing one or more of the keys listed in                                     \sphinxurl{https://plot.ly/python/reference/\#layout-colorscale}.

\item {} 
\sphinxstylestrong{showscale} (bool) \textendash{} determines whether or not a colorbar is displayed for a trace.

\item {} 
\sphinxstylestrong{marker\_size} (int) \textendash{} sets the marker size (in px).

\end{itemize}

\item[{Returns}] \leavevmode
list of volcano plot figures within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}volcanoplot}\PYG{p}{(}\PYG{n}{results}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fc}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{2.0}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{range\PYGZus{}x}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{range\PYGZus{}y}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{[}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}axis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{colorscale}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Blues}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}                                 \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{showscale}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{marker\PYGZus{}size}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{7}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{run\_volcano() (in module analytics\_core.viz.viz)@\spxentry{run\_volcano()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.run_volcano}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{run\_volcano}}}{\emph{data}, \emph{identifier}, \emph{args=\{\textquotesingle{}alpha\textquotesingle{}: 0.05}, \emph{\textquotesingle{}colorscale\textquotesingle{}: \textquotesingle{}Blues\textquotesingle{}}, \emph{\textquotesingle{}fc\textquotesingle{}: 2}, \emph{\textquotesingle{}marker\_size\textquotesingle{}: 8}, \emph{\textquotesingle{}num\_annotations\textquotesingle{}: 10}, \emph{\textquotesingle{}showscale\textquotesingle{}: False}, \emph{\textquotesingle{}x\_title\textquotesingle{}: \textquotesingle{}log2FC\textquotesingle{}}, \emph{\textquotesingle{}y\_title\textquotesingle{}: \textquotesingle{}\sphinxhyphen{}log10(pvalue)\textquotesingle{}\}}}{}
This function parsers the regulation data from statistical tests and creates volcano plots for all distinct group comparisons. Significant hits with lowest adjusted p\sphinxhyphen{}values are highlighed.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with format: ‘identifier’, ‘group1’, ‘group2’, ‘mean(group1’, ‘mean(group2)’, ‘log2FC’, ‘std\_error’, ‘tail’, ‘t\sphinxhyphen{}statistics’, ‘padj\_THSD’,                                                 ‘effsize’, ‘efftype’, ‘FC’, ‘rejected’, ‘F\sphinxhyphen{}statistics’, ‘pvalue’, ‘padj’, ‘correction’, ‘\sphinxhyphen{}log10 pvalue’ and ‘Method’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{alpha} (float) \textendash{} adjusted p\sphinxhyphen{}value threshold for significant hits.

\item {} 
\sphinxstylestrong{fc} (float) \textendash{} fold change threshold.

\item {} 
\sphinxstylestrong{colorscale} (str or dict) \textendash{} name of predefined plotly colorscale or dictionary containing one or more of the keys listed in                                     \sphinxurl{https://plot.ly/python/reference/\#layout-colorscale}.

\item {} 
\sphinxstylestrong{showscale} (bool) \textendash{} determines whether or not a colorbar is displayed for a trace.

\item {} 
\sphinxstylestrong{marker\_size} (int) \textendash{} sets the marker size (in px).

\item {} 
\sphinxstylestrong{x\_title} (str) \textendash{} plot x axis title.

\item {} 
\sphinxstylestrong{y\_title} (str) \textendash{} plot y axis title.

\item {} 
\sphinxstylestrong{num\_annotations} (int) \textendash{} number of hits to be highlighted (if num\_annotations = 10, highlights 10 hits with lowest adjusted p\sphinxhyphen{}value).

\end{itemize}

\item[{Returns}] \leavevmode
list of volcano plot figures within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{run\PYGZus{}volcano}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{volvano data}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{alpha}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fc}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{2.0}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{colorscale}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Blues}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{showscale}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{marker\PYGZus{}size}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{6}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{log2FC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}                             \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZhy{}log10(pvalue)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{num\PYGZus{}annotations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{10}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_heatmapplot() (in module analytics\_core.viz.viz)@\spxentry{get\_heatmapplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_heatmapplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_heatmapplot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function plots a simple Heatmap.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} is a Pandas DataFrame with the shape of the heatmap where index corresponds to rows                 and column names corresponds to columns, values in the heatmap corresponds to the row values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} is the id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{format} (str) \textendash{} defines the format of the input dataframe.

\item {} 
\sphinxstylestrong{source} (str) \textendash{} name of the column containing the source.

\item {} 
\sphinxstylestrong{target} (str) \textendash{} name of the column containing the target.

\item {} 
\sphinxstylestrong{values} (str) \textendash{} name of the column containing the values to be plotted.

\item {} 
\sphinxstylestrong{title} (str) \textendash{} title of the figure.

\end{itemize}

\item[{Returns}] \leavevmode
heatmap figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}heatmapplot}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{heatmap}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{format}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{edgelist}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{source}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{node1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{target}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{node2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{score}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Heatmap Plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_complex\_heatmapplot() (in module analytics\_core.viz.viz)@\spxentry{get\_complex\_heatmapplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_complex_heatmapplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_complex\_heatmapplot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
\end{fulllineitems}

\index{get\_notebook\_network\_pyvis() (in module analytics\_core.viz.viz)@\spxentry{get\_notebook\_network\_pyvis()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_notebook_network_pyvis}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_notebook\_network\_pyvis}}}{\emph{graph}, \emph{args=\{\}}}{}
This function converts a Networkx graph into a PyVis graph supporting Jupyter notebook embedding.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{graph}}) \textendash{} networkX graph.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{height} (int) \textendash{} network canvas height.

\item {} 
\sphinxstylestrong{width} (int) \textendash{} network canvas width.

\end{itemize}

\item[{Returns}] \leavevmode
PyVis graph.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}notebook\PYGZus{}network\PYGZus{}pyvis}\PYG{p}{(}\PYG{n}{graph}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{height}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{100}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{width}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{100}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_notebook\_network\_web() (in module analytics\_core.viz.viz)@\spxentry{get\_notebook\_network\_web()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_notebook_network_web}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_notebook\_network\_web}}}{\emph{graph}, \emph{args}}{}
This function converts a networkX graph into a webweb interactive network in a browser.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{graph}}) \textendash{} networkX graph.

\item[{Returns}] \leavevmode
web network.

\end{description}\end{quote}

\end{fulllineitems}

\index{network\_to\_tables() (in module analytics\_core.viz.viz)@\spxentry{network\_to\_tables()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.network_to_tables}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{network\_to\_tables}}}{\emph{graph}}{}
Creates the graph edge list and node list and returns them as separate Pandas DataFrames.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{graph}} \textendash{} networkX graph used to construct the Pandas DataFrame.

\item[{Returns}] \leavevmode
two Pandas DataFrames.

\end{description}\end{quote}

\end{fulllineitems}

\index{generate\_configuration\_tree() (in module analytics\_core.viz.viz)@\spxentry{generate\_configuration\_tree()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.generate_configuration_tree}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{generate\_configuration\_tree}}}{\emph{report\_pipeline}, \emph{dataset\_type}}{}
This function retrieves the analysis pipeline from a dataset .yml file and creates a Cytoscape network, organized hierarchically.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{report\_pipeline}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary with dataset type analysis and visualization pipeline (conversion of .yml files to python dictionary).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dataset\_type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} type of dataset (‘clinical’, ‘proteomics’, ‘DNAseq’, ‘RNAseq’, ‘multiomics’).

\end{itemize}

\item[{Returns}] \leavevmode
new Dash div with title and Cytoscape network, summarizing analysis pipeline.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_network() (in module analytics\_core.viz.viz)@\spxentry{get\_network()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_network}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_network}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function filters an input dataframe based on a threshold score and builds a cytoscape network. For more information on     ‘node\_size’ parameter, visit \sphinxurl{https://networkx.github.io/documentation/networkx-1.10/reference/generated/networkx.algorithms.centrality.betweenness\_centrality.html} and     \sphinxurl{https://networkx.github.io/documentation/networkx-1.10/reference/generated/networkx.algorithms.centrality.eigenvector\_centrality\_numpy.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} long\sphinxhyphen{}format pandas dataframe with at least three columns: source node, target node and value (e.g. weight, score).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{source} (str) \textendash{} name of the column containing the source.

\item {} 
\sphinxstylestrong{target} (str) \textendash{} name of the column containing the target.

\item {} 
\sphinxstylestrong{cutoff} (float) \textendash{} value threshold for network building.

\item {} 
\sphinxstylestrong{cutoff\_abs} (bool) \textendash{} if True will take both positive and negative sides of the cutoff value.

\item {} 
\sphinxstylestrong{values} (str) \textendash{} name of the column containing the values to be plotted.

\item {} 
\sphinxstylestrong{node\_size} (str) \textendash{} method used to determine node radius (‘betweenness’, ‘ev\_centrality’, ‘degree’).

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\item {} 
\sphinxstylestrong{color\_weight} (bool) \textendash{} if True, edges in network are colored red if score \textgreater{} 0 and blue if score \textless{} 0.

\end{itemize}

\item[{Returns}] \leavevmode
dictionary with the network in multiple formats: jupyetr\sphinxhyphen{}notebook compatible, web brower compatibles, data table, and json.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}network}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{network}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{source}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{node1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{target}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{node2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{cutoff\PYGZus{}abs}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{weight}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}                             \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{node\PYGZus{}size}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{degree}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Network Figure}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{color\PYGZus{}weight}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_network\_style() (in module analytics\_core.viz.viz)@\spxentry{get\_network\_style()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_network_style}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_network\_style}}}{\emph{node\_colors}, \emph{color\_edges}}{}
This function uses a dictionary of nodes and colors and creates a stylesheet and layout for a network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{node\_colors}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary with node names as keys and colors as values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{color\_edges}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, add edge coloring to stylesheet (red for positive width, blue for negative).

\end{itemize}

\item[{Returns}] \leavevmode
stylesheet (list of dictionaries specifying the style for a group of elements, a class of elements, or a single element) and                 layout (dictionary specifying how the nodes should be positioned on the canvas).

\end{description}\end{quote}

\end{fulllineitems}

\index{visualize\_notebook\_network() (in module analytics\_core.viz.viz)@\spxentry{visualize\_notebook\_network()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.visualize_notebook_network}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{visualize\_notebook\_network}}}{\emph{network}, \emph{notebook\_type=\textquotesingle{}jupyter\textquotesingle{}}, \emph{layout=\{\textquotesingle{}height\textquotesingle{}: \textquotesingle{}700px\textquotesingle{}}, \emph{\textquotesingle{}width\textquotesingle{}: \textquotesingle{}100\%\textquotesingle{}\}}}{}
This function returns a Cytoscape network visualization for Jupyter notebooks
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{network}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#tuple}{\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}}) \textendash{} tuple with two dictionaries: network data and stylesheet (see get\_network(data, identifier, args)).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{notebook\_type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} the type of notebook where the network will be visualized (currently only jupyter notebook is supported)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{layout}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} specific layout properties (see \sphinxurl{https://dash.plot.ly/cytoscape/layout})

\end{itemize}

\item[{Returns}] \leavevmode
cyjupyter.cytoscape.Cytoscape object

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{net} \PYG{o}{=} \PYG{n}{get\PYGZus{}network}\PYG{p}{(}\PYG{n}{clincorr}\PYG{o}{.}\PYG{n}{dropna}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{corr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{source}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{node1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{target}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{node2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{cutoff\PYGZus{}abs}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,}
                                                    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{weight}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{node\PYGZus{}size}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{degree}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Network Figure}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{color\PYGZus{}weight}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\PYG{n}{visualize\PYGZus{}notebook\PYGZus{}network}\PYG{p}{(}\PYG{n}{net}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{notebook}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{notebook\PYGZus{}type}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{jupyter}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{layout}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{width}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{100}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{height}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{700px}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{visualize\_notebook\_path() (in module analytics\_core.viz.viz)@\spxentry{visualize\_notebook\_path()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.visualize_notebook_path}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{visualize\_notebook\_path}}}{\emph{path}, \emph{notebook\_type=\textquotesingle{}jupyter\textquotesingle{}}}{}
This function returns a Cytoscape network visualization for Jupyter notebooks
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{object}} (\sphinxstyleliteralemphasis{\sphinxupquote{path}}) \textendash{} dash\_html\_components object with the cytoscape network (returned by get\_cytoscape\_network())

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{notebook\_type}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} the type of notebook where the network will be visualized (currently only jupyter notebook is supported)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{layout}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} specific layout properties (see \sphinxurl{https://dash.plot.ly/cytoscape/layout})

\end{itemize}

\item[{Returns}] \leavevmode
cyjupyter.cytoscape.Cytoscape object

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{net} \PYG{o}{=} \PYG{n}{get\PYGZus{}cytoscape\PYGZus{}network}\PYG{p}{(}\PYG{n}{G}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{corr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Cytoscape path}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{stylesheet}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{stylesheet}\PYG{p}{,}
                                                    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{layout}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{n}{layout}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\PYG{n}{visualize\PYGZus{}notebook\PYGZus{}path}\PYG{p}{(}\PYG{n}{net}\PYG{p}{,} \PYG{n}{notebook\PYGZus{}type}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{jupyter}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_pca\_plot() (in module analytics\_core.viz.viz)@\spxentry{get\_pca\_plot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_pca_plot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_pca\_plot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function creates a pca plot with scores and top “args{[}‘loadings’{]}” loadings.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#tuple}{\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}}) \textendash{} tuple with two pandas dataframes: scores and loadings.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{loadings} (int) \textendash{} number of features with highest loading values to be displayed in the pca plot

\item {} 
\sphinxstylestrong{title} (str) \textendash{} title of the figure

\item {} 
\sphinxstylestrong{x\_title} (str) \textendash{} plot x axis title

\item {} 
\sphinxstylestrong{y\_title} (str) \textendash{} plot y axis title

\item {} 
\sphinxstylestrong{height} (int) \textendash{} plot height

\item {} 
\sphinxstylestrong{width} (int) \textendash{} plot width

\end{itemize}

\item[{Returns}] \leavevmode
PCA figure within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}pca\PYGZus{}plot}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pca}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{loadings}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{15}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{PCA Plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{PC1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{PC2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{height}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{100}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{width}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{100}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_sankey\_plot() (in module analytics\_core.viz.viz)@\spxentry{get\_sankey\_plot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_sankey_plot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_sankey\_plot}}}{\emph{data}, \emph{identifier}, \emph{args=\{\textquotesingle{}font\textquotesingle{}: 12}, \emph{\textquotesingle{}height\textquotesingle{}: 800}, \emph{\textquotesingle{}orientation\textquotesingle{}: \textquotesingle{}h\textquotesingle{}}, \emph{\textquotesingle{}source\textquotesingle{}: \textquotesingle{}source\textquotesingle{}}, \emph{\textquotesingle{}source\_colors\textquotesingle{}: \textquotesingle{}source\_colors\textquotesingle{}}, \emph{\textquotesingle{}target\textquotesingle{}: \textquotesingle{}target\textquotesingle{}}, \emph{\textquotesingle{}target\_colors\textquotesingle{}: \textquotesingle{}target\_colors\textquotesingle{}}, \emph{\textquotesingle{}title\textquotesingle{}: \textquotesingle{}Sankey plot\textquotesingle{}}, \emph{\textquotesingle{}valueformat\textquotesingle{}: \textquotesingle{}.0f\textquotesingle{}}, \emph{\textquotesingle{}weight\textquotesingle{}: \textquotesingle{}weight\textquotesingle{}}, \emph{\textquotesingle{}width\textquotesingle{}: 800\}}}{}
This function generates a Sankey plot in Plotly.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} Pandas DataFrame with the format: source  target  weight.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{source} (str) \textendash{} name of the column containing the source

\item {} 
\sphinxstylestrong{target} (str) \textendash{} name of the column containing the target

\item {} 
\sphinxstylestrong{weight} (str) \textendash{} name of the column containing the weight

\item {} 
\sphinxstylestrong{source\_colors} (str) \textendash{} name of the column in data that contains the colors of each source item

\item {} 
\sphinxstylestrong{target\_colors} (str) \textendash{} name of the column in data that contains the colors of each target item

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title

\item {} 
\sphinxstylestrong{orientation} (str) \textendash{} whether to plot horizontal (‘h’) or vertical (‘v’)

\item {} 
\sphinxstylestrong{valueformat} (str) \textendash{} how to show the value (‘.0f’)

\item {} 
\sphinxstylestrong{width} (int) \textendash{} plot width

\item {} 
\sphinxstylestrong{height} (int) \textendash{} plot height

\item {} 
\sphinxstylestrong{font} (int) \textendash{} font size

\end{itemize}

\item[{Returns}] \leavevmode
dcc.Graph

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}sankey\PYGZus{}plot}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sankeyplot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{source}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{source}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{target}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{target}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{weight}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{weight}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{source\PYGZus{}colors}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{source\PYGZus{}colors}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}                                 \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{target\PYGZus{}colors}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{target\PYGZus{}colors}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{orientation}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{h}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{valueformat}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{.0f}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{width}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{800}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{height}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{800}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{font}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{12}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Sankey plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_table() (in module analytics\_core.viz.viz)@\spxentry{get\_table()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_table}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_table}}}{\emph{data}, \emph{identifier}, \emph{title}, \emph{colors=(\textquotesingle{}\#C2D4FF\textquotesingle{}}, \emph{\textquotesingle{}\#F5F8FF\textquotesingle{})}, \emph{subset=None}, \emph{plot\_attr=\{\textquotesingle{}font\textquotesingle{}: 12}, \emph{\textquotesingle{}height\textquotesingle{}: 2500}, \emph{\textquotesingle{}width\textquotesingle{}: 1500\}}, \emph{subplot=False}}{}
This function converts a pandas dataframe into an interactive table for viewing, editing and exploring large datasets. For more information visit \sphinxurl{https://dash.plot.ly/datatable}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} table title.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subset}} \textendash{} selects columns from dataframe to be used. If None, the entire dataframe is used.

\end{itemize}

\item[{Returns}] \leavevmode
new Dash div containing title and interactive table.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}table}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{table}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Table Figure}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{subset} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_multi\_table() (in module analytics\_core.viz.viz)@\spxentry{get\_multi\_table()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_multi_table}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_multi\_table}}}{\emph{data}, \emph{identifier}, \emph{title}}{}
\end{fulllineitems}

\index{get\_violinplot() (in module analytics\_core.viz.viz)@\spxentry{get\_violinplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_violinplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_violinplot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function creates a violin plot for all columns in the input dataframe.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with samples as rows and dependent variables as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{drop\_cols} (list) \textendash{} column labels to be dropped from the dataframe.

\item {} 
\sphinxstylestrong{group} (str) \textendash{} name of the column containing the group.

\end{itemize}

\item[{Returns}] \leavevmode
list of violion plots within the \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}violinplot}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{violinplot, args=}\PYG{l+s+s1}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{drop\PYGZus{}cols}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{:[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{sample}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{, }\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{subject}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{], }\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{n}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZcb{})}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{create\_violinplot() (in module analytics\_core.viz.viz)@\spxentry{create\_violinplot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.create_violinplot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{create\_violinplot}}}{\emph{df}, \emph{variable}, \emph{group\_col=\textquotesingle{}group\textquotesingle{}}}{}
This function creates traces for a simple violin plot.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe with samples as rows and dependent variables as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{variable}} (\sphinxstyleliteralemphasis{\sphinxupquote{(}}\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}\sphinxstyleliteralemphasis{\sphinxupquote{)}}) \textendash{} name of the column with the dependent variable.

\end{itemize}

\item[{Pram (str) group\_col}] \leavevmode
name of the column containing the group.

\item[{Returns}] \leavevmode
list of traces to be used as data for plotly figure.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{create\PYGZus{}violinplot}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{prptein a}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{group\PYGZus{}col}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_clustergrammer\_plot() (in module analytics\_core.viz.viz)@\spxentry{get\_clustergrammer\_plot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_clustergrammer_plot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_clustergrammer\_plot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function takes a pandas dataframe, calculates clustering, and generates the visualization json.
For more information visit \sphinxurl{https://github.com/MaayanLab/clustergrammer-py}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} long\sphinxhyphen{}format pandas dataframe with columns ‘node1’ (source), ‘node2’ (target) and ‘weight’

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{format} (str) \textendash{} defines if dataframe needs to be converted from ‘edgelist’ to matrix

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title

\end{itemize}

\item[{Returns}] \leavevmode
Dash Div with heatmap plot from Clustergrammer web\sphinxhyphen{}based tool

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_parallel\_plot() (in module analytics\_core.viz.viz)@\spxentry{get\_parallel\_plot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_parallel_plot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_parallel\_plot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
This function creates a parallel coordinates plot, with sample groups as the different dimensions.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with groups as rows and dependent variables as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{group} (str) \textendash{} name of the column containing the groups.

\item {} 
\sphinxstylestrong{zscore} (bool) \textendash{} if True, calculates the z score of each values in the row, relative to the row mean and standard deviation.

\item {} 
\sphinxstylestrong{color} (str) \textendash{} line color.

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\end{itemize}

\item[{Returns}] \leavevmode
parallel plot figure within \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{} .

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}parallel\PYGZus{}plot}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{parallel plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{zscore}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{color}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{blue}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parallel Plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_WGCNAPlots() (in module analytics\_core.viz.viz)@\spxentry{get\_WGCNAPlots()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_WGCNAPlots}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_WGCNAPlots}}}{\emph{data}, \emph{identifier}}{}
Takes data from runWGCNA function and builds WGCNA plots.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} tuple with multiple pandas dataframes.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} is the id used to identify the div where the figure will be generated.

\end{itemize}

\item[{Returns}] \leavevmode
list of dcc.Graph.

\end{description}\end{quote}

\end{fulllineitems}

\index{getMapperFigure() (in module analytics\_core.viz.viz)@\spxentry{getMapperFigure()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.getMapperFigure}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{getMapperFigure}}}{\emph{data}, \emph{identifier}, \emph{title}}{}
This function uses the KeplerMapper python package to visualize high\sphinxhyphen{}dimensional data and generate a FigureWidget that can be shown or editted.
This method is suitable for use in Jupyter notebooks. For more information visit \sphinxurl{https://kepler-mapper.scikit-tda.org/reference/stubs/kmapper.plotlyviz.plotlyviz.html}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} dictionary. Simplicial complex output from the KeplerMapper map method.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} plot title.

\end{itemize}

\item[{Returns}] \leavevmode
plotly FigureWidget within \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{} .

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_2\_venn\_diagram() (in module analytics\_core.viz.viz)@\spxentry{get\_2\_venn\_diagram()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_2_venn_diagram}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_2\_venn\_diagram}}}{\emph{data}, \emph{identifier}, \emph{cond1}, \emph{cond2}, \emph{args}}{}
This function extracts the exlusive features in cond1 and cond2 and their common features, and build a two\sphinxhyphen{}circle venn diagram.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with features as rows and group identifiers as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cond1}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of first group.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cond2}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier of second group.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{colors} (dict) \textendash{} dictionary with cond1 and cond2 as keys, and color codes as values.

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\end{itemize}

\item[{Returns}] \leavevmode
two\sphinxhyphen{}circle venn diagram figure within \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}2\PYGZus{}venn\PYGZus{}diagram}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{venn2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{cond1}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{cond2}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{color}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{blue}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{red}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}                                     \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Two\PYGZhy{}circle Venn diagram}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{plot\_2\_venn\_diagram() (in module analytics\_core.viz.viz)@\spxentry{plot\_2\_venn\_diagram()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.plot_2_venn_diagram}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{plot\_2\_venn\_diagram}}}{\emph{cond1}, \emph{cond2}, \emph{unique1}, \emph{unique2}, \emph{intersection}, \emph{identifier}, \emph{args}}{}
This function creates a simple non area\sphinxhyphen{}weighted two\sphinxhyphen{}circle venn diagram.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cond1}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} label of the first circle.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cond2}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} label of the second circle.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{unique1}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of features exclusive to cond1.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{unique2}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} number of features exclusive to cond2.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Parm int intersection}] \leavevmode
number of features common to cond1 and cond2.

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{colors} (dict) \textendash{} dictionary with cond1 and cond2 as keys, and color codes as values.

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\end{itemize}

\item[{Returns}] \leavevmode
two\sphinxhyphen{}circle venn diagram figure within \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{plot\PYGZus{}2\PYGZus{}venn\PYGZus{}diagram}\PYG{p}{(}\PYG{n}{cond1}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{cond2}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{unique1}\PYG{o}{=}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{unique2}\PYG{o}{=}\PYG{l+m+mi}{15}\PYG{p}{,} \PYG{n}{intersection}\PYG{o}{=}\PYG{l+m+mi}{8}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{vennplot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}                                     \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{color}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{blue}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{group2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{red}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{title}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Two\PYGZhy{}circle Venn diagram}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_wordcloud() (in module analytics\_core.viz.viz)@\spxentry{get\_wordcloud()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_wordcloud}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_wordcloud}}}{\emph{data}, \emph{identifier}, \emph{args=\{\textquotesingle{}height\textquotesingle{}: 700}, \emph{\textquotesingle{}margin\textquotesingle{}: 1}, \emph{\textquotesingle{}max\_font\_size\textquotesingle{}: 100}, \emph{\textquotesingle{}max\_words\textquotesingle{}: 400}, \emph{\textquotesingle{}stopwords\textquotesingle{}: {[}{]}}, \emph{\textquotesingle{}width\textquotesingle{}: 700\}}}{}
This function generates a Wordcloud based on the natural text in a pandas dataframe column.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} \textendash{} pandas dataframe with columns: ‘PMID’, ‘abstract’, ‘authors’, ‘date’, ‘journal’, ‘keywords’, ‘title’, ‘url’, ‘Proteins’, ‘Diseases’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{text\_col} (str) \textendash{} name of column containing the natural text used to generate the wordcloud.

\item {} 
\sphinxstylestrong{stopwords} (list) \textendash{} list of words that will be eliminated.

\item {} 
\sphinxstylestrong{max\_words} (int) \textendash{} maximum number of words.

\item {} 
\sphinxstylestrong{max\_font\_size} (int) \textendash{} maximum font size for the largest word.

\item {} 
\sphinxstylestrong{margin} (int) \textendash{} plot margin size.

\item {} 
\sphinxstylestrong{width} (int) \textendash{} width of the plot.

\item {} 
\sphinxstylestrong{height} (int) \textendash{} height of the plot.

\item {} 
\sphinxstylestrong{title} (str) \textendash{} plot title.

\end{itemize}

\item[{Returns}] \leavevmode
wordcloud figure within \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{get\PYGZus{}wordcloud}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,} \PYG{n}{identifier}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wordcloud}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{stopwords}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{BACKGROUND}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CONCLUSION}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{RESULT}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{METHOD}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CONCLUSIONS}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{RESULTS}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{METHODS}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,}                                 \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{max\PYGZus{}words}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+m+mi}{400}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{max\PYGZus{}font\PYGZus{}size}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{width}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{700}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{height}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{700}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{margin}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{get\_cytoscape\_network() (in module analytics\_core.viz.viz)@\spxentry{get\_cytoscape\_network()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_cytoscape_network}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_cytoscape\_network}}}{\emph{net}, \emph{identifier}, \emph{args}}{}
This function creates a Cytoscpae network in dash. For more information visit \sphinxurl{https://dash.plot.ly/cytoscape}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{net}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary in which each element (key) is defined by a dictionary with ‘id’ and ‘label’                     (if it is a node) or ‘source’, ‘target’ and ‘label’ (if it is an edge).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} is the id used to identify the div where the figure will be generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} see below.

\end{itemize}

\item[{Arguments}] \leavevmode\begin{itemize}
\item {} 
\sphinxstylestrong{title} (str) \textendash{} title of the figure.

\item {} 
\sphinxstylestrong{stylesheet} (list{[}dict{]}) \textendash{} specifies the style for a group of elements, a class of elements, or a single element                                         (accepts two keys ‘selector’ and ‘style’).

\item {} 
\sphinxstylestrong{layout} (dict) \textendash{} specifies how the nodes should be positioned on the screen.

\end{itemize}

\item[{Returns}] \leavevmode
network figure within \textless{}div id=”\_dash\sphinxhyphen{}app\sphinxhyphen{}content”\textgreater{}.

\end{description}\end{quote}

\end{fulllineitems}

\index{save\_DASH\_plot() (in module analytics\_core.viz.viz)@\spxentry{save\_DASH\_plot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.save_DASH_plot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{save\_DASH\_plot}}}{\emph{plot}, \emph{name}, \emph{plot\_format=\textquotesingle{}svg\textquotesingle{}}, \emph{directory=\textquotesingle{}.\textquotesingle{}}}{}
This function saves a plotly figure to a specified directory, in a determined format.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{plot}} \textendash{} plotly figure (dictionary with data and layout)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} name of the figure

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{plot\_format}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} suffix of the saved file (‘svg’, ‘pdf’, ‘png’, ‘jpeg’, ‘jpg’)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{directory}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} folder where figure is to be saved

\end{itemize}

\item[{Returns}] \leavevmode
figure saved in directory

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{save\PYGZus{}DASH\PYGZus{}plot}\PYG{p}{(}\PYG{n}{plot}\PYG{p}{,} \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Plot example}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{plot\PYGZus{}format}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{svg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{directory}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{/data/plots}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{mpl\_to\_plotly() (in module analytics\_core.viz.viz)@\spxentry{mpl\_to\_plotly()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.mpl_to_plotly}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{mpl\_to\_plotly}}}{\emph{fig}, \emph{ci=True}, \emph{legend=True}}{}
\end{fulllineitems}

\index{get\_km\_plot() (in module analytics\_core.viz.viz)@\spxentry{get\_km\_plot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_km_plot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_km\_plot}}}{\emph{data}, \emph{identifier}, \emph{args}}{}
\end{fulllineitems}

\index{get\_polar\_plot() (in module analytics\_core.viz.viz)@\spxentry{get\_polar\_plot()}\spxextra{in module analytics\_core.viz.viz}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.viz.get_polar_plot}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.viz.}}\sphinxbfcode{\sphinxupquote{get\_polar\_plot}}}{\emph{df}, \emph{identifier}, \emph{args}}{}
This function creates a Polar plot with data aggregated for a given group.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} (\sphinxstyleliteralemphasis{\sphinxupquote{dataframe}}) \textendash{} dataframe with the data to plot

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{identifier}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} identifier to be used in the app

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{args}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary containing the arguments needed to plot the figure (value\_col (value to aggregate), group\_col (group by), color\_col (color by))

\end{itemize}

\item[{Returns}] \leavevmode
Dash Graph

\end{description}\end{quote}
\begin{description}
\item[{Example::}] \leavevmode
figure = get\_polar\_plot(df, identifier=’polar’, args=\{‘value\_col’:’intensity’, ‘group\_col’:’modifier’, ‘color\_col’:’group’\})

\end{description}

\end{fulllineitems}



\paragraph{WGCNA viz}
\label{\detokenize{_autosummary/analytics_core.viz:module-analytics_core.viz.wgcnaFigures}}\label{\detokenize{_autosummary/analytics_core.viz:wgcna-viz}}\index{analytics\_core.viz.wgcnaFigures (module)@\spxentry{analytics\_core.viz.wgcnaFigures}\spxextra{module}}\index{get\_module\_color\_annotation() (in module analytics\_core.viz.wgcnaFigures)@\spxentry{get\_module\_color\_annotation()}\spxextra{in module analytics\_core.viz.wgcnaFigures}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.wgcnaFigures.get_module_color_annotation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.wgcnaFigures.}}\sphinxbfcode{\sphinxupquote{get\_module\_color\_annotation}}}{\emph{map\_list}, \emph{col\_annotation=False}, \emph{row\_annotation=False}, \emph{bygene=False}, \emph{module\_colors={[}{]}}, \emph{dendrogram={[}{]}}}{}
This function takes a list of values, converts them into colors, and creates a new plotly object to be used as an annotation.
Options module\_colors and dendrogram only apply when map\_list is a list of experimental features used in module eigenegenes calculation.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{map\_list}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} dendrogram leaf labels.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{col\_annotation}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, adds color annotations as a row.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{row\_annotation}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, adds color annotations as a column.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{bygene}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} determines wether annotation colors have to be reordered to match dendrogram leaf labels.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{module\_colors}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} dendrogram leaf module color.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dendrogram}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dendrogram represented as a plotly object figure.

\end{itemize}

\item[{Returns}] \leavevmode
Plotly object figure.

\end{description}\end{quote}

\begin{sphinxadmonition}{note}{Note:}
map\_list and module\_colors must have the same length.
\end{sphinxadmonition}

\end{fulllineitems}

\index{get\_heatmap() (in module analytics\_core.viz.wgcnaFigures)@\spxentry{get\_heatmap()}\spxextra{in module analytics\_core.viz.wgcnaFigures}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.wgcnaFigures.get_heatmap}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.wgcnaFigures.}}\sphinxbfcode{\sphinxupquote{get\_heatmap}}}{\emph{df}, \emph{colorscale=None}, \emph{color\_missing=True}}{}
This function plots a simple Plotly heatmap.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe containing experimental data, with samples/subjects as rows and features as columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{colorscale}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{{[}}}\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{{]}}}) \textendash{} heatmap colorscale (e.g. {[}{[}0,’\#67a9cf’{]},{[}0.5,’\#f7f7f7’{]},{[}1,’\#ef8a62’{]}{]}). If colorscale is not defined, will take {[}{[}0, ‘rgb(255,255,255)’{]}, {[}1, ‘rgb(255,51,0)’{]}{]} as default.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{color\_missing}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if set to True, plots missing values as grey in the heatmap.

\end{itemize}

\item[{Returns}] \leavevmode
Plotly object figure.

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_labeled\_heatmap() (in module analytics\_core.viz.wgcnaFigures)@\spxentry{plot\_labeled\_heatmap()}\spxextra{in module analytics\_core.viz.wgcnaFigures}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.wgcnaFigures.plot_labeled_heatmap}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.wgcnaFigures.}}\sphinxbfcode{\sphinxupquote{plot\_labeled\_heatmap}}}{\emph{df, textmatrix, title, colorscale={[}{[}0, \textquotesingle{}rgb(0,255,0)\textquotesingle{}{]}, {[}0.5, \textquotesingle{}rgb(255,255,255)\textquotesingle{}{]}, {[}1, \textquotesingle{}rgb(255,0,0)\textquotesingle{}{]}{]}, width=1200, height=800, row\_annotation=False, col\_annotation=False}}{}
This function plots a simple Plotly heatmap with column and/or row annotations and heatmap annotations.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{df}} \textendash{} pandas dataframe containing data to be plotted in the heatmap.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{textmatrix}} \textendash{} pandas dataframe with heatmap annotations as values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} the title of the figure.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{colorscale}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{{[}}}\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{{]}}}) \textendash{} heatmap colorscale (e.g. {[}{[}0,’rgb(0,255,0)’{]},{[}0.5,’rgb(255,255,255)’{]},{[}1,’rgb(255,0,0)’{]}{]})

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{width}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} the width of the figure.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{height}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} the height of the figure.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{row\_annotation}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, adds a color\sphinxhyphen{}coded column at the left of the heatmap.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{col\_annotation}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if True, adds a color\sphinxhyphen{}coded row at the bottom of the heatmap.

\end{itemize}

\item[{Returns}] \leavevmode
Plotly object figure.

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_dendrogram\_guidelines() (in module analytics\_core.viz.wgcnaFigures)@\spxentry{plot\_dendrogram\_guidelines()}\spxextra{in module analytics\_core.viz.wgcnaFigures}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.wgcnaFigures.plot_dendrogram_guidelines}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.wgcnaFigures.}}\sphinxbfcode{\sphinxupquote{plot\_dendrogram\_guidelines}}}{\emph{Z\_tree}, \emph{dendrogram}}{}
This function takes a dendrogram tree dictionary and its plotly object and creates shapes to be plotted as vertical dashed lines in the dendrogram.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{Z\_tree}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#dict}{\sphinxstyleliteralemphasis{\sphinxupquote{dict}}}) \textendash{} dictionary of data structures computed to render the dendrogram. Keys: ‘icoords’, ‘dcoords’, ‘ivl’ and ‘leaves’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dendrogram}} \textendash{} dendrogram represented as a plotly object figure.

\end{itemize}

\item[{Returns}] \leavevmode
List of dictionaries.

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_intramodular\_correlation() (in module analytics\_core.viz.wgcnaFigures)@\spxentry{plot\_intramodular\_correlation()}\spxextra{in module analytics\_core.viz.wgcnaFigures}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.wgcnaFigures.plot_intramodular_correlation}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.wgcnaFigures.}}\sphinxbfcode{\sphinxupquote{plot\_intramodular\_correlation}}}{\emph{MM}, \emph{FS}, \emph{feature\_module\_df}, \emph{title}, \emph{width=1000}, \emph{height=800}}{}
This function uses the Feature significance and Module Membership measures, and plots a multi\sphinxhyphen{}scatter plot of all modules against all clinical traits.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{MM}} \textendash{} pandas dataframe with module membership data

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{FS}} \textendash{} pandas dataframe with feature significance data

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{feature\_module\_df}} \textendash{} pandas DataFrame of experimental features and module colors (use mode=’dataframe’ in get\_FeaturesPerModule)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} plot title

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{width}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} plot width

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{height}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} plot height

\end{itemize}

\item[{Returns}] \leavevmode
Plotly object figure.

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{plot} \PYG{o}{=} \PYG{n}{plot\PYGZus{}intramodular\PYGZus{}correlation}\PYG{p}{(}\PYG{n}{MM}\PYG{p}{,} \PYG{n}{FS}\PYG{p}{,} \PYG{n}{feature\PYGZus{}module\PYGZus{}df}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Plot}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{width}\PYG{o}{=}\PYG{l+m+mi}{1000}\PYG{p}{,} \PYG{n}{height}\PYG{o}{=}\PYG{l+m+mi}{800}\PYG{p}{)}\PYG{p}{:}
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
There is a limit in the number of subplots one can make in Plotly. This function limits the number of modules shown to 5.
\end{sphinxadmonition}

\end{fulllineitems}

\index{plot\_complex\_dendrogram() (in module analytics\_core.viz.wgcnaFigures)@\spxentry{plot\_complex\_dendrogram()}\spxextra{in module analytics\_core.viz.wgcnaFigures}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.wgcnaFigures.plot_complex_dendrogram}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.wgcnaFigures.}}\sphinxbfcode{\sphinxupquote{plot\_complex\_dendrogram}}}{\emph{dendro\_df}, \emph{subplot\_df}, \emph{title}, \emph{dendro\_labels={[}{]}}, \emph{distfun=\textquotesingle{}euclidean\textquotesingle{}}, \emph{linkagefun=\textquotesingle{}average\textquotesingle{}}, \emph{hang=0.04}, \emph{subplot=\textquotesingle{}module colors\textquotesingle{}}, \emph{subplot\_colorscale={[}{]}}, \emph{color\_missingvals=True}, \emph{row\_annotation=False}, \emph{col\_annotation=False}, \emph{width=1000}, \emph{height=800}}{}
This function plots a dendrogram with a subplot below that can be a heatmap (annotated or not) or module colors.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dendro\_df}} \textendash{} pandas dataframe containing data used to generate dendrogram, columns will result in dendrogram leaves.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subplot\_df}} \textendash{} pandas dataframe containing data used to generate plot below dendrogram.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{title}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} the title of the figure.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dendro\_labels}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} list of strings for dendrogram leaf nodes labels.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{distfun}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} distance measure to be used (‘euclidean‘, ‘maximum‘, ‘manhattan‘, ‘canberra‘, ‘binary‘, ‘minkowski‘ or ‘jaccard‘).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{linkagefun}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} hierarchical/agglomeration method to be used (‘single‘, ‘complete‘, ‘average‘, ‘weighted‘, ‘centroid‘, ‘median‘ or ‘ward‘).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{hang}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} height at which the dendrogram leaves should be placed.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subplot}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} type of plot to be shown below the dendrogram (´module colors´ or ´heatmap´).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{subplot\_colorscale}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} colorscale to be used in the subplot.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{color\_missingvals}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if set to \sphinxtitleref{True}, plots missing values as grey in the heatmap.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{row\_annotation}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if \sphinxtitleref{True}, adds a color\sphinxhyphen{}coded column at the left of the heatmap.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{col\_annotation}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#bool}{\sphinxstyleliteralemphasis{\sphinxupquote{bool}}}) \textendash{} if \sphinxtitleref{True}, adds a color\sphinxhyphen{}coded row at the bottom of the heatmap.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{width}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} the width of the figure.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{height}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} the height of the figure.

\end{itemize}

\item[{Returns}] \leavevmode
Plotly object figure.

\end{description}\end{quote}

\end{fulllineitems}



\paragraph{Dendrogram}
\label{\detokenize{_autosummary/analytics_core.viz:module-analytics_core.viz.Dendrogram}}\label{\detokenize{_autosummary/analytics_core.viz:dendrogram}}\index{analytics\_core.viz.Dendrogram (module)@\spxentry{analytics\_core.viz.Dendrogram}\spxextra{module}}\index{plot\_dendrogram() (in module analytics\_core.viz.Dendrogram)@\spxentry{plot\_dendrogram()}\spxextra{in module analytics\_core.viz.Dendrogram}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.Dendrogram.plot_dendrogram}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.Dendrogram.}}\sphinxbfcode{\sphinxupquote{plot\_dendrogram}}}{\emph{Z\_dendrogram}, \emph{cutoff\_line=True}, \emph{value=15}, \emph{orientation=\textquotesingle{}bottom\textquotesingle{}}, \emph{hang=30}, \emph{hide\_labels=False}, \emph{labels=None}, \emph{colorscale=None}, \emph{hovertext=None}, \emph{color\_threshold=None}}{}
Modified version of Plotly \_dendrogram.py that returns a dendrogram Plotly figure object with cutoff line.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{Z\_dendrogram}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} Matrix of observations as array of arrays

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cutoff\_line}} (\sphinxstyleliteralemphasis{\sphinxupquote{boolean}}) \textendash{} plot distance cutoff line

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{value}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}\sphinxstyleliteralemphasis{\sphinxupquote{ or }}\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} dendrogram distance for cutoff line

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{orientation}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} ‘top’, ‘right’, ‘bottom’, or ‘left’

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{hang}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} dendrogram distance of leaf lines

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{hide\_labels}} (\sphinxstyleliteralemphasis{\sphinxupquote{boolean}}) \textendash{} show leaf labels

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{labels}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} List of axis category labels(observation labels)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{colorscale}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} Optional colorscale for dendrogram tree

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{hovertext}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{{[}}}\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}\sphinxstyleliteralemphasis{\sphinxupquote{{]}}}) \textendash{} List of hovertext for constituent traces of dendrogram
clusters

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{color\_threshold}} (\sphinxstyleliteralemphasis{\sphinxupquote{double}}) \textendash{} Value at which the separation of clusters will be made

\end{itemize}

\item[{Returns}] \leavevmode
Plotly figure object

\end{description}\end{quote}

Example:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{figure} \PYG{o}{=} \PYG{n}{plot\PYGZus{}dendrogram}\PYG{p}{(}\PYG{n}{dendro\PYGZus{}tree}\PYG{p}{,} \PYG{n}{hang}\PYG{o}{=}\PYG{l+m+mf}{0.9}\PYG{p}{,} \PYG{n}{cutoff\PYGZus{}line}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{Dendrogram (class in analytics\_core.viz.Dendrogram)@\spxentry{Dendrogram}\spxextra{class in analytics\_core.viz.Dendrogram}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.Dendrogram.Dendrogram}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{analytics\_core.viz.Dendrogram.}}\sphinxbfcode{\sphinxupquote{Dendrogram}}}{\emph{Z\_dendrogram}, \emph{orientation=\textquotesingle{}bottom\textquotesingle{}}, \emph{hang=1}, \emph{hide\_labels=False}, \emph{labels=None}, \emph{colorscale=None}, \emph{hovertext=None}, \emph{color\_threshold=None}, \emph{width=inf}, \emph{height=inf}, \emph{xaxis=\textquotesingle{}xaxis\textquotesingle{}}, \emph{yaxis=\textquotesingle{}yaxis\textquotesingle{}}}{}
Bases: \sphinxhref{https://docs.python.org/3.6/library/functions.html\#object}{\sphinxcode{\sphinxupquote{object}}}

Refer to plot\_dendrogram() for docstring.
\index{get\_color\_dict() (analytics\_core.viz.Dendrogram.Dendrogram method)@\spxentry{get\_color\_dict()}\spxextra{analytics\_core.viz.Dendrogram.Dendrogram method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.Dendrogram.Dendrogram.get_color_dict}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_color\_dict}}}{\emph{colorscale}}{}
Returns colorscale used for dendrogram tree clusters.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{colorscale}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} colors to use for the plot in rgb format

\item[{Return (dict)}] \leavevmode
default colors mapped to the user colorscale

\end{description}\end{quote}

\end{fulllineitems}

\index{set\_axis\_layout() (analytics\_core.viz.Dendrogram.Dendrogram method)@\spxentry{set\_axis\_layout()}\spxextra{analytics\_core.viz.Dendrogram.Dendrogram method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.Dendrogram.Dendrogram.set_axis_layout}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_axis\_layout}}}{\emph{axis\_key}, \emph{hide\_labels}}{}
Sets and returns default axis object for dendrogram figure.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{axis\_key}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#str}{\sphinxstyleliteralemphasis{\sphinxupquote{str}}}) \textendash{} E.g., ‘xaxis’, ‘xaxis1’, ‘yaxis’, yaxis1’, etc.

\item[{Return (dict)}] \leavevmode
An axis\_key dictionary with set parameters.

\end{description}\end{quote}

\end{fulllineitems}

\index{set\_figure\_layout() (analytics\_core.viz.Dendrogram.Dendrogram method)@\spxentry{set\_figure\_layout()}\spxextra{analytics\_core.viz.Dendrogram.Dendrogram method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.Dendrogram.Dendrogram.set_figure_layout}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_figure\_layout}}}{\emph{width}, \emph{height}, \emph{hide\_labels}}{}
Sets and returns default layout object for dendrogram figure.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{width}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} plot width

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{height}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#int}{\sphinxstyleliteralemphasis{\sphinxupquote{int}}}) \textendash{} plot height

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{hide\_labels}} (\sphinxstyleliteralemphasis{\sphinxupquote{boolean}}) \textendash{} show leaf labels

\end{itemize}

\item[{Returns}] \leavevmode
Plotly layout

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_dendrogram\_traces() (analytics\_core.viz.Dendrogram.Dendrogram method)@\spxentry{get\_dendrogram\_traces()}\spxextra{analytics\_core.viz.Dendrogram.Dendrogram method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.Dendrogram.Dendrogram.get_dendrogram_traces}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dendrogram\_traces}}}{\emph{Z\_dendrogram}, \emph{hang}, \emph{colorscale}, \emph{hovertext}, \emph{color\_threshold}}{}
Calculates all the elements needed for plotting a dendrogram.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{Z\_dendrogram}} (\sphinxstyleliteralemphasis{\sphinxupquote{ndarray}}) \textendash{} Matrix of observations as array of arrays

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{hang}} (\sphinxhref{https://docs.python.org/3.6/library/functions.html\#float}{\sphinxstyleliteralemphasis{\sphinxupquote{float}}}) \textendash{} dendrogram distance of leaf lines

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{colorscale}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} Color scale for dendrogram tree clusters

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{hovertext}} (\sphinxhref{https://docs.python.org/3.6/library/stdtypes.html\#list}{\sphinxstyleliteralemphasis{\sphinxupquote{list}}}) \textendash{} List of hovertext for constituent traces of dendrogram

\end{itemize}

\item[{Return (tuple)}] \leavevmode
Contains all the traces in the following order:
\begin{enumerate}
\sphinxsetlistlabels{\alph}{enumi}{enumii}{}{.}%
\item {} 
trace\_list: List of Plotly trace objects for dendrogram tree

\item {} 
icoord: All X points of the dendrogram tree as array of arrays with length 4

\item {} 
dcoord: All Y points of the dendrogram tree as array of arrays with length 4

\item {} 
ordered\_labels: leaf labels in the order they are going to appear on the plot

\item {} 
Z\_dendrogram{[}‘leaves’{]}: left\sphinxhyphen{}to\sphinxhyphen{}right traversal of the leaves

\end{enumerate}

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\paragraph{Colors}
\label{\detokenize{_autosummary/analytics_core.viz:module-analytics_core.viz.color_list}}\label{\detokenize{_autosummary/analytics_core.viz:colors}}\index{analytics\_core.viz.color\_list (module)@\spxentry{analytics\_core.viz.color\_list}\spxextra{module}}
Code for handling color names and RGB codes.

This module is part of Swampy, and used in Think Python and
Think Complexity, by Allen Downey.

\sphinxurl{http://greenteapress.com}

Copyright 2013 Allen B. Downey.
Distributed under the GNU General Public License at gnu.org/licenses/gpl.html.
\index{make\_color\_dict() (in module analytics\_core.viz.color\_list)@\spxentry{make\_color\_dict()}\spxextra{in module analytics\_core.viz.color\_list}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.color_list.make_color_dict}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.color\_list.}}\sphinxbfcode{\sphinxupquote{make\_color\_dict}}}{\emph{colors=\textquotesingle{}\textbackslash{}n141 211 199\textbackslash{}t\textbackslash{}tturquoise\textbackslash{}n31   120 180\textbackslash{}t\textbackslash{}tblue\textbackslash{}n139  69  19\textbackslash{}t\textbackslash{}tsaddlebrown\textbackslash{}n177  89  40\textbackslash{}t\textbackslash{}tbrown\textbackslash{}n51 160   44\textbackslash{}t\textbackslash{}tgreen\textbackslash{}n255 237   111\textbackslash{}t\textbackslash{}tyellow\textbackslash{}n173 255  47\textbackslash{}t\textbackslash{}tgreenyellow\textbackslash{}n255   0   0\textbackslash{}t\textbackslash{}tred\textbackslash{}n255 255 255\textbackslash{}t\textbackslash{}twhite\textbackslash{}n0   0   0\textbackslash{}t\textbackslash{}tblack\textbackslash{}n255 192 203\textbackslash{}t\textbackslash{}tpink\textbackslash{}n255   0 255\textbackslash{}t\textbackslash{}tmagenta\textbackslash{}n160  32 240\textbackslash{}t\textbackslash{}tpurple\textbackslash{}n210 180 140\textbackslash{}t\textbackslash{}ttan\textbackslash{}n250 128 114\textbackslash{}t\textbackslash{}tsalmon\textbackslash{}n166 206 227\textbackslash{}t\textbackslash{}tcyan\textbackslash{}n25  25 112\textbackslash{}t\textbackslash{}tmidnightblue\textbackslash{}n224 255 255\textbackslash{}t\textbackslash{}tlightcyan\textbackslash{}n153 153 153 \textbackslash{}t\textbackslash{}tgrey60\textbackslash{}n144 238 144\textbackslash{}t\textbackslash{}tlightgreen\textbackslash{}n255 255 224\textbackslash{}t\textbackslash{}tlightyellow\textbackslash{}n65 105 225\textbackslash{}t\textbackslash{}troyalblue\textbackslash{}n139   0   0\textbackslash{}t\textbackslash{}tdarkred\textbackslash{}n0 100   0\textbackslash{}t\textbackslash{}tdarkgreen\textbackslash{}n0 206 209\textbackslash{}t\textbackslash{}tdarkturquoise\textbackslash{}n169 169 169\textbackslash{}t\textbackslash{}tdarkgrey\textbackslash{}n255 165   0\textbackslash{}t\textbackslash{}torange\textbackslash{}n255 140   0\textbackslash{}t\textbackslash{}tdarkorange\textbackslash{}n135 206 235\textbackslash{}t\textbackslash{}tskyblue\textbackslash{}n70 130 180\textbackslash{}t\textbackslash{}tsteelblue\textbackslash{}n175 238 238\textbackslash{}t\textbackslash{}tpaleturquoise\textbackslash{}n238 130 238\textbackslash{}t\textbackslash{}tviolet\textbackslash{}n85 107  47\textbackslash{}t\textbackslash{}tdarkolivegreen\textbackslash{}n139   0 139\textbackslash{}t\textbackslash{}tdarkmagenta\textbackslash{}n190 190 190\textbackslash{}t\textbackslash{}tgray\textbackslash{}n190 190 190\textbackslash{}t\textbackslash{}tgrey\textbackslash{}n\textquotesingle{}}}{}
Returns a dictionary that maps color names to RGB strings.

The format of RGB strings is ‘\#RRGGBB’.

\end{fulllineitems}

\index{read\_colors() (in module analytics\_core.viz.color\_list)@\spxentry{read\_colors()}\spxextra{in module analytics\_core.viz.color\_list}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.color_list.read_colors}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.color\_list.}}\sphinxbfcode{\sphinxupquote{read\_colors}}}{}{}
Returns color information in two data structures.

The format of RGB strings is ‘\#RRGGBB’.

color\_dict: map from color name to RGB string
rgbs: list of (rgb, names) pairs, where rgb is an RGB code and             names is a sorted list of color names

\end{fulllineitems}

\index{invert\_dict() (in module analytics\_core.viz.color\_list)@\spxentry{invert\_dict()}\spxextra{in module analytics\_core.viz.color\_list}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core.viz:analytics_core.viz.color_list.invert_dict}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.viz.color\_list.}}\sphinxbfcode{\sphinxupquote{invert\_dict}}}{\emph{d}}{}
Returns a dictionary that maps from values to lists of keys.

d: dict

returns: dict

\end{fulllineitems}



\subsubsection{R wrapper}
\label{\detokenize{_autosummary/analytics_core:module-analytics_core.R_wrapper}}\label{\detokenize{_autosummary/analytics_core:r-wrapper}}\index{analytics\_core.R\_wrapper (module)@\spxentry{analytics\_core.R\_wrapper}\spxextra{module}}\index{call\_Rpackage() (in module analytics\_core.R\_wrapper)@\spxentry{call\_Rpackage()}\spxextra{in module analytics\_core.R\_wrapper}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.R_wrapper.call_Rpackage}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.R\_wrapper.}}\sphinxbfcode{\sphinxupquote{call\_Rpackage}}}{\emph{call=\textquotesingle{}function\textquotesingle{}}, \emph{designation=\textquotesingle{}aov\textquotesingle{}}}{}
\end{fulllineitems}

\index{R\_matrix2Py\_matrix() (in module analytics\_core.R\_wrapper)@\spxentry{R\_matrix2Py\_matrix()}\spxextra{in module analytics\_core.R\_wrapper}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.R_wrapper.R_matrix2Py_matrix}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.R\_wrapper.}}\sphinxbfcode{\sphinxupquote{R\_matrix2Py\_matrix}}}{\emph{r\_matrix}, \emph{index}, \emph{columns}}{}
\end{fulllineitems}



\subsubsection{Analytics factory}
\label{\detokenize{_autosummary/analytics_core:module-analytics_core.analytics_factory}}\label{\detokenize{_autosummary/analytics_core:analytics-factory}}\index{analytics\_core.analytics\_factory (module)@\spxentry{analytics\_core.analytics\_factory}\spxextra{module}}\index{Analysis (class in analytics\_core.analytics\_factory)@\spxentry{Analysis}\spxextra{class in analytics\_core.analytics\_factory}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.analytics_factory.Analysis}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{analytics\_core.analytics\_factory.}}\sphinxbfcode{\sphinxupquote{Analysis}}}{\emph{identifier}, \emph{analysis\_type}, \emph{args}, \emph{data}, \emph{result=None}}{}
Bases: \sphinxhref{https://docs.python.org/3.6/library/functions.html\#object}{\sphinxcode{\sphinxupquote{object}}}
\index{identifier() (analytics\_core.analytics\_factory.Analysis property)@\spxentry{identifier()}\spxextra{analytics\_core.analytics\_factory.Analysis property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.analytics_factory.Analysis.identifier}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{identifier}}}
\end{fulllineitems}

\index{analysis\_type() (analytics\_core.analytics\_factory.Analysis property)@\spxentry{analysis\_type()}\spxextra{analytics\_core.analytics\_factory.Analysis property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.analytics_factory.Analysis.analysis_type}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{analysis\_type}}}
\end{fulllineitems}

\index{args() (analytics\_core.analytics\_factory.Analysis property)@\spxentry{args()}\spxextra{analytics\_core.analytics\_factory.Analysis property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.analytics_factory.Analysis.args}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{args}}}
\end{fulllineitems}

\index{data() (analytics\_core.analytics\_factory.Analysis property)@\spxentry{data()}\spxextra{analytics\_core.analytics\_factory.Analysis property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.analytics_factory.Analysis.data}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{data}}}
\end{fulllineitems}

\index{result() (analytics\_core.analytics\_factory.Analysis property)@\spxentry{result()}\spxextra{analytics\_core.analytics\_factory.Analysis property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.analytics_factory.Analysis.result}}\pysigline{\sphinxbfcode{\sphinxupquote{property }}\sphinxbfcode{\sphinxupquote{result}}}
\end{fulllineitems}

\index{generate\_result() (analytics\_core.analytics\_factory.Analysis method)@\spxentry{generate\_result()}\spxextra{analytics\_core.analytics\_factory.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.analytics_factory.Analysis.generate_result}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_result}}}{}{}
\end{fulllineitems}

\index{get\_plot() (analytics\_core.analytics\_factory.Analysis method)@\spxentry{get\_plot()}\spxextra{analytics\_core.analytics\_factory.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.analytics_factory.Analysis.get_plot}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_plot}}}{\emph{name}, \emph{identifier}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsubsection{Utils}
\label{\detokenize{_autosummary/analytics_core:module-analytics_core.utils}}\label{\detokenize{_autosummary/analytics_core:utils}}\index{analytics\_core.utils (module)@\spxentry{analytics\_core.utils}\spxextra{module}}\index{generate\_html() (in module analytics\_core.utils)@\spxentry{generate\_html()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.generate_html}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{generate\_html}}}{\emph{network}}{}
This method gets the data structures supporting the nodes, edges,
and options and updates the pyvis html template holding the visualization.
\begin{quote}\begin{description}
\end{description}\end{quote}

\end{fulllineitems}

\index{append\_to\_list() (in module analytics\_core.utils)@\spxentry{append\_to\_list()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.append_to_list}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{append\_to\_list}}}{\emph{mylist}, \emph{myappend}}{}
\end{fulllineitems}

\index{neo4j\_path\_to\_networkx() (in module analytics\_core.utils)@\spxentry{neo4j\_path\_to\_networkx()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.neo4j_path_to_networkx}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{neo4j\_path\_to\_networkx}}}{\emph{paths}, \emph{key=\textquotesingle{}path\textquotesingle{}}}{}
\end{fulllineitems}

\index{neo4j\_schema\_to\_networkx() (in module analytics\_core.utils)@\spxentry{neo4j\_schema\_to\_networkx()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.neo4j_schema_to_networkx}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{neo4j\_schema\_to\_networkx}}}{\emph{schema}}{}
\end{fulllineitems}

\index{networkx\_to\_cytoscape() (in module analytics\_core.utils)@\spxentry{networkx\_to\_cytoscape()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.networkx_to_cytoscape}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{networkx\_to\_cytoscape}}}{\emph{graph}}{}
\end{fulllineitems}

\index{networkx\_to\_gml() (in module analytics\_core.utils)@\spxentry{networkx\_to\_gml()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.networkx_to_gml}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{networkx\_to\_gml}}}{\emph{graph}, \emph{path}}{}
\end{fulllineitems}

\index{json\_network\_to\_gml() (in module analytics\_core.utils)@\spxentry{json\_network\_to\_gml()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.json_network_to_gml}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{json\_network\_to\_gml}}}{\emph{graph\_json}, \emph{path}}{}
\end{fulllineitems}

\index{json\_network\_to\_networkx() (in module analytics\_core.utils)@\spxentry{json\_network\_to\_networkx()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.json_network_to_networkx}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{json\_network\_to\_networkx}}}{\emph{graph\_json}}{}
\end{fulllineitems}

\index{get\_clustergrammer\_link() (in module analytics\_core.utils)@\spxentry{get\_clustergrammer\_link()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.get_clustergrammer_link}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{get\_clustergrammer\_link}}}{\emph{net}, \emph{filename=None}}{}
\end{fulllineitems}

\index{generator\_to\_dict() (in module analytics\_core.utils)@\spxentry{generator\_to\_dict()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.generator_to_dict}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{generator\_to\_dict}}}{\emph{genvar}}{}
\end{fulllineitems}

\index{parse\_html() (in module analytics\_core.utils)@\spxentry{parse\_html()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.parse_html}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{parse\_html}}}{\emph{html\_snippet}}{}
\end{fulllineitems}

\index{hex2rgb() (in module analytics\_core.utils)@\spxentry{hex2rgb()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.hex2rgb}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{hex2rgb}}}{\emph{color}}{}
\end{fulllineitems}

\index{get\_rgb\_colors() (in module analytics\_core.utils)@\spxentry{get\_rgb\_colors()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.get_rgb_colors}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{get\_rgb\_colors}}}{\emph{n}}{}
\end{fulllineitems}

\index{get\_hex\_colors() (in module analytics\_core.utils)@\spxentry{get\_hex\_colors()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.get_hex_colors}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{get\_hex\_colors}}}{\emph{n}}{}
\end{fulllineitems}

\index{getMedlineAbstracts() (in module analytics\_core.utils)@\spxentry{getMedlineAbstracts()}\spxextra{in module analytics\_core.utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/analytics_core:analytics_core.utils.getMedlineAbstracts}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{analytics\_core.utils.}}\sphinxbfcode{\sphinxupquote{getMedlineAbstracts}}}{\emph{idList}}{}
\end{fulllineitems}



\subsection{Notebooks}
\label{\detokenize{_autosummary/notebooks:notebooks}}\label{\detokenize{_autosummary/notebooks::doc}}

\subsubsection{vis.py}
\label{\detokenize{_autosummary/notebooks:module-notebooks.development.vis}}\label{\detokenize{_autosummary/notebooks:vis-py}}\index{notebooks.development.vis (module)@\spxentry{notebooks.development.vis}\spxextra{module}}\index{vis\_network() (in module notebooks.development.vis)@\spxentry{vis\_network()}\spxextra{in module notebooks.development.vis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/notebooks:notebooks.development.vis.vis_network}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{notebooks.development.vis.}}\sphinxbfcode{\sphinxupquote{vis\_network}}}{\emph{nodes}, \emph{edges}, \emph{physics=False}}{}
\end{fulllineitems}

\index{draw() (in module notebooks.development.vis)@\spxentry{draw()}\spxextra{in module notebooks.development.vis}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{_autosummary/notebooks:notebooks.development.vis.draw}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{notebooks.development.vis.}}\sphinxbfcode{\sphinxupquote{draw}}}{\emph{graph}, \emph{options}, \emph{physics=False}, \emph{limit=100}}{}
\end{fulllineitems}



\chapter{Project Info}
\label{\detokenize{index:project-info}}

\section{Credits}
\label{\detokenize{MANIFEST:credits}}\label{\detokenize{MANIFEST::doc}}

\subsection{Development Leads}
\label{\detokenize{MANIFEST:development-leads}}\begin{itemize}
\item {} 
Alberto Santos Delgado (\sphinxhref{https://github.com/albsantosdel}{@albsantosdel})

\item {} 
Ana Rita Colaço (\sphinxhref{https://github.com/arcolaco}{@arcolaco})

\item {} 
Annelaura Bach Nielsen (\sphinxhref{https://github.com/annelaura}{@aannelaura})

\end{itemize}


\subsection{Core Committers}
\label{\detokenize{MANIFEST:core-committers}}

\subsection{Contributors}
\label{\detokenize{MANIFEST:contributors}}

\section{Backers}
\label{\detokenize{MANIFEST:backers}}
We would like to thank the following people for supporting us in our efforts to maintain and improve the Clinical Knowledge Graph:
\begin{itemize}
\item {} 
\item {} 
\end{itemize}


\section{Contributing}
\label{\detokenize{MANIFEST:contributing}}
Contributions are welcome, and they are greatly appreciated! Every little bit helps, and credit will always be given.
\begin{itemize}
\item {} 
Types of Contributions {\hyperref[\detokenize{MANIFEST:types-of-contributions}]{\sphinxcrossref{Types of Contributions}}}

\item {} 
Contributor Setup {\hyperref[\detokenize{MANIFEST:setting-up-the-code-for-local-development}]{\sphinxcrossref{Setting Up the Code for Local Development}}}

\item {} 
Contributor Guidelines {\hyperref[\detokenize{MANIFEST:contributor-guidelines}]{\sphinxcrossref{Contributor Guidelines}}}

\item {} 
Core Committer Guide {\hyperref[\detokenize{MANIFEST:core-committer-guide}]{\sphinxcrossref{Core Committer Guide}}}

\end{itemize}


\subsection{Types of Contributions}
\label{\detokenize{MANIFEST:types-of-contributions}}
You can contribute in many ways:


\subsubsection{Create Analysis or Visualization methods}
\label{\detokenize{MANIFEST:create-analysis-or-visualization-methods}}
If you develop new ways of analysing or visualizing data, please feel free to add to the Analytics Core.


\subsubsection{Report Bugs}
\label{\detokenize{MANIFEST:report-bugs}}
Report bugs at \sphinxurl{https://github.com/MannLabs/CKG/issues}.

If you are reporting a bug, please include:
\begin{itemize}
\item {} 
Your operating system name and version.

\item {} 
Any details about your local setup that might be helpful in troubleshooting.

\item {} 
If you can, provide detailed steps to reproduce the bug.

\item {} 
If you don’t have steps to reproduce the bug, just note your observations in as much detail as you can. Questions to start a discussion about the issue are welcome.

\end{itemize}


\subsubsection{Fix Bugs}
\label{\detokenize{MANIFEST:fix-bugs}}
Look through the GitHub issues for bugs. Anything tagged with “bug” is open to whoever wants to implement it.


\subsubsection{Implement Features}
\label{\detokenize{MANIFEST:implement-features}}
Look through the GitHub issues for features. Anything tagged with “enhancement” and “please\sphinxhyphen{}help” is open to whoever wants to implement it.

Please do not combine multiple feature enhancements into a single pull request.


\subsubsection{Write Documentation}
\label{\detokenize{MANIFEST:write-documentation}}
The Clinical Knowledge Graph could always use more documentation, whether as part of the official docs, in docstrings, or even on the web in blog posts, articles, and such.

If you want to review your changes on the documentation locally, you can do:

This will compile the documentation, open it in your browser and start watching the files for changes, recompiling as you save.


\subsubsection{Submit Feedback}
\label{\detokenize{MANIFEST:submit-feedback}}
The best way to send feedback is to file an issue at \sphinxurl{https://github.com/MannLabs/CKG/issues}.

If you are proposing a feature:
\begin{itemize}
\item {} 
Explain in detail how it would work.

\item {} 
Keep the scope as narrow as possible, to make it easier to implement.

\item {} 
Remember that this is a volunteer\sphinxhyphen{}driven project, and that contributions are welcome :)

\end{itemize}


\subsection{Setting Up the Code for Local Development}
\label{\detokenize{MANIFEST:setting-up-the-code-for-local-development}}
Here’s how to set up \sphinxcode{\sphinxupquote{CKG}} for local development.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
Fork the \sphinxcode{\sphinxupquote{CKG}} repo on GitHub.

\item {} 
Clone your fork locally:

\end{enumerate}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{2}
\item {} 
Install your local copy into a virtualenv. Assuming you have virtualenvwrapper installed, this is how you set up your fork for local development:

\end{enumerate}

FINISH THIS PART!!!!!!
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{3}
\item {} 
Create a branch for local development:

\end{enumerate}

Now you can make your changes locally.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{4}
\item {} 
When you’re done making changes, check that your changes pass ……

\item {} 
Commit your changes and push your branch to GitHub:

\end{enumerate}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\setcounter{enumi}{6}
\item {} 
Submit a pull request through the GitHub website.

\end{enumerate}


\subsection{Contributor Guidelines}
\label{\detokenize{MANIFEST:contributor-guidelines}}

\subsubsection{Pull Request Guidelines}
\label{\detokenize{MANIFEST:pull-request-guidelines}}
Before you submit a pull request, check that it meets these guidelines:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
If the pull request adds functionality, the docs should be updated. Put your new functionality into a function with a docstring, and describe it.

\item {} 
The pull request should work for Python 3.5, 3.6 and 3.7.

\end{enumerate}


\subsubsection{Coding Standards}
\label{\detokenize{MANIFEST:coding-standards}}\begin{itemize}
\item {} 
PEP8

\item {} 
Functions over classes except in tests

\item {} 
Quotes via \sphinxurl{http://stackoverflow.com/a/56190/5549}
\begin{itemize}
\item {} 
Use double quotes around strings that are used for interpolation or that are natural language messages

\item {} 
Use single quotes for small symbol\sphinxhyphen{}like strings (but break the rules if the strings contain quotes)

\item {} 
Use triple double quotes for docstrings and raw string literals for regular expressions even if they aren’t needed.

\item {} 
Example:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{LIGHT\PYGZus{}MESSAGES} \PYG{o}{=} \PYG{p}{\PYGZob{}}
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{English}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{There are }\PYG{l+s+si}{\PYGZpc{}(number\PYGZus{}of\PYGZus{}lights)s}\PYG{l+s+s2}{ lights.}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pirate}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}  \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Arr! Thar be }\PYG{l+s+si}{\PYGZpc{}(number\PYGZus{}of\PYGZus{}lights)s}\PYG{l+s+s2}{ lights.}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{\PYGZcb{}}
\PYG{k}{def} \PYG{n+nf}{lights\PYGZus{}message}\PYG{p}{(}\PYG{n}{language}\PYG{p}{,} \PYG{n}{number\PYGZus{}of\PYGZus{}lights}\PYG{p}{)}\PYG{p}{:}
    \PYG{l+s+sd}{\PYGZdq{}\PYGZdq{}\PYGZdq{}Return a language\PYGZhy{}appropriate string reporting the light count.\PYGZdq{}\PYGZdq{}\PYGZdq{}}
    \PYG{k}{return} \PYG{n}{LIGHT\PYGZus{}MESSAGES}\PYG{p}{[}\PYG{n}{language}\PYG{p}{]} \PYG{o}{\PYGZpc{}} \PYG{n+nb}{locals}\PYG{p}{(}\PYG{p}{)}
\PYG{k}{def} \PYG{n+nf}{is\PYGZus{}pirate}\PYG{p}{(}\PYG{n}{message}\PYG{p}{)}\PYG{p}{:}
    \PYG{l+s+sd}{\PYGZdq{}\PYGZdq{}\PYGZdq{}Return True if the given message sounds piratical.\PYGZdq{}\PYGZdq{}\PYGZdq{}}
    \PYG{k}{return} \PYG{n}{re}\PYG{o}{.}\PYG{n}{search}\PYG{p}{(}\PYG{l+s+sa}{r}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{(?i)(arr|avast|yohoho)!}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{message}\PYG{p}{)} \PYG{o+ow}{is} \PYG{o+ow}{not} \PYG{n+nb+bp}{None}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
Write new code in Python 3.

\end{itemize}


\subsection{Core Committer Guide}
\label{\detokenize{MANIFEST:core-committer-guide}}

\subsubsection{Vision and Scope}
\label{\detokenize{MANIFEST:vision-and-scope}}
Core committers, use this section to:
\begin{itemize}
\item {} 
Guide your instinct and decisions as a core committer

\item {} 
Limit the codebase from growing infinitely

\end{itemize}


\paragraph{Command\sphinxhyphen{}Line and API Accessible}
\label{\detokenize{MANIFEST:command-line-and-api-accessible}}\begin{itemize}
\item {} 
Provides command\sphinxhyphen{}line utilities that launch a dash app to browse projects, statistics and others, create new users, and import and load data into the database.

\item {} 
Extremely easy to use without having to think too hard

\item {} 
Flexible for more complex use via optional arguments

\end{itemize}


\paragraph{Extensible}
\label{\detokenize{MANIFEST:extensible}}
Being extendable by people with different ideas.
\begin{itemize}
\item {} 
Entirely function\sphinxhyphen{}based

\item {} 
Aim for statelessness

\item {} 
Lets anyone write more opinionated tools

\end{itemize}

Freedom for CKG users to build and extend.
\begin{itemize}
\item {} 
Community\sphinxhyphen{}based project, all contributions to improve and/or extend the code are welcome.

\end{itemize}


\paragraph{Inclusive}
\label{\detokenize{MANIFEST:inclusive}}\begin{itemize}
\item {} 
Cross\sphinxhyphen{}platform support.

\item {} 
Fixing Windows bugs even if it’s a pain, to allow for use by the entire community.

\end{itemize}


\subsubsection{Process: Pull Requests}
\label{\detokenize{MANIFEST:process-pull-requests}}
If a pull request is untriaged:
\begin{itemize}
\item {} 
Look at the roadmap

\item {} 
Set it for the milestone where it makes the most sense

\item {} 
Add it to the roadmap

\end{itemize}

How to prioritize pull requests, from most to least important:
\begin{itemize}
\item {} 
Fixes for broken code. Broken means broken on any supported platform or Python version.

\item {} 
Features.

\item {} 
Bug fixes.

\item {} 
Major edits to docs.

\item {} 
Extra tests to cover corner cases.

\item {} 
Minor edits to docs.

\end{itemize}

Ensure that each pull request meets all requirements in \sphinxhref{https://gist.github.com/audreyr/4feef90445b9680475f2}{checklist}.


\subsubsection{Process: Issues}
\label{\detokenize{MANIFEST:process-issues}}
If an issue is a bug that needs an urgent fix, mark it for the next patch release.
Then either fix it or mark as please\sphinxhyphen{}help.

For other issues: encourage friendly discussion, moderate debate, offer your thoughts.

New features require a +1 from 2 other core committers (besides yourself).


\subsubsection{Process: Pull Request merging and HISTORY.md maintenance}
\label{\detokenize{MANIFEST:process-pull-request-merging-and-history-md-maintenance}}
If you merge a pull request, you’re responsible for updating \sphinxcode{\sphinxupquote{AUTHORS.rst}} and \sphinxcode{\sphinxupquote{HISTORY.rst}}

When you’re processing the first change after a release, create boilerplate following the existing pattern:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZsh{}\PYGZsh{} x.y.z (Development)

The goals of this release are TODO: release summary of features

Features:

* Feature description, thanks to [@contributor](https://github.com/contributor) (\PYGZsh{}PR).

Bug Fixes:

* Bug fix description, thanks to [@contributor](https://github.com/contributor) (\PYGZsh{}PR).

Other changes:

* Description of the change, thanks to [@contributor](https://github.com/contributor) (\PYGZsh{}PR).
\end{sphinxVerbatim}


\subsubsection{Process: Accepting New Features Pull Requests}
\label{\detokenize{MANIFEST:process-accepting-new-features-pull-requests}}\begin{itemize}
\item {} 
Run the feature to generate the output.

\item {} 
Attempt to include it in the standard pipeline and run an example project dataset.

\item {} 
Merge the feature in.

\item {} 
Update the history file.

\end{itemize}

note: Adding features doesn’t give authors credit.


\subsubsection{Process: Your own code changes}
\label{\detokenize{MANIFEST:process-your-own-code-changes}}
All code changes, regardless of who does them, need to be reviewed and merged by someone else.
This rule applies to all the core committers.

Exceptions:
\begin{itemize}
\item {} 
Minor corrections and fixes to pull requests submitted by others.

\item {} 
While making a formal release, the release manager can make necessary, appropriate changes.

\item {} 
Small documentation changes that reinforce existing subject matter. Most commonly being, but not limited to spelling and grammar corrections.

\end{itemize}


\subsubsection{Responsibilities}
\label{\detokenize{MANIFEST:responsibilities}}\begin{itemize}
\item {} 
Ensure cross\sphinxhyphen{}platform compatibility for every change that’s accepted. Windows, Mac, Debian \& Ubuntu Linux.

\item {} 
Ensure that code that goes into core meets all requirements in this checklist: \sphinxurl{https://gist.github.com/audreyr/4feef90445b9680475f2}

\item {} 
Create issues for any major changes and enhancements that you wish to make. Discuss things transparently and get community feedback.

\item {} 
Keep feature versions as small as possible, preferably one new feature per version.

\item {} 
Be welcoming to newcomers and encourage diverse new contributors from all backgrounds. Look at \sphinxtitleref{Code of Conduct :ref:code\sphinxhyphen{}of\sphinxhyphen{}conduct}.

\end{itemize}


\section{History}
\label{\detokenize{MANIFEST:history}}

\subsection{1.0b0 (2020\sphinxhyphen{}05\sphinxhyphen{}11)}
\label{\detokenize{MANIFEST:b0-2020-05-11}}\begin{itemize}
\item {} 
First release on GitHub.

\end{itemize}

Beta version of CKG for trial under real conditions, by community users.


\section{Code of Conduct}
\label{\detokenize{MANIFEST:code-of-conduct}}\label{\detokenize{MANIFEST:id2}}
Everyone interacting in the Clinical Knowledge Graph codebases, issue trackers,
chat rooms, and mailing lists is expected to follow the \sphinxhref{https://www.pypa.io/en/latest/code-of-conduct/}{PyPA Code of Conduct}.


\chapter{Index}
\label{\detokenize{index:index}}\begin{itemize}
\item {} 
\DUrole{xref,std,std-ref}{genindex}

\item {} 
\DUrole{xref,std,std-ref}{modindex}

\item {} 
\DUrole{xref,std,std-ref}{search}

\end{itemize}


\renewcommand{\indexname}{Python Module Index}
\begin{sphinxtheindex}
\let\bigletter\sphinxstyleindexlettergroup
\bigletter{a}
\item\relax\sphinxstyleindexentry{analytics\_core.analytics.analytics}\sphinxstyleindexpageref{_autosummary/analytics_core.analytics:\detokenize{module-analytics_core.analytics.analytics}}
\item\relax\sphinxstyleindexentry{analytics\_core.analytics.wgcnaAnalysis}\sphinxstyleindexpageref{_autosummary/analytics_core.analytics:\detokenize{module-analytics_core.analytics.wgcnaAnalysis}}
\item\relax\sphinxstyleindexentry{analytics\_core.analytics\_factory}\sphinxstyleindexpageref{_autosummary/analytics_core:\detokenize{module-analytics_core.analytics_factory}}
\item\relax\sphinxstyleindexentry{analytics\_core.R\_wrapper}\sphinxstyleindexpageref{_autosummary/analytics_core:\detokenize{module-analytics_core.R_wrapper}}
\item\relax\sphinxstyleindexentry{analytics\_core.utils}\sphinxstyleindexpageref{_autosummary/analytics_core:\detokenize{module-analytics_core.utils}}
\item\relax\sphinxstyleindexentry{analytics\_core.viz.color\_list}\sphinxstyleindexpageref{_autosummary/analytics_core.viz:\detokenize{module-analytics_core.viz.color_list}}
\item\relax\sphinxstyleindexentry{analytics\_core.viz.Dendrogram}\sphinxstyleindexpageref{_autosummary/analytics_core.viz:\detokenize{module-analytics_core.viz.Dendrogram}}
\item\relax\sphinxstyleindexentry{analytics\_core.viz.viz}\sphinxstyleindexpageref{_autosummary/analytics_core.viz:\detokenize{module-analytics_core.viz.viz}}
\item\relax\sphinxstyleindexentry{analytics\_core.viz.wgcnaFigures}\sphinxstyleindexpageref{_autosummary/analytics_core.viz:\detokenize{module-analytics_core.viz.wgcnaFigures}}
\indexspace
\bigletter{g}
\item\relax\sphinxstyleindexentry{graphdb\_builder.builder.builder}\sphinxstyleindexpageref{_autosummary/graphdb_builder.builder:\detokenize{module-graphdb_builder.builder.builder}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.builder.create\_user}\sphinxstyleindexpageref{_autosummary/graphdb_builder.builder:\detokenize{module-graphdb_builder.builder.create_user}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.builder.importer}\sphinxstyleindexpageref{_autosummary/graphdb_builder.builder:\detokenize{module-graphdb_builder.builder.importer}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.builder.loader}\sphinxstyleindexpageref{_autosummary/graphdb_builder.builder:\detokenize{module-graphdb_builder.builder.loader}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.builder\_utils}\sphinxstyleindexpageref{_autosummary/graphdb_builder:\detokenize{module-graphdb_builder.builder_utils}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.databases\_controller}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases:\detokenize{module-graphdb_builder.databases.databases_controller}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.cancerGenomeInterpreterParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.cancerGenomeInterpreterParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.corumParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.corumParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.disgenetParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.disgenetParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.drugBankParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.drugBankParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.drugGeneInteractionDBParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.drugGeneInteractionDBParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.exposomeParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.exposomeParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.foodbParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.foodbParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.goaParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.goaParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.gwasCatalogParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.gwasCatalogParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.hgncParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.hgncParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.hmdbParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.hmdbParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.hpaParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.hpaParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.intactParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.intactParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.jensenlabParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.jensenlabParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.mutationDsParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.mutationDsParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.oncokbParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.oncokbParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.pathwayCommonsParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.pathwayCommonsParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.pfamParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.pfamParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.pspParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.pspParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.reactomeParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.reactomeParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.refseqParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.refseqParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.siderParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.siderParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.smpdbParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.smpdbParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.stringParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.stringParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.textminingParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.textminingParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.databases.parsers.uniprotParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.databases.parsers:\detokenize{module-graphdb_builder.databases.parsers.uniprotParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.experiments.experiments\_controller}\sphinxstyleindexpageref{_autosummary/graphdb_builder.experiments:\detokenize{module-graphdb_builder.experiments.experiments_controller}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.experiments.parsers.clinicalParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.experiments.parsers:\detokenize{module-graphdb_builder.experiments.parsers.clinicalParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.experiments.parsers.proteomicsParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.experiments.parsers:\detokenize{module-graphdb_builder.experiments.parsers.proteomicsParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.experiments.parsers.wesParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.experiments.parsers:\detokenize{module-graphdb_builder.experiments.parsers.wesParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.mapping}\sphinxstyleindexpageref{_autosummary/graphdb_builder:\detokenize{module-graphdb_builder.mapping}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.ontologies.ontologies\_controller}\sphinxstyleindexpageref{_autosummary/graphdb_builder.ontologies:\detokenize{module-graphdb_builder.ontologies.ontologies_controller}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.ontologies.parsers.icdParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.ontologies.parsers:\detokenize{module-graphdb_builder.ontologies.parsers.icdParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.ontologies.parsers.oboParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.ontologies.parsers:\detokenize{module-graphdb_builder.ontologies.parsers.oboParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.ontologies.parsers.reflectParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.ontologies.parsers:\detokenize{module-graphdb_builder.ontologies.parsers.reflectParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.ontologies.parsers.snomedParser}\sphinxstyleindexpageref{_autosummary/graphdb_builder.ontologies.parsers:\detokenize{module-graphdb_builder.ontologies.parsers.snomedParser}}
\item\relax\sphinxstyleindexentry{graphdb\_builder.users.users\_controller}\sphinxstyleindexpageref{_autosummary/graphdb_builder.users:\detokenize{module-graphdb_builder.users.users_controller}}
\item\relax\sphinxstyleindexentry{graphdb\_connector.connector}\sphinxstyleindexpageref{_autosummary/graphdb_connector:\detokenize{module-graphdb_connector.connector}}
\item\relax\sphinxstyleindexentry{graphdb\_connector.query\_utils}\sphinxstyleindexpageref{_autosummary/graphdb_connector:\detokenize{module-graphdb_connector.query_utils}}
\indexspace
\bigletter{n}
\item\relax\sphinxstyleindexentry{notebooks.development.vis}\sphinxstyleindexpageref{_autosummary/notebooks:\detokenize{module-notebooks.development.vis}}
\indexspace
\bigletter{r}
\item\relax\sphinxstyleindexentry{report\_manager.apps.dataUpload}\sphinxstyleindexpageref{_autosummary/report_manager.apps:\detokenize{module-report_manager.apps.dataUpload}}
\item\relax\sphinxstyleindexentry{report\_manager.apps.homepageStats}\sphinxstyleindexpageref{_autosummary/report_manager.apps:\detokenize{module-report_manager.apps.homepageStats}}
\item\relax\sphinxstyleindexentry{report\_manager.apps.imports}\sphinxstyleindexpageref{_autosummary/report_manager.apps:\detokenize{module-report_manager.apps.imports}}
\item\relax\sphinxstyleindexentry{report\_manager.apps.initialApp}\sphinxstyleindexpageref{_autosummary/report_manager.apps:\detokenize{module-report_manager.apps.initialApp}}
\item\relax\sphinxstyleindexentry{report\_manager.apps.projectCreation}\sphinxstyleindexpageref{_autosummary/report_manager.apps:\detokenize{module-report_manager.apps.projectCreation}}
\item\relax\sphinxstyleindexentry{report\_manager.dataset}\sphinxstyleindexpageref{_autosummary/report_manager:\detokenize{module-report_manager.dataset}}
\item\relax\sphinxstyleindexentry{report\_manager.knowledge}\sphinxstyleindexpageref{_autosummary/report_manager:\detokenize{module-report_manager.knowledge}}
\item\relax\sphinxstyleindexentry{report\_manager.project}\sphinxstyleindexpageref{_autosummary/report_manager:\detokenize{module-report_manager.project}}
\item\relax\sphinxstyleindexentry{report\_manager.report}\sphinxstyleindexpageref{_autosummary/report_manager:\detokenize{module-report_manager.report}}
\item\relax\sphinxstyleindexentry{report\_manager.user}\sphinxstyleindexpageref{_autosummary/report_manager:\detokenize{module-report_manager.user}}
\item\relax\sphinxstyleindexentry{report\_manager.utils}\sphinxstyleindexpageref{_autosummary/report_manager:\detokenize{module-report_manager.utils}}
\item\relax\sphinxstyleindexentry{report\_manager.worker}\sphinxstyleindexpageref{_autosummary/report_manager:\detokenize{module-report_manager.worker}}
\end{sphinxtheindex}

\renewcommand{\indexname}{Index}
\printindex
\end{document}