

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>graphdb_builder.databases.parsers.cancerGenomeInterpreterParser &mdash; ClinicalKnowledgeGraph 1.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../../../../_static/jquery.js"></script>
        <script type="text/javascript" src="../../../../_static/underscore.js"></script>
        <script type="text/javascript" src="../../../../_static/doctools.js"></script>
        <script type="text/javascript" src="../../../../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../../index.html" class="icon icon-home"> ClinicalKnowledgeGraph
          

          
          </a>

          
            
            
              <div class="version">
                1.0.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../INTRO.html">Clinical Knowledge Graph</a></li>
</ul>
<p class="caption"><span class="caption-text">First steps</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../intro/getting-started-with-requirements.html">Getting Started with some Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../intro/getting-started-with-neo4j.html">Getting Started with Neo4j</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../intro/getting-started-with-build.html">Getting Started with the CKG Build</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../intro/getting-started-with-windows.html">Getting started with Windows</a></li>
</ul>
<p class="caption"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../getting_started/connect-to-ckg.html">Connecting to the Clinical Knowledge Graph database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../getting_started/create-new-user.html">Create a new user in the graph database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../getting_started/create-new-project.html">Create a new project in the database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../getting_started/upload-data.html">Upload project experimental data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../getting_started/data-analysis-config.html">Define data analysis parameters</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../getting_started/access-report.html">Accessing the analysis report</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../getting_started/notifications.html">Report notifications</a></li>
</ul>
<p class="caption"><span class="caption-text">Project Report</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../project_report/project-report.html">Generating a project report</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../project_report/project-tabs.html">Project report tabs</a></li>
</ul>
<p class="caption"><span class="caption-text">CKG Builder</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../ckg_builder/ontologies.html">Ontology sources and raw file parsers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../ckg_builder/databases.html">Biomedical databases and resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../ckg_builder/experiments.html">Parsing experimental data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../ckg_builder/graphdb-builder.html">Building the graph database from one Python module</a></li>
</ul>
<p class="caption"><span class="caption-text">Advanced Features</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../advanced_features/import-statistics.html">Clinical Knowledge Graph Statistics: Imports</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../advanced_features/graphdb-statistics.html">Clinical Knowledge Graph Statistics: Database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../advanced_features/ckg-notebooks.html">Using Jupyter Notebooks with the Clinical Knowledge Graph</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../advanced_features/ckg-queries.html">Retrieving data from the Clinical Knowledge Graph database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../advanced_features/standard-analysis.html">Standardizing the data analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../advanced_features/visualization-plots.html">Visualization plots</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../advanced_features/R-interface.html">Python - R interface</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../system_requirements.html">System Requirements</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../src.html">CKG package API Reference</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../MANIFEST.html">Credits</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../MANIFEST.html#backers">Backers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../MANIFEST.html#contributing">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../MANIFEST.html#history">History</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../MANIFEST.html#code-of-conduct">Code of Conduct</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">ClinicalKnowledgeGraph</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../../../index.html">Module code</a> &raquo;</li>
        
      <li>graphdb_builder.databases.parsers.cancerGenomeInterpreterParser</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <h1>Source code for graphdb_builder.databases.parsers.cancerGenomeInterpreterParser</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">os.path</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="k">import</span> <span class="n">defaultdict</span>
<span class="kn">import</span> <span class="nn">zipfile</span>
<span class="kn">from</span> <span class="nn">graphdb_builder</span> <span class="k">import</span> <span class="n">builder_utils</span><span class="p">,</span> <span class="n">mapping</span> <span class="k">as</span> <span class="n">mp</span>
<span class="kn">import</span> <span class="nn">re</span>

<span class="c1">#######################################</span>
<span class="c1">#   The Cancer Genome Interpreter     # </span>
<span class="c1">#######################################</span>
<div class="viewcode-block" id="parser"><a class="viewcode-back" href="../../../../_autosummary/graphdb_builder.databases.parsers.html#graphdb_builder.databases.parsers.cancerGenomeInterpreterParser.parser">[docs]</a><span class="k">def</span> <span class="nf">parser</span><span class="p">(</span><span class="n">databases_directory</span><span class="p">,</span> <span class="n">download</span><span class="o">=</span><span class="kc">True</span><span class="p">):</span>
    <span class="n">variant_regex</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;(\D\d+\D)$&quot;</span>
    <span class="n">regex</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;(chr\d+)\:g\.(\d+)(\w)&gt;(\w)&quot;</span>
    <span class="n">config</span> <span class="o">=</span> <span class="n">builder_utils</span><span class="o">.</span><span class="n">get_config</span><span class="p">(</span><span class="n">config_name</span><span class="o">=</span><span class="s2">&quot;cancerGenomeInterpreterConfig.yml&quot;</span><span class="p">,</span> <span class="n">data_type</span><span class="o">=</span><span class="s1">&#39;databases&#39;</span><span class="p">)</span>
    <span class="n">url</span> <span class="o">=</span> <span class="n">config</span><span class="p">[</span><span class="s1">&#39;cancerBiomarkers_url&#39;</span><span class="p">]</span>
    <span class="n">entities_header</span> <span class="o">=</span> <span class="n">config</span><span class="p">[</span><span class="s1">&#39;entities_header&#39;</span><span class="p">]</span>
    <span class="n">relationships_headers</span> <span class="o">=</span> <span class="n">config</span><span class="p">[</span><span class="s1">&#39;relationships_headers&#39;</span><span class="p">]</span>
    <span class="n">amino_acids</span> <span class="o">=</span> <span class="n">config</span><span class="p">[</span><span class="s1">&#39;amino_acids&#39;</span><span class="p">]</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="n">mp</span><span class="o">.</span><span class="n">getMappingFromOntology</span><span class="p">(</span><span class="n">ontology</span><span class="o">=</span><span class="s2">&quot;Disease&quot;</span><span class="p">,</span> <span class="n">source</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span>
    <span class="n">drugmapping</span> <span class="o">=</span> <span class="n">mp</span><span class="o">.</span><span class="n">getMappingForEntity</span><span class="p">(</span><span class="s2">&quot;Drug&quot;</span><span class="p">)</span>
    <span class="n">protein_mapping</span> <span class="o">=</span> <span class="n">mp</span><span class="o">.</span><span class="n">getMultipleMappingForEntity</span><span class="p">(</span><span class="s2">&quot;Protein&quot;</span><span class="p">)</span>

    <span class="n">fileName</span> <span class="o">=</span> <span class="n">config</span><span class="p">[</span><span class="s1">&#39;cancerBiomarkers_variant_file&#39;</span><span class="p">]</span>
    <span class="n">relationships</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">set</span><span class="p">)</span>
    <span class="n">entities</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
    <span class="n">directory</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">databases_directory</span><span class="p">,</span> <span class="s2">&quot;CancerGenomeInterpreter&quot;</span><span class="p">)</span>
    <span class="n">builder_utils</span><span class="o">.</span><span class="n">checkDirectory</span><span class="p">(</span><span class="n">directory</span><span class="p">)</span>
    <span class="n">zipFile</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">directory</span><span class="p">,</span> <span class="n">url</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;/&#39;</span><span class="p">)[</span><span class="o">-</span><span class="mi">1</span><span class="p">])</span>

    <span class="k">if</span> <span class="n">download</span><span class="p">:</span>
        <span class="n">builder_utils</span><span class="o">.</span><span class="n">downloadDB</span><span class="p">(</span><span class="n">url</span><span class="p">,</span> <span class="n">directory</span><span class="p">)</span>
    <span class="k">with</span> <span class="n">zipfile</span><span class="o">.</span><span class="n">ZipFile</span><span class="p">(</span><span class="n">zipFile</span><span class="p">)</span> <span class="k">as</span> <span class="n">z</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">fileName</span> <span class="ow">in</span> <span class="n">z</span><span class="o">.</span><span class="n">namelist</span><span class="p">():</span>
            <span class="k">with</span> <span class="n">z</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="n">fileName</span><span class="p">,</span> <span class="s1">&#39;r&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">responses</span><span class="p">:</span>
                <span class="n">first</span> <span class="o">=</span> <span class="kc">True</span>
                <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">responses</span><span class="p">:</span>
                    <span class="k">if</span> <span class="n">first</span><span class="p">:</span>
                        <span class="n">first</span> <span class="o">=</span> <span class="kc">False</span>
                        <span class="k">continue</span>
                    <span class="n">data</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">decode</span><span class="p">(</span><span class="s1">&#39;utf-8&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">rstrip</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\r\n</span><span class="s2">&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span> 
                    <span class="n">gene_variant</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)</span>
                    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">gene_variant</span><span class="p">)</span> <span class="o">&lt;</span> <span class="mi">2</span><span class="p">:</span>
                        <span class="k">continue</span>
                    <span class="n">gene</span> <span class="o">=</span> <span class="n">gene_variant</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
                    <span class="n">variants</span> <span class="o">=</span> <span class="n">gene_variant</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)</span>
                    <span class="c1">#alterationType = data[1]</span>
                    <span class="n">response</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span>
                    <span class="n">drugs</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">)</span>
                    <span class="c1">#status = data[11].split(&#39;;&#39;)</span>
                    <span class="n">evidence</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">12</span><span class="p">]</span>
                    <span class="n">tumors</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">16</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">)</span>
                    <span class="n">publications</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">17</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">)</span>
                    <span class="n">identifier</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">21</span><span class="p">]</span>
                    <span class="n">prot_variant</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="mi">22</span><span class="p">]</span>
                    <span class="n">matches</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">identifier</span><span class="p">)</span>
                    <span class="n">alternative_names</span> <span class="o">=</span> <span class="p">[</span><span class="n">identifier</span><span class="p">]</span>
                    <span class="k">if</span> <span class="n">matches</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                        <span class="n">cpra</span> <span class="o">=</span> <span class="n">matches</span><span class="o">.</span><span class="n">groups</span><span class="p">()</span>
                        <span class="n">chromosome</span><span class="p">,</span> <span class="n">position</span><span class="p">,</span> <span class="n">reference</span><span class="p">,</span> <span class="n">alternative</span> <span class="o">=</span> <span class="n">cpra</span>
                        <span class="n">variant</span> <span class="o">=</span> <span class="n">chromosome</span><span class="o">+</span><span class="s2">&quot;:g.&quot;</span><span class="o">+</span><span class="n">position</span><span class="o">+</span><span class="n">reference</span><span class="o">+</span><span class="s2">&quot;&gt;&quot;</span><span class="o">+</span><span class="n">alternative</span>
                        <span class="k">if</span> <span class="n">prot_variant</span> <span class="o">!=</span> <span class="s2">&quot;&quot;</span><span class="p">:</span>
                            <span class="n">prot_variant</span> <span class="o">=</span> <span class="n">prot_variant</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span>
                            <span class="n">alternative_names</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">prot_variant</span><span class="p">)</span>

                    <span class="n">valid_variants</span> <span class="o">=</span> <span class="p">[]</span>
                    <span class="k">if</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">protein_mapping</span><span class="p">:</span>
                        <span class="k">for</span> <span class="n">protein</span> <span class="ow">in</span> <span class="n">protein_mapping</span><span class="p">[</span><span class="n">gene</span><span class="p">]:</span>
                            <span class="k">for</span> <span class="n">variant</span> <span class="ow">in</span> <span class="n">variants</span><span class="p">:</span>
                                <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">search</span><span class="p">(</span><span class="n">variant_regex</span><span class="p">,</span> <span class="n">variant</span><span class="p">)</span>
                                <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
                                    <span class="k">if</span> <span class="n">variant</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="ow">in</span> <span class="n">amino_acids</span> <span class="ow">and</span> <span class="n">variant</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="ow">in</span> <span class="n">amino_acids</span><span class="p">:</span>
                                        <span class="n">valid_variant</span> <span class="o">=</span> <span class="n">protein</span> <span class="o">+</span> <span class="s1">&#39;_p.&#39;</span> <span class="o">+</span> <span class="n">amino_acids</span><span class="p">[</span><span class="n">variant</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">+</span> <span class="s1">&#39;&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">variant</span><span class="p">[</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">])</span> <span class="o">+</span> <span class="n">amino_acids</span><span class="p">[</span><span class="n">variant</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]]</span>
                                        <span class="n">valid_variants</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">valid_variant</span><span class="p">)</span>
                                        <span class="n">entities</span><span class="o">.</span><span class="n">add</span><span class="p">((</span><span class="n">valid_variant</span><span class="p">,</span> <span class="s2">&quot;Clinically_relevant_variant&quot;</span><span class="p">,</span>  <span class="s2">&quot;,&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">alternative_names</span><span class="p">),</span> <span class="n">chromosome</span><span class="p">,</span> <span class="n">position</span><span class="p">,</span> <span class="n">reference</span><span class="p">,</span> <span class="n">alternative</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="s2">&quot;CGI&quot;</span><span class="p">))</span>
                                        <span class="n">relationships</span><span class="p">[</span><span class="s2">&quot;known_variant_is_clinically_relevant&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">add</span><span class="p">((</span><span class="n">valid_variant</span><span class="p">,</span> <span class="n">valid_variant</span><span class="p">,</span> <span class="s2">&quot;KNOWN_VARIANT_IS_CLINICALLY_RELEVANT&quot;</span><span class="p">,</span> <span class="s2">&quot;CGI&quot;</span><span class="p">))</span>

                    <span class="k">for</span> <span class="n">drug</span> <span class="ow">in</span> <span class="n">drugs</span><span class="p">:</span>
                        <span class="k">if</span> <span class="n">drug</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span> <span class="ow">in</span> <span class="n">drugmapping</span><span class="p">:</span>
                            <span class="n">drug</span> <span class="o">=</span> <span class="n">drugmapping</span><span class="p">[</span><span class="n">drug</span><span class="o">.</span><span class="n">lower</span><span class="p">()]</span>
                        <span class="k">elif</span> <span class="n">drug</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span> <span class="ow">in</span> <span class="n">drugmapping</span><span class="p">:</span>
                            <span class="n">drug</span> <span class="o">=</span> <span class="n">drugmapping</span><span class="p">[</span><span class="n">drug</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">lower</span><span class="p">()]</span>
                        <span class="k">elif</span> <span class="s2">&quot; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">drug</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)[</span><span class="mi">1</span><span class="p">:])</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span> <span class="ow">in</span> <span class="n">drugmapping</span><span class="p">:</span>
                            <span class="n">drug</span> <span class="o">=</span> <span class="n">drugmapping</span><span class="p">[</span><span class="s2">&quot; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">drug</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)[</span><span class="mi">1</span><span class="p">:])</span><span class="o">.</span><span class="n">lower</span><span class="p">()]</span>
                        <span class="n">relationships</span><span class="p">[</span><span class="s2">&quot;targets&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">add</span><span class="p">((</span><span class="n">drug</span><span class="p">,</span> <span class="n">gene</span><span class="p">,</span> <span class="s2">&quot;CURATED_TARGETS&quot;</span><span class="p">,</span> <span class="n">evidence</span><span class="p">,</span> <span class="n">response</span><span class="p">,</span> <span class="s2">&quot;,&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">tumors</span><span class="p">),</span> <span class="s2">&quot;curated&quot;</span><span class="p">,</span> <span class="s2">&quot;CGI&quot;</span><span class="p">))</span>

                        <span class="k">for</span> <span class="n">valid_variant</span> <span class="ow">in</span> <span class="n">valid_variants</span><span class="p">:</span>
                            <span class="n">relationships</span><span class="p">[</span><span class="s2">&quot;targets_clinically_relevant_variant&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">add</span><span class="p">((</span><span class="n">drug</span><span class="p">,</span> <span class="n">valid_variant</span><span class="p">,</span> <span class="s2">&quot;TARGETS_CLINICALLY_RELEVANT_VARIANT&quot;</span><span class="p">,</span> <span class="n">evidence</span><span class="p">,</span> <span class="n">response</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">tumors</span><span class="p">),</span> <span class="s2">&quot;curated&quot;</span><span class="p">,</span> <span class="s2">&quot;CGI&quot;</span><span class="p">))</span>

                    <span class="k">for</span> <span class="n">tumor</span> <span class="ow">in</span> <span class="n">tumors</span><span class="p">:</span>                         
                        <span class="k">if</span> <span class="n">tumor</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span> <span class="ow">in</span> <span class="n">mapping</span><span class="p">:</span>
                            <span class="n">tumor</span> <span class="o">=</span> <span class="n">mapping</span><span class="p">[</span><span class="n">tumor</span><span class="o">.</span><span class="n">lower</span><span class="p">()]</span>
                            <span class="k">for</span> <span class="n">valid_variant</span> <span class="ow">in</span> <span class="n">valid_variants</span><span class="p">:</span>
                                <span class="n">relationships</span><span class="p">[</span><span class="s2">&quot;associated_with&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">add</span><span class="p">((</span><span class="n">valid_variant</span><span class="p">,</span> <span class="n">tumor</span><span class="p">,</span> <span class="s2">&quot;ASSOCIATED_WITH&quot;</span><span class="p">,</span> <span class="s2">&quot;curated&quot;</span><span class="p">,</span><span class="s2">&quot;curated&quot;</span><span class="p">,</span> <span class="s2">&quot;CGI&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">publications</span><span class="p">)))</span>

    <span class="n">builder_utils</span><span class="o">.</span><span class="n">remove_directory</span><span class="p">(</span><span class="n">directory</span><span class="p">)</span>

    <span class="k">return</span> <span class="p">(</span><span class="n">entities</span><span class="p">,</span> <span class="n">relationships</span><span class="p">,</span> <span class="n">entities_header</span><span class="p">,</span> <span class="n">relationships_headers</span><span class="p">)</span></div>
</pre></div>

           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Alberto Santos, Ana Rita Cola√ßo, Annelaura B. Nielsen

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>