{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "from graphdb_connector import connector"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('assets/experiments.xlsx', 'rb') as f:\n",
    "    proteomics_data = pd.read_excel(f, sheet_name=\"Proteomics\", header=1)\n",
    "    phosphoSTY_data = pd.read_excel(f, sheet_name=\"Phospho-pSTY\", header=1)\n",
    "    phosphoY_data = pd.read_excel(f, sheet_name=\"Phospho-pY\", header=1)\n",
    "    acetylomics_data = pd.read_excel(f, sheet_name=\"Acetylomics-acK\", header=1)\n",
    "    rnaseq_data = pd.read_excel(f, sheet_name=\"RNA-seq\", header=1)\n",
    "with open('assets/samples.xlsx', 'rb') as f:\n",
    "    clinical_data = pd.read_excel(f, header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>entry_name</th>\n",
       "      <th>accession_number</th>\n",
       "      <th>MB278</th>\n",
       "      <th>MB282</th>\n",
       "      <th>MB275</th>\n",
       "      <th>MB247</th>\n",
       "      <th>MB091</th>\n",
       "      <th>MB234</th>\n",
       "      <th>MB199</th>\n",
       "      <th>...</th>\n",
       "      <th>MB287</th>\n",
       "      <th>MB099</th>\n",
       "      <th>MB266</th>\n",
       "      <th>MB104</th>\n",
       "      <th>MB248</th>\n",
       "      <th>x1M6</th>\n",
       "      <th>MB106</th>\n",
       "      <th>MB270</th>\n",
       "      <th>MB018</th>\n",
       "      <th>MB206</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SYNE2</td>\n",
       "      <td>nesprin-2 isoform 5 GN=SYNE2</td>\n",
       "      <td>NP_878918.2</td>\n",
       "      <td>0.091456</td>\n",
       "      <td>2.150935</td>\n",
       "      <td>-0.331464</td>\n",
       "      <td>-1.433293</td>\n",
       "      <td>2.092454</td>\n",
       "      <td>0.887642</td>\n",
       "      <td>1.550818</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.964865</td>\n",
       "      <td>3.083386</td>\n",
       "      <td>-1.909874</td>\n",
       "      <td>-0.524604</td>\n",
       "      <td>-2.140114</td>\n",
       "      <td>-0.348677</td>\n",
       "      <td>-0.923848</td>\n",
       "      <td>0.506956</td>\n",
       "      <td>1.068076</td>\n",
       "      <td>-0.048876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SYNE1</td>\n",
       "      <td>nesprin-1 isoform 1 GN=SYNE1</td>\n",
       "      <td>NP_892006.3</td>\n",
       "      <td>2.435026</td>\n",
       "      <td>0.820542</td>\n",
       "      <td>-1.851959</td>\n",
       "      <td>-2.981811</td>\n",
       "      <td>2.046466</td>\n",
       "      <td>-3.366614</td>\n",
       "      <td>-0.559521</td>\n",
       "      <td>...</td>\n",
       "      <td>1.499885</td>\n",
       "      <td>0.387276</td>\n",
       "      <td>2.243569</td>\n",
       "      <td>-1.679982</td>\n",
       "      <td>-1.156270</td>\n",
       "      <td>-1.667173</td>\n",
       "      <td>-1.281532</td>\n",
       "      <td>3.313708</td>\n",
       "      <td>-0.696709</td>\n",
       "      <td>2.082124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SYNE2</td>\n",
       "      <td>nesprin-2 isoform 4 GN=SYNE2</td>\n",
       "      <td>NP_878917.1</td>\n",
       "      <td>-0.337245</td>\n",
       "      <td>1.821656</td>\n",
       "      <td>0.221618</td>\n",
       "      <td>-1.492311</td>\n",
       "      <td>1.525269</td>\n",
       "      <td>1.065754</td>\n",
       "      <td>1.029620</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.820821</td>\n",
       "      <td>2.440397</td>\n",
       "      <td>-0.972687</td>\n",
       "      <td>0.562076</td>\n",
       "      <td>-2.452002</td>\n",
       "      <td>-0.112415</td>\n",
       "      <td>-1.003560</td>\n",
       "      <td>0.765873</td>\n",
       "      <td>0.931591</td>\n",
       "      <td>0.557188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1 GN=PLEC</td>\n",
       "      <td>NP_958782.1</td>\n",
       "      <td>0.614473</td>\n",
       "      <td>2.960855</td>\n",
       "      <td>-1.004028</td>\n",
       "      <td>-0.033725</td>\n",
       "      <td>2.687743</td>\n",
       "      <td>0.046718</td>\n",
       "      <td>-0.952974</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.543326</td>\n",
       "      <td>-0.676344</td>\n",
       "      <td>0.604675</td>\n",
       "      <td>-2.558485</td>\n",
       "      <td>-1.503658</td>\n",
       "      <td>-1.543326</td>\n",
       "      <td>-1.279489</td>\n",
       "      <td>-0.161007</td>\n",
       "      <td>2.420232</td>\n",
       "      <td>1.873585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1c GN=PLEC</td>\n",
       "      <td>NP_000436.2</td>\n",
       "      <td>0.628763</td>\n",
       "      <td>2.979443</td>\n",
       "      <td>-1.002101</td>\n",
       "      <td>-0.036535</td>\n",
       "      <td>2.700518</td>\n",
       "      <td>0.052558</td>\n",
       "      <td>-0.940199</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.557045</td>\n",
       "      <td>-0.691168</td>\n",
       "      <td>0.615325</td>\n",
       "      <td>-2.579303</td>\n",
       "      <td>-1.506194</td>\n",
       "      <td>-1.566190</td>\n",
       "      <td>-1.304015</td>\n",
       "      <td>-0.147953</td>\n",
       "      <td>2.420232</td>\n",
       "      <td>1.906170</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  geneSymbol                     entry_name accession_number     MB278  \\\n",
       "0      SYNE2   nesprin-2 isoform 5 GN=SYNE2      NP_878918.2  0.091456   \n",
       "1      SYNE1   nesprin-1 isoform 1 GN=SYNE1      NP_892006.3  2.435026   \n",
       "2      SYNE2   nesprin-2 isoform 4 GN=SYNE2      NP_878917.1 -0.337245   \n",
       "3       PLEC      plectin isoform 1 GN=PLEC      NP_958782.1  0.614473   \n",
       "4       PLEC     plectin isoform 1c GN=PLEC      NP_000436.2  0.628763   \n",
       "\n",
       "      MB282     MB275     MB247     MB091     MB234     MB199  ...     MB287  \\\n",
       "0  2.150935 -0.331464 -1.433293  2.092454  0.887642  1.550818  ... -0.964865   \n",
       "1  0.820542 -1.851959 -2.981811  2.046466 -3.366614 -0.559521  ...  1.499885   \n",
       "2  1.821656  0.221618 -1.492311  1.525269  1.065754  1.029620  ... -0.820821   \n",
       "3  2.960855 -1.004028 -0.033725  2.687743  0.046718 -0.952974  ... -1.543326   \n",
       "4  2.979443 -1.002101 -0.036535  2.700518  0.052558 -0.940199  ... -1.557045   \n",
       "\n",
       "      MB099     MB266     MB104     MB248      x1M6     MB106     MB270  \\\n",
       "0  3.083386 -1.909874 -0.524604 -2.140114 -0.348677 -0.923848  0.506956   \n",
       "1  0.387276  2.243569 -1.679982 -1.156270 -1.667173 -1.281532  3.313708   \n",
       "2  2.440397 -0.972687  0.562076 -2.452002 -0.112415 -1.003560  0.765873   \n",
       "3 -0.676344  0.604675 -2.558485 -1.503658 -1.543326 -1.279489 -0.161007   \n",
       "4 -0.691168  0.615325 -2.579303 -1.506194 -1.566190 -1.304015 -0.147953   \n",
       "\n",
       "      MB018     MB206  \n",
       "0  1.068076 -0.048876  \n",
       "1 -0.696709  2.082124  \n",
       "2  0.931591  0.557188  \n",
       "3  2.420232  1.873585  \n",
       "4  2.420232  1.906170  \n",
       "\n",
       "[5 rows x 48 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteomics_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequenceVariablyModifiedLocations</th>\n",
       "      <th>variableSites</th>\n",
       "      <th>accession_number</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>entry_name</th>\n",
       "      <th>accessionNumber_VMsites_numVMsitesPresent_numVMsitesLocalizedBest_earliestVMsiteAA_latestVMsiteAA</th>\n",
       "      <th>MB278</th>\n",
       "      <th>MB282</th>\n",
       "      <th>MB275</th>\n",
       "      <th>MB247</th>\n",
       "      <th>...</th>\n",
       "      <th>MB287</th>\n",
       "      <th>MB099</th>\n",
       "      <th>MB266</th>\n",
       "      <th>MB104</th>\n",
       "      <th>MB248</th>\n",
       "      <th>x1M6</th>\n",
       "      <th>MB106</th>\n",
       "      <th>MB270</th>\n",
       "      <th>MB018</th>\n",
       "      <th>MB206</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FS(1.0)PEVPGQK</td>\n",
       "      <td>S91s</td>\n",
       "      <td>NP_005900.2</td>\n",
       "      <td>MAP1B</td>\n",
       "      <td>microtubule-associated protein 1B isoform 1 G...</td>\n",
       "      <td>NP_005900.2_S91s._1_1_91_91</td>\n",
       "      <td>0.343731</td>\n",
       "      <td>1.654637</td>\n",
       "      <td>0.962653</td>\n",
       "      <td>-1.030472</td>\n",
       "      <td>...</td>\n",
       "      <td>0.714805</td>\n",
       "      <td>-0.528415</td>\n",
       "      <td>0.355929</td>\n",
       "      <td>-0.034688</td>\n",
       "      <td>-0.999658</td>\n",
       "      <td>-1.234229</td>\n",
       "      <td>-1.526479</td>\n",
       "      <td>0.867438</td>\n",
       "      <td>-0.780989</td>\n",
       "      <td>-0.232932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AS(0.0)LT(0.0)LFCPEEGDWKN(0.50)S(0.50)N(0.50)L...</td>\n",
       "      <td>S199s N200n</td>\n",
       "      <td>NP_005900.2</td>\n",
       "      <td>MAP1B</td>\n",
       "      <td>microtubule-associated protein 1B isoform 1 G...</td>\n",
       "      <td>NP_005900.2_S199s_1_0_199_199</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.769758</td>\n",
       "      <td>2.896013</td>\n",
       "      <td>-0.209092</td>\n",
       "      <td>-0.406924</td>\n",
       "      <td>-1.272480</td>\n",
       "      <td>-0.737773</td>\n",
       "      <td>0.534315</td>\n",
       "      <td>-0.988661</td>\n",
       "      <td>0.490170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IAELEEEQS(0.99)QGS(0.0)T(0.0)T(0.0)NS(0.0)DWMK</td>\n",
       "      <td>S336s</td>\n",
       "      <td>NP_005900.2</td>\n",
       "      <td>MAP1B</td>\n",
       "      <td>microtubule-associated protein 1B isoform 1 G...</td>\n",
       "      <td>NP_005900.2_S336s._1_1_336_336</td>\n",
       "      <td>-0.601529</td>\n",
       "      <td>0.035859</td>\n",
       "      <td>2.789795</td>\n",
       "      <td>-1.124151</td>\n",
       "      <td>...</td>\n",
       "      <td>1.690103</td>\n",
       "      <td>-0.887889</td>\n",
       "      <td>2.538215</td>\n",
       "      <td>0.010599</td>\n",
       "      <td>-0.652194</td>\n",
       "      <td>-0.785418</td>\n",
       "      <td>-1.595935</td>\n",
       "      <td>0.285298</td>\n",
       "      <td>-0.981562</td>\n",
       "      <td>1.099845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IAELEEEQS(0.0)QGS(0.99)T(0.0)T(0.0)NS(0.0)DWMK</td>\n",
       "      <td>S339s</td>\n",
       "      <td>NP_005900.2</td>\n",
       "      <td>MAP1B</td>\n",
       "      <td>microtubule-associated protein 1B isoform 1 G...</td>\n",
       "      <td>NP_005900.2_S339s._1_1_339_339</td>\n",
       "      <td>-0.484790</td>\n",
       "      <td>0.594235</td>\n",
       "      <td>2.474716</td>\n",
       "      <td>-0.871217</td>\n",
       "      <td>...</td>\n",
       "      <td>0.942737</td>\n",
       "      <td>-0.706246</td>\n",
       "      <td>2.281311</td>\n",
       "      <td>0.187894</td>\n",
       "      <td>-0.659626</td>\n",
       "      <td>-0.432235</td>\n",
       "      <td>-1.296504</td>\n",
       "      <td>-0.011544</td>\n",
       "      <td>-1.059660</td>\n",
       "      <td>0.526629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NLIS(1.0)PDLGVVFLNVPENLK</td>\n",
       "      <td>S351s</td>\n",
       "      <td>NP_005900.2</td>\n",
       "      <td>MAP1B</td>\n",
       "      <td>microtubule-associated protein 1B isoform 1 G...</td>\n",
       "      <td>NP_005900.2_S351s._1_1_351_351</td>\n",
       "      <td>-0.160516</td>\n",
       "      <td>-0.472997</td>\n",
       "      <td>1.470897</td>\n",
       "      <td>-1.002368</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sequenceVariablyModifiedLocations variableSites  \\\n",
       "0                                     FS(1.0)PEVPGQK         S91s    \n",
       "1  AS(0.0)LT(0.0)LFCPEEGDWKN(0.50)S(0.50)N(0.50)L...  S199s N200n    \n",
       "2     IAELEEEQS(0.99)QGS(0.0)T(0.0)T(0.0)NS(0.0)DWMK        S336s    \n",
       "3     IAELEEEQS(0.0)QGS(0.99)T(0.0)T(0.0)NS(0.0)DWMK        S339s    \n",
       "4                           NLIS(1.0)PDLGVVFLNVPENLK        S351s    \n",
       "\n",
       "  accession_number geneSymbol  \\\n",
       "0      NP_005900.2      MAP1B   \n",
       "1      NP_005900.2      MAP1B   \n",
       "2      NP_005900.2      MAP1B   \n",
       "3      NP_005900.2      MAP1B   \n",
       "4      NP_005900.2      MAP1B   \n",
       "\n",
       "                                          entry_name  \\\n",
       "0   microtubule-associated protein 1B isoform 1 G...   \n",
       "1   microtubule-associated protein 1B isoform 1 G...   \n",
       "2   microtubule-associated protein 1B isoform 1 G...   \n",
       "3   microtubule-associated protein 1B isoform 1 G...   \n",
       "4   microtubule-associated protein 1B isoform 1 G...   \n",
       "\n",
       "  accessionNumber_VMsites_numVMsitesPresent_numVMsitesLocalizedBest_earliestVMsiteAA_latestVMsiteAA  \\\n",
       "0                        NP_005900.2_S91s._1_1_91_91                                                  \n",
       "1                      NP_005900.2_S199s_1_0_199_199                                                  \n",
       "2                     NP_005900.2_S336s._1_1_336_336                                                  \n",
       "3                     NP_005900.2_S339s._1_1_339_339                                                  \n",
       "4                     NP_005900.2_S351s._1_1_351_351                                                  \n",
       "\n",
       "      MB278     MB282     MB275     MB247  ...     MB287     MB099     MB266  \\\n",
       "0  0.343731  1.654637  0.962653 -1.030472  ...  0.714805 -0.528415  0.355929   \n",
       "1       NaN       NaN       NaN       NaN  ...       NaN -0.769758  2.896013   \n",
       "2 -0.601529  0.035859  2.789795 -1.124151  ...  1.690103 -0.887889  2.538215   \n",
       "3 -0.484790  0.594235  2.474716 -0.871217  ...  0.942737 -0.706246  2.281311   \n",
       "4 -0.160516 -0.472997  1.470897 -1.002368  ...       NaN       NaN       NaN   \n",
       "\n",
       "      MB104     MB248      x1M6     MB106     MB270     MB018     MB206  \n",
       "0 -0.034688 -0.999658 -1.234229 -1.526479  0.867438 -0.780989 -0.232932  \n",
       "1 -0.209092 -0.406924 -1.272480 -0.737773  0.534315 -0.988661  0.490170  \n",
       "2  0.010599 -0.652194 -0.785418 -1.595935  0.285298 -0.981562  1.099845  \n",
       "3  0.187894 -0.659626 -0.432235 -1.296504 -0.011544 -1.059660  0.526629  \n",
       "4       NaN       NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "\n",
       "[5 rows x 51 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phosphoSTY_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "accessions = set(proteomics_data['accession_number'].values.tolist())\n",
    "accessions.update(phosphoSTY_data['accession_number'].values.tolist())\n",
    "accessions.update(phosphoY_data['accession_number'].values.tolist())\n",
    "accessions.update(acetylomics_data['accession_number'].values.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping_query = 'MATCH (t:Transcript)-[:TRANSLATED_INTO]-(p:Protein) WHERE (t.id IN [{}] OR p.name IN [{}]) AND NOT p.id=~\".*-.*\" RETURN t.id, p.id ORDER BY t.id,p.id'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "driver = connector.getGraphDatabaseConnectionConfiguration()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "identifiers = ['\"{}\"'.format(i) for i in list(accessions)]\n",
    "names = ['\"{}\"'.format(g) for g in proteomics_data['geneSymbol'].values.tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping = connector.getCursorData(driver, mapping_query.format(\",\".join(identifiers), \",\".join(names)))\n",
    "mapping = mapping.drop_duplicates(subset=\"t.id\", keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping = mapping.set_index(\"t.id\").to_dict(orient='dict')[\"p.id\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "proteomics_data[\"identifier\"] = [mapping[p] if p in mapping else p for p in proteomics_data[\"accession_number\"].tolist()]\n",
    "phosphoSTY_data[\"identifier\"] = [mapping[p] if p in mapping else p for p in phosphoSTY_data[\"accession_number\"].tolist()]\n",
    "phosphoY_data[\"identifier\"] = [mapping[p] if p in mapping else p for p in phosphoY_data[\"accession_number\"].tolist()]\n",
    "acetylomics_data[\"identifier\"] = [mapping[p] if p in mapping else p for p in acetylomics_data[\"accession_number\"].tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>entry_name</th>\n",
       "      <th>accession_number</th>\n",
       "      <th>MB278</th>\n",
       "      <th>MB282</th>\n",
       "      <th>MB275</th>\n",
       "      <th>MB247</th>\n",
       "      <th>MB091</th>\n",
       "      <th>MB234</th>\n",
       "      <th>MB199</th>\n",
       "      <th>...</th>\n",
       "      <th>MB099</th>\n",
       "      <th>MB266</th>\n",
       "      <th>MB104</th>\n",
       "      <th>MB248</th>\n",
       "      <th>x1M6</th>\n",
       "      <th>MB106</th>\n",
       "      <th>MB270</th>\n",
       "      <th>MB018</th>\n",
       "      <th>MB206</th>\n",
       "      <th>identifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11795</th>\n",
       "      <td>MYC</td>\n",
       "      <td>myc proto-oncogene protein GN=MYC</td>\n",
       "      <td>NP_002458.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.953058</td>\n",
       "      <td>0.170398</td>\n",
       "      <td>-0.943038</td>\n",
       "      <td>4.655508</td>\n",
       "      <td>-0.487767</td>\n",
       "      <td>3.401069</td>\n",
       "      <td>-2.236698</td>\n",
       "      <td>4.270717</td>\n",
       "      <td>-4.278292</td>\n",
       "      <td>P01106</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      geneSymbol                          entry_name accession_number  MB278  \\\n",
       "11795        MYC   myc proto-oncogene protein GN=MYC      NP_002458.2    NaN   \n",
       "\n",
       "       MB282  MB275  MB247  MB091  MB234  MB199  ...     MB099     MB266  \\\n",
       "11795    NaN    NaN    NaN    NaN    NaN    NaN  ... -1.953058  0.170398   \n",
       "\n",
       "          MB104     MB248      x1M6     MB106     MB270     MB018     MB206  \\\n",
       "11795 -0.943038  4.655508 -0.487767  3.401069 -2.236698  4.270717 -4.278292   \n",
       "\n",
       "       identifier  \n",
       "11795      P01106  \n",
       "\n",
       "[1 rows x 49 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteomics_data[proteomics_data['identifier'] == 'P01106']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "phosphoSTY_data[\"variableSites\"] = [m.split(' ')[0][0:-1]+\"-p\" for m in phosphoSTY_data[\"variableSites\"].tolist()]\n",
    "phosphoY_data[\"variableSites\"] = [m.split(' ')[0][0:-1]+\"-p\" for m in phosphoY_data[\"variableSites\"].tolist()]\n",
    "acetylomics_data[\"variableSites\"] = [m.split(' ')[0][0:-1]+\"-ac\" for m in acetylomics_data[\"variableSites\"].tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "phosphoSTY_data[\"identifier\"] = phosphoSTY_data[\"identifier\"] +\"_\"+phosphoSTY_data[\"variableSites\"]\n",
    "phosphoY_data[\"identifier\"] = phosphoY_data[\"identifier\"] +\"_\"+phosphoY_data[\"variableSites\"]\n",
    "acetylomics_data[\"identifier\"] = acetylomics_data[\"identifier\"] +\"_\"+acetylomics_data[\"variableSites\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaseq_data = rnaseq_data.set_index(\"Transcript / Genesym\")\n",
    "rnaseq_data.index.name = \"identifier\"\n",
    "rnaseq_data = rnaseq_data.loc[:,rnaseq_data.columns.isin(clinical_data['sample'].tolist())].transpose().join(clinical_data.set_index('sample')[['group']]).reset_index()\n",
    "proteomics_data = proteomics_data.set_index(\"identifier\")[clinical_data['sample'].tolist()].transpose().join(clinical_data.set_index('sample')[['group']]).reset_index()\n",
    "phosphoSTY_data = phosphoSTY_data.set_index(\"identifier\")[clinical_data['sample'].tolist()].transpose().join(clinical_data.set_index('sample')[['group']]).reset_index()\n",
    "phosphoY_data = phosphoY_data.set_index(\"identifier\")[clinical_data['sample'].tolist()].transpose().join(clinical_data.set_index('sample')[['group']]).reset_index()\n",
    "acetylomics_data = acetylomics_data.set_index(\"identifier\")[clinical_data['sample'].tolist()].transpose().join(clinical_data.set_index('sample')[['group']]).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "proteomics_data.to_csv(path_or_buf=\"assets/proteomics.tsv\", sep='\\t', header=True, index=False, quotechar='\"', line_terminator='\\n', escapechar='\\\\')\n",
    "phosphoSTY_data.to_csv(path_or_buf=\"assets/phosphoSTY.tsv\", sep='\\t', header=True, index=False, quotechar='\"', line_terminator='\\n', escapechar='\\\\')\n",
    "phosphoY_data.to_csv(path_or_buf=\"assets/phosphoY.tsv\", sep='\\t', header=True, index=False, quotechar='\"', line_terminator='\\n', escapechar='\\\\')\n",
    "acetylomics_data.to_csv(path_or_buf=\"assets/acetylomics.tsv\", sep='\\t', header=True, index=False, quotechar='\"', line_terminator='\\n', escapechar='\\\\')\n",
    "rnaseq_data.to_csv(path_or_buf=\"assets/rnaseq.tsv\", sep='\\t', header=True, index=False, quotechar='\"', line_terminator='\\n', escapechar='\\\\')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Q6MZP0</th>\n",
       "      <th>Q8NF91</th>\n",
       "      <th>Q8WXH0</th>\n",
       "      <th>Q15149</th>\n",
       "      <th>Q15149</th>\n",
       "      <th>Q15149</th>\n",
       "      <th>Q15149</th>\n",
       "      <th>P58107</th>\n",
       "      <th>B4DTV0</th>\n",
       "      <th>...</th>\n",
       "      <th>A0A0A0MR82</th>\n",
       "      <th>Q9HD23</th>\n",
       "      <th>P78325</th>\n",
       "      <th>O43763</th>\n",
       "      <th>P28749</th>\n",
       "      <th>A0A024QZT7</th>\n",
       "      <th>Q96MC2</th>\n",
       "      <th>W5XKT8</th>\n",
       "      <th>Q5VZ18</th>\n",
       "      <th>group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MB018</td>\n",
       "      <td>1.068076</td>\n",
       "      <td>-0.696709</td>\n",
       "      <td>0.931591</td>\n",
       "      <td>2.420232</td>\n",
       "      <td>2.420232</td>\n",
       "      <td>2.436102</td>\n",
       "      <td>2.429754</td>\n",
       "      <td>1.690194</td>\n",
       "      <td>0.122202</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.620531</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.797946</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.290428</td>\n",
       "      <td>1.607669</td>\n",
       "      <td>GR3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MB095</td>\n",
       "      <td>-2.623296</td>\n",
       "      <td>1.433604</td>\n",
       "      <td>-2.755208</td>\n",
       "      <td>-0.515201</td>\n",
       "      <td>-0.522668</td>\n",
       "      <td>-0.515201</td>\n",
       "      <td>-0.522668</td>\n",
       "      <td>-0.398223</td>\n",
       "      <td>-0.983114</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.658141</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.707383</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GR3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MB106</td>\n",
       "      <td>-0.923848</td>\n",
       "      <td>-1.281532</td>\n",
       "      <td>-1.003560</td>\n",
       "      <td>-1.279489</td>\n",
       "      <td>-1.304015</td>\n",
       "      <td>-1.304015</td>\n",
       "      <td>-1.301972</td>\n",
       "      <td>0.318850</td>\n",
       "      <td>-2.362762</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.443529</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.339289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.504846</td>\n",
       "      <td>-0.686754</td>\n",
       "      <td>GR3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MB170</td>\n",
       "      <td>-2.355968</td>\n",
       "      <td>1.185584</td>\n",
       "      <td>-2.055772</td>\n",
       "      <td>0.250797</td>\n",
       "      <td>0.252697</td>\n",
       "      <td>0.241297</td>\n",
       "      <td>0.246997</td>\n",
       "      <td>0.294496</td>\n",
       "      <td>0.203297</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.029845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.075999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GR3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB226</td>\n",
       "      <td>0.077085</td>\n",
       "      <td>-3.470951</td>\n",
       "      <td>-0.535310</td>\n",
       "      <td>-1.351123</td>\n",
       "      <td>-1.353264</td>\n",
       "      <td>-1.359688</td>\n",
       "      <td>-1.359688</td>\n",
       "      <td>-1.571671</td>\n",
       "      <td>-2.201195</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.668067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.246243</td>\n",
       "      <td>1.181965</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GR3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 13127 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   index    Q6MZP0    Q8NF91    Q8WXH0    Q15149    Q15149    Q15149  \\\n",
       "0  MB018  1.068076 -0.696709  0.931591  2.420232  2.420232  2.436102   \n",
       "1  MB095 -2.623296  1.433604 -2.755208 -0.515201 -0.522668 -0.515201   \n",
       "2  MB106 -0.923848 -1.281532 -1.003560 -1.279489 -1.304015 -1.304015   \n",
       "3  MB170 -2.355968  1.185584 -2.055772  0.250797  0.252697  0.241297   \n",
       "4  MB226  0.077085 -3.470951 -0.535310 -1.351123 -1.353264 -1.359688   \n",
       "\n",
       "     Q15149    P58107    B4DTV0  ...  A0A0A0MR82    Q9HD23    P78325  \\\n",
       "0  2.429754  1.690194  0.122202  ...         NaN  0.620531       NaN   \n",
       "1 -0.522668 -0.398223 -0.983114  ...         NaN       NaN -2.658141   \n",
       "2 -1.301972  0.318850 -2.362762  ...         NaN -0.443529       NaN   \n",
       "3  0.246997  0.294496  0.203297  ...         NaN       NaN -4.029845   \n",
       "4 -1.359688 -1.571671 -2.201195  ...         NaN -0.668067       NaN   \n",
       "\n",
       "     O43763    P28749  A0A024QZT7    Q96MC2    W5XKT8    Q5VZ18  group  \n",
       "0 -2.797946       NaN         NaN       NaN -0.290428  1.607669    GR3  \n",
       "1       NaN       NaN         NaN -1.707383       NaN       NaN    GR3  \n",
       "2  0.339289       NaN         NaN       NaN -0.504846 -0.686754    GR3  \n",
       "3       NaN       NaN         NaN -0.075999       NaN       NaN    GR3  \n",
       "4  0.246243  1.181965         NaN       NaN       NaN       NaN    GR3  \n",
       "\n",
       "[5 rows x 13127 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteomics_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
