

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Proteomics data analysis parameters &mdash; ClinicalKnowledgeGraph 1.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../../_static/jquery.js"></script>
        <script type="text/javascript" src="../../_static/underscore.js"></script>
        <script type="text/javascript" src="../../_static/doctools.js"></script>
        <script type="text/javascript" src="../../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Multiomics data analysis parameters" href="multiomics.html" />
    <link rel="prev" title="Clinical data analysis parameters" href="clinical-data.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../index.html" class="icon icon-home"> ClinicalKnowledgeGraph
          

          
          </a>

          
            
            
              <div class="version">
                1.0.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../../INTRO.html">Clinical Knowledge Graph</a></li>
</ul>
<p class="caption"><span class="caption-text">First steps</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../intro/getting-started-with-requirements.html">Getting Started with some Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../intro/getting-started-with-neo4j.html">Getting Started with Neo4j</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../intro/getting-started-with-build.html">Getting Started with the CKG Build</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../intro/getting-started-with-windows.html">Getting started with Windows</a></li>
</ul>
<p class="caption"><span class="caption-text">Getting started</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../connect-to-ckg.html">Connecting to the Clinical Knowledge Graph database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../create-new-user.html">Create a new user in the graph database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../create-new-project.html">Create a new project in the database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../upload-data.html">Upload project experimental data</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../data-analysis-config.html">Define data analysis parameters</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="clinical-data.html">Clinical data analysis parameters</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Proteomics data analysis parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="multiomics.html">Multiomics data analysis parameters</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../access-report.html">Accessing the analysis report</a></li>
<li class="toctree-l1"><a class="reference internal" href="../notifications.html">Report notifications</a></li>
</ul>
<p class="caption"><span class="caption-text">Project Report</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../project_report/project-report.html">Generating a project report</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../project_report/project-tabs.html">Project report tabs</a></li>
</ul>
<p class="caption"><span class="caption-text">CKG Builder</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../ckg_builder/ontologies.html">Ontology sources and raw file parsers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../ckg_builder/databases.html">Biomedical databases and resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../ckg_builder/experiments.html">Parsing experimental data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../ckg_builder/graphdb-builder.html">Building the graph database from one Python module</a></li>
</ul>
<p class="caption"><span class="caption-text">Advanced Features</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../advanced_features/import-statistics.html">Clinical Knowledge Graph Statistics: Imports</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../advanced_features/graphdb-statistics.html">Clinical Knowledge Graph Statistics: Database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../advanced_features/ckg-notebooks.html">Using Jupyter Notebooks with the Clinical Knowledge Graph</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../advanced_features/ckg-queries.html">Retrieving data from the Clinical Knowledge Graph database</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../advanced_features/standard-analysis.html">Standardizing the data analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../advanced_features/visualization-plots.html">Visualization plots</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../advanced_features/R-interface.html">Python - R interface</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../system_requirements.html">System Requirements</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../src.html">CKG package API Reference</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../MANIFEST.html">Credits</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../MANIFEST.html#backers">Backers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../MANIFEST.html#contributing">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../MANIFEST.html#history">History</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../MANIFEST.html#code-of-conduct">Code of Conduct</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">ClinicalKnowledgeGraph</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../data-analysis-config.html">Define data analysis parameters</a> &raquo;</li>
        
      <li>Proteomics data analysis parameters</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../../_sources/getting_started/data_settings/proteomics.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="proteomics-data-analysis-parameters">
<span id="proteomics-data-conf-file"></span><h1>Proteomics data analysis parameters<a class="headerlink" href="#proteomics-data-analysis-parameters" title="Permalink to this headline">¶</a></h1>
<p>Similarly to the <a class="reference internal" href="clinical-data.html#clinical-data-conf-file"><span class="std std-ref">Clinical data analysis parameters</span></a>, the Proteomics configuration file is divided into sections: <code class="docutils literal notranslate"><span class="pre">args</span></code>, <code class="docutils literal notranslate"><span class="pre">overview</span></code>, <code class="docutils literal notranslate"><span class="pre">data</span> <span class="pre">exploration</span></code>, <code class="docutils literal notranslate"><span class="pre">data</span> <span class="pre">associations</span></code> and <code class="docutils literal notranslate"><span class="pre">enrichment</span></code>.</p>
<div class="figure align-right" id="id1">
<a class="reference internal image-reference" href="../../_images/proteomics_config_1.png"><img alt="../../_images/proteomics_config_1.png" src="../../_images/proteomics_config_1.png" style="width: 350px;" /></a>
<p class="caption"><span class="caption-text">Proteomics configuration file</span><a class="headerlink" href="#id1" title="Permalink to this image">¶</a></p>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">args</span></code> section contains the parameters used to process the proteomics data extracted from the CKG database, which includes the filtering of proteins according to the determined threshold, as well as the imputation of all missing values. These parameters include:</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>imputation</strong>: boolean. Set to <em>True</em> if missing values shall be imputed.</p></li>
<li><p><strong>imputation_method</strong>: method for missing values imputation (“KNN”, “distribuition”, or “mixed”).</p></li>
<li><p><strong>missing_method</strong>:</p></li>
<li><p><strong>missing_per_group</strong>: boolean. If <em>True</em>, proteins are filtered based on valid values per group; if <em>False</em> filter across all samples.</p></li>
<li><p><strong>missing_max</strong>: maximum ratio of missing/valid values to be filtered. (e.g. <strong>0.3</strong> filters all proteins with more than 30% missing values).</p></li>
<li><p><strong>min_valid</strong>: minimum number of required valid values to keep a protein.</p></li>
<li><p><strong>value_col</strong>: column label containing expression values.</p></li>
<li><p><strong>index</strong>: column labels to be be kept as index identifiers.</p></li>
</ul>
</div></blockquote>
<p>The result is a Pandas dataframe, stored as “processed”, where columns are protein identifiers (UniprotID~GeneName) and analytical samples are rows, group and subject identifier are kept as columns as well.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>We advise to change only <strong>imputation</strong>, <strong>imputation_method</strong>, <strong>missing_method</strong>, <strong>missing_per_group</strong>, and <strong>missing_max</strong> or <strong>min_valid</strong>.</p>
</div>
<div class="figure align-right" id="id2">
<a class="reference internal image-reference" href="../../_images/proteomics_config_2.png"><img alt="../../_images/proteomics_config_2.png" src="../../_images/proteomics_config_2.png" style="width: 350px;" /></a>
<p class="caption"><span class="caption-text">Proteomics configuration file (<em>continuation</em>)</span><a class="headerlink" href="#id2" title="Permalink to this image">¶</a></p>
</div>
<p>The second section (<code class="docutils literal notranslate"><span class="pre">overview</span></code>) corresponds to basic statistics and includes:</p>
<blockquote>
<div><ul class="simple">
<li><p>Data summary statistics (<strong>overview statistics</strong>)</p></li>
<li><p>Number of peptides per sample (<strong>peptides</strong>)</p></li>
<li><p>Number of proteins per sample (<strong>proteins</strong>)</p></li>
<li><p>Number of modifications per sample (<strong>modifications</strong>)</p></li>
<li><p>Dynamic range per group (<strong>ranking</strong>)</p></li>
<li><p>Coefficient of variation per group (<strong>coefficient_variation</strong>)</p></li>
</ul>
</div></blockquote>
<p>In the <code class="docutils literal notranslate"><span class="pre">data</span> <span class="pre">exploration</span></code> section, we look at the sample stratification (<strong>stratification</strong>), differentially expressed proteins (<strong>regulation</strong>), and protein-protein correlation network (<strong>correlation</strong>). In this section, you can choose to modify parameters like <code class="docutils literal notranslate"><span class="pre">alpha</span></code> (FDR value), <code class="docutils literal notranslate"><span class="pre">s0</span></code> (artificial within groups variance) or <code class="docutils literal notranslate"><span class="pre">fc</span></code> (fold-change), to better fit your data and experimental design. Likewise, the correlation network is, by default, set to show correlations above 50% (<code class="docutils literal notranslate"><span class="pre">cutoff:</span> <span class="pre">0.5</span></code>), this too can be changed.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">data</span> <span class="pre">associations</span></code> section includes analyses that correlated differentially expressed proteins to proteins (<strong>interaction_network</strong>), drugs (<strong>drug_associations</strong>), diseases (<strong>disease_associations</strong>) and publications (<strong>literature_associations</strong>). All these associations are directly queried from the CKG database.</p>
<p>The last section in the default analysis pipeline corresponds to the <code class="docutils literal notranslate"><span class="pre">enrichment</span></code> analysis, and includes both Gene Ontology (<strong>go_enrichment</strong>) and Pathway (<strong>pathway_enrichment</strong>) enrichment analyses, using Fisher’s exact test (<code class="docutils literal notranslate"><span class="pre">method:</span> <span class="pre">fisher</span></code>)</p>
<p>Likewise in the <a class="reference internal" href="clinical-data.html#clinical-data-conf-file"><span class="std std-ref">Clinical data analysis parameters</span></a>, within each analysis, specific parameters are defined:</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>description</strong>: Definition of the analysis used.</p></li>
<li><p><strong>data</strong>: defines on which dataset dataframe the analysis will be ran (e.g. “clinical variables”, “original”, “processed”).</p></li>
<li><p><strong>analyses</strong>: which statistical analysis to run on the <code class="docutils literal notranslate"><span class="pre">data</span></code>. These functions are called from the mocdule <code class="docutils literal notranslate"><span class="pre">analytics_factory.py</span></code>.</p></li>
<li><p><strong>plots</strong>: which plot to use to show the results of <strong>analyses</strong>. Functions also called from the mocdule <code class="docutils literal notranslate"><span class="pre">analytics_factory.py</span></code>.</p></li>
<li><p><strong>store_analysis</strong>: boolean. True if the dataframe resulting from <strong>analyses</strong> is to be stored.</p></li>
<li><p><strong>args</strong>: all arguments necessary for <strong>analyses</strong> and <strong>plots</strong>.</p></li>
</ul>
</div></blockquote>
<p>You can modify the analysis parameters just by changing the respective parameters within the configuration file. Remember to consult the modules <code class="docutils literal notranslate"><span class="pre">analytics.py</span></code> and <code class="docutils literal notranslate"><span class="pre">viz.py</span></code>, to learn more about the arguments of each function.
If you would like to add a specific analysis step to the default pipeline, remember to add a call to the function in <code class="docutils literal notranslate"><span class="pre">analytics_factory.py</span></code>.</p>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="multiomics.html" class="btn btn-neutral float-right" title="Multiomics data analysis parameters" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="clinical-data.html" class="btn btn-neutral float-left" title="Clinical data analysis parameters" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Alberto Santos, Ana Rita Colaço, Annelaura B. Nielsen

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>